---
title: "Cadot et al. data analysis in R"
author: "Selma Cadot and Klaus Schlaeppi"
date: '`r Sys.Date()`'
output:
  pdf_document:
    toc: yes
    toc_depth: 3
  word_document:
    toc: yes
  html_document:
    toc: yes
---
   
\pagebreak

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE, fig.align="center")
options(tinytex.verbose=TRUE)
```

```{r libraries, echo=F, message=F, warning=F}

# libraries
library(reshape)
library(sciplot)
library(vegan)
library(phyloseq)
library(ggplot2)
library(plyr)
library(dplyr)
library(pander)
library(edgeR)
library(knitr)
library(coin)
library(gplots)
library(gridExtra) 
library(emmeans)
library(cowplot)
library(grid)   
library(gtable) 
library(gridGraphics)
library(multcomp)
library(stringi)
library(Hmisc)
library(stringr)
library(multipanelfigure)
library(patchwork)
library(broom)

# functions
source("functions/fun_axis_label_pco.R")
source("functions/fun_maplot.R")
source("functions/fun_merge_samples_mean.R")
source("functions/fun_norm_edgeR_obj.R")
source("functions/fun_pairwise_adonis.R")
source("functions/fun_plotting_fig6.R") 
source("functions/fun_plotting_figS10.R") 
source("functions/fun_union_intersect_merge.R")
source("functions/fun_vennDia.R")
source("functions/fun_percent_abundance.R")
source("functions/variability_table.R")
source("functions/ordination_paperBX.R")
  
# grey theme for plots (deactivated by cowplot)
theme_set(theme_gray())
# prepare text for CAP multiplots  
# text_graph() function in "ordination_paperBX.R" file
text_null = text_graph("")
text_rhizo = text_graph("Rhizosphere",srt= 90, size=6)
text_bacteria = text_graph("Bacteria", size=6)
text_root = text_graph("Root", srt=90, size=6)
text_fungi = text_graph("Fungi", size=6)

```


# Data import

## Design file

```{r DESIGN import, warning=F, echo=F,  message=F}
DESIGN <- read.delim("input/Cadot_et_al_SupplementaryData_S1.txt", header=T, row.names=NULL)

## define combined factors
DESIGN$compartment_location_genotype <- as.factor(paste(DESIGN$compartment, DESIGN$location,  DESIGN$genotype, sep="_"))
DESIGN$groups <- as.factor(paste(DESIGN$compartment, DESIGN$location, DESIGN$background, DESIGN$genotype, sep="_"))
DESIGN$location_background_genotype <- as.factor(paste(DESIGN$location, DESIGN$background, DESIGN$genotype, sep="_"))
DESIGN$location_genotype <- as.factor(paste(DESIGN$location, DESIGN$genotype, sep="_"))
DESIGN$background_genotype <- as.factor(paste(DESIGN$background, DESIGN$genotype, sep="_"))
DESIGN$location_compartment <- as.factor(paste(DESIGN$location, DESIGN$compartment, sep="_"))
DESIGN$compartment_background <- as.factor(paste(DESIGN$compartment, DESIGN$background, sep="_"))
DESIGN$genotype_background <- as.factor(paste(DESIGN$genotype, DESIGN$background, sep="_"))


##### LOCATION
## define order of levels 

DESIGN$location <- factor(DESIGN$location, levels=c("Changins", "Aurora", "Reckenholz") )
## define vector with colors 

DESIGN$cols_loc <- as.character(DESIGN$location)
DESIGN[DESIGN$location=="Changins", ]$cols_loc <- "goldenrod2"
DESIGN[DESIGN$location=="Aurora", ]$cols_loc <- "dodgerblue2"
DESIGN[DESIGN$location=="Reckenholz", ]$cols_loc <- "firebrick2"

## collapsed color vector for each level
temp <- data.frame(DESIGN$location, DESIGN$cols_loc)
temp <- plyr::ddply(temp, .variables="DESIGN.cols_loc", .fun=unique)   #library(plyr)
level_cols_loc <- as.character(temp[,2])
names(level_cols_loc) <- temp[,1]
level_cols_loc <- level_cols_loc[levels(DESIGN$location)]


##### COMPARTMENT
## define order of levels 
DESIGN$compartment <- factor(DESIGN$compartment, levels=c("soil", "rhizo", "root") )
# levels(DESIGN$compartment) <- c("Soil", "Rhizosphere", "Root")

## define vector with colors
DESIGN$cols_comp <- as.character(DESIGN$compartment)
DESIGN[DESIGN$compartment=="root", ]$cols_comp <- "seagreen3"
DESIGN[DESIGN$compartment=="rhizo", ]$cols_comp <- "tan3"
DESIGN[DESIGN$compartment=="soil", ]$cols_comp <- "tan4"

## collapsed color vector for each level
temp <- data.frame(DESIGN$compartment, DESIGN$cols_comp)
temp <- plyr::ddply(temp, .variables="DESIGN.cols_comp", .fun=unique)   #library(plyr)
level_cols_comp <- as.character(temp[,2])
names(level_cols_comp) <- temp[,1]

## Define vector with line type for COMPARTMENT
DESIGN$lty_comp <- as.character(DESIGN$compartment)
DESIGN[DESIGN$compartment=="root", ]$lty_comp <- 1
DESIGN[DESIGN$compartment=="rhizo", ]$lty_comp <- 2
DESIGN[DESIGN$compartment=="soil", ]$lty_comp <- 3


##### GENOTYPE
## define order of levels 
DESIGN$genotype <- factor(DESIGN$genotype, levels=c( "no", "WT", "bx1", "bx2", "bx6") )


##### BACKGROUNDS & GENOTYPES
## define order and name of levels 
DESIGN$background_genotype <- factor(DESIGN$background_genotype, 
                                     levels=c( "no_no", "B73_WT", "B73_bx1",
                                               "W22_WT","W22_bx1", "W22_bx2", "W22_bx6") )
levels(DESIGN$background_genotype) <- c("no", "B73", "bx1(B73)",
                                        "W22", "bx1(W22)", "bx2(W22)", "bx6(W22)")

## define vector with colors
DESIGN$cols_bg_gen <- as.character(DESIGN$background_genotype)
DESIGN[DESIGN$background_genotype=="no", ]$cols_bg_gen <- "grey"
DESIGN[DESIGN$background_genotype=="B73", ]$cols_bg_gen <- "gold2"
DESIGN[DESIGN$background_genotype=="bx1(B73)", ]$cols_bg_gen <- "seagreen"
DESIGN[DESIGN$background_genotype=="W22", ]$cols_bg_gen <- "lightgoldenrod1"
DESIGN[DESIGN$background_genotype=="bx1(W22)", ]$cols_bg_gen <- "darkseagreen4"
DESIGN[DESIGN$background_genotype=="bx2(W22)", ]$cols_bg_gen <- "darkseagreen3"
DESIGN[DESIGN$background_genotype=="bx6(W22)", ]$cols_bg_gen <- "darkseagreen2"

## collapsed color vector for each level
temp <- data.frame(DESIGN$background_genotype, DESIGN$cols_bg_gen)
temp <- plyr::ddply(temp, .variables="DESIGN.background_genotype", .fun=unique)   #library(plyr)
level_cols_bg_gen <- as.character(temp[,2])
names(level_cols_bg_gen) <- temp[,1]


##### LOCATION & BACKGROUNDS & GENOTYPES
## define order of levels 
DESIGN$location_background_genotype <- factor(DESIGN$location_background_genotype, 
                                       levels=c( "Changins_no_no", "Changins_B73_WT", "Changins_B73_bx1",
                                                 "Aurora_W22_WT", "Aurora_W22_bx1", "Aurora_W22_bx2", "Aurora_W22_bx6",
                                                 "Reckenholz_B73_WT","Reckenholz_B73_bx1", "Reckenholz_W22_WT", "Reckenholz_W22_bx1") )
## define vector with colors
DESIGN$cols_loc_bg_gen <- as.character(DESIGN$location_background_genotype)
DESIGN[DESIGN$location_background_genotype=="Changins_no_no", ]$cols_loc_bg_gen <- "grey"
DESIGN[DESIGN$location_background_genotype=="Changins_B73_WT", ]$cols_loc_bg_gen <- "gold2"
DESIGN[DESIGN$location_background_genotype=="Reckenholz_B73_WT", ]$cols_loc_bg_gen <- "gold2"
DESIGN[DESIGN$location_background_genotype=="Changins_B73_bx1", ]$cols_loc_bg_gen <- "seagreen"
DESIGN[DESIGN$location_background_genotype=="Reckenholz_B73_bx1", ]$cols_loc_bg_gen <- "seagreen"
DESIGN[DESIGN$location_background_genotype=="Aurora_W22_WT", ]$cols_loc_bg_gen <- "lightgoldenrod1"
DESIGN[DESIGN$location_background_genotype=="Reckenholz_W22_WT", ]$cols_loc_bg_gen <- "lightgoldenrod1"
DESIGN[DESIGN$location_background_genotype=="Aurora_W22_bx1", ]$cols_loc_bg_gen <- "darkseagreen4"
DESIGN[DESIGN$location_background_genotype=="Reckenholz_W22_bx1", ]$cols_loc_bg_gen <- "darkseagreen4"
DESIGN[DESIGN$location_background_genotype=="Aurora_W22_bx2", ]$cols_loc_bg_gen <- "darkseagreen3"
DESIGN[DESIGN$location_background_genotype=="Aurora_W22_bx6", ]$cols_loc_bg_gen <- "darkseagreen2"

## collapsed color vector for each level
temp <- data.frame(DESIGN$location_background_genotype, DESIGN$cols_loc_bg_gen)
temp <- plyr::ddply(temp, .variables="DESIGN.location_background_genotype", .fun=unique)   #library(plyr)
level_cols_loc_bg_gen <- as.character(temp[,2])
names(level_cols_loc_bg_gen) <- temp[,1]
```

```{r b and fDESIGN, warning=F, echo=F,  message=F}
# define bDESIGN, remove lost samples
bDESIGN <- DESIGN[! DESIGN$sample_ID_bac=="sample_lost",]
rownames(bDESIGN) <- bDESIGN$sample_ID_bac

# define fDESIGN, remove lost samples
fDESIGN <- DESIGN[ ! DESIGN$sample_ID_fun=="sample_lost", ]
rownames(fDESIGN) <- fDESIGN$sample_ID_fun
```

Importing design file input/Cadot_et_al_SupplementaryData_S1.txt


## OTU table bacteria

```{r bDAT import, warning=F, echo=F,  message=F}
all_bDAT <- read.delim("input/zh_ch_us_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt", header=T, row.names=1, sep="\t")

# rename ZOTUs to bOTUs for bacteria OTUs
rownames(all_bDAT) <- gsub("ZOTU", "bOTU", rownames(all_bDAT))

## sorting otu_table with sample order of bDESIGN file
bDAT <- all_bDAT[, rownames(bDESIGN) ]
# dim(bDAT)
```

Importing count file input/zh_ch_us_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt


## OTU table fungi

```{r fDAT import, warning=F, echo=F,  message=F, include=F}
all_fDAT <- read.delim("input/zh_ch_us_ITS_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt", header=T, row.names=1, sep="\t")

# rename ZOTUs to fOTUs for fungi OTUs
rownames(all_fDAT) <- gsub("ZOTU", "fOTU", rownames(all_fDAT))   

## sorting otu_table with sample order of fDESIGN file
fDAT <- all_fDAT[, rownames(fDESIGN) ]
# dim(fDAT)
```

Importing OTU file input/zh_ch_us_ITS_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt

## Taxonomy table bacteria

```{r bTAX import, warning=F, echo=F, include=F, message=F}
b_DAT_phylo <- import_qiime(otufilename = "input/zh_ch_us_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt") 
bTAX <- data.frame(tax_table(b_DAT_phylo), stringsAsFactors = FALSE)
colnames(bTAX) <- c("kingdom", "phylum", "class", "order", "family", "genus", "species")
bTAX[bTAX=="Blastocatellaceae_Subgroup_4"] <- "Blastocatellaceae"
bTAX[bTAX=="Clostridiaceae_1"] <- "Clostridiaceae"
bTAX[bTAX=="uncultured_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Chloroflexi_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Acidobacteria_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Acidobacterium_sp."] <- "unassigned"
bTAX[bTAX=="uncultured_actinobacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Armatimonadetes_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Bacteroidetes_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Bellilinea_sp."] <- "unassigned"
bTAX[bTAX=="uncultured_Carnobacterium_sp."] <- "unassigned"
bTAX[bTAX=="uncultured_delta_proteobacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Desulfuromonadales_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Gemmatimonadetes_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Geobacter_sp."] <- "unassigned"
bTAX[bTAX=="uncultured_Planctomycetaceae_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Planctomycetales_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Rhodocyclaceae_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_soil_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Termite_group_1_bacterium"] <- "unassigned"
bTAX[bTAX=="uncultured_Verrucomicrobia_bacterium"] <- "unassigned"
bTAX[is.na(bTAX)] <- "unassigned"

# rename ZOTUs to fOTUs for bacteria OTUs
rownames(bTAX) <- gsub("ZOTU", "bOTU", rownames(bTAX))   

## define OTUs for removal (Eukaryotes, Cyanobacteria, Mitrochondria)
# table(bTAX$kingdom)
# unique(bTAX$kingdom)
# table(bTAX$phylum)
unique(bTAX$phylum)
r1 <- rownames(bTAX[bTAX$phylum=="Cyanobacteria",])   # cannot be safely discriminated from plant organelles
# table(bTAX$family)
# unique(bTAX$family)
r2 <- rownames(bTAX[bTAX$family=="mitochondria",])
otus_to_remove <- unique(r1)
 
## remove these from otu and tax table
bDAT <- bDAT[ -which(rownames(bDAT) %in% otus_to_remove) , ]
bTAX <- bTAX[ rownames(bDAT) ,]
# 
## add OTU_ID to taxonomy file
bTAX$OTU_ID <- rownames(bTAX)
 
### Defining OTU colors by phylum (using the taxonomy file)
bTAX$labels <- as.character(bTAX$phylum)

### create separate taxonomy label specifying classes of Proteobacteria
bTAX$class <- as.factor(bTAX$class)
bTAX[ bTAX$class=="Alphaproteobacteria", ]$labels <- "Alphaproteobacteria"
bTAX[ bTAX$class=="Betaproteobacteria", ]$labels <- "Betaproteobacteria"
bTAX[ bTAX$class=="Gammaproteobacteria", ]$labels <- "Gammaproteobacteria"
bTAX[ bTAX$class=="Deltaproteobacteria", ]$labels <- "Deltaproteobacteria"
# table(bTAX$labels)
bTAX$labels <- as.factor(bTAX$labels)
# levels(bTAX$labels)

### vector of colors for abundant phyla (and classes for Proteobacteria)
bTAX$cols_phyla <- as.character(bTAX$labels)
bTAX$cols_phyla <- "lightgrey"    #low abundant
bTAX[ bTAX$labels=="Acidobacteria" , ]$cols_phyla <- "lightsalmon4"
bTAX[ bTAX$labels=="Actinobacteria" , ]$cols_phyla <- "indianred2"
bTAX[ bTAX$labels=="Alphaproteobacteria" , ]$cols_phyla <- "palegreen1"
bTAX[ bTAX$labels=="Betaproteobacteria" , ]$cols_phyla <- "palegreen3"
bTAX[ bTAX$labels=="Gammaproteobacteria" , ]$cols_phyla <- "palegreen2"
bTAX[ bTAX$labels=="Deltaproteobacteria" , ]$cols_phyla <- "palegreen4"
bTAX[ bTAX$labels=="Bacteroidetes" , ]$cols_phyla <- "steelblue1"
bTAX[ bTAX$labels=="Firmicutes" , ]$cols_phyla <- "tan1"
bTAX[ bTAX$labels=="Chloroflexi" , ]$cols_phyla <- "gold1"
bTAX[ bTAX$labels=="Gemmatimonadetes", ]$cols_phyla <- "peachpuff3"
bTAX[ bTAX$labels=="Verrucomicrobia", ]$cols_phyla <- "orchid3"
bTAX[ bTAX$labels=="Saccharibacteria", ]$cols_phyla <- "steelblue3"
bTAX[ bTAX$labels=="unassigned", ]$cols_phyla <- "dimgrey"
# table(bTAX$cols_phyla)

## collapsed color vector for each level
temp <- data.frame(bTAX$labels, bTAX$cols_phyla)
temp <- plyr::ddply(temp, .variables="bTAX.labels", .fun=unique)   #library(plyr)
bTAX_level_cols_phyla <- as.character(temp[,2])
names(bTAX_level_cols_phyla) <- temp[,1]
```

Importing taxonomy file input/zh_ch_us_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt

## Taxonomy table fungi

```{r fTAX import, warning=F, echo=F,  message=F, include=F}
f_DAT_phylo <- import_qiime(otufilename = "input/zh_ch_us_ITS_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt" ) 
fTAX <- data.frame(tax_table(f_DAT_phylo), stringsAsFactors = FALSE)
colnames(fTAX) <- c("kingdom", "phylum", "class", "order", "family", "genus", "species")
fTAX[is.na(fTAX)] <- "unassigned"
fTAX[fTAX=="Ascomycota_fam_Incertae_sedis"] <- "Ascomycota inc. sed."
fTAX[fTAX=="Saccharomycetales_fam_Incertae_sedis"] <- "Saccharomycetales inc. sed."
fTAX[fTAX=="Hypocreales_fam_Incertae_sedis"] <- "Hypocreales inc. sed."
fTAX[fTAX=="Rhizophydiales_fam_Incertae_sedis"] <- "Rhizophydiales inc. sed."

# rename ZOTUs to fOTUs for fungi OTUs
rownames(fTAX) <- gsub("ZOTU","fOTU", rownames(fTAX))

## define OTUs for removal (Eukaryotes, Cyanobacteria, Mitrochondria)
# table(fTAX$kingdom)
# unique(fTAX$kingdom)
# r1 <- rownames(fTAX[fTAX$kingdom=="Eukaryota_kgd_Incertae_sedis" |
#                           fTAX$kingdom=="Protista"| fTAX$kingdom=="Unassignable" | 
#                           fTAX$kingdom=="Plantae"| fTAX$kingdom=="Protozoa"| 
#                           fTAX$kingdom=="Unclassified" | fTAX$kingdom=="Animalia",])

# table(fTAX$phylum)
# unique(fTAX$phylum)
#r2 <- rownames(fTAX[fTAX$phylum=="Cyanobacteria",])
# table(fTAX$family)
# unique(fTAX$family)
#r3 <- rownames(fTAX[fTAX$family=="Mitochondria",])
#otus_to_remove <- unique(r1)

## remove these from otu and tax table
#fDAT <- fDAT[ -which(rownames(fDAT) %in% otus_to_remove) , ]
#fTAX <- fTAX[ rownames(fDAT) ,]


## add OTU_ID to taxonomy file
fTAX$OTU_ID <- rownames(fTAX)

fTAX$phylum <- as.factor(fTAX$phylum)
levels(fTAX$phylum)[levels(fTAX$phylum) < 2 ] <- "unassigned"
levels(fTAX$phylum)

### Defining fOTU colors by phylum (using the taxonomy file)
fTAX$labels <- as.character(fTAX$phylum)
fTAX$labels <- as.factor(fTAX$labels)
levels(fTAX$labels)

## vector of colors for abundant phyla 
fTAX$cols_phyla <- fTAX$labels
fTAX$cols_phyla <- "lightgrey"
fTAX[fTAX$labels=="Ascomycota", ]$cols_phyla <- "dodgerblue2"
fTAX[fTAX$labels=="Basidiomycota", ]$cols_phyla <- "firebrick1"
fTAX[fTAX$labels=="Glomeromycota", ]$cols_phyla <- "seagreen4"
fTAX[fTAX$labels=="Chytridiomycota", ]$cols_phyla <- "goldenrod2"
fTAX[fTAX$labels=="Mortierellomycota" , ]$cols_phyla <- "mediumorchid1"
fTAX[fTAX$labels=="unassigned", ]$cols_phyla <- "dimgrey"
# table(fTAX$cols_phyla)

## collapsed color vector for each level
temp <- data.frame(fTAX$labels, fTAX$cols_phyla)
temp <- plyr::ddply(temp, .variables="fTAX.labels", .fun=unique)   #library(plyr)
fTAX_level_cols_phyla <- as.character(temp[,2])
names(fTAX_level_cols_phyla) <- temp[,1]
```

Importing taxonomy file input/zh_ch_us_ITS_trimmed_qfiltered_renamed_ZOTU_Count_Sintax.txt


\pagebreak

# Description of all data

To decide on how to normalize the data we followed the recommendation of Weiss et al. (2017, Microbiome Journal) and we inspected whether there are differences in sequencing depths between the different sample groups utilizing the non-parametric Kruskal-Wallis Test. 

## Sequencing depth bacteria 

### FAQs: sum, range, median

```{r seq numbers bacteria, echo=F, message=F, warning=F}
sum(colSums(bDAT)[rownames(bDESIGN)])
range(colSums(bDAT)[rownames(bDESIGN)])
median(colSums(bDAT)[rownames(bDESIGN)])

## define rarefication threshold
# sort(colSums(bDAT), decr=T)
sort(colSums(bDAT), decr=F)[1:5]
b_rare <- 9000
```
The five samples with lowest sequencing depth are shown. We define the rarefication threshold for the bacteria data to `r b_rare` sequences per sample. 

### plotting and stats

```{r seq numbers per group bacteria, fig.height=3, fig.width=5, echo=F, message=F, warning=F}
## data
df_bac <- data.frame(col_sum=colSums(bDAT), colnames(bDAT), 
                      groups=bDESIGN$groups, rownames(bDESIGN), 
                      cols_gen=bDESIGN$cols_bg_gen, location=bDESIGN$location, 
                      location_background_genotype=bDESIGN$location_background_genotype,
                      compartment=bDESIGN$compartment, location_background=bDESIGN$location_background,
                      genotype=bDESIGN$genotype, background_genotype=bDESIGN$background_genotype)
df_bac$compartment <- factor(df_bac$compartment, levels=c("soil", "rhizo", "root") )
df_bac$location <- factor(df_bac$location, levels=c("Changins", "Reckenholz", "Aurora") )

point <- scales::format_format(big.mark=" ", decimal.mark=",", scientific=FALSE)

ylim1=boxplot.stats(df_bac$col_sum)$stats[c(1, 5)]

## Boxplot with ggplot for faceting by compartments
seq_nr_bac <- ggplot(data=df_bac, aes(x=location, y=col_sum, fill=background_genotype)) + 
  geom_boxplot(position=position_dodge2(width=0.75, preserve="single")) +
  facet_grid(. ~ compartment,
  scales='free_y') +
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size=12),
        legend.position="none") + 
  ylab("Sequencing depth") +
  scale_fill_manual(values=level_cols_bg_gen) +
  scale_y_continuous(labels=point) +
  coord_cartesian(ylim=ylim1*1.17)


print(seq_nr_bac)     

## stats 
bac_seq_krusk <- kruskal_test(colSums(bDAT) ~ bDESIGN$groups)    # library(coin)
bac_seq_krusk

# bDESIGN$bac_seqs <- colSums(bDAT)[rownames(bDESIGN)]
# p <- bargraph.CI(bDESIGN$groups, bDESIGN$bac_seqs, las=2, col=level_cols_bg_gen)
# range(p$vals)
# range(p$vals)[2]/range(p$vals)[1]

```

\pagebreak

## Sequencing depth fungi 

### FAQs: sum, range, median

```{r seq numbers fungi, echo=F, message=F, warning=F}
sum(colSums(fDAT)[rownames(fDESIGN)])
range(colSums(fDAT)[rownames(fDESIGN)])
median(colSums(fDAT)[rownames(fDESIGN)])

## define rarefication threshold
# sort(colSums(fDAT), decr=T)
sort(colSums(fDAT), decr=F)[1:6]
f_rare <- 4000
```
The five samples with lowest sequencing depth are shown. We define the rarefication threshold for the fungi data to `r f_rare` sequences per sample. 

### plotting and stats

```{r seq numbers per group fungi, fig.height=3, fig.width=5, echo=F, message=F, warning=F}
## data
df_fun <- data.frame(col_sum=colSums(fDAT), colnames(fDAT), 
                     genotype=fDESIGN$genotype, groups=fDESIGN$groups, 
                     rownames(fDESIGN), cols_bg_gen=fDESIGN$cols_bg_gen, location=fDESIGN$location, 
                     location_background_genotype=fDESIGN$location_background_genotype,
                     compartment=fDESIGN$compartment, location_background=fDESIGN$location_background,
                     background_genotype=fDESIGN$background_genotype)
df_fun$compartment <- factor(df_fun$compartment, levels=c("soil", "rhizo", "root") )
df_fun$location <- factor(df_fun$location, levels=c("Changins", "Reckenholz", "Aurora") )

ylim2=boxplot.stats(df_fun$col_sum)$stats[c(1, 5)]

## Boxplot with ggplot for faceting by compartments
seq_nr_fun <- ggplot(data=df_fun, aes(x=location, y=col_sum, fill=background_genotype)) + 
  geom_boxplot(position=position_dodge2(width=0.75, preserve="single") ) + 
  facet_grid(. ~ compartment,
  scales='free_y') +
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size=12),
        legend.position="none") + 
  ylab("") +
  scale_fill_manual(values=level_cols_bg_gen) +
  scale_y_continuous(labels=point) +
  coord_cartesian(ylim=ylim2*1.05)  # scale y limits based on ylim1  

print(seq_nr_fun)     
                 
## stats
fDAT_seq_krusk <- kruskal_test(colSums(fDAT) ~ fDESIGN$groups)  # library(coin)
fDAT_seq_krusk

# fDESIGN$fDAT_seqs <- colSums(fDAT)[rownames(fDESIGN)]
# b <- bargraph.CI(fDESIGN$groups, fDESIGN$fDAT_seqs, las=2, cex=0.7, ylim=c(0,40000), col=level_cols_groups)
# range(b$vals)
# range(b$vals)[2]/range(b$vals)[1]

```
  
*CONCLUSION:* Because we find significant differences in sequencing depths between the different sample groups (mainly between compartment and location), we will normalize the data by rarefication for diversity comparisions (see Weiss et al. (2017), Microbiome Journal).


## Figure S3 | Sequencing depth

```{r Figure S3 Boxplot sequences, echo=F, warning=F, message=F}
# library(cowplot)
prow <- plot_grid(seq_nr_bac + theme(legend.position="none")
                  #+ panel_border(colour="black")
                  , seq_nr_fun + theme(legend.position="none")
                  # + panel_border(colour="black")
                  , align='vh', labels=c("A", "B"),
                  hjust=-1, nrow=1)

legend_b <- get_legend(seq_nr_bac + theme(legend.position="bottom") +
                       labs(fill="genotype (background)"))

p <- plot_grid( prow, legend_b, nrow=2, rel_heights=c(1, .1))

dir.create("figures")
pdf(file="figures/Fig_S3_sequencing_depth.pdf", width=12, height=6)
p
noshow <- dev.off()

```





\pagebreak

# Rarefaction analysis

## Bacteria  

```{r prepare_rarefaction_plot_16S, include=FALSE, eval=F}
# default plot is all in black, with black lines for the number of samples
rarefaction_bac <- rarecurve(t(bDAT), step=500, 10000, 
                             xlab="number of sequences", ylab="number of OTUs", label=FALSE)
```

```{r rarefaction_plot, fig.width=5, fig.height=4, echo=F, eval=F}
Nmax <- sapply(rarefaction_bac, function(x) max(attr(x, "Subsample")))
Smax <- sapply(rarefaction_bac, max)

plot(c(1,30000), c(1, max(Smax)), xlab="Sequencing depth", main="Bacteria",
     ylab="number of bOTUs", type="n", las=1,ylim=c(0, 6000))
abline(v=b_rare)
for (i in seq_along(rarefaction_bac)) {
  N <- attr(rarefaction_bac[[i]], "Subsample")
  lines(N, rarefaction_bac[[i]], col=bDESIGN$cols_comp[i]
  )
}

legend("topleft", legend=names(level_cols_comp), pch=15,
       col=level_cols_comp, bty="n")
```

### Fungi  

```{r prepare_rarefaction_plot_ITS, include=FALSE, eval=F}
# default plot is all in black, with black lines for the number of samples
rarefaction_fun <- rarecurve(t(fDAT), step=500, 10000, 
                             xlab="number of sequences", ylab="number of OTUs", label=FALSE)
```

```{r rarefaction_plot fun, fig.width=5, fig.height=4, echo=F, eval=F}
Nmax <- sapply(rarefaction_fun, function(x) max(attr(x, "Subsample")))
Smax <- sapply(rarefaction_fun, max)

plot(c(1, 30000), c(1, max(Smax)), xlab="Sequencing depth", main="Fungi",
     ylab="number of fOTUs", type="n", las=1, ylim=c(0, 1000))
abline(v=f_rare)
for (i in seq_along(rarefaction_fun)) {
  N <- attr(rarefaction_fun[[i]], "Subsample")
  lines(N, rarefaction_fun[[i]], col=fDESIGN$cols_comp[i]
  )
}

legend("topleft", legend=names(level_cols_comp), pch=15,
       col=level_cols_comp, bty="n")
```

## Figure S4 | Rarefaction analysis

```{r print Figure S4, message=FALSE, warning=F, echo=F, eval=F}
pdf(file="figures/Fig_S4_rarefication.pdf", width=12, height=6)
par(mfrow=c(1,2), mar=c(4,4,4,4), oma=c(2,2,4,2))

### BACTERIA
Nmax <- sapply(rarefaction_bac, function(x) max(attr(x, "Subsample")))
Smax <- sapply(rarefaction_bac, max)

plot(c(1,30000), c(1, max(Smax)), xlab="Sequencing depth", main="Bacteria",
     ylab="number of bOTUs", type="n", las=1, ylim=c(0, 6000))
abline(v=b_rare)
for (i in seq_along(rarefaction_bac)) {
  N <- attr(rarefaction_bac[[i]], "Subsample")
  lines(N, rarefaction_bac[[i]], col=bDESIGN$cols_comp[i]
  )
}

legend("topleft", legend=names(level_cols_comp), pch=15,
       col=level_cols_comp, bty="n")


### FUNGI
Nmax <- sapply(rarefaction_fun, function(x) max(attr(x, "Subsample")))
Smax <- sapply(rarefaction_fun, max)

plot(c(1, 30000), c(1, max(Smax)), xlab="Sequencing depth", main="Fungi",
     ylab="number of fOTUs", type="n", las=1, ylim=c(0, 1000))
abline(v=f_rare)
for (i in seq_along(rarefaction_fun)) {
  N <- attr(rarefaction_fun[[i]], "Subsample")
  lines(N, rarefaction_fun[[i]], col=fDESIGN$cols_comp[i]
  )
}

legend("topleft", legend=names(level_cols_comp), pch=15,
       col=level_cols_comp, bty="n")

noshow <- dev.off()
```


\pagebreak

# Phylum level taxonomy analysis

## Bacteria

### Data display

```{r bDAT rare, warning=F, echo=F, fig.height=3.5, fig.width=4.5}
### Data normalization
# indices of outlier
index_outliers <- which(colSums(bDAT) < b_rare)
bDAT_for_rare <- bDAT[ ,-index_outliers]
bDESIGN_rare <- bDESIGN[-index_outliers, ]
stopifnot(identical(rownames(bDESIGN_rare), colnames(bDAT_for_rare)))

# rarefication with library(vegan)
set.seed(74500)     # 74500 = zip code of Evian
bDAT_rare <- t(rrarefy(t(bDAT_for_rare), b_rare)) 
bDAT_rare <- bDAT_rare[rowSums(bDAT_rare) > 0,]  # removal of rows with 0 values
# dim(bDAT_rare)
# colSums(bDAT_rare)
```

Final number of samples after normalization   
```{r, echo=F}
table1 <- table(bDESIGN_rare$compartment, bDESIGN_rare$location_background_genotype)
pander(table1, caption="Bacteria profiles")
```


```{r phyla all bacteria, warning=F, echo=F, fig.height=7, fig.width=10}
### use PHYLOSEQ to plot the data
## for faster calculation in tax_glom
bTAX2 <- bTAX[,c("labels", "kingdom", "phylum", "class", "order", "family", "genus", "species", "OTU_ID", "cols_phyla")]

## library(phyloseq) with all data, rarefied
bPHYSEQ <- phyloseq(sample_data(bDESIGN_rare),
                    otu_table(bDAT_rare, taxa_are_rows=T),
                    tax_table(as.matrix(bTAX2[rownames(bDAT_rare),])) )

## agglomerate data by phylum and class (for Proteobacteria) using factor 'labels'
bPHYSEQ_phyla <- tax_glom(bPHYSEQ, "labels") # merge by 'labels': all Phyla plus classes of Proteobacteria ?? NArm=F
bPHYSEQ_phyla <- transform_sample_counts(bPHYSEQ_phyla, function(x) 100 * x/sum(x)) 

## melt phyloseq object
bPHYSEQ_phyla_melt <- psmelt(bPHYSEQ_phyla) 
bPHYSEQ_phyla_melt <- bPHYSEQ_phyla_melt[,c("Sample", "Abundance", "compartment", "location", "groups", "labels", "background_genotype", "location_compartment")] # subset


### Defining OTU colors by phylum and class
bPHYSEQ_phyla_melt$cols <- as.character(bPHYSEQ_phyla_melt$labels)
# attributing previously assigned colors
# bTAX_level_cols_phyla
for(i in names(bTAX_level_cols_phyla)){
  bPHYSEQ_phyla_melt[bPHYSEQ_phyla_melt$labels==paste(i), ]$cols <- bTAX_level_cols_phyla[paste(i)]
}

### Defining high and low abundant Phyla
# Phyla with MEAN abundances higher than 1% relative abundances
bPHYSEQ_phyla_abu <-  rownames(otu_table(bPHYSEQ_phyla))[apply(otu_table(bPHYSEQ_phyla), 1, mean) > 1]
bPHYSEQ_phyla_abuP <- tax_table(bPHYSEQ_phyla)[rownames(tax_table(bPHYSEQ_phyla)) %in% bPHYSEQ_phyla_abu, "labels"]
# bPHYSEQ_phyla_abuP
# Phyla with MEAN abundances lower than 1% relative abundances
bPHYSEQ_phyla_low <-  rownames(otu_table(bPHYSEQ_phyla))[apply(otu_table(bPHYSEQ_phyla), 1, mean) < 1]
bPHYSEQ_phyla_lowP <- tax_table(bPHYSEQ_phyla)[rownames(tax_table(bPHYSEQ_phyla)) %in% bPHYSEQ_phyla_low, "labels"]
# bPHYSEQ_phyla_lowP


### subsetting the color vector to abundant phylum and classes
# delete label name of low-abundant phyla (for plot) and put them at the bottom
bPHYSEQ_phyla_melt$labels_2 <- as.character(bPHYSEQ_phyla_melt$labels)
bPHYSEQ_phyla_melt[bPHYSEQ_phyla_melt$labels_2 %in% bPHYSEQ_phyla_lowP, ]$labels_2 <- "Low abundant phyla"
bPHYSEQ_phyla_melt$labels_2 <- as.factor(bPHYSEQ_phyla_melt$labels_2)
bPHYSEQ_phyla_melt$labels_2 <- factor(bPHYSEQ_phyla_melt$labels_2, levels= c("Acidobacteria", "Actinobacteria", 
                                                                             "Alphaproteobacteria", "Betaproteobacteria", "Gammaproteobacteria", "Deltaproteobacteria",
                                                                             "Bacteroidetes", "Firmicutes","Chloroflexi", "Saccharibacteria", 
                                                                             "Gemmatimonadetes", "Verrucomicrobia", "unassigned", "Low abundant phyla", ""))

### color matrix for plot
col_class <- bPHYSEQ_phyla_melt$cols
names(col_class) <- bPHYSEQ_phyla_melt$labels_2


### Define plotting factors
bPHYSEQ_phyla_melt$compartment <- factor(bPHYSEQ_phyla_melt$compartment, levels= c("soil","rhizo","root"))
bPHYSEQ_phyla_melt$background_genotype <- as.character(bPHYSEQ_phyla_melt$background_genotype)
bPHYSEQ_phyla_melt$samples <- paste(bPHYSEQ_phyla_melt$background_genotype, bPHYSEQ_phyla_melt$Sample, sep="_")
bPHYSEQ_phyla_melt$samples <- as.factor(bPHYSEQ_phyla_melt$samples)


### PLOT library(ggplot2)
bTAX_phyla_plot <- ggplot(bPHYSEQ_phyla_melt, aes_string(x="samples", y="Abundance", fill="labels_2")) + 
  geom_bar(stat="identity") + 
  xlab("") + 
  ylab("Relative abundance [%]") +
  scale_colour_manual(values=col_class) +
  scale_fill_manual(values=col_class) +
  guides(fill=guide_legend(title="Phylum")) + 
  facet_grid(. ~ compartment + location, 
             labeller=label_parsed, 
             scale="free_x", 
             space="free_x") +
  geom_text(data=bPHYSEQ_phyla_melt, aes(label=background_genotype, y=-10), angle=90, check_overlap=TRUE) + 
  theme_minimal() +
  theme(axis.text.x=element_blank(), axis.ticks.x=element_blank())

print(bTAX_phyla_plot) 
```




\pagebreak

### GLM stats

#### Analysis by location and compartment (all samples)  

```{r GLM stats bacteria phyla, echo=F, warning=F}
## dataset
# head(bPHYSEQ_phyla_melt)
# dim(bPHYSEQ_phyla_melt)

# ### 1 phylum template code
# bPHYSEQ_phyla_melt_temp <- bPHYSEQ_phyla_melt[bPHYSEQ_phyla_melt$labels=="Actinobacteria", ]
# bPHYSEQ_phyla_melt_temp <- bPHYSEQ_phyla_melt[bPHYSEQ_phyla_melt$labels=="Acidobacteria", ]
# bPHYSEQ_phyla_melt_temp <- bPHYSEQ_phyla_melt[bPHYSEQ_phyla_melt$labels=="Gammaproteobacteria", ]
# head(bPHYSEQ_phyla_melt_temp)
# dim(bPHYSEQ_phyla_melt_temp)
# ## GLM
# # fit_temp <- glm(Abundance ~ location + compartment, family=quasipoisson, data=bPHYSEQ_phyla_melt_temp)
# # fit_temp <- glm.nb(as.integer(Abundance*100) ~ location + compartment, data=bPHYSEQ_phyla_melt_temp)
# fit_temp <- glm(as.integer(Abundance*100) ~ location + compartment, family=negative.binomial(theta=1), data=bPHYSEQ_phyla_melt_temp)
# summary(fit_temp)
# fit_temp2 <- anova(fit_temp, test="F")
# fit_temp2
# ## posthoc
# summary(glht(fit_temp, linfct=mcp(location="Tukey")))
# cld(glht(fit_temp, linfct=mcp(location="Tukey")) , decreasing=TRUE)  # letters
# summary(glht(fit_temp, linfct=mcp(compartment="Tukey")))

## GLM-loop for each phylum, both factors
bPHYLA_stats <- c()
bPHYSEQ_phyla_melt$labels <- as.factor(bPHYSEQ_phyla_melt$labels)
for(i in levels(bPHYSEQ_phyla_melt$labels) ) {
  bPHYSEQ_phyla_melt_temp <- bPHYSEQ_phyla_melt[bPHYSEQ_phyla_melt$labels==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location + compartment, family=negative.binomial(theta=1), data=bPHYSEQ_phyla_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  loc_tukey_temp <- cld(glht(fit_temp, linfct=mcp(location="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  comp_tukey_temp <- cld(glht(fit_temp, linfct=mcp(compartment="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  bPHYLA_stats <- rbind(bPHYLA_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2:3], 4), scientific=F, nsmall=2, justify="left"),
                                      format( round( fit_temp2$`Pr(>F)`[2:3] , 4), scientific=F, nsmall=4),
                                      paste(loc_tukey_temp$mcletters$Letters),
                                      paste(comp_tukey_temp$mcletters$Letters) )  )
}
# arrange data frame
colnames(bPHYLA_stats) <- c("phylum", "F_location", "F_compartment", "P_location", "P_compartment", "Ch", "Re", "Au", "soil", "rhizo", "root")
rownames(bPHYLA_stats) <- bPHYLA_stats[,1]
bPHYLA_stats <- data.frame(bPHYLA_stats)
bPHYLA_stats[, 2:5] <- sapply(bPHYLA_stats[, 2:5], as.character)
# multiple hypotheis testing
bPHYLA_stats$Padj_location <- format(round(p.adjust(bPHYLA_stats$P_location, method="BH", n=nrow(bPHYLA_stats) ), 4), scientific=F)  
bPHYLA_stats$Padj_compartment <- format(round(p.adjust(bPHYLA_stats$P_compartment, method="BH", n=nrow(bPHYLA_stats) ), 4), scientific=F) 
# head(bPHYLA_stats)

## display stats in legend names
# subset to abundant phyla
bPHYLA_stats_abu <- bPHYLA_stats[ get_taxa_unique(bPHYSEQ_phyla_abuP) , ]
bPHYLA_stats_abu$loc_legend <- ifelse( bPHYLA_stats_abu$Padj_location < 0.05 , "L*","") 
bPHYLA_stats_abu$comp_legend <- ifelse( bPHYLA_stats_abu$Padj_compartment < 0.05 , "C*","")
# head(bPHYLA_stats_abu)

## to be used in the Phyla plot 
bPHYLA_stats_abu$stat_legend <- paste(rownames(bPHYLA_stats_abu), bPHYLA_stats_abu[,"loc_legend"], bPHYLA_stats_abu[,"comp_legend"], sep=" ")
# bPHYLA_stats_abu$stat_legend
```

##### Average abundances by location

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per location. Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. The columns with the posthoc Tukey test results is indicated with "_T". Different letters are used for groups that differ significantly in their abundances between locations (alpha=0.01).

```{r average abundances phyla bacteria by loc, echo=F, warning=F}
### Percentages by location
bPHYSEQ_phyla_loc <- t(merge_samples(bPHYSEQ_phyla, "location")) # merge samples in groups if mean of all samples in one group
bPHYSEQ_phyla_loc <- transform_sample_counts(bPHYSEQ_phyla_loc, function(x) 100 * x/sum(x)) # express in %RA
bPHYSEQ_phyla_loc <- prune_taxa(taxa_names(bPHYSEQ_phyla_abuP), bPHYSEQ_phyla_loc) # prune to only abundant phyla 
taxa_names(bPHYSEQ_phyla_loc) <- tax_table(bPHYSEQ_phyla_loc)[,"labels"]

bPHYLA_loc <- as(otu_table(bPHYSEQ_phyla_loc), "matrix")
bPHYLA_stats_temp <- bPHYLA_stats[rownames(bPHYLA_loc),]
bPHYLA_loc_stats <- data.frame(F=bPHYLA_stats_temp[,"F_location"], Padj=bPHYLA_stats_temp[,"Padj_location"],
                               Ch_=bPHYLA_loc[,"Changins"], Ch_T=as.character(bPHYLA_stats_temp[,"Ch"]),
                               Re_=bPHYLA_loc[,"Reckenholz"], Re_T=as.character(bPHYLA_stats_temp[,"Re"]),
                               Au_=bPHYLA_loc[,"Aurora"], Au_T=as.character(bPHYLA_stats_temp[,"Au"]) )
pander(bPHYLA_loc_stats)
```

\pagebreak

##### Average abundances by compartment 

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per compartment. Compartments are abbreviated with So=Soil, Rh=Rhizosphere and Ro=Root. The columns with the posthoc Tukey test results is indicated with "_T". Different letters are used for groups that differ significantly in their abundances between compartments (alpha=0.01).

```{r average abundances phyla bacteria by comp, echo=F, warning=F}
### Percentages by compartment
bPHYSEQ_phyla_comp <- t(merge_samples(bPHYSEQ_phyla, "compartment")) # merge samples in groups if mean of all samples in one group
bPHYSEQ_phyla_comp <- transform_sample_counts(bPHYSEQ_phyla_comp, function(x) 100 * x/sum(x)) # express in %RA
bPHYSEQ_phyla_comp <- prune_taxa(taxa_names(bPHYSEQ_phyla_abuP), bPHYSEQ_phyla_comp) # prune to only abundant phyla 
taxa_names(bPHYSEQ_phyla_comp) <- tax_table(bPHYSEQ_phyla_comp)[,"labels"]

bPHYLA_comp <- as(otu_table(bPHYSEQ_phyla_comp), "matrix")
bPHYLA_stats_temp <- bPHYLA_stats[rownames(bPHYLA_comp),]
bPHYLA_comp_stats <- data.frame(F=bPHYLA_stats_temp[,"F_compartment"], Padj=bPHYLA_stats_temp[,"Padj_compartment"], 
                                Soil=bPHYLA_comp[,"soil"], So_T=as.character(bPHYLA_stats_temp[,"soil"]),
                                Rhizo=bPHYLA_comp[,"rhizo"], Rh_T=as.character(bPHYLA_stats_temp[,"rhizo"]),
                                Root=bPHYLA_comp[,"root"], Ro_=as.character(bPHYLA_stats_temp[,"root"]) )
pander(bPHYLA_comp_stats)
```

\pagebreak

#### Analysis by sample groups (reported in Table S3)

```{r GLM stats 2 bacteria phyla, echo=F, warning=F}
## GLM-loop for each phylum, all factors
# library(emmeans)
bPHYLA_stats <- c()
for(i in levels(bPHYSEQ_phyla_melt$labels) ) {
  bPHYSEQ_phyla_melt_temp <- bPHYSEQ_phyla_melt[bPHYSEQ_phyla_melt$labels==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_compartment, family=negative.binomial(theta=1), data=bPHYSEQ_phyla_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_compartment")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdefghi", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bPHYLA_stats <- rbind(bPHYLA_stats, c(paste(i), format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"), format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4),
tukey_temp_letters$.group ))
}

# arrange data frame
colnames(bPHYLA_stats) <- c("phylum", "F", "P", 
                            "Changins_rhizo", "Changins_root", "Changins_soil",
                            "Aurora_rhizo"  , "Aurora_root"  , "Aurora_soil",
                            "Reckenholz_rhizo"  , "Reckenholz_root"  , "Reckenholz_soil")
rownames(bPHYLA_stats) <- bPHYLA_stats[,1]
bPHYLA_stats <- data.frame(bPHYLA_stats)
bPHYLA_stats[, 2:3] <- sapply(bPHYLA_stats[, 2:3], as.character)
# multiple hypotheis testing
bPHYLA_stats$Padj <- format(round(p.adjust(bPHYLA_stats$P, method="BH", n=nrow(bPHYLA_stats) ), 4), scientific=F)  
# head(bPHYLA_stats)
```

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * compartment). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Compartments are abbreviated with So=Soil, Rh=Rhizosphere and Ro=Root. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r average abundances phyla bacteria by sample groups, echo=F, warning=F}
### Percentages by sample groups (location and compartment)
bPHYSEQ_phyla_groups <- t(merge_samples(bPHYSEQ_phyla, "location_compartment")) # merge samples in groups if mean of all samples in one group
bPHYSEQ_phyla_groups <- transform_sample_counts(bPHYSEQ_phyla_groups, function(x) 100 * x/sum(x)) # express in %RA
#delete bPHYSEQ_phyla_groups <- prune_taxa(taxa_names(bPHYSEQ_phyla_abuP), bPHYSEQ_phyla_groups) # prune to only abundant phyla 
taxa_names(bPHYSEQ_phyla_groups) <- tax_table(bPHYSEQ_phyla_groups)[,"labels"]

bPHYLA_groups <- as(otu_table(bPHYSEQ_phyla_groups), "matrix")
bPHYLA_stats_temp <- bPHYLA_stats[rownames(bPHYLA_groups),]
bPHYLA_group_stats <- data.frame(F=bPHYLA_stats_temp[,"F"], Padj=bPHYLA_stats_temp[,"Padj"], 
                                 Ch_So=bPHYLA_groups[,"Changins_soil"], Ch_So_T=as.character(bPHYLA_stats_temp[,"Changins_soil"]),
                                 Re_So=bPHYLA_groups[,"Reckenholz_soil"], Re_So_T=as.character(bPHYLA_stats_temp[,"Reckenholz_soil"]),
                                 Au_So=bPHYLA_groups[,"Aurora_soil"], Au_So_T=as.character(bPHYLA_stats_temp[,"Aurora_soil"]),
                                 Ch_Rh=bPHYLA_groups[,"Changins_rhizo"], Ch_Rh_T=as.character(bPHYLA_stats_temp[,"Changins_rhizo"]), 
                                 Re_Rh=bPHYLA_groups[,"Reckenholz_rhizo"], Re_Rh_T=as.character(bPHYLA_stats_temp[,"Reckenholz_rhizo"]),
                                 Au_Rh=bPHYLA_groups[,"Aurora_rhizo"], Au_Rh_T=as.character(bPHYLA_stats_temp[,"Aurora_rhizo"]),
                                 Ch_Ro=bPHYLA_groups[,"Changins_root"], Ch_Ro_T=as.character(bPHYLA_stats_temp[,"Changins_root"]),
                                 Re_Ro=bPHYLA_groups[,"Reckenholz_root"], Re_Ro_T=as.character(bPHYLA_stats_temp[,"Reckenholz_root"]), 
                                 Au_Ro=bPHYLA_groups[,"Aurora_root"], Au_Ro_T=as.character(bPHYLA_stats_temp[,"Aurora_root"]) )

pander(bPHYLA_group_stats[ get_taxa_unique(bPHYSEQ_phyla_abuP), ])

dir.create("tables")
write.csv(bPHYLA_group_stats[ c( get_taxa_unique(bPHYSEQ_phyla_abuP), get_taxa_unique(bPHYSEQ_phyla_lowP) ), ], 
          file="tables/Table_S3_Phylum_stats_bacteria.csv")
```


\pagebreak

#### BX-effect in SOIL samples (reported in Table S6)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r soil BX effect glm phyla bacteria, echo=F, warning=F}
# subsetting to soil samples, remove Changins soil samples (not sampled by genotype)
bPHYSEQ_phyla_soil <- prune_samples(sample_data(bPHYSEQ_phyla)$compartment=="soil", bPHYSEQ_phyla)
bPHYSEQ_phyla_soil <- prune_samples(sample_data(bPHYSEQ_phyla_soil)$genotype=="WT" | 
                                      sample_data(bPHYSEQ_phyla_soil)$genotype=="bx1", bPHYSEQ_phyla_soil)
# sample_data(bPHYSEQ_phyla_soil)$genotype
# sample_data(bPHYSEQ_phyla_soil)$location

## melt phyloseq object
bPHYSEQ_phyla_soil_melt <- psmelt(bPHYSEQ_phyla_soil) 
bPHYSEQ_phyla_soil_melt <- bPHYSEQ_phyla_soil_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "labels")] # subset

# ### 1 phylum template code
# bPHYSEQ_phyla_soil_melt_temp <- bPHYSEQ_phyla_soil_melt[bPHYSEQ_phyla_soil_melt$labels=="Actinobacteria", ]
# bPHYSEQ_phyla_soil_melt_temp <- bPHYSEQ_phyla_soil_melt[bPHYSEQ_phyla_soil_melt$labels=="Acidobacteria", ]
# head(bPHYSEQ_phyla_soil_melt_temp)
# dim(bPHYSEQ_phyla_soil_melt_temp)
# ## GLM
# fit_temp <- glm.nb(as.integer(Abundance*100) ~ location_genotype, data=bPHYSEQ_phyla_soil_melt_temp)
# summary(fit_temp)
# fit_temp2 <- anova(fit_temp, test="F")
# fit_temp2
# tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
# tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
# tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
# tukey_temp_letters

## GLM-loop for each phylum
# library(emmeans)
bPHYLA_soil_stats <- c()
bPHYSEQ_phyla_soil_melt$labels <- as.factor(bPHYSEQ_phyla_soil_melt$labels)
for(i in levels(bPHYSEQ_phyla_soil_melt$labels) ) {
  bPHYSEQ_phyla_soil_melt_temp <- bPHYSEQ_phyla_soil_melt[bPHYSEQ_phyla_soil_melt$labels==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=bPHYSEQ_phyla_soil_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bPHYLA_soil_stats <- rbind(bPHYLA_soil_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                      format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                      tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(bPHYLA_soil_stats) <- c("phylum", "F", "P",
                                 "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(bPHYLA_soil_stats) <- bPHYLA_soil_stats[,1]
bPHYLA_soil_stats <- data.frame(bPHYLA_soil_stats)
bPHYLA_soil_stats[, 2:3] <- sapply(bPHYLA_soil_stats[, 2:3], as.character)
# multiple hypotheis testing
bPHYLA_soil_stats$Padj <- format(round(p.adjust(bPHYLA_soil_stats$P, method="BH", n=nrow(bPHYLA_soil_stats) ), 4), scientific=F)  
# head(bPHYLA_soil_stats)

### abundances by genotype at each location
bPHYLA_soil_RAs_geno <- t(merge_samples(bPHYSEQ_phyla_soil, "location_genotype")) # merge samples in groups if mean of all samples in one group
bPHYLA_soil_RAs_geno <- transform_sample_counts(bPHYLA_soil_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(bPHYLA_soil_RAs_geno) <- tax_table(bPHYLA_soil_RAs_geno)[,"labels"]
bPHYLA_soil_RAs_geno <- as(otu_table(bPHYLA_soil_RAs_geno), "matrix")
bPHYLA_soil_RAs_geno <- format( bPHYLA_soil_RAs_geno[rownames(bPHYLA_soil_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
bPHYLA_soil_table <- data.frame(F=bPHYLA_soil_stats[,"F"], Padj=bPHYLA_soil_stats[,"Padj"], 
                                Au_WT=bPHYLA_soil_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(bPHYLA_soil_stats[,"Aurora_WT"]),
                                Au_bx=bPHYLA_soil_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(bPHYLA_soil_stats[,"Aurora_bx1"]),
                                Re_WT=bPHYLA_soil_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(bPHYLA_soil_stats[,"Reckenholz_WT"]),
                                Re_bx=bPHYLA_soil_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(bPHYLA_soil_stats[,"Reckenholz_bx1"]))

pander(bPHYLA_soil_table[ get_taxa_unique(bPHYSEQ_phyla_abuP), ])

write.csv(bPHYLA_soil_table[ c( get_taxa_unique(bPHYSEQ_phyla_abuP), get_taxa_unique(bPHYSEQ_phyla_lowP) ), ],
          file="tables/Table_S6_Phylum_BX_stats_soil_bacteria.csv")
```


\pagebreak

#### BX-effect RHIZOSPHERE samples (reported in Table S6)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r rhizo BX effect glm phyla bacteria, echo=F, warning=F}
# subsetting to rhizo samples
bPHYSEQ_phyla_rhizo <- prune_samples(sample_data(bPHYSEQ_phyla)$compartment=="rhizo", bPHYSEQ_phyla)
bPHYSEQ_phyla_rhizo <- prune_samples(sample_data(bPHYSEQ_phyla_rhizo)$genotype=="WT" | 
                                      sample_data(bPHYSEQ_phyla_rhizo)$genotype=="bx1", bPHYSEQ_phyla_rhizo)
# sample_data(bPHYSEQ_phyla_rhizo)$genotype
# sample_data(bPHYSEQ_phyla_rhizo)$location

## melt phyloseq object
bPHYSEQ_phyla_rhizo_melt <- psmelt(bPHYSEQ_phyla_rhizo) 
bPHYSEQ_phyla_rhizo_melt <- bPHYSEQ_phyla_rhizo_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "labels")] # subset

## GLM-loop for each phylum
# library(emmeans)
bPHYLA_rhizo_stats <- c()
bPHYSEQ_phyla_rhizo_melt$labels <- as.factor(bPHYSEQ_phyla_rhizo_melt$labels)
for(i in levels(bPHYSEQ_phyla_rhizo_melt$labels) ) {
  bPHYSEQ_phyla_rhizo_melt_temp <- bPHYSEQ_phyla_rhizo_melt[bPHYSEQ_phyla_rhizo_melt$labels==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=bPHYSEQ_phyla_rhizo_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdef", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bPHYLA_rhizo_stats <- rbind(bPHYLA_rhizo_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                      format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                      tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(bPHYLA_rhizo_stats) <- c("phylum", "F", "P", "Changins_bx1", "Changins_WT",
                                 "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(bPHYLA_rhizo_stats) <- bPHYLA_rhizo_stats[,1]
bPHYLA_rhizo_stats <- data.frame(bPHYLA_rhizo_stats)
bPHYLA_rhizo_stats[, 2:3] <- sapply(bPHYLA_rhizo_stats[, 2:3], as.character)
# multiple hypotheis testing
bPHYLA_rhizo_stats$Padj <- format(round(p.adjust(bPHYLA_rhizo_stats$P, method="BH", n=nrow(bPHYLA_rhizo_stats) ), 4), scientific=F)  
# head(bPHYLA_rhizo_stats)

### abundances by genotype at each location
bPHYLA_rhizo_RAs_geno <- t(merge_samples(bPHYSEQ_phyla_rhizo, "location_genotype")) # merge samples in groups if mean of all samples in one group
bPHYLA_rhizo_RAs_geno <- transform_sample_counts(bPHYLA_rhizo_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(bPHYLA_rhizo_RAs_geno) <- tax_table(bPHYLA_rhizo_RAs_geno)[,"labels"]
bPHYLA_rhizo_RAs_geno <- as(otu_table(bPHYLA_rhizo_RAs_geno), "matrix")
bPHYLA_rhizo_RAs_geno <- format( bPHYLA_rhizo_RAs_geno[rownames(bPHYLA_rhizo_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
bPHYLA_rhizo_table <- data.frame(F=bPHYLA_rhizo_stats[,"F"], Padj=bPHYLA_rhizo_stats[,"Padj"], 
                                Ch_WT=bPHYLA_rhizo_RAs_geno[,"Changins_WT"], Ch_WT_T=as.character(bPHYLA_rhizo_stats[,"Changins_WT"]),
                                Ch_bx=bPHYLA_rhizo_RAs_geno[,"Changins_bx1"], Ch_bx_T=as.character(bPHYLA_rhizo_stats[,"Changins_bx1"]),
                                Au_WT=bPHYLA_rhizo_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(bPHYLA_rhizo_stats[,"Aurora_WT"]),
                                Au_bx=bPHYLA_rhizo_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(bPHYLA_rhizo_stats[,"Aurora_bx1"]),
                                Re_WT=bPHYLA_rhizo_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(bPHYLA_rhizo_stats[,"Reckenholz_WT"]),
                                Re_bx=bPHYLA_rhizo_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(bPHYLA_rhizo_stats[,"Reckenholz_bx1"]))

pander(bPHYLA_rhizo_table[ get_taxa_unique(bPHYSEQ_phyla_abuP), ])

write.csv(bPHYLA_rhizo_table[ c( get_taxa_unique(bPHYSEQ_phyla_abuP), get_taxa_unique(bPHYSEQ_phyla_lowP) ), ],
          file="tables/Table_S6_Phylum_BX_stats_rhizo_bacteria.csv")

# temp <- bPHYSEQ_phyla_rhizo_melt[bPHYSEQ_phyla_rhizo_melt$labels=="Verrucomicrobia", ]
# p <- bargraph.CI(temp$location, group=temp$genotype, temp$Abundance, las=2, legend=T)
```


\pagebreak

#### BX-effect ROOT samples (reported in Table S6)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r root BX effect glm phyla bacteria, echo=F, warning=F}
# subsetting to root samples
bPHYSEQ_phyla_root <- prune_samples(sample_data(bPHYSEQ_phyla)$compartment=="root", bPHYSEQ_phyla)
bPHYSEQ_phyla_root <- prune_samples(sample_data(bPHYSEQ_phyla_root)$genotype=="WT" | 
                                       sample_data(bPHYSEQ_phyla_root)$genotype=="bx1", bPHYSEQ_phyla_root)
# sample_data(bPHYSEQ_phyla_root)$genotype
# sample_data(bPHYSEQ_phyla_root)$location

## melt phyloseq object
bPHYSEQ_phyla_root_melt <- psmelt(bPHYSEQ_phyla_root) 
bPHYSEQ_phyla_root_melt <- bPHYSEQ_phyla_root_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "labels")] # subset

## GLM-loop for each phylum
# library(emmeans)
bPHYLA_root_stats <- c()
bPHYSEQ_phyla_root_melt$labels <- as.factor(bPHYSEQ_phyla_root_melt$labels)
for(i in levels(bPHYSEQ_phyla_root_melt$labels) ) {
  bPHYSEQ_phyla_root_melt_temp <- bPHYSEQ_phyla_root_melt[bPHYSEQ_phyla_root_melt$labels==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=bPHYSEQ_phyla_root_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdef", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bPHYLA_root_stats <- rbind(bPHYLA_root_stats, c(paste(i), 
                                                    format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                                    format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                                    tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(bPHYLA_root_stats) <- c("phylum", "F", "P", "Changins_bx1", "Changins_WT",
                                  "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(bPHYLA_root_stats) <- bPHYLA_root_stats[,1]
bPHYLA_root_stats <- data.frame(bPHYLA_root_stats)
bPHYLA_root_stats[, 2:3] <- sapply(bPHYLA_root_stats[, 2:3], as.character)
# multiple hypotheis testing
bPHYLA_root_stats$Padj <- format(round(p.adjust(bPHYLA_root_stats$P, method="BH", n=nrow(bPHYLA_root_stats) ), 4), scientific=F)  
# head(bPHYLA_root_stats)

### abundances by genotype at each location
bPHYLA_root_RAs_geno <- t(merge_samples(bPHYSEQ_phyla_root, "location_genotype")) # merge samples in groups if mean of all samples in one group
bPHYLA_root_RAs_geno <- transform_sample_counts(bPHYLA_root_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(bPHYLA_root_RAs_geno) <- tax_table(bPHYLA_root_RAs_geno)[,"labels"]
bPHYLA_root_RAs_geno <- as(otu_table(bPHYLA_root_RAs_geno), "matrix")
bPHYLA_root_RAs_geno <- format( bPHYLA_root_RAs_geno[rownames(bPHYLA_root_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
bPHYLA_root_table <- data.frame(F=bPHYLA_root_stats[,"F"], Padj=bPHYLA_root_stats[,"Padj"], 
                                 Ch_WT=bPHYLA_root_RAs_geno[,"Changins_WT"], Ch_WT_T=as.character(bPHYLA_root_stats[,"Changins_WT"]),
                                 Ch_bx=bPHYLA_root_RAs_geno[,"Changins_bx1"], Ch_bx_T=as.character(bPHYLA_root_stats[,"Changins_bx1"]),
                                 Au_WT=bPHYLA_root_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(bPHYLA_root_stats[,"Aurora_WT"]),
                                 Au_bx=bPHYLA_root_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(bPHYLA_root_stats[,"Aurora_bx1"]),
                                 Re_WT=bPHYLA_root_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(bPHYLA_root_stats[,"Reckenholz_WT"]),
                                 Re_bx=bPHYLA_root_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(bPHYLA_root_stats[,"Reckenholz_bx1"]))

pander(bPHYLA_root_table[ get_taxa_unique(bPHYSEQ_phyla_abuP), ])

write.csv(bPHYLA_root_table[ c( get_taxa_unique(bPHYSEQ_phyla_abuP), get_taxa_unique(bPHYSEQ_phyla_lowP) ), ],
          file="tables/Table_S6_Phylum_BX_stats_root_bacteria.csv")
```



## Fungi

### Data display

```{r fDAT rare, warning=F, echo=F, fig.height=3.5, fig.width=4.5}
### Data normalization
# indices of outlier
index_outliers <- which(colSums(fDAT) < f_rare)
if (length(index_outliers) > 0) { 
  fDAT_for_rare <- fDAT[-index_outliers, ]
  fDESIGN_rare <- fDESIGN[-index_outliers, ]
} else { 
  fDAT_for_rare <- fDAT 
  fDESIGN_rare <- fDESIGN}

stopifnot(identical(rownames(fDESIGN_rare), colnames(fDAT_for_rare)))

# rarefication with library(vegan)
set.seed(74500)     # 74500 = zip code of Evian
fDAT_rare <- t(rrarefy(t(fDAT_for_rare), f_rare)) 
fDAT_rare <- fDAT_rare[rowSums(fDAT_rare) > 0,]  # removal of rows with 0 values
# dim(bDAT_rare)
# colSums(bDAT_rare)
```

Final number of samples after normalization   
```{r, echo=F}
table1 <- table(fDESIGN_rare$compartment, fDESIGN_rare$location_background_genotype)
pander(table1, caption="Fungi profiles")
```


\pagebreak


```{r phyla all fungi, warning=F, echo=F, fig.height=7, fig.width=10}
### use PHYLOSEQ to plot the data

## library(phyloseq) with all data, rarefied
fPHYSEQ <- phyloseq(sample_data(fDESIGN_rare),
                    otu_table(fDAT_rare, taxa_are_rows=T),
                    tax_table(as.matrix(fTAX[rownames(fDAT_rare),])) )

## agglomerate data by phylum 
fPHYSEQ_phyla <- tax_glom(fPHYSEQ, "phylum") # merge by 'phylum' 
fPHYSEQ_phyla <- transform_sample_counts(fPHYSEQ_phyla, function(x) 100 * x/sum(x)) 

## melt phyloseq object
fPHYSEQ_phyla_melt <- psmelt(fPHYSEQ_phyla) 
fPHYSEQ_phyla_melt <- fPHYSEQ_phyla_melt[,c("Sample", "Abundance", "compartment", "location", "location_compartment", "groups", "phylum", "background_genotype")] # subset


### Defining OTU colors by phylum
fPHYSEQ_phyla_melt$labels <- as.factor(fPHYSEQ_phyla_melt$phylum)
fPHYSEQ_phyla_melt$cols <- as.character(fPHYSEQ_phyla_melt$phylum)
# attributing previously assigned colors
# fTAX_level_cols_phyla
for(i in names(fTAX_level_cols_phyla)[names(fTAX_level_cols_phyla) %in% levels(fPHYSEQ_phyla_melt$labels)]){
  fPHYSEQ_phyla_melt[fPHYSEQ_phyla_melt$labels==paste(i), ]$cols <- fTAX_level_cols_phyla[paste(i)]
}

### Defining high and low abundant Phyla
# Phyla with MEAN abundances higher than 1% relative abundances
fPHYSEQ_phyla_abu <-  rownames(otu_table(fPHYSEQ_phyla))[apply(otu_table(fPHYSEQ_phyla), 1, mean) > 1]
fPHYSEQ_phyla_abuP <- tax_table(fPHYSEQ_phyla)[rownames(tax_table(fPHYSEQ_phyla)) %in% fPHYSEQ_phyla_abu, "phylum"]
# fPHYSEQ_phyla_abuP
# Phyla with MEAN abundances lower than 1% relative abundances
fPHYSEQ_phyla_low <-  rownames(otu_table(fPHYSEQ_phyla))[apply(otu_table(fPHYSEQ_phyla), 1, mean) < 1]
fPHYSEQ_phyla_lowP <- tax_table(fPHYSEQ_phyla)[rownames(tax_table(fPHYSEQ_phyla)) %in% fPHYSEQ_phyla_low, "phylum"]
# fPHYSEQ_phyla_lowP


### subsetting the color vector to abundant phylum and classes
# delete label name of low-abundant phyla (for plot) and put them at the bottom
fPHYSEQ_phyla_melt$labels_2 <- as.character(fPHYSEQ_phyla_melt$labels)
fPHYSEQ_phyla_melt[fPHYSEQ_phyla_melt$labels_2 %in% fPHYSEQ_phyla_lowP, ]$labels_2 <- "Low abundant phyla"
fPHYSEQ_phyla_melt$labels_2 <- as.factor(fPHYSEQ_phyla_melt$labels_2)
fPHYSEQ_phyla_melt$labels_2 <- factor(fPHYSEQ_phyla_melt$labels_2, levels= c("Ascomycota", "Basidiomycota", "Chytridiomycota",
                                                                             "Glomeromycota", "Mortierellomycota", "unassigned", "Low abundant phyla"))

### color matrix for plot
col_class <- fPHYSEQ_phyla_melt$cols
names(col_class) <- fPHYSEQ_phyla_melt$labels_2


### Define plotting factors
fPHYSEQ_phyla_melt$compartment <- factor(fPHYSEQ_phyla_melt$compartment, levels= c("soil","rhizo","root"))
fPHYSEQ_phyla_melt$background_genotype <- as.character(fPHYSEQ_phyla_melt$background_genotype)
fPHYSEQ_phyla_melt$samples <- paste(fPHYSEQ_phyla_melt$background_genotype, fPHYSEQ_phyla_melt$Sample, sep="_")
fPHYSEQ_phyla_melt$samples <- as.factor(fPHYSEQ_phyla_melt$samples)


### PLOT library(ggplot2)
fTAX_phyla_plot <- ggplot(fPHYSEQ_phyla_melt, aes_string(x="samples", y="Abundance", fill="labels_2")) + 
  geom_bar(stat="identity") + 
  xlab("") + 
  ylab("Relative abundance [%]") +
  scale_colour_manual(values=col_class) +
  scale_fill_manual(values=col_class) +
  guides(fill=guide_legend(title="Phylum")) + 
  facet_grid(. ~ compartment + location, 
             labeller=label_parsed, 
             scale="free_x", 
             space="free_x") +
  geom_text(data=fPHYSEQ_phyla_melt, aes(label=background_genotype, y=-10), angle=90, check_overlap=TRUE) + 
  theme_minimal() +
  theme(axis.text.x=element_blank(), axis.ticks.x=element_blank())

print(fTAX_phyla_plot) 
```


## Figure S5 | Phylum level taxonomy

```{r Figure S5 phyla barplot, echo=F, warning=F}
#library(cowplot)
bfTAX_phyla_plot <- plot_grid(
           bTAX_phyla_plot,
           fTAX_phyla_plot,
           align='vh',
           labels=c("A", "B"),
           hjust=-1,
           nrow=2,
           ncol=1
           )

# legend_b <- get_legend(rar_plot + theme(legend.position="right"))
# p <- plot_grid( prow, legend_b, ncol=2, rel_widths=c(1, .5))

pdf("figures/Fig_S5_phyla_both_barplot.pdf", width=10, height=10)
bfTAX_phyla_plot 
noshow <- dev.off()
```


\pagebreak

### GLM stats

#### Analysis by location and compartment (all samples)  

```{r GLM stats fungi phyla, echo=F, warning=F}
## dataset
# head(fPHYSEQ_phyla_melt)
# dim(fPHYSEQ_phyla_melt)

# ### 1 phylum template code, see bacteria

## GLM-loop for each phylum, both factors
fPHYLA_stats <- c()
fPHYSEQ_phyla_melt$phylum <- as.factor(fPHYSEQ_phyla_melt$phylum)
for(i in levels(fPHYSEQ_phyla_melt$phylum) ) {
  fPHYSEQ_phyla_melt_temp <- fPHYSEQ_phyla_melt[fPHYSEQ_phyla_melt$phylum==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location + compartment, family=negative.binomial(theta=1), data=fPHYSEQ_phyla_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  loc_tukey_temp <- cld(glht(fit_temp, linfct=mcp(location="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  comp_tukey_temp <- cld(glht(fit_temp, linfct=mcp(compartment="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  fPHYLA_stats <- rbind(fPHYLA_stats, c(paste(i), 
                                        format( round( fit_temp2$F[2:3], 4), scientific=F, nsmall=2, justify="left"),
                                        format( round( fit_temp2$`Pr(>F)`[2:3] , 4), scientific=F, nsmall=4),
                                        paste(loc_tukey_temp$mcletters$Letters),
                                        paste(comp_tukey_temp$mcletters$Letters) )  )
}
# arrange data frame
colnames(fPHYLA_stats) <- c("phylum", "F_location", "F_compartment", "P_location", "P_compartment", "Ch", "Re", "Au", "soil", "rhizo", "root")
rownames(fPHYLA_stats) <- fPHYLA_stats[,1]
fPHYLA_stats <- data.frame(fPHYLA_stats)
fPHYLA_stats[, 2:5] <- sapply(fPHYLA_stats[, 2:5], as.character)
# multiple hypotheis testing
fPHYLA_stats$Padj_location <- format(round(p.adjust(fPHYLA_stats$P_location, method="BH", n=nrow(fPHYLA_stats) ), 4), scientific=F)  
fPHYLA_stats$Padj_compartment <- format(round(p.adjust(fPHYLA_stats$P_compartment, method="BH", n=nrow(fPHYLA_stats) ), 4), scientific=F) 
# head(fPHYLA_stats)

## display stats in legend names
# subset to abundant phyla
fPHYLA_stats_abu <- fPHYLA_stats[ get_taxa_unique(fPHYSEQ_phyla_abuP) , ]
fPHYLA_stats_abu$loc_legend <- ifelse( fPHYLA_stats_abu$Padj_location < 0.05 , "L*","") 
fPHYLA_stats_abu$comp_legend <- ifelse( fPHYLA_stats_abu$Padj_compartment < 0.05 , "C*","")
# head(fPHYLA_stats_abu)

## to be used in the Phyla plot 
fPHYLA_stats_abu$stat_legend <- paste(rownames(fPHYLA_stats_abu), fPHYLA_stats_abu[,"loc_legend"], fPHYLA_stats_abu[,"comp_legend"], sep=" ")
# fPHYLA_stats_abu$stat_legend
```


##### Average abundances per location

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per location. Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. The columns with the posthoc Tukey test results is indicated with "_T". Different letters are used for groups that differ significantly in their abundances between locations (alpha=0.01).

```{r average abundances phyla fungi by loc, echo=F, warning=F}
### Percentages by location
fPHYSEQ_phyla_loc <- t(merge_samples(fPHYSEQ_phyla, "location")) # merge samples in groups if mean of all samples in one group
fPHYSEQ_phyla_loc <- transform_sample_counts(fPHYSEQ_phyla_loc, function(x) 100 * x/sum(x)) # express in %RA
fPHYSEQ_phyla_loc <- prune_taxa(taxa_names(fPHYSEQ_phyla_abuP), fPHYSEQ_phyla_loc) # prune to only abundant phyla 
taxa_names(fPHYSEQ_phyla_loc) <- tax_table(fPHYSEQ_phyla_loc)[,"phylum"]

fPHYLA_loc <- as(otu_table(fPHYSEQ_phyla_loc), "matrix")
fPHYLA_stats_temp <- fPHYLA_stats[rownames(fPHYLA_loc),]
fPHYLA_loc_stats <- data.frame(F=fPHYLA_stats_temp[,"F_location"], Padj=fPHYLA_stats_temp[,"Padj_location"],
                               Ch=fPHYLA_loc[,"Changins"], Ch_T=as.character(fPHYLA_stats_temp[,"Ch"]),
                               Re=fPHYLA_loc[,"Reckenholz"], Re_T=as.character(fPHYLA_stats_temp[,"Re"]),
                               Au=fPHYLA_loc[,"Aurora"], Au_T=as.character(fPHYLA_stats_temp[,"Au"]) )
pander(fPHYLA_loc_stats)
```

##### Average abundances by compartment 

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per compartment. Compartments are abbreviated with So=Soil, Rh=Rhizosphere and Ro=Root. The columns with the posthoc Tukey test results is indicated with "_T". Different letters are used for groups that differ significantly in their abundances between compartments (alpha=0.01).

```{r average abundances phyla fungi by comp, echo=F, warning=F}
### Percentages by compartment
fPHYSEQ_phyla_comp <- t(merge_samples(fPHYSEQ_phyla, "compartment")) # merge samples in groups if mean of all samples in one group
fPHYSEQ_phyla_comp <- transform_sample_counts(fPHYSEQ_phyla_comp, function(x) 100 * x/sum(x)) # express in %RA
fPHYSEQ_phyla_comp <- prune_taxa(taxa_names(fPHYSEQ_phyla_abuP), fPHYSEQ_phyla_comp) # prune to only abundant phyla 
taxa_names(fPHYSEQ_phyla_comp) <- tax_table(fPHYSEQ_phyla_comp)[,"phylum"]

fPHYLA_comp <- as(otu_table(fPHYSEQ_phyla_comp), "matrix")
fPHYLA_stats_temp <- fPHYLA_stats[rownames(fPHYLA_comp),]
fPHYLA_comp_stats <- data.frame(F=fPHYLA_stats_temp[,"F_compartment"], Padj=fPHYLA_stats_temp[,"Padj_compartment"], 
                                So=fPHYLA_comp[,"soil"], So_T=as.character(fPHYLA_stats_temp[,"soil"]),
                                Rh=fPHYLA_comp[,"rhizo"], Rh_T=as.character(fPHYLA_stats_temp[,"rhizo"]),
                                Ro=fPHYLA_comp[,"root"], Ro_T=as.character(fPHYLA_stats_temp[,"root"]) )
pander(fPHYLA_comp_stats)
```


\pagebreak

#### Analysis by sample groups (reported in Table S3)

```{r GLM stats 2 fungi phyla, echo=F, warning=F}
## GLM-loop for each phylum, all factors
# library(emmeans)
fPHYLA_stats <- c()
for(i in levels(fPHYSEQ_phyla_melt$phylum) ) {
  fPHYSEQ_phyla_melt_temp <- fPHYSEQ_phyla_melt[fPHYSEQ_phyla_melt$phylum==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_compartment, family=negative.binomial(theta=1), data=fPHYSEQ_phyla_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_compartment")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdefghi", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  fPHYLA_stats <- rbind(fPHYLA_stats, c(paste(i), 
                                        format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                        format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4),
                                        tukey_temp_letters$.group )  )
}

# arrange data frame
colnames(fPHYLA_stats) <- c("phylum", "F", "P", 
                            "Changins_rhizo", "Changins_root", "Changins_soil",
                            "Aurora_rhizo"  , "Aurora_root"  , "Aurora_soil",
                            "Reckenholz_rhizo"  , "Reckenholz_root"  , "Reckenholz_soil")
rownames(fPHYLA_stats) <- fPHYLA_stats[,1]
fPHYLA_stats <- data.frame(fPHYLA_stats)
fPHYLA_stats[, 2:3] <- sapply(fPHYLA_stats[, 2:3], as.character)
# multiple hypotheis testing
fPHYLA_stats$Padj <- format(round(p.adjust(fPHYLA_stats$P, method="BH", n=nrow(fPHYLA_stats) ), 4), scientific=F)  
# head(fPHYLA_stats)
```

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * compartment). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Compartments are abbreviated with So=Soil, Rh=Rhizosphere and Ro=Root. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r average abundances phyla fungi by sample groups, echo=F, warning=F}
### Percentages by sample groups (location and compartment)
fPHYSEQ_phyla_groups <- t(merge_samples(fPHYSEQ_phyla, "location_compartment")) # merge samples in groups if mean of all samples in one group
fPHYSEQ_phyla_groups <- transform_sample_counts(fPHYSEQ_phyla_groups, function(x) 100 * x/sum(x)) # express in %RA
#delete fPHYSEQ_phyla_groups <- prune_taxa(taxa_names(fPHYSEQ_phyla_abuP), fPHYSEQ_phyla_groups) # prune to only abundant phyla 
taxa_names(fPHYSEQ_phyla_groups) <- tax_table(fPHYSEQ_phyla_groups)[,"phylum"]

fPHYLA_groups <- as(otu_table(fPHYSEQ_phyla_groups), "matrix")
fPHYLA_stats_temp <- fPHYLA_stats[rownames(fPHYLA_groups),]
fPHYLA_group_stats <- data.frame(F=fPHYLA_stats_temp[,"F"], Padj=fPHYLA_stats_temp[,"Padj"], 
                                 Ch_So=fPHYLA_groups[,"Changins_soil"], Ch_So_T=as.character(fPHYLA_stats_temp[,"Changins_soil"]),
                                 Re_So=fPHYLA_groups[,"Reckenholz_soil"], Re_So_T=as.character(fPHYLA_stats_temp[,"Reckenholz_soil"]),
                                 Au_So=fPHYLA_groups[,"Aurora_soil"], Au_So_T=as.character(fPHYLA_stats_temp[,"Aurora_soil"]),
                                 Ch_Rh=fPHYLA_groups[,"Changins_rhizo"], Ch_Rh_T=as.character(fPHYLA_stats_temp[,"Changins_rhizo"]), 
                                 Re_Rh=fPHYLA_groups[,"Reckenholz_rhizo"], Re_Rh_T=as.character(fPHYLA_stats_temp[,"Reckenholz_rhizo"]),
                                 Au_Rh=fPHYLA_groups[,"Aurora_rhizo"], Au_Rh_T=as.character(fPHYLA_stats_temp[,"Aurora_rhizo"]),
                                 Ch_Ro=fPHYLA_groups[,"Changins_root"], Ch_Ro_T=as.character(fPHYLA_stats_temp[,"Changins_root"]),
                                 Re_Ro=fPHYLA_groups[,"Reckenholz_root"], Re_Ro_T=as.character(fPHYLA_stats_temp[,"Reckenholz_root"]), 
                                 Au_Ro=fPHYLA_groups[,"Aurora_root"], Au_Ro_T=as.character(fPHYLA_stats_temp[,"Aurora_root"]) )
pander(fPHYLA_group_stats[ get_taxa_unique(fPHYSEQ_phyla_abuP), ])

# levels(fPHYSEQ_phyla_melt$location_compartment)

### Table S3
write.csv(fPHYLA_group_stats[c( get_taxa_unique(fPHYSEQ_phyla_abuP), get_taxa_unique(fPHYSEQ_phyla_lowP) ), ],
          file="tables/Table_S3_Phylum_stats_fungi.csv")
```


\pagebreak

#### BX-effect in SOIL samples (reported in Table S6)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r soil BX effect glm phyla fungi, echo=F, warning=F}
# subsetting to soil samples, remove Changins soil samples (not sampled by genotype)
fPHYSEQ_phyla_soil <- prune_samples(sample_data(fPHYSEQ_phyla)$compartment=="soil", fPHYSEQ_phyla)
fPHYSEQ_phyla_soil <- prune_samples(sample_data(fPHYSEQ_phyla_soil)$genotype=="WT" | 
                                      sample_data(fPHYSEQ_phyla_soil)$genotype=="bx1", fPHYSEQ_phyla_soil)
# sample_data(fPHYSEQ_phyla_soil)$genotype
# sample_data(fPHYSEQ_phyla_soil)$location

## melt phyloseq object
fPHYSEQ_phyla_soil_melt <- psmelt(fPHYSEQ_phyla_soil) 
fPHYSEQ_phyla_soil_melt <- fPHYSEQ_phyla_soil_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "phylum")] # subset

# ### 1 phylum template code, see bacteria

## GLM-loop for each phylum
# library(emmeans)
fPHYLA_soil_stats <- c()
fPHYSEQ_phyla_soil_melt$phylum <- as.factor(fPHYSEQ_phyla_soil_melt$phylum)
for(i in levels(fPHYSEQ_phyla_soil_melt$phylum) ) {
  fPHYSEQ_phyla_soil_melt_temp <- fPHYSEQ_phyla_soil_melt[fPHYSEQ_phyla_soil_melt$phylum==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=fPHYSEQ_phyla_soil_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  fPHYLA_soil_stats <- rbind(fPHYLA_soil_stats, c(paste(i), 
                                                  format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                                  format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                                  tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(fPHYLA_soil_stats) <- c("phylum", "F", "P",
                                 "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(fPHYLA_soil_stats) <- fPHYLA_soil_stats[,1]
fPHYLA_soil_stats <- data.frame(fPHYLA_soil_stats)
fPHYLA_soil_stats[, 2:3] <- sapply(fPHYLA_soil_stats[, 2:3], as.character)
# multiple hypotheis testing
fPHYLA_soil_stats$Padj <- format(round(p.adjust(fPHYLA_soil_stats$P, method="BH", n=nrow(fPHYLA_soil_stats) ), 4), scientific=F)  
# head(fPHYLA_soil_stats)

### Percentages by genotype at each location
fPHYLA_soil_RAs_geno <- t(merge_samples(fPHYSEQ_phyla_soil, "location_genotype")) # merge samples in groups if mean of all samples in one group
fPHYLA_soil_RAs_geno <- transform_sample_counts(fPHYLA_soil_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(fPHYLA_soil_RAs_geno) <- tax_table(fPHYLA_soil_RAs_geno)[,"phylum"]
fPHYLA_soil_RAs_geno <- as(otu_table(fPHYLA_soil_RAs_geno), "matrix")
fPHYLA_soil_RAs_geno <- format( fPHYLA_soil_RAs_geno[rownames(fPHYLA_soil_stats), ], scientific=F, nsmall=4 ) 

fPHYLA_soil_table <- data.frame(F=fPHYLA_soil_stats[,"F"], Padj=fPHYLA_soil_stats[,"Padj"], 
                                AuWT=fPHYLA_soil_RAs_geno[,"Aurora_WT"], AuWT_T=as.character(fPHYLA_soil_stats[,"Aurora_WT"]),
                                Aubx=fPHYLA_soil_RAs_geno[,"Aurora_bx1"], Aubx_T=as.character(fPHYLA_soil_stats[,"Aurora_bx1"]),
                                ZhWT=fPHYLA_soil_RAs_geno[,"Reckenholz_WT"], ZhWT_T=as.character(fPHYLA_soil_stats[,"Reckenholz_WT"]),
                                Zhbx=fPHYLA_soil_RAs_geno[,"Reckenholz_bx1"], Zhbx_T=as.character(fPHYLA_soil_stats[,"Reckenholz_bx1"]))

pander(fPHYLA_soil_table[ get_taxa_unique(fPHYSEQ_phyla_abuP), ])

write.csv(fPHYLA_soil_table[ c( get_taxa_unique(fPHYSEQ_phyla_abuP), get_taxa_unique(fPHYSEQ_phyla_lowP) ), ],
          file="tables/Table_S6_Phylum_BX_stats_soil_fungi.csv")
```


\pagebreak

#### BX-effect in RHIZOSPHERE samples (reported in Table S6)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r rhizo BX effect glm phyla fungi, echo=F, warning=F}
# subsetting to rhizo samples
fPHYSEQ_phyla_rhizo <- prune_samples(sample_data(fPHYSEQ_phyla)$compartment=="rhizo", fPHYSEQ_phyla)
fPHYSEQ_phyla_rhizo <- prune_samples(sample_data(fPHYSEQ_phyla_rhizo)$genotype=="WT" | 
                                       sample_data(fPHYSEQ_phyla_rhizo)$genotype=="bx1", fPHYSEQ_phyla_rhizo)
# sample_data(fPHYSEQ_phyla_rhizo)$genotype
# sample_data(fPHYSEQ_phyla_rhizo)$location

## melt phyloseq object
fPHYSEQ_phyla_rhizo_melt <- psmelt(fPHYSEQ_phyla_rhizo) 
fPHYSEQ_phyla_rhizo_melt <- fPHYSEQ_phyla_rhizo_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "phylum")] # subset

## GLM-loop for each phylum
# library(emmeans)
fPHYLA_rhizo_stats <- c()
fPHYSEQ_phyla_rhizo_melt$phylum <- as.factor(fPHYSEQ_phyla_rhizo_melt$phylum)
for(i in levels(fPHYSEQ_phyla_rhizo_melt$phylum) ) {
  fPHYSEQ_phyla_rhizo_melt_temp <- fPHYSEQ_phyla_rhizo_melt[fPHYSEQ_phyla_rhizo_melt$phylum==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=fPHYSEQ_phyla_rhizo_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdef", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  fPHYLA_rhizo_stats <- rbind(fPHYLA_rhizo_stats, c(paste(i), 
                                                    format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                                    format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                                    tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(fPHYLA_rhizo_stats) <- c("phylum", "F", "P", "Changins_bx1", "Changins_WT",
                                  "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(fPHYLA_rhizo_stats) <- fPHYLA_rhizo_stats[,1]
fPHYLA_rhizo_stats <- data.frame(fPHYLA_rhizo_stats)
fPHYLA_rhizo_stats[, 2:3] <- sapply(fPHYLA_rhizo_stats[, 2:3], as.character)
# multiple hypotheis testing
fPHYLA_rhizo_stats$Padj <- format(round(p.adjust(fPHYLA_rhizo_stats$P, method="BH", n=nrow(fPHYLA_rhizo_stats) ), 4), scientific=F)  
# head(fPHYLA_rhizo_stats)

### abundances by genotype at each location
fPHYLA_rhizo_RAs_geno <- t(merge_samples(fPHYSEQ_phyla_rhizo, "location_genotype")) # merge samples in groups if mean of all samples in one group
fPHYLA_rhizo_RAs_geno <- transform_sample_counts(fPHYLA_rhizo_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(fPHYLA_rhizo_RAs_geno) <- tax_table(fPHYLA_rhizo_RAs_geno)[,"phylum"]
fPHYLA_rhizo_RAs_geno <- as(otu_table(fPHYLA_rhizo_RAs_geno), "matrix")
fPHYLA_rhizo_RAs_geno <- format( fPHYLA_rhizo_RAs_geno[rownames(fPHYLA_rhizo_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
fPHYLA_rhizo_table <- data.frame(F=fPHYLA_rhizo_stats[,"F"], Padj=fPHYLA_rhizo_stats[,"Padj"], 
                                 ChWT=fPHYLA_rhizo_RAs_geno[,"Changins_WT"], ChWT_T=as.character(fPHYLA_rhizo_stats[,"Changins_WT"]),
                                 Chbx=fPHYLA_rhizo_RAs_geno[,"Changins_bx1"], Chbx_T=as.character(fPHYLA_rhizo_stats[,"Changins_bx1"]),
                                 AuWT=fPHYLA_rhizo_RAs_geno[,"Aurora_WT"], AuWT_T=as.character(fPHYLA_rhizo_stats[,"Aurora_WT"]),
                                 Aubx=fPHYLA_rhizo_RAs_geno[,"Aurora_bx1"], Aubx_T=as.character(fPHYLA_rhizo_stats[,"Aurora_bx1"]),
                                 ZhWT=fPHYLA_rhizo_RAs_geno[,"Reckenholz_WT"], ZhWT_T=as.character(fPHYLA_rhizo_stats[,"Reckenholz_WT"]),
                                 Zhbx=fPHYLA_rhizo_RAs_geno[,"Reckenholz_bx1"], Zhbx_T=as.character(fPHYLA_rhizo_stats[,"Reckenholz_bx1"]))

pander(fPHYLA_rhizo_table[ get_taxa_unique(fPHYSEQ_phyla_abuP), ])

write.csv(fPHYLA_rhizo_table[ c( get_taxa_unique(fPHYSEQ_phyla_abuP), get_taxa_unique(fPHYSEQ_phyla_lowP) ), ],
          file="tables/Table_S6_Phylum_BX_stats_rhizo_fungi.csv")

# temp <- fPHYSEQ_phyla_rhizo_melt[fPHYSEQ_phyla_rhizo_melt$phylum=="Glomeromycota", ]
# p <- bargraph.CI(temp$location, group=temp$genotype, temp$Abundance, las=2, legend=T)
```


\pagebreak

#### BX-effect ROOT samples (reported in Table S6)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r root BX effect glm phyla fungi, echo=F, warning=F}
# subsetting to root samples
fPHYSEQ_phyla_root <- prune_samples(sample_data(fPHYSEQ_phyla)$compartment=="root", fPHYSEQ_phyla)
fPHYSEQ_phyla_root <- prune_samples(sample_data(fPHYSEQ_phyla_root)$genotype=="WT" | 
                                       sample_data(fPHYSEQ_phyla_root)$genotype=="bx1", fPHYSEQ_phyla_root)
# sample_data(fPHYSEQ_phyla_root)$genotype
# sample_data(fPHYSEQ_phyla_root)$location

## melt phyloseq object
fPHYSEQ_phyla_root_melt <- psmelt(fPHYSEQ_phyla_root) 
fPHYSEQ_phyla_root_melt <- fPHYSEQ_phyla_root_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "phylum")] # subset

## GLM-loop for each phylum
# library(emmeans)
fPHYLA_root_stats <- c()
fPHYSEQ_phyla_root_melt$phylum <- as.factor(fPHYSEQ_phyla_root_melt$phylum)
for(i in levels(fPHYSEQ_phyla_root_melt$phylum) ) {
  fPHYSEQ_phyla_root_melt_temp <- fPHYSEQ_phyla_root_melt[fPHYSEQ_phyla_root_melt$phylum==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=fPHYSEQ_phyla_root_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdef", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  fPHYLA_root_stats <- rbind(fPHYLA_root_stats, c(paste(i), 
                                                    format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                                    format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                                    tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(fPHYLA_root_stats) <- c("phylum", "F", "P", "Changins_bx1", "Changins_WT",
                                  "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(fPHYLA_root_stats) <- fPHYLA_root_stats[,1]
fPHYLA_root_stats <- data.frame(fPHYLA_root_stats)
fPHYLA_root_stats[, 2:3] <- sapply(fPHYLA_root_stats[, 2:3], as.character)
# multiple hypotheis testing
fPHYLA_root_stats$Padj <- format(round(p.adjust(fPHYLA_root_stats$P, method="BH", n=nrow(fPHYLA_root_stats) ), 4), scientific=F)  
# head(fPHYLA_root_stats)

### abundances by genotype at each location
fPHYLA_root_RAs_geno <- t(merge_samples(fPHYSEQ_phyla_root, "location_genotype")) # merge samples in groups if mean of all samples in one group
fPHYLA_root_RAs_geno <- transform_sample_counts(fPHYLA_root_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(fPHYLA_root_RAs_geno) <- tax_table(fPHYLA_root_RAs_geno)[,"phylum"]
fPHYLA_root_RAs_geno <- as(otu_table(fPHYLA_root_RAs_geno), "matrix")
fPHYLA_root_RAs_geno <- format( fPHYLA_root_RAs_geno[rownames(fPHYLA_root_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
fPHYLA_root_table <- data.frame(F=fPHYLA_root_stats[,"F"], Padj=fPHYLA_root_stats[,"Padj"], 
                                 ChWT=fPHYLA_root_RAs_geno[,"Changins_WT"], ChWT_T=as.character(fPHYLA_root_stats[,"Changins_WT"]),
                                 Chbx=fPHYLA_root_RAs_geno[,"Changins_bx1"], Chbx_T=as.character(fPHYLA_root_stats[,"Changins_bx1"]),
                                 AuWT=fPHYLA_root_RAs_geno[,"Aurora_WT"], AuWT_T=as.character(fPHYLA_root_stats[,"Aurora_WT"]),
                                 Aubx=fPHYLA_root_RAs_geno[,"Aurora_bx1"], Aubx_T=as.character(fPHYLA_root_stats[,"Aurora_bx1"]),
                                 ZhWT=fPHYLA_root_RAs_geno[,"Reckenholz_WT"], ZhWT_T=as.character(fPHYLA_root_stats[,"Reckenholz_WT"]),
                                 Zhbx=fPHYLA_root_RAs_geno[,"Reckenholz_bx1"], Zhbx_T=as.character(fPHYLA_root_stats[,"Reckenholz_bx1"]))

pander(fPHYLA_root_table[ get_taxa_unique(fPHYSEQ_phyla_abuP), ])

write.csv(fPHYLA_root_table[ c( get_taxa_unique(fPHYSEQ_phyla_abuP), get_taxa_unique(fPHYSEQ_phyla_lowP) ), ],
          file="tables/Table_S6_Phylum_BX_stats_root_fungi.csv")

# temp <- fPHYSEQ_phyla_root_melt[fPHYSEQ_phyla_root_melt$phylum=="Glomeromycota", ]
# p <- bargraph.CI(temp$location, group=temp$genotype, temp$Abundance, las=2, legend=T)
```














### Family level

```{r family all bacteria, warning=F, echo=F, fig.height=7, fig.width=10}
### use PHYLOSEQ to plot the data
## for faster calculation in tax_glom

## library(phyloseq) with all data, rarefied
bPHYSEQ <- phyloseq(sample_data(bDESIGN_rare),
                    otu_table(bDAT_rare, taxa_are_rows=T),
                    tax_table(as.matrix(bTAX2[rownames(bDAT_rare),])) )

## agglomerate data by family and class (for Proteobacteria) using factor 'family'
bPHYSEQ_family <- tax_glom(bPHYSEQ, "family") # merge by 'family': all family plus classes of Proteobacteria ?? NArm=F
bPHYSEQ_family <- transform_sample_counts(bPHYSEQ_family, function(x) 100 * x/sum(x)) 

bPHYSEQ_family = subset_taxa(bPHYSEQ_family, family != "unassigned" & family != "uncultured" & family != "Unknown_Family")

### Defining high and low abundant family
# family with MEAN abundances higher than 1% relative abundances
family_abu <-  rownames(otu_table(bPHYSEQ_family))[apply(otu_table(bPHYSEQ_family), 1, mean) > 1]
family_abuP <- tax_table(bPHYSEQ_family)[rownames(tax_table(bPHYSEQ_family)) %in% family_abu, "family"]
# bPHYSEQ_family_abuP
# family with MEAN abundances lower than 1% relative abundances
family_low <-  rownames(otu_table(bPHYSEQ_family))[apply(otu_table(bPHYSEQ_family), 1, mean) < 1]
family_lowP <- tax_table(bPHYSEQ_family)[rownames(tax_table(bPHYSEQ_family)) %in% family_low, "family"]


# For abundant family
bPHYSEQ_family_abuF <- prune_taxa(taxa_names(family_abuP), bPHYSEQ_family) # prune to only abundant family 

## melt phyloseq object: abundant families
bPHYSEQ_family_melt <- psmelt(bPHYSEQ_family) 
bPHYSEQ_family_melt <- bPHYSEQ_family_melt[,c("OTU","Sample", "Abundance", "compartment", "location", "groups", "family", "background_genotype", "location_compartment")] # subset

### subsetting the color vector to abundant family and classes
# delete label name of low-abundant family (for plot) and put them at the bottom
bPHYSEQ_family_melt$family <- as.factor(as.character(bPHYSEQ_family_melt$family))
bPHYSEQ_family_melt$compartment <- factor(bPHYSEQ_family_melt$compartment, levels= c("soil","rhizo","root"))
bPHYSEQ_family_melt$background_genotype <- as.character(bPHYSEQ_family_melt$background_genotype)
bPHYSEQ_family_melt$samples <- paste(bPHYSEQ_family_melt$background_genotype, bPHYSEQ_family_melt$Sample, sep="_")
bPHYSEQ_family_melt$samples <- as.factor(bPHYSEQ_family_melt$samples)

```



### GLM stats on family

#### Analysis by location and compartment (all samples)  

```{r GLM stats bacteria family, echo=F, warning=F}
## dataset
# head(bPHYSEQ_family_melt)
# dim(bPHYSEQ_family_melt)

## GLM-loop for each family, both factors
bfamily_stats <- c()
for(i in levels(bPHYSEQ_family_melt$family) ) {
  bPHYSEQ_family_melt_temp <- bPHYSEQ_family_melt[bPHYSEQ_family_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location + compartment, family=negative.binomial(theta=1), data=bPHYSEQ_family_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  loc_tukey_temp <- cld(glht(fit_temp, linfct=mcp(location="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  comp_tukey_temp <- cld(glht(fit_temp, linfct=mcp(compartment="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  bfamily_stats <- rbind(bfamily_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2:3], 4), scientific=F, nsmall=2, justify="left"),
                                      format( round( fit_temp2$`Pr(>F)`[2:3] , 4), scientific=F, nsmall=4),
                                      paste(loc_tukey_temp$mcletters$Letters),
                                      paste(comp_tukey_temp$mcletters$Letters) )  )
}
# arrange data frame
colnames(bfamily_stats) <- c("family", "F_location", "F_compartment", "P_location", "P_compartment", "Ch", "Re", "Au", "soil", "rhizo", "root")
rownames(bfamily_stats) <- bfamily_stats[,1]
bfamily_stats <- data.frame(bfamily_stats)
bfamily_stats[, 2:5] <- sapply(bfamily_stats[, 2:5], as.character)
# multiple hypotheis testing
bfamily_stats$Padj_location <- format(round(p.adjust(bfamily_stats$P_location, method="BH", n=nrow(bfamily_stats) ), 4), scientific=F)  
bfamily_stats$Padj_compartment <- format(round(p.adjust(bfamily_stats$P_compartment, method="BH", n=nrow(bfamily_stats) ), 4), scientific=F) 
# head(bfamily_stats)

## display stats in legend names
# subset to abundant family
bfamily_stats_abu <- bfamily_stats[ get_taxa_unique(family_abuP) , ]
bfamily_stats_abu$loc_legend <- ifelse( bfamily_stats_abu$Padj_location < 0.05 , "L*","") 
bfamily_stats_abu$comp_legend <- ifelse( bfamily_stats_abu$Padj_compartment < 0.05 , "C*","")
# head(bfamily_stats_abu)

## to be used in the family plot 
bfamily_stats_abu$stat_legend <- paste(rownames(bfamily_stats_abu), bfamily_stats_abu[,"loc_legend"], bfamily_stats_abu[,"comp_legend"], sep=" ")
# bfamily_stats_abu$stat_legend
```

##### Average abundances by location

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per location. Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. The columns with the posthoc Tukey test results is indicated with "_T". Different letters are used for groups that differ significantly in their abundances between locations (alpha=0.01).

```{r average abundances family bacteria by loc all, echo=F, warning=F}

### Percentages by location
bPHYSEQ_family_loc <- t(merge_samples(bPHYSEQ_family, "location")) # merge samples in groups if mean of all samples in one group
bPHYSEQ_family_loc <- transform_sample_counts(bPHYSEQ_family_loc, function(x) 100 * x/sum(x, na.rm=T)) # express in %RA

taxa_names(bPHYSEQ_family_loc) <- tax_table(bPHYSEQ_family_loc)[,"family"]

bfamily_loc <- as(otu_table(bPHYSEQ_family_loc), "matrix")
bfamily_stats_temp <- bfamily_stats[rownames(bfamily_loc),]
bfamily_loc_stats <- data.frame(F=bfamily_stats_temp[,"F_location"], Padj=bfamily_stats_temp[,"Padj_location"],
                               Ch_=bfamily_loc[,"Changins"], Ch_T=as.character(bfamily_stats_temp[,"Ch"]),
                               Re_=bfamily_loc[,"Reckenholz"], Re_T=as.character(bfamily_stats_temp[,"Re"]),
                               Au_=bfamily_loc[,"Aurora"], Au_T=as.character(bfamily_stats_temp[,"Au"]) )
# pander(bfamily_loc_stats)

```

##### Average abundances by compartment 


```{r average abundances family bacteria by comp all, echo=F, warning=F}

### Percentages by location
bPHYSEQ_family_comp <- t(merge_samples(bPHYSEQ_family, "compartment")) # merge samples in groups if mean of all samples in one group
bPHYSEQ_family_comp <- transform_sample_counts(bPHYSEQ_family_comp, function(x) 100 * x/sum(x, na.rm=T)) # express in %RA
taxa_names(bPHYSEQ_family_comp) <- tax_table(bPHYSEQ_family_comp)[,"family"]

bfamily_comp <- as(otu_table(bPHYSEQ_family_comp), "matrix")
bfamily_stats_temp <- bfamily_stats[rownames(bfamily_comp),]
bfamily_comp_stats <- data.frame(F=bfamily_stats_temp[,"F_compartment"], Padj=bfamily_stats_temp[,"Padj_compartment"],
                               Soil_=bfamily_comp[,"soil"], Ch_T=as.character(bfamily_stats_temp[,"Ch"]),
                               Rhizo_=bfamily_comp[,"rhizo"], Re_T=as.character(bfamily_stats_temp[,"Re"]),
                               Root_=bfamily_comp[,"root"], Au_T=as.character(bfamily_stats_temp[,"Au"]) )
# pander(bfamily_comp_stats)

```


#### Analysis by sample groups (reported in Table S3)

```{r GLM stats 2 bacteria family, echo=F, warning=F}
## GLM-loop for each family, all factors
# library(emmeans)
bfamily_stats <- c()
for(i in levels(bPHYSEQ_family_melt$family) ) {
  bPHYSEQ_family_melt_temp <- bPHYSEQ_family_melt[bPHYSEQ_family_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_compartment, family=negative.binomial(theta=1), data=bPHYSEQ_family_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_compartment")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdefghi", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bfamily_stats <- rbind(bfamily_stats, c(paste(i), format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"), format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4),
tukey_temp_letters$.group ))
}

# arrange data frame
colnames(bfamily_stats) <- c("family", "F", "P", 
                            "Changins_rhizo", "Changins_root", "Changins_soil",
                            "Aurora_rhizo", "Aurora_root", "Aurora_soil",
                            "Reckenholz_rhizo", "Reckenholz_root", "Reckenholz_soil")
rownames(bfamily_stats) <- bfamily_stats[,1]
bfamily_stats <- data.frame(bfamily_stats)
bfamily_stats[, 2:3] <- sapply(bfamily_stats[, 2:3], as.character)
# multiple hypotheis testing
bfamily_stats$Padj <- format(round(p.adjust(bfamily_stats$P, method="BH", n=nrow(bfamily_stats) ), 4), scientific=F)  
# head(bfamily_stats)

```

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * compartment). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Compartments are abbreviated with So=Soil, Rh=Rhizosphere and Ro=Root. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).


```{r average abundances family bacteria by sample groups all, echo=F, warning=F}

### Percentages by sample groups (location and compartment)
bPHYSEQ_family_groups <- t(merge_samples(bPHYSEQ_family, "location_compartment")) # merge samples in groups if mean of all samples in one group
bPHYSEQ_family_groups <- transform_sample_counts(bPHYSEQ_family_groups, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(bPHYSEQ_family_groups) <- tax_table(bPHYSEQ_family_groups)[,"family"]

bfamily_groups <- as(otu_table(bPHYSEQ_family_groups), "matrix")
bfamily_stats_temp <- bfamily_stats[rownames(bfamily_groups),]
bfamily_group_stats <- data.frame(F=bfamily_stats_temp[,"F"], Padj=bfamily_stats_temp[,"Padj"], 
                                 Ch_So=bfamily_groups[,"Changins_soil"], Ch_So_T=as.character(bfamily_stats_temp[,"Changins_soil"]),
                                 Re_So=bfamily_groups[,"Reckenholz_soil"], Re_So_T=as.character(bfamily_stats_temp[,"Reckenholz_soil"]),
                                 Au_So=bfamily_groups[,"Aurora_soil"], Au_So_T=as.character(bfamily_stats_temp[,"Aurora_soil"]),
                                 Ch_Rh=bfamily_groups[,"Changins_rhizo"], Ch_Rh_T=as.character(bfamily_stats_temp[,"Changins_rhizo"]), 
                                 Re_Rh=bfamily_groups[,"Reckenholz_rhizo"], Re_Rh_T=as.character(bfamily_stats_temp[,"Reckenholz_rhizo"]),
                                 Au_Rh=bfamily_groups[,"Aurora_rhizo"], Au_Rh_T=as.character(bfamily_stats_temp[,"Aurora_rhizo"]),
                                 Ch_Ro=bfamily_groups[,"Changins_root"], Ch_Ro_T=as.character(bfamily_stats_temp[,"Changins_root"]),
                                 Re_Ro=bfamily_groups[,"Reckenholz_root"], Re_Ro_T=as.character(bfamily_stats_temp[,"Reckenholz_root"]), 
                                 Au_Ro=bfamily_groups[,"Aurora_root"], Au_Ro_T=as.character(bfamily_stats_temp[,"Aurora_root"]) )

# pander(bfamily_group_stats[ get_taxa_unique(family_abuP), ])

bfamily_group_stats <- bfamily_group_stats[c(family_abuP, family_lowP),]

write.csv(bfamily_group_stats#[ c( get_taxa_unique(bPHYSEQ_family_abuP)
                                  #, get_taxa_unique(bPHYSEQ_family_lowP) 
                              #    ), ]
          ,file="tables/Table_S2_family_stats_bacteria.csv")
```

#### BX-effect (reported in Table S7)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).


#### in SOIL samples 

```{r soil BX effect glm family bacteria all, echo=F, warning=F}
# subsetting to soil samples, remove Changins soil samples (not sampled by genotype)
bPHYSEQ_family_soil <- prune_samples(sample_data(bPHYSEQ_family)$compartment=="soil", bPHYSEQ_family)
bPHYSEQ_family_soil <- prune_samples(sample_data(bPHYSEQ_family_soil)$genotype=="WT" | 
                                      sample_data(bPHYSEQ_family_soil)$genotype=="bx1", bPHYSEQ_family_soil)

## melt phyloseq object
bPHYSEQ_family_soil_melt <- psmelt(bPHYSEQ_family_soil) 
bPHYSEQ_family_soil_melt <- bPHYSEQ_family_soil_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "family")] # subset

bPHYSEQ_family_soil_melt$family <- factor(bPHYSEQ_family_soil_melt$family)

## GLM-loop for each family
# library(emmeans)
bfamily_soil_stats <- c()
bPHYSEQ_family_soil_melt$family <- as.factor(bPHYSEQ_family_soil_melt$family)
for(i in levels(bPHYSEQ_family_soil_melt$family) ) {
  bPHYSEQ_family_soil_melt_temp <- bPHYSEQ_family_soil_melt[bPHYSEQ_family_soil_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=bPHYSEQ_family_soil_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bfamily_soil_stats <- rbind(bfamily_soil_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                      format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                      tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(bfamily_soil_stats) <- c("family", "F", "P",
                                 "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(bfamily_soil_stats) <- bfamily_soil_stats[,1]
bfamily_soil_stats <- data.frame(bfamily_soil_stats)
bfamily_soil_stats[, 2:3] <- sapply(bfamily_soil_stats[, 2:3], as.character)
# multiple hypotheis testing
bfamily_soil_stats$Padj <- format(round(p.adjust(bfamily_soil_stats$P, method="BH", n=nrow(bfamily_soil_stats) ), 4), scientific=F)  
# head(bfamily_soil_stats)

### abundances by genotype at each location
bfamily_soil_RAs_geno <- t(merge_samples(bPHYSEQ_family_soil, "location_genotype")) # merge samples in groups if mean of all samples in one group
bfamily_soil_RAs_geno <- transform_sample_counts(bfamily_soil_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA

taxa_names(bfamily_soil_RAs_geno) <- tax_table(bfamily_soil_RAs_geno)[,"family"]
bfamily_soil_RAs_geno <- as(otu_table(bfamily_soil_RAs_geno), "matrix")
bfamily_soil_RAs_geno <- format( bfamily_soil_RAs_geno[rownames(bfamily_soil_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
bfamily_soil_table <- data.frame(F=bfamily_soil_stats[,"F"], Padj=bfamily_soil_stats[,"Padj"], 
                                Au_WT=bfamily_soil_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(bfamily_soil_stats[,"Aurora_WT"]),
                                Au_bx=bfamily_soil_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(bfamily_soil_stats[,"Aurora_bx1"]),
                                Re_WT=bfamily_soil_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(bfamily_soil_stats[,"Reckenholz_WT"]),
                                Re_bx=bfamily_soil_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(bfamily_soil_stats[,"Reckenholz_bx1"]))

#pander(bfamily_soil_table[ get_taxa_unique(family_abuP), ])


bfamily_soil_table <- bfamily_soil_table[c(family_abuP, family_lowP),]

write.csv(bfamily_soil_table#[ c( get_taxa_unique(bPHYSEQ_family_abuP)
                                 #, get_taxa_unique(bPHYSEQ_family_lowP) 
                             #    ), ]
          ,file="tables/Table_S7_family_BX_stats_soil_bacteria.csv")
```

#### on RHIZOSPHERE samples


```{r rhizo BX effect glm family bacteria all, echo=F, warning=F}
# subsetting to rhizo samples, remove Changins rhizo samples (not sampled by genotype)
bPHYSEQ_family_rhizo <- prune_samples(sample_data(bPHYSEQ_family)$compartment=="rhizo", bPHYSEQ_family)
bPHYSEQ_family_rhizo <- prune_samples(sample_data(bPHYSEQ_family_rhizo)$genotype=="WT" | 
                                      sample_data(bPHYSEQ_family_rhizo)$genotype=="bx1", bPHYSEQ_family_rhizo)

## melt phyloseq object
bPHYSEQ_family_rhizo_melt <- psmelt(bPHYSEQ_family_rhizo) 
bPHYSEQ_family_rhizo_melt <- bPHYSEQ_family_rhizo_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "family")] # subset

## GLM-loop for each family
# library(emmeans)
bfamily_rhizo_stats <- c()
bPHYSEQ_family_rhizo_melt$family <- as.factor(bPHYSEQ_family_rhizo_melt$family)
for(i in levels(bPHYSEQ_family_rhizo_melt$family) ) {
  bPHYSEQ_family_rhizo_melt_temp <- bPHYSEQ_family_rhizo_melt[bPHYSEQ_family_rhizo_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=bPHYSEQ_family_rhizo_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bfamily_rhizo_stats <- rbind(bfamily_rhizo_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"),
                                      format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) ,
                                      tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(bfamily_rhizo_stats) <- c("family", "F", "P", "Changins_bx1", "Changins_WT", "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")

rownames(bfamily_rhizo_stats) <- bfamily_rhizo_stats[,1]
bfamily_rhizo_stats <- data.frame(bfamily_rhizo_stats)
bfamily_rhizo_stats[, 2:3] <- sapply(bfamily_rhizo_stats[, 2:3], as.character)
# multiple hypotheis testing
bfamily_rhizo_stats$Padj <- format(round(p.adjust(bfamily_rhizo_stats$P, method="BH", n=nrow(bfamily_rhizo_stats) ), 4), scientific=F)  
# head(bfamily_rhizo_stats)

### abundances by genotype at each location
bfamily_rhizo_RAs_geno <- t(merge_samples(bPHYSEQ_family_rhizo, "location_genotype")) # merge samples in groups if mean of all samples in one group
bfamily_rhizo_RAs_geno <- transform_sample_counts(bfamily_rhizo_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA


taxa_names(bfamily_rhizo_RAs_geno) <- tax_table(bfamily_rhizo_RAs_geno)[,"family"]
bfamily_rhizo_RAs_geno <- as(otu_table(bfamily_rhizo_RAs_geno), "matrix")
bfamily_rhizo_RAs_geno <- format( bfamily_rhizo_RAs_geno[rownames(bfamily_rhizo_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
bfamily_rhizo_table <- data.frame(F=bfamily_rhizo_stats[,"F"], Padj=bfamily_rhizo_stats[,"Padj"], 
                                Ch_WT=bfamily_rhizo_RAs_geno[,"Changins_WT"], Ch_WT_T=as.character(bfamily_rhizo_stats[,"Changins_WT"]),
                                Ch_bx=bfamily_rhizo_RAs_geno[,"Changins_bx1"], Ch_bx_T=as.character(bfamily_rhizo_stats[,"Changins_bx1"]),
                                Au_WT=bfamily_rhizo_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(bfamily_rhizo_stats[,"Aurora_WT"]),
                                Au_bx=bfamily_rhizo_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(bfamily_rhizo_stats[,"Aurora_bx1"]),
                                Re_WT=bfamily_rhizo_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(bfamily_rhizo_stats[,"Reckenholz_WT"]),
                                Re_bx=bfamily_rhizo_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(bfamily_rhizo_stats[,"Reckenholz_bx1"]))

#pander(bfamily_rhizo_table)

#pander(bfamily_rhizo_table[ get_taxa_unique(bPHYSEQ_family_abuP), ])
bfamily_rhizo_table <- bfamily_rhizo_table[c(family_abuP,family_lowP),]


write.csv(bfamily_rhizo_table#[ c( get_taxa_unique(bPHYSEQ_family_abuP)
                                 #, get_taxa_unique(bPHYSEQ_family_lowP) 
                              #   ), ]
          ,file="tables/Table_S7_family_BX_stats_rhizo_bacteria.csv")
```

#### on ROOT samples 

```{r root BX effect glm family bacteria all, echo=F, warning=F}
# subsetting to root samples, remove Changins root samples (not sampled by genotype)
bPHYSEQ_family_root <- prune_samples(sample_data(bPHYSEQ_family)$compartment=="root", bPHYSEQ_family)
bPHYSEQ_family_root <- prune_samples(sample_data(bPHYSEQ_family_root)$genotype=="WT" | 
                                      sample_data(bPHYSEQ_family_root)$genotype=="bx1", bPHYSEQ_family_root)

## melt phyloseq object
bPHYSEQ_family_root_melt <- psmelt(bPHYSEQ_family_root) 
bPHYSEQ_family_root_melt <- bPHYSEQ_family_root_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "family")] # subset

## GLM-loop for each family
# library(emmeans)
bfamily_root_stats <- c()
bPHYSEQ_family_root_melt$family <- as.factor(bPHYSEQ_family_root_melt$family)
for(i in levels(bPHYSEQ_family_root_melt$family) ) {
  bPHYSEQ_family_root_melt_temp <- bPHYSEQ_family_root_melt[bPHYSEQ_family_root_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=bPHYSEQ_family_root_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  bfamily_root_stats <- rbind(bfamily_root_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"), format( round( fit_temp2$`Pr(>F)`[2], 4), scientific=F, nsmall=4), tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(bfamily_root_stats) <- c("family", "F", "P", "Changins_bx1", "Changins_WT", "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")

rownames(bfamily_root_stats) <- bfamily_root_stats[,1]
bfamily_root_stats <- data.frame(bfamily_root_stats)
bfamily_root_stats[, 2:3] <- sapply(bfamily_root_stats[, 2:3], as.character)
# multiple hypotheis testing
bfamily_root_stats$Padj <- format(round(p.adjust(bfamily_root_stats$P, method="BH", n=nrow(bfamily_root_stats) ), 4), scientific=F)  
# head(bfamily_root_stats)

### abundances by genotype at each location
bfamily_root_RAs_geno <- t(merge_samples(bPHYSEQ_family_root, "location_genotype")) # merge samples in groups if mean of all samples in one group
bfamily_root_RAs_geno <- transform_sample_counts(bfamily_root_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA


taxa_names(bfamily_root_RAs_geno) <- tax_table(bfamily_root_RAs_geno)[,"family"]
bfamily_root_RAs_geno <- as(otu_table(bfamily_root_RAs_geno), "matrix")
bfamily_root_RAs_geno <- format( bfamily_root_RAs_geno[rownames(bfamily_root_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
bfamily_root_table <- data.frame(F=bfamily_root_stats[,"F"], Padj=bfamily_root_stats[,"Padj"], 
                                 Ch_WT=bfamily_root_RAs_geno[,"Changins_WT"], Ch_WT_T=as.character(bfamily_root_stats[,"Changins_WT"]),
                                Ch_bx=bfamily_root_RAs_geno[,"Changins_bx1"], Ch_bx_T=as.character(bfamily_root_stats[,"Changins_bx1"]),
                                Au_WT=bfamily_root_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(bfamily_root_stats[,"Aurora_WT"]),
                                Au_bx=bfamily_root_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(bfamily_root_stats[,"Aurora_bx1"]),
                                Re_WT=bfamily_root_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(bfamily_root_stats[,"Reckenholz_WT"]),
                                Re_bx=bfamily_root_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(bfamily_root_stats[,"Reckenholz_bx1"]))

#pander(bfamily_root_table)

#pander(bfamily_root_table[ get_taxa_unique(bPHYSEQ_family_abuP), ])

bfamily_root_table <- bfamily_root_table[c(family_abuP, family_lowP),]


write.csv(bfamily_root_table#[ c( get_taxa_unique(bPHYSEQ_family_abuP)
                                 #, get_taxa_unique(bPHYSEQ_family_lowP) 
                                # ), ]
         , file="tables/Table_S7_family_BX_stats_root_bacteria.csv")
```



\pagebreak

## Fungi

### Data display


```{r family all fungi, warning=F, echo=F, fig.height=7, fig.width=10}
### use PHYLOSEQ to plot the data
## for faster calculation in tax_glom

fTAX2 <- fTAX[,c( "kingdom","phylum", "family", "class", "order", "genus", "species", "OTU_ID")]

## library(phyloseq) with all data, rarefied
fPHYSEQ <- phyloseq(sample_data(fDESIGN_rare),
                    otu_table(fDAT_rare, taxa_are_rows=T),
                    tax_table(as.matrix(fTAX2[rownames(fDAT_rare),])) )

## agglomerate data by family and class (for Proteofungi) using factor 'family'
fPHYSEQ_family <- tax_glom(fPHYSEQ, "family") # merge by 'family': all family plus classes of Proteofungi ?? NArm=F
fPHYSEQ_family <- transform_sample_counts(fPHYSEQ_family, function(x) 100 * x/sum(x)) 

fPHYSEQ_family = subset_taxa(fPHYSEQ_family, family != "unassigned" & family != "uncultured" & family != "Unknown_Family")

### Defining high and low abundant family
# family with MEAN abundances higher than 1% relative abundances
ffamily_abu <-  rownames(otu_table(fPHYSEQ_family))[apply(otu_table(fPHYSEQ_family), 1, mean) > 1]
ffamily_abuP <- tax_table(fPHYSEQ_family)[rownames(tax_table(fPHYSEQ_family)) %in% ffamily_abu, "family"]
# fPHYSEQ_family_abuP
# family with MEAN abundances lower than 1% relative abundances
ffamily_low <-  rownames(otu_table(fPHYSEQ_family))[apply(otu_table(fPHYSEQ_family), 1, mean) < 1]
ffamily_lowP <- tax_table(fPHYSEQ_family)[rownames(tax_table(fPHYSEQ_family)) %in% ffamily_low, "family"]


# For abundant family
fPHYSEQ_family_abuF <- prune_taxa(taxa_names(ffamily_abuP), fPHYSEQ_family) # prune to only abundant family 

## melt phyloseq object: abundant families
fPHYSEQ_family_melt <- psmelt(fPHYSEQ_family) 
fPHYSEQ_family_melt <- fPHYSEQ_family_melt[,c("OTU","Sample", "Abundance", "compartment", "location", "groups", "family", "background_genotype", "location_compartment")] # subset

### subsetting the color vector to abundant family and classes
# delete label name of low-abundant family (for plot) and put them at the bottom
fPHYSEQ_family_melt$family <- as.factor(as.character(fPHYSEQ_family_melt$family))
fPHYSEQ_family_melt$compartment <- factor(fPHYSEQ_family_melt$compartment, levels= c("soil","rhizo","root"))
fPHYSEQ_family_melt$background_genotype <- as.character(fPHYSEQ_family_melt$background_genotype)
fPHYSEQ_family_melt$samples <- paste(fPHYSEQ_family_melt$background_genotype, fPHYSEQ_family_melt$Sample, sep="_")
fPHYSEQ_family_melt$samples <- as.factor(fPHYSEQ_family_melt$samples)

```

### GLM stats on family

#### Analysis by location and compartment (all samples)  

```{r GLM stats fungi family, echo=F, warning=F}
## dataset

## GLM-loop for each family, both factors
ffamily_stats <- c()
for(i in levels(fPHYSEQ_family_melt$family) ) {
  fPHYSEQ_family_melt_temp <- fPHYSEQ_family_melt[fPHYSEQ_family_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location + compartment, family=negative.binomial(theta=1), data=fPHYSEQ_family_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  loc_tukey_temp <- cld(glht(fit_temp, linfct=mcp(location="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  comp_tukey_temp <- cld(glht(fit_temp, linfct=mcp(compartment="Tukey")) , decreasing=TRUE, alpha=0.01)  # letters
  ffamily_stats <- rbind(ffamily_stats, c(paste(i), 
                                      format( round( fit_temp2$F[2:3], 4), scientific=F, nsmall=2, justify="left"),
                                      format( round( fit_temp2$`Pr(>F)`[2:3] , 4), scientific=F, nsmall=4),
                                      paste(loc_tukey_temp$mcletters$Letters),
                                      paste(comp_tukey_temp$mcletters$Letters) )  )
}
# arrange data frame
colnames(ffamily_stats) <- c("family", "F_location", "F_compartment", "P_location", "P_compartment", "Ch", "Re", "Au", "soil", "rhizo", "root")
rownames(ffamily_stats) <- ffamily_stats[,1]
ffamily_stats <- data.frame(ffamily_stats)
ffamily_stats[, 2:5] <- sapply(ffamily_stats[, 2:5], as.character)
# multiple hypotheis testing
ffamily_stats$Padj_location <- format(round(p.adjust(ffamily_stats$P_location, method="BH", n=nrow(ffamily_stats) ), 4), scientific=F)  
ffamily_stats$Padj_compartment <- format(round(p.adjust(ffamily_stats$P_compartment, method="BH", n=nrow(ffamily_stats) ), 4), scientific=F) 
# head(ffamily_stats)

## display stats in legend names
# subset to abundant family
ffamily_stats_abu <- ffamily_stats[ get_taxa_unique(ffamily_abuP) , ]
ffamily_stats_abu$loc_legend <- ifelse( ffamily_stats_abu$Padj_location < 0.05 , "L*","") 
ffamily_stats_abu$comp_legend <- ifelse( ffamily_stats_abu$Padj_compartment < 0.05 , "C*","")
# head(ffamily_stats_abu)

## to be used in the family plot 
ffamily_stats_abu$stat_legend <- paste(rownames(ffamily_stats_abu), ffamily_stats_abu[,"loc_legend"], ffamily_stats_abu[,"comp_legend"], sep=" ")
# ffamily_stats_abu$stat_legend
```

##### Average abundances by location

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per location. Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. The columns with the posthoc Tukey test results is indicated with "_T". Different letters are used for groups that differ significantly in their abundances between locations (alpha=0.01).


```{r average abundances family fungi by loc all, echo=F, warning=F}

### Percentages by location
fPHYSEQ_family_loc <- t(merge_samples(fPHYSEQ_family, "location")) # merge samples in groups if mean of all samples in one group
fPHYSEQ_family_loc <- transform_sample_counts(fPHYSEQ_family_loc, function(x) 100 * x/sum(x, na.rm=T)) # express in %RA
taxa_names(fPHYSEQ_family_loc) <- tax_table(fPHYSEQ_family_loc)[,"family"]

ffamily_loc <- as(otu_table(fPHYSEQ_family_loc), "matrix")
ffamily_stats_temp <- ffamily_stats[rownames(ffamily_loc),]
ffamily_loc_stats <- data.frame(F=ffamily_stats_temp[,"F_location"], Padj=ffamily_stats_temp[,"Padj_location"],
                               Ch_=ffamily_loc[,"Changins"], Ch_T=as.character(ffamily_stats_temp[,"Ch"]),
                               Re_=ffamily_loc[,"Reckenholz"], Re_T=as.character(ffamily_stats_temp[,"Re"]),
                               Au_=ffamily_loc[,"Aurora"], Au_T=as.character(ffamily_stats_temp[,"Au"]) )
# pander(ffamily_loc_stats)

```

##### Average abundances by compartment 

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per compartment. Compartments are abbreviated with So=Soil, Rh=Rhizosphere and Ro=Root. The columns with the posthoc Tukey test results is indicated with "_T". Different letters are used for groups that differ significantly in their abundances between compartments (alpha=0.01).


```{r average abundances family fungi by comp all, echo=F, warning=F}

### Percentages by location
fPHYSEQ_family_comp <- t(merge_samples(fPHYSEQ_family, "compartment")) # merge samples in groups if mean of all samples in one group
fPHYSEQ_family_comp <- transform_sample_counts(fPHYSEQ_family_comp, function(x) 100 * x/sum(x, na.rm=T)) # express in %RA
taxa_names(fPHYSEQ_family_comp) <- tax_table(fPHYSEQ_family_comp)[,"family"]

ffamily_comp <- as(otu_table(fPHYSEQ_family_comp), "matrix")
ffamily_stats_temp <- ffamily_stats[rownames(ffamily_comp),]
ffamily_comp_stats <- data.frame(F=ffamily_stats_temp[,"F_compartment"], Padj=ffamily_stats_temp[,"Padj_compartment"],
                               Soil_=ffamily_comp[,"soil"], Ch_T=as.character(ffamily_stats_temp[,"Ch"]),
                               Rhizo_=ffamily_comp[,"rhizo"], Re_T=as.character(ffamily_stats_temp[,"Re"]),
                               Root_=ffamily_comp[,"root"], Au_T=as.character(ffamily_stats_temp[,"Au"]) )
# pander(ffamily_comp_stats)

```


\pagebreak

#### Analysis by sample groups (reported in Table S3)

```{r GLM stats 2 fungi family, echo=F, warning=F}
## GLM-loop for each family, all factors
# library(emmeans)
ffamily_stats <- c()
for(i in levels(fPHYSEQ_family_melt$family) ) {
  fPHYSEQ_family_melt_temp <- fPHYSEQ_family_melt[fPHYSEQ_family_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_compartment, family=negative.binomial(theta=1), data=fPHYSEQ_family_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_compartment")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcdefghi", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  ffamily_stats <- rbind(ffamily_stats, c(paste(i), format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"), format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4),
tukey_temp_letters$.group ))
}

# arrange data frame
colnames(ffamily_stats) <- c("family", "F", "P", 
                            "Changins_rhizo", "Changins_root", "Changins_soil",
                            "Aurora_rhizo"  , "Aurora_root"  , "Aurora_soil",
                            "Reckenholz_rhizo"  , "Reckenholz_root"  , "Reckenholz_soil")
rownames(ffamily_stats) <- ffamily_stats[,1]
ffamily_stats <- data.frame(ffamily_stats)
ffamily_stats[, 2:3] <- sapply(ffamily_stats[, 2:3], as.character)
# multiple hypotheis testing
ffamily_stats$Padj <- format(round(p.adjust(ffamily_stats$P, method="BH", n=nrow(ffamily_stats) ), 4), scientific=F)  
# head(ffamily_stats)
```

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * compartment). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Compartments are abbreviated with So=Soil, Rh=Rhizosphere and Ro=Root. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).


```{r average abundances family fungi by sample groups all, echo=F, warning=F}

### Percentages by sample groups (location and compartment)
fPHYSEQ_family_groups <- t(merge_samples(fPHYSEQ_family, "location_compartment")) # merge samples in groups if mean of all samples in one group
fPHYSEQ_family_groups <- transform_sample_counts(fPHYSEQ_family_groups, function(x) 100 * x/sum(x)) # express in %RA
taxa_names(fPHYSEQ_family_groups) <- tax_table(fPHYSEQ_family_groups)[,"family"]

ffamily_groups <- as(otu_table(fPHYSEQ_family_groups), "matrix")
ffamily_stats_temp <- ffamily_stats[rownames(ffamily_groups),]
ffamily_group_stats <- data.frame(F=ffamily_stats_temp[,"F"], Padj=ffamily_stats_temp[,"Padj"], 
                                 Ch_So=ffamily_groups[,"Changins_soil"], Ch_So_T=as.character(ffamily_stats_temp[,"Changins_soil"]),
                                 Re_So=ffamily_groups[,"Reckenholz_soil"], Re_So_T=as.character(ffamily_stats_temp[,"Reckenholz_soil"]),
                                 Au_So=ffamily_groups[,"Aurora_soil"], Au_So_T=as.character(ffamily_stats_temp[,"Aurora_soil"]),
                                 Ch_Rh=ffamily_groups[,"Changins_rhizo"], Ch_Rh_T=as.character(ffamily_stats_temp[,"Changins_rhizo"]), 
                                 Re_Rh=ffamily_groups[,"Reckenholz_rhizo"], Re_Rh_T=as.character(ffamily_stats_temp[,"Reckenholz_rhizo"]),
                                 Au_Rh=ffamily_groups[,"Aurora_rhizo"], Au_Rh_T=as.character(ffamily_stats_temp[,"Aurora_rhizo"]),
                                 Ch_Ro=ffamily_groups[,"Changins_root"], Ch_Ro_T=as.character(ffamily_stats_temp[,"Changins_root"]),
                                 Re_Ro=ffamily_groups[,"Reckenholz_root"], Re_Ro_T=as.character(ffamily_stats_temp[,"Reckenholz_root"]), 
                                 Au_Ro=ffamily_groups[,"Aurora_root"], Au_Ro_T=as.character(ffamily_stats_temp[,"Aurora_root"]) )

# pander(ffamily_group_stats[ get_taxa_unique(family_abuP), ])

ffamily_group_stats <- ffamily_group_stats[c(ffamily_abuP, ffamily_lowP),]
#length(fPHYSEQ_family_abuP)


write.csv(ffamily_group_stats#[ c( get_taxa_unique(fPHYSEQ_family_abuP)
                                  #, get_taxa_unique(fPHYSEQ_family_lowP) 
                              #    ), ]
          ,file="tables/Table_S3_family_stats_fungi.csv")
```


\pagebreak

#### BX-effect in SOIL samples (reported in Table S7)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).


```{r soil BX effect glm family fungi all, echo=F, warning=F}
# subsetting to soil samples, remove Changins soil samples (not sampled by genotype)
fPHYSEQ_family_soil <- prune_samples(sample_data(fPHYSEQ_family)$compartment=="soil", fPHYSEQ_family)
fPHYSEQ_family_soil <- prune_samples(sample_data(fPHYSEQ_family_soil)$genotype=="WT" | 
                                      sample_data(fPHYSEQ_family_soil)$genotype=="bx1", fPHYSEQ_family_soil)

## melt phyloseq object
fPHYSEQ_family_soil_melt <- psmelt(fPHYSEQ_family_soil) 
fPHYSEQ_family_soil_melt <- fPHYSEQ_family_soil_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "family")] # subset


## GLM-loop for each family
# library(emmeans)
ffamily_soil_stats <- c()
fPHYSEQ_family_soil_melt$family <- as.factor(fPHYSEQ_family_soil_melt$family)
for(i in levels(fPHYSEQ_family_soil_melt$family) ) {
  fPHYSEQ_family_soil_melt_temp <- fPHYSEQ_family_soil_melt[fPHYSEQ_family_soil_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=fPHYSEQ_family_soil_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  ffamily_soil_stats <- rbind(ffamily_soil_stats, c(paste(i), format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"), format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) , tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(ffamily_soil_stats) <- c("family", "F", "P",
                                 "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")
rownames(ffamily_soil_stats) <- ffamily_soil_stats[,1]
ffamily_soil_stats <- data.frame(ffamily_soil_stats)
ffamily_soil_stats[, 2:3] <- sapply(ffamily_soil_stats[, 2:3], as.character)
# multiple hypotheis testing
ffamily_soil_stats$Padj <- format(round(p.adjust(ffamily_soil_stats$P, method="BH", n=nrow(ffamily_soil_stats) ), 4), scientific=F)  
# head(ffamily_soil_stats)

### abundances by genotype at each location
ffamily_soil_RAs_geno <- t(merge_samples(fPHYSEQ_family_soil, "location_genotype")) # merge samples in groups if mean of all samples in one group
ffamily_soil_RAs_geno <- transform_sample_counts(ffamily_soil_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA

taxa_names(ffamily_soil_RAs_geno) <- tax_table(ffamily_soil_RAs_geno)[,"family"]
ffamily_soil_RAs_geno <- as(otu_table(ffamily_soil_RAs_geno), "matrix")
ffamily_soil_RAs_geno <- format( ffamily_soil_RAs_geno[rownames(ffamily_soil_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
ffamily_soil_table <- data.frame(F=ffamily_soil_stats[,"F"], Padj=ffamily_soil_stats[,"Padj"], 
                                Au_WT=ffamily_soil_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(ffamily_soil_stats[,"Aurora_WT"]),
                                Au_bx=ffamily_soil_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(ffamily_soil_stats[,"Aurora_bx1"]),
                                Re_WT=ffamily_soil_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(ffamily_soil_stats[,"Reckenholz_WT"]),
                                Re_bx=ffamily_soil_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(ffamily_soil_stats[,"Reckenholz_bx1"]))

# pander(ffamily_soil_table[ get_taxa_unique(family_abuP), ])

ffamily_soil_table <- ffamily_soil_table[c(ffamily_abuP, ffamily_lowP),]


write.csv(ffamily_soil_table#[ c( get_taxa_unique(fPHYSEQ_family_abuP)
                                 #, get_taxa_unique(fPHYSEQ_family_lowP) 
                             #    ), ]
          ,file="tables/Table_S7_family_BX_stats_soil_fungi.csv")
```



\pagebreak

#### BX-effect RHIZOSPHERE samples (reported in Table S7)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).


```{r rhizo BX effect glm family fungi all, echo=F, warning=F}
# subsetting to rhizo samples, remove Changins rhizo samples (not sampled by genotype)
fPHYSEQ_family_rhizo <- prune_samples(sample_data(fPHYSEQ_family)$compartment=="rhizo", fPHYSEQ_family)
fPHYSEQ_family_rhizo <- prune_samples(sample_data(fPHYSEQ_family_rhizo)$genotype=="WT" | 
                                      sample_data(fPHYSEQ_family_rhizo)$genotype=="bx1", fPHYSEQ_family_rhizo)

## melt phyloseq object
fPHYSEQ_family_rhizo_melt <- psmelt(fPHYSEQ_family_rhizo) 
fPHYSEQ_family_rhizo_melt <- fPHYSEQ_family_rhizo_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "family")] # subset

## GLM-loop for each family
# library(emmeans)
ffamily_rhizo_stats <- c()
fPHYSEQ_family_rhizo_melt$family <- as.factor(fPHYSEQ_family_rhizo_melt$family)
for(i in levels(fPHYSEQ_family_rhizo_melt$family) ) {
  fPHYSEQ_family_rhizo_melt_temp <- fPHYSEQ_family_rhizo_melt[fPHYSEQ_family_rhizo_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=fPHYSEQ_family_rhizo_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  ffamily_rhizo_stats <- rbind(ffamily_rhizo_stats, c(paste(i), format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"), format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4) , tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(ffamily_rhizo_stats) <- c("family", "F", "P", "Changins_bx1", "Changins_WT", "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")

rownames(ffamily_rhizo_stats) <- ffamily_rhizo_stats[,1]
ffamily_rhizo_stats <- data.frame(ffamily_rhizo_stats)
ffamily_rhizo_stats[, 2:3] <- sapply(ffamily_rhizo_stats[, 2:3], as.character)
# multiple hypotheis testing
ffamily_rhizo_stats$Padj <- format(round(p.adjust(ffamily_rhizo_stats$P, method="BH", n=nrow(ffamily_rhizo_stats) ), 4), scientific=F)  
# head(ffamily_rhizo_stats)

### abundances by genotype at each location
ffamily_rhizo_RAs_geno <- t(merge_samples(fPHYSEQ_family_rhizo, "location_genotype")) # merge samples in groups if mean of all samples in one group
ffamily_rhizo_RAs_geno <- transform_sample_counts(ffamily_rhizo_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA

taxa_names(ffamily_rhizo_RAs_geno) <- tax_table(ffamily_rhizo_RAs_geno)[,"family"]
ffamily_rhizo_RAs_geno <- as(otu_table(ffamily_rhizo_RAs_geno), "matrix")
ffamily_rhizo_RAs_geno <- format( ffamily_rhizo_RAs_geno[rownames(ffamily_rhizo_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
ffamily_rhizo_table <- data.frame(F=ffamily_rhizo_stats[,"F"], Padj=ffamily_rhizo_stats[,"Padj"], 
                                  Ch_WT=ffamily_rhizo_RAs_geno[,"Changins_WT"], Ch_WT_T=as.character(ffamily_rhizo_stats[,"Changins_WT"]),
                                Ch_bx=ffamily_rhizo_RAs_geno[,"Changins_bx1"], Ch_bx_T=as.character(ffamily_rhizo_stats[,"Changins_bx1"]),
                                Au_WT=ffamily_rhizo_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(ffamily_rhizo_stats[,"Aurora_WT"]),
                                Au_bx=ffamily_rhizo_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(ffamily_rhizo_stats[,"Aurora_bx1"]),
                                Re_WT=ffamily_rhizo_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(ffamily_rhizo_stats[,"Reckenholz_WT"]),
                                Re_bx=ffamily_rhizo_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(ffamily_rhizo_stats[,"Reckenholz_bx1"]))

#pander(ffamily_rhizo_table)

#pander(ffamily_rhizo_table[ get_taxa_unique(fPHYSEQ_family_abuP), ])

ffamily_rhizo_table <- ffamily_rhizo_table[c(ffamily_abuP,ffamily_lowP),]

write.csv(ffamily_rhizo_table#[ c( get_taxa_unique(fPHYSEQ_family_abuP)
                                 #, get_taxa_unique(fPHYSEQ_family_lowP) 
                              #   ), ]
          ,file="tables/Table_S7_family_BX_stats_rhizo_fungi.csv")
```

\pagebreak

#### BX-effect ROOT samples (reported in Table S7)

The table reports the F statistic together with the corresponding P value (adjusted for multiple hypothesis testing, following Bonferroni Hochberg). The table also displays the relative average abundances [%] per sample group (location * genotype). Locations are abbreviated with Ch=Changins, Au=Aurora and Zh=Reckenholz. Genotypes are abbreviated with WT=wildtype and bx for the bx1 mutant. The columns with the posthoc Tukey test results is indicated with "_T" for each sample group. Different letters are used for groups that differ significantly in their abundances (alpha=0.01).

```{r root BX effect glm family fungi all, echo=F, warning=F}
# subsetting to root samples, remove Changins root samples (not sampled by genotype)
fPHYSEQ_family_root <- prune_samples(sample_data(fPHYSEQ_family)$compartment=="root", fPHYSEQ_family)
fPHYSEQ_family_root <- prune_samples(sample_data(fPHYSEQ_family_root)$genotype=="WT" | 
                                      sample_data(fPHYSEQ_family_root)$genotype=="bx1", fPHYSEQ_family_root)

## melt phyloseq object
fPHYSEQ_family_root_melt <- psmelt(fPHYSEQ_family_root) 
fPHYSEQ_family_root_melt <- fPHYSEQ_family_root_melt[, c("Sample", "Abundance", "compartment", "location", "genotype", "location_genotype", "family")] # subset

## GLM-loop for each family
# library(emmeans)
ffamily_root_stats <- c()
fPHYSEQ_family_root_melt$family <- as.factor(fPHYSEQ_family_root_melt$family)
for(i in levels(fPHYSEQ_family_root_melt$family) ) {
  fPHYSEQ_family_root_melt_temp <- fPHYSEQ_family_root_melt[fPHYSEQ_family_root_melt$family==paste(i), ]
  fit_temp <- glm(as.integer(Abundance*100) ~ location_genotype, family=negative.binomial(theta=1), data=fPHYSEQ_family_root_melt_temp)
  fit_temp2 <- anova(fit_temp, test="F")
  tukey_temp <- emmeans(fit_temp, "location_genotype")  # library(emmeans)
  tukey_temp_letters <- cld(tukey_temp, Letter="abcd", alpha=0.01)
  tukey_temp_letters <- tukey_temp_letters[order(as.numeric(rownames(tukey_temp_letters))),]
  ffamily_root_stats <- rbind(ffamily_root_stats, c(paste(i), format( round( fit_temp2$F[2], 4), scientific=F, nsmall=2, justify="left"), format( round( fit_temp2$`Pr(>F)`[2] , 4), scientific=F, nsmall=4), tukey_temp_letters$.group ) )
}

# arrange data frame
colnames(ffamily_root_stats) <- c("family", "F", "P", "Changins_bx1", "Changins_WT", "Aurora_bx1", "Aurora_WT", "Reckenholz_bx1", "Reckenholz_WT")

rownames(ffamily_root_stats) <- ffamily_root_stats[,1]
ffamily_root_stats <- data.frame(ffamily_root_stats)
ffamily_root_stats[, 2:3] <- sapply(ffamily_root_stats[, 2:3], as.character)
# multiple hypotheis testing
ffamily_root_stats$Padj <- format(round(p.adjust(ffamily_root_stats$P, method="BH", n=nrow(ffamily_root_stats) ), 4), scientific=F)  
# head(ffamily_root_stats)

### abundances by genotype at each location
ffamily_root_RAs_geno <- t(merge_samples(fPHYSEQ_family_root, "location_genotype")) # merge samples in groups if mean of all samples in one group
ffamily_root_RAs_geno <- transform_sample_counts(ffamily_root_RAs_geno, function(x) 100 * x/sum(x)) # express in %RA

taxa_names(ffamily_root_RAs_geno) <- tax_table(ffamily_root_RAs_geno)[,"family"]
ffamily_root_RAs_geno <- as(otu_table(ffamily_root_RAs_geno), "matrix")
ffamily_root_RAs_geno <- format( ffamily_root_RAs_geno[rownames(ffamily_root_stats), ], scientific=F, nsmall=4 ) 

# combine stats and abundances
ffamily_root_table <- data.frame(F=ffamily_root_stats[,"F"], Padj=ffamily_root_stats[,"Padj"], 
                                  Ch_WT=ffamily_root_RAs_geno[,"Changins_WT"], Ch_WT_T=as.character(ffamily_root_stats[,"Changins_WT"]),
                                Ch_bx=ffamily_root_RAs_geno[,"Changins_bx1"], Ch_bx_T=as.character(ffamily_root_stats[,"Changins_bx1"]),
                                Au_WT=ffamily_root_RAs_geno[,"Aurora_WT"], Au_WT_T=as.character(ffamily_root_stats[,"Aurora_WT"]),
                                Au_bx=ffamily_root_RAs_geno[,"Aurora_bx1"], Au_bx_T=as.character(ffamily_root_stats[,"Aurora_bx1"]),
                                Re_WT=ffamily_root_RAs_geno[,"Reckenholz_WT"], Re_WT_T=as.character(ffamily_root_stats[,"Reckenholz_WT"]),
                                Re_bx=ffamily_root_RAs_geno[,"Reckenholz_bx1"], Re_bx_T=as.character(ffamily_root_stats[,"Reckenholz_bx1"]))

#pander(ffamily_root_table)

#pander(ffamily_root_table[ get_taxa_unique(fPHYSEQ_family_abuP), ])
ffamily_root_table <- ffamily_root_table[c(ffamily_abuP,ffamily_lowP),]

write.csv(ffamily_root_table#[ c( get_taxa_unique(fPHYSEQ_family_abuP)
                                 #, get_taxa_unique(fPHYSEQ_family_lowP) 
                                # ), ]
         , file="tables/Table_S7_family_BX_stats_root_fungi.csv")
```


\pagebreak

# Alpha diversity

## Shannon diversity Bacteria

Method: we rarefy the bacterial dataset to the sequencing depth of `r b_rare` and calculated Shannon diversity in each sample. This was repeated 100 times and the mean value from the 100 iterations was taken for statistical analysis between the different samples.  

### Analysis by location and compartment (reported in Table S4)

The table reports the Shannon diversity index together with the results form the pair-wise Tukey test results. Different letters are used for groups that differ significantly in their abundances (alpha=0.05).

```{r a-div stats bacteria, warning=FALSE, echo=F, message=FALSE, eval=T}
diversity_all_bac_mat <- c()

for (i in 1:100) {
  # generate rarefied community
  temp_mat <- t(rrarefy(t(bDAT), b_rare))
  # from Jost 2006: D=exp(H), exponential of shannon entropy
  shannon <- vegan::diversity(temp_mat, index="shannon", MARGIN=2)
  d <- exp(shannon)
  diversity_all_bac_mat <- cbind(diversity_all_bac_mat, d)
}

# calculate means of subsamples
otu_diversity_all_bac <- rowMeans(diversity_all_bac_mat)

# put them all in one dataframe
alpha_diversity_bac_summary <- cbind(otu_diversity_all_bac, bDESIGN)


### F test for sample groups by location and compartment
alpha_bac_all_aov <- aov(log(otu_diversity_all_bac) ~ location_compartment, data=alpha_diversity_bac_summary)
alpha_bac_all_aov_sum <- summary(alpha_bac_all_aov)
# pander(alpha_bac_all_aov_sum, caption="Shannon diversity bacteria")

### pair-wise tests
alpha_bac_all_aov_tukey <- emmeans(alpha_bac_all_aov, "location_compartment")  # library(emmeans)
alpha_bac_all_aov_tukey_letters <- cld(alpha_bac_all_aov_tukey, Letter="abcdefghi", alpha=0.05)
# alpha_bac_all_aov_tukey_letters

### table
# stats
alpha_bac_all_aov_tukey_letters2 <- alpha_bac_all_aov_tukey_letters[1:7]
row.names(alpha_bac_all_aov_tukey_letters2) <- alpha_bac_all_aov_tukey_letters2[,1]
alpha_bac_all_aov_tukey_letters2 <- alpha_bac_all_aov_tukey_letters2[levels(bDESIGN$location_compartment),]
# means
alpha_bac_all_means <- aggregate(alpha_diversity_bac_summary$otu_diversity_all_bac, 
                                 list(alpha_diversity_bac_summary$location_compartment), mean)
row.names(alpha_bac_all_means) <- alpha_bac_all_means[,1]
alpha_bac_all_means <- alpha_bac_all_means[levels(bDESIGN$location_compartment),]
# table
alpha_bac_all_table <- cbind(Shannon=alpha_bac_all_means[,2], alpha_bac_all_aov_tukey_letters2[,2:7])
alpha_bac_all_table <- alpha_bac_all_table[,c(1,7)]

pander(alpha_bac_all_table)

write.csv(alpha_bac_all_table, file="tables/Table_S4_Alpha_stats_bacteria.csv")
```

### BX-effect in SOIL samples (reported in Table S8)

The table reports the F-test between WT and bx1 soil samples. To be consistent with the previous ANOVA statistics, we kept using the aov() function but de-facto this is a T-test comparing WT and bx1 samples.  

```{r a-div stats bacteria soil, warning=F, echo=F, message=F, eval=T}
# subsetting to soil samples, remove Changins soil samples (not sampled by genotype)
alpha_diversity_bac_summary_soil <- droplevels(alpha_diversity_bac_summary[alpha_diversity_bac_summary$compartment=="soil", ])
alpha_diversity_bac_summary_soil <- droplevels(alpha_diversity_bac_summary_soil[alpha_diversity_bac_summary_soil$genotype=="WT" |
                                                                                  alpha_diversity_bac_summary_soil$genotype=="bx1", ])

### F test for genotype (becomes a T test)
alpha_bac_soil_aov <- aov(log(otu_diversity_all_bac) ~ genotype, data=alpha_diversity_bac_summary_soil)
alpha_bac_soil_aov_sum <- summary(alpha_bac_soil_aov)
pander(alpha_bac_soil_aov_sum, caption="soil samples: Shannon diversity bacteria")

effect_size_bac_soil <- c("effect_size_bac_soil", round(alpha_bac_soil_aov_sum[[1]]$`Sum Sq`[1] / (alpha_bac_soil_aov_sum[[1]]$`Sum Sq`[1] + alpha_bac_soil_aov_sum[[1]]$`Sum Sq`[2]),5))
```

### BX-effect in RHIZOSPHERE samples (reported in Table S8)

The table reports the F-test between WT and bx1 rhizosphere samples. To be consistent with the previous ANOVA statistics, we kept using the aov() function but de-facto this is a T-test comparing WT and bx1 samples.  

```{r a-div stats bacteria rhizo, warning=F, echo=F, message=F, eval=T}
# subsetting to rhizo samples, remove Changins rhizo samples (not sampled by genotype)
alpha_diversity_bac_summary_rhizo <- droplevels(alpha_diversity_bac_summary[alpha_diversity_bac_summary$compartment=="rhizo", ])
alpha_diversity_bac_summary_rhizo <- droplevels(alpha_diversity_bac_summary_rhizo[alpha_diversity_bac_summary_rhizo$genotype=="WT" |
                                                                                  alpha_diversity_bac_summary_rhizo$genotype=="bx1", ])

### F test for genotype (becomes a T test)
alpha_bac_rhizo_aov <- aov(log(otu_diversity_all_bac) ~ genotype, data=alpha_diversity_bac_summary_rhizo)
alpha_bac_rhizo_aov_sum <- summary(alpha_bac_rhizo_aov)
pander(alpha_bac_rhizo_aov_sum, caption="rhizo samples: Shannon diversity bacteria")

effect_size_bac_rhizo <- c("effect_size_bac_rhizo", round(alpha_bac_rhizo_aov_sum[[1]]$`Sum Sq`[1] / (alpha_bac_rhizo_aov_sum[[1]]$`Sum Sq`[1] + alpha_bac_rhizo_aov_sum[[1]]$`Sum Sq`[2]),5))
effect_size_bac_rhizo
```

### BX-effect in ROOT samples (reported in Table S8)

The table reports the F-test between WT and bx1 root samples. To be consistent with the previous ANOVA statistics, we kept using the aov() function but de-facto this is a T-test comparing WT and bx1 samples.  

```{r a-div stats bacteria root, warning=F, echo=F, message=F, eval=T}
# subsetting to root samples, remove Changins root samples (not sampled by genotype)
alpha_diversity_bac_summary_root <- droplevels(alpha_diversity_bac_summary[alpha_diversity_bac_summary$compartment=="root", ])
alpha_diversity_bac_summary_root <- droplevels(alpha_diversity_bac_summary_root[alpha_diversity_bac_summary_root$genotype=="WT" |
                                                                                  alpha_diversity_bac_summary_root$genotype=="bx1", ])

### F test for genotype (becomes a T test)
alpha_bac_root_aov <- aov(log(otu_diversity_all_bac) ~ genotype, data=alpha_diversity_bac_summary_root)
alpha_bac_root_aov_sum <- summary(alpha_bac_root_aov)
pander(alpha_bac_root_aov_sum, caption="root samples: Shannon diversity bacteria")

effect_size_bac_root <- c("effect_size_bac_root", round(alpha_bac_root_aov_sum[[1]]$`Sum Sq`[1] / (alpha_bac_root_aov_sum[[1]]$`Sum Sq`[1] + alpha_bac_root_aov_sum[[1]]$`Sum Sq`[2]),5))
effect_size_bac_root
```

### plot as in Fig. S6 (all compartments)

```{r a-div plots bacteria, warning=F, echo=F, fig.width=5, fig.height=3.5, eval=T}
# library(ggplot2)
p_div_bac <- ggplot(alpha_diversity_bac_summary, aes(y=otu_diversity_all_bac, x=location, fill=background_genotype)) +
  geom_boxplot(position= position_dodge2(width=0.75, preserve="single")) + 
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size=12)) + 
  scale_fill_manual(values=level_cols_bg_gen) +
  facet_grid(~compartment, scales="free_x") +
  ylab("Shannon diversity index") +
  xlab("")

print(p_div_bac)
```


\pagebreak

## Shannon diversity Fungi

Method: we rarefy the fungal dataset to the sequencing depth of `r f_rare` and calculated Shannon diversity in each sample. This was repeated 100 times and the mean value from the 100 iterations was taken for statistical analysis between the different samples.  

### Analysis by location and compartment (reported in Table S4)

The table reports the Shannon diversity index together with the results form the pair-wise Tukey test results. Different letters are used for groups that differ significantly in their abundances (alpha=0.05).

```{r a-div stats fungi, warning=F, echo=F, message=F, eval=T}
diversity_all_fDAT_mat <- c()

for (i in 1:100){
  # generate rarefied community
  temp_mat <- t(rrarefy(t(fDAT), f_rare))
  # from Jost 2006: D=exp(H), exponential of shannon entropy
  shannon <- vegan::diversity(temp_mat, index="shannon", MARGIN=2)
  d <- exp(shannon)
  diversity_all_fDAT_mat <- cbind(diversity_all_fDAT_mat, d)
}

# calculate means of subsamples
otu_diversity_all_fun <- rowMeans(diversity_all_fDAT_mat)

# put them all in one dataframe
alpha_diversity_fDAT_summary <- cbind(otu_diversity_all_fun, fDESIGN)


### F test for sample groups by location and compartment
alpha_fDAT_all_aov <- aov(log(otu_diversity_all_fun) ~ location_compartment, data=alpha_diversity_fDAT_summary)
alpha_fDAT_all_aov_sum <- summary(alpha_fDAT_all_aov)
# pander(alpha_fDAT_all_aov_sum, caption="Shannon diversity fungi")

### pair-wise tests
alpha_fDAT_all_aov_tukey <- emmeans(alpha_fDAT_all_aov, "location_compartment")  # library(emmeans)
alpha_fDAT_all_aov_tukey_letters <- cld(alpha_fDAT_all_aov_tukey, Letter="abcdefghi", alpha=0.05)
# alpha_fDAT_all_aov_tukey_letters

### table
# stats
alpha_fDAT_all_aov_tukey_letters2 <- alpha_fDAT_all_aov_tukey_letters[1:7]
row.names(alpha_fDAT_all_aov_tukey_letters2) <- alpha_fDAT_all_aov_tukey_letters2[,1]
alpha_fDAT_all_aov_tukey_letters2 <- alpha_fDAT_all_aov_tukey_letters2[levels(fDESIGN$location_compartment),]
# means
alpha_fDAT_all_means <- aggregate(alpha_diversity_fDAT_summary$otu_diversity_all_fun, 
                                 list(alpha_diversity_fDAT_summary$location_compartment), mean)
row.names(alpha_fDAT_all_means) <- alpha_fDAT_all_means[,1]
alpha_fDAT_all_means <- alpha_fDAT_all_means[levels(fDESIGN$location_compartment),]
# table
alpha_fDAT_all_table <- cbind(Shannon=alpha_fDAT_all_means[,2], alpha_fDAT_all_aov_tukey_letters2[,2:7])
alpha_fDAT_all_table <- alpha_fDAT_all_table[,c(1,7)]

pander(alpha_fDAT_all_table)

write.csv(alpha_fDAT_all_table, file="tables/Table_S4_Alpha_stats_fungi.csv")
```

### BX-effect in SOIL samples (reported in Table S8)

The table reports the F-test between WT and bx1 soil samples. To be consistent with the previous ANOVA statistics, we kept using the aov() function but de-facto this is a T-test comparing WT and bx1 samples.  

```{r a-div stats fungi soil, warning=F, echo=F, message=F, eval=T}
# subsetting to soil samples, remove Changins soil samples (not sampled by genotype)
alpha_diversity_fDAT_summary_soil <- droplevels(alpha_diversity_fDAT_summary[alpha_diversity_fDAT_summary$compartment=="soil", ])
alpha_diversity_fDAT_summary_soil <- droplevels(alpha_diversity_fDAT_summary_soil[alpha_diversity_fDAT_summary_soil$genotype=="WT" |
                                                                                  alpha_diversity_fDAT_summary_soil$genotype=="bx1", ])

### F test for genotype (becomes a T test)
alpha_fDAT_soil_aov <- aov(log(otu_diversity_all_fun) ~ genotype, data=alpha_diversity_fDAT_summary_soil)
alpha_fDAT_soil_aov_sum <- summary(alpha_fDAT_soil_aov)
pander(alpha_fDAT_soil_aov_sum, caption="soil samples: Shannon diversity fungi")

effect_size_fun_soil <- c("effect_size_fun_soil", round(alpha_fDAT_soil_aov_sum[[1]]$`Sum Sq`[1] / (alpha_fDAT_soil_aov_sum[[1]]$`Sum Sq`[1] + alpha_fDAT_soil_aov_sum[[1]]$`Sum Sq`[2]),5))
effect_size_fun_soil
```

### BX-effect in RHIZOSPHERE samples (reported in Table S8)

The table reports the F-test between WT and bx1 rhizosphere samples. To be consistent with the previous ANOVA statistics, we kept using the aov() function but de-facto this is a T-test comparing WT and bx1 samples.  

```{r a-div stats fungi rhizo, warning=F, echo=F, message=F, eval=T}
# subsetting to rhizo samples, remove Changins rhizo samples (not sampled by genotype)
alpha_diversity_fDAT_summary_rhizo <- droplevels(alpha_diversity_fDAT_summary[alpha_diversity_fDAT_summary$compartment=="rhizo", ])
alpha_diversity_fDAT_summary_rhizo <- droplevels(alpha_diversity_fDAT_summary_rhizo[alpha_diversity_fDAT_summary_rhizo$genotype=="WT" |
                                                                                    alpha_diversity_fDAT_summary_rhizo$genotype=="bx1", ])

### F test for genotype (becomes a T test)
alpha_fDAT_rhizo_aov <- aov(log(otu_diversity_all_fun) ~ genotype, data=alpha_diversity_fDAT_summary_rhizo)
alpha_fDAT_rhizo_aov_sum <- summary(alpha_fDAT_rhizo_aov)
pander(alpha_fDAT_rhizo_aov_sum, caption="rhizo samples: Shannon diversity fungi")

effect_size_fun_rhizo <- c("effect_size_fun_rhizo", round(alpha_fDAT_rhizo_aov_sum[[1]]$`Sum Sq`[1] / (alpha_fDAT_rhizo_aov_sum[[1]]$`Sum Sq`[1] + alpha_fDAT_rhizo_aov_sum[[1]]$`Sum Sq`[2]),5))
effect_size_fun_rhizo
```

### BX-effect in ROOT samples (reported in Table S8)

The table reports the F-test between WT and bx1 root samples. To be consistent with the previous ANOVA statistics, we kept using the aov() function but de-facto this is a T-test comparing WT and bx1 samples.  

```{r a-div stats fungi root, warning=F, echo=F, message=F, eval=T, fig.height=3, fig.width=3.5}
# subsetting to root samples, remove Changins root samples (not sampled by genotype)
alpha_diversity_fDAT_summary_root <- droplevels(alpha_diversity_fDAT_summary[alpha_diversity_fDAT_summary$compartment=="root", ])
alpha_diversity_fDAT_summary_root <- droplevels(alpha_diversity_fDAT_summary_root[alpha_diversity_fDAT_summary_root$genotype=="WT" |
                                                                                  alpha_diversity_fDAT_summary_root$genotype=="bx1", ])

### F test for genotype (becomes a T test)
alpha_fDAT_root_aov <- aov(log(otu_diversity_all_fun) ~ genotype, data=alpha_diversity_fDAT_summary_root)
alpha_fDAT_root_aov_sum <- summary(alpha_fDAT_root_aov)
pander(alpha_fDAT_root_aov_sum, caption="root samples: Shannon diversity fungi")

effect_size_fun_root <- c("effect_size_fun_root", round(alpha_fDAT_root_aov_sum[[1]]$`Sum Sq`[1] / (alpha_fDAT_root_aov_sum[[1]]$`Sum Sq`[1] + alpha_fDAT_root_aov_sum[[1]]$`Sum Sq`[2]),5))
effect_size_fun_root

### plot
level_cols_loc_bg_gen2 <- level_cols_loc_bg_gen[c(2:5,8:11)]

par(mar=c(9,4,1,4))
bargraph.CI(alpha_diversity_fDAT_summary_root$location_background_genotype, alpha_diversity_fDAT_summary_root$otu_diversity_all_fun, 
            las=2, col=level_cols_loc_bg_gen2, cex.leg=0.7,
            ylab="Shannon diverstiy")
```
*CONCLUSION:* We find a consistant and significant enrichment in Shannon diversity in WT compared to bx1 root samples. This suggests that the secretion of BXs enriches fungal diversity in roots.    


### plot as in Fig. S6 (all compartments)

```{r a-div plots fungi, warning=F, echo=F, fig.width=5, fig.height=3.5, eval=T}
p_div_fun <- ggplot(alpha_diversity_fDAT_summary, aes(y=otu_diversity_all_fun, x=location, fill=background_genotype)) +
  geom_boxplot(position= position_dodge2(width=0.75, preserve="single")) + 
  theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size=12)) + 
  scale_fill_manual(values=level_cols_bg_gen) +
  facet_grid(~compartment, scales="free_x") +
  ylab("")+
  xlab("")

print(p_div_fun)
```


## Table S8 | Alpha diversity by compartments

```{r a-div stats table, echo=F, eval=T}
capture.output(pander(alpha_bac_soil_aov_sum, caption="soil samples: Shannon diversity bacteria"),
               effect_size_bac_soil,
               pander(alpha_bac_rhizo_aov_sum, caption="rhizo samples: Shannon diversity bacteria"),
               effect_size_bac_rhizo,
               pander(alpha_bac_root_aov_sum, caption="root samples: Shannon diversity bacteria"),
               effect_size_bac_root,
               pander(alpha_fDAT_soil_aov_sum, caption="soil samples: Shannon diversity fungi"),
               effect_size_fun_soil,
               pander(alpha_fDAT_rhizo_aov_sum, caption="rhizo samples: Shannon diversity fungi"),
               effect_size_fun_rhizo,
               pander(alpha_fDAT_root_aov_sum, caption="root samples: Shannon diversity fungi"),
               effect_size_fun_root,
               file="tables/Table_S8_Alpha_BX_statistics.doc")
```


## Figure S6 | Alpha diversity

```{r Figure and Table Alpha diversity, echo=F, warning=F, message=F, eval=T}
# library(cowplot)
prow <- plot_grid(p_div_bac + theme(legend.position="none"),
                  p_div_fun + theme(legend.position="none"),
                  align='vh', labels=c("A", "B"),
                  hjust=-1, ncol=2)

legend_b <- get_legend(p_div_bac + theme(legend.position="bottom") +
                       labs(fill="genotype (background)"))

p <- plot_grid( prow, legend_b, nrow=2, rel_heights=c(1, .1) )

pdf(file="figures/Fig_S6_alph_div.pdf", width=12, height=6)
p
noshow <- dev.off()
```




\pagebreak

# Beta diversity

For all beta diversity analyses we use the rarefied data as input and we employ Bray-Curtis distances for PERMANOVA and PCoA.

## Analysis by location and compartment (all samples)  

### Table S5 | PERMANOVA (all samples)

We use the function 'adonis()' from the package vegan and ask for compartment and location effects using the formula [~ compartment + location].    

```{r permanova compartment and location, warning=F, echo=F, eval=T}

# library(vegan)
bDAT_rare_dist <- vegdist(t(bDAT_rare), method = "bray")
bDAT_rare_dist_paov <- adonis(bDAT_rare_dist ~ compartment + location, data=bDESIGN[colnames(bDAT_rare),], permutations = 999)
pander(bDAT_rare_dist_paov$aov.tab[1:6], caption = "Bacteria")

fDAT_rare_dist <- vegdist(t(fDAT_rare), method="bray")
fDAT_rare_dist_paov <- adonis(fDAT_rare_dist ~ compartment + location, data=fDESIGN[colnames(fDAT_rare),], permutations = 999)
pander(fDAT_rare_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               pander(fDAT_rare_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S5_PERMANOVA_stats_all_samples.doc")

```

\pagebreak

### Figure 1 | PCoA (all samples)

We use the phyloseq package to perform the ordination analysis. 

```{r ordination all bacteria, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T, results="hide"}
# the phyloseq object was defined above
# bPHYSEQ
# source("functions/axis_label_pco.R")
# with color according to locations, shape according to sample type
bPHYSEQ_PCoA_bray <- phyloseq::ordinate(bPHYSEQ, method="PCoA", distance="bray")
p0 <- phyloseq::plot_ordination(bPHYSEQ, bPHYSEQ_PCoA_bray, shape="compartment", color="location", title="Bacteria")
p0 <- p0 + geom_point(size=3)#, alpha=0.75)
p0 <- p0 + scale_color_manual(values=level_cols_loc)
strivar <- axis_label_pco(bPHYSEQ_PCoA_bray)
p0 <- p0 + xlab(strivar[1]) + ylab(strivar[2])
bPCoA_all <- p0
par(mar=c(4,4,4,4), oma=c(4,4,4,4))
print(bPCoA_all)

```

```{r ordination all fungi, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T, results="hide"}
# the phyloseq object was defined above
# fPHYSEQ

# with color according to locations, shape according to sample type
fPHYSEQ_PCoA_bray <- phyloseq::ordinate(fPHYSEQ, method="PCoA", distance="bray")
p1 <- phyloseq::plot_ordination(fPHYSEQ, fPHYSEQ_PCoA_bray, shape="compartment", color="location", title="Fungi")
p1 <- p1 + geom_point(size=3)#, alpha=0.75)
p1 <- p1 + scale_color_manual(values=level_cols_loc)
strivar <- axis_label_pco(fPHYSEQ_PCoA_bray)
p1 <- p1 + xlab(strivar[1]) + ylab(strivar[2])
fPCoA_all <- p1
par(mar=c(4,4,4,4), oma=c(4,4,4,4))
print(fPCoA_all)

```

```{r Figure 1 PCoA all, echo=F, warning=F, eval=T}
# library(cowplot)
# https://stackoverflow.com/questions/13649473/add-a-common-legend-for-combined-ggplots
prow <- plot_grid(bPCoA_all 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  fPCoA_all 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
           align='vh',
           rel_widths=c(1,1),
           labels=c("", ""),
           hjust=-1,
           nrow=1
           )

legend_b <- get_legend(bPCoA_all
                       + theme(legend.position="right"))

Fig_1 <- plot_grid( prow, legend_b, ncol=2, rel_widths=c(1, .2))

pdf("figures/Fig_1_PCoA_all.pdf", width=9, height=3.7)
Fig_1
noshow <- dev.off()
```




\pagebreak

## Q1: Exudation effects by compartments? (all samples)   

To answer the first research question – How does BX exudation affect the different compartments? – we analyzed the three compartments separately. We conducted taxonomy, alpha- and beta diversity analyses and we limited these comparisons to WT and bx1 mutant lines, as these plant lines were present at all locations.

### SOIL samples | only use WT and bx1 lines, removing Changins soil samples (not sampled by genotype) 

```{r subsetting to soil samples, echo=F, warning=F, message=F, eval=T}
# the phyloseq object was defined above
# bPHYSEQ
# fPHYSEQ

### Bacteria
bPHYSEQ_soil <- prune_samples(sample_data(bPHYSEQ)$compartment=="soil", bPHYSEQ)
bPHYSEQ_soil <- prune_samples(sample_data(bPHYSEQ_soil)$genotype=="WT" | 
                                      sample_data(bPHYSEQ_soil)$genotype=="bx1", bPHYSEQ_soil)
# sample_data(bPHYSEQ_soil)$genotype
# sample_data(bPHYSEQ_soil)$location

### Fungi
fPHYSEQ_soil <- prune_samples(sample_data(fPHYSEQ)$compartment=="soil", fPHYSEQ)
fPHYSEQ_soil <- prune_samples(sample_data(fPHYSEQ_soil)$genotype=="WT" | 
                                sample_data(fPHYSEQ_soil)$genotype=="bx1", fPHYSEQ_soil)
```

#### Table S9 | PERMANOVA (soil samples)

We use the function 'adonis()' from the package vegan and ask for location and genotype effects using the formula [~ location + genotype].

```{r PERMANOVA soil samples to test for genotype, warning=F, echo=F, eval=T}
bDAT_rare_soil <- otu_table(bPHYSEQ_soil)
# head(bDAT_rare_soil)
# dim(bDAT_rare_soil)
# bDESIGN[colnames(bDAT_rare_soil),1:5]  
fDAT_rare_soil <- otu_table(fPHYSEQ_soil)

# library(vegan)
bDAT_rare_soil_dist <- vegdist(t(bDAT_rare_soil), method="bray")
bDAT_rare_soil_dist_paov <- adonis(bDAT_rare_soil_dist ~ location * genotype, data=bDESIGN[colnames(bDAT_rare_soil),], permutations=999)
pander(bDAT_rare_soil_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_soil_dist <- vegdist(t(fDAT_rare_soil), method="bray")
fDAT_rare_soil_dist_paov <- adonis(fDAT_rare_soil_dist ~ location * genotype, data=fDESIGN[colnames(fDAT_rare_soil),], permutations=999)
pander(fDAT_rare_soil_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_soil_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               file="tables/Table_S8_PERMANOVA_BX_stats_soil_bacteria.doc")
capture.output(pander(fDAT_rare_soil_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S8_PERMANOVA_BX_stats_soil_fungi.doc")
```

#### PCoA ordination (soil samples)

```{r ordination soil bacteria, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T}

# custom function for phyloseq ordination
# source("functions/ordination_paperBX.R")
bCAP_soil <- ordination_paperBX(PHYSEQ_object=bPHYSEQ_soil, 
                      model= ~genotype*location, method ="CAP", title_var=T,
                      shape="genotype", color="location", 
                      shape_values= c(15,0), color_values=level_cols_loc)


```

```{r ordination soil fungi, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T}

fCAP_soil <- ordination_paperBX(PHYSEQ_object=fPHYSEQ_soil, 
                      model= ~genotype*location, method ="CAP", 
                      shape="genotype", color="location", title_var=T,
                      shape_values= c(15,0), color_values=level_cols_loc)

```

```{r Figure PCoA soil, echo=F, warning=F, fig.height=3, fig.width=6, eval=T}
prow <- plot_grid(text_bacteria, text_fungi,
  bCAP_soil 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  fCAP_soil 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
           align='vh',
           rel_widths=c(1,1),rel_heights = c(0.2,1),
           labels=c("A", "B"),
           hjust=-1,
           nrow=2
           )

legend_b <- get_legend(bCAP_soil
                       + theme(legend.position="right"))

Fig_soil <- plot_grid( prow, legend_b, ncol=2, rel_widths=c(1, .2))

print(Fig_soil)
```


\pagebreak

### RHIZOSPHERE samples | only use WT and bx1 lines 

```{r subsetting to rhizosphere samples, echo=F, warning=F, message=F, eval=T}
### Bacteria
bPHYSEQ_rhizo <- prune_samples(sample_data(bPHYSEQ)$compartment=="rhizo", bPHYSEQ)
bPHYSEQ_rhizo <- prune_samples(sample_data(bPHYSEQ_rhizo)$genotype=="WT" | 
                                sample_data(bPHYSEQ_rhizo)$genotype=="bx1", bPHYSEQ_rhizo)
### Fungi
fPHYSEQ_rhizo <- prune_samples(sample_data(fPHYSEQ)$compartment=="rhizo", fPHYSEQ)
fPHYSEQ_rhizo <- prune_samples(sample_data(fPHYSEQ_rhizo)$genotype=="WT" | 
                                sample_data(fPHYSEQ_rhizo)$genotype=="bx1", fPHYSEQ_rhizo)
```

#### Table S8 | PERMANOVA (rhizosphere samples)

We use the function 'adonis()' from the package vegan and ask for location and genotype effects using the formula [~ location + genotype].

```{r PERMANOVA rhizo samples to test for genotype, warning=F, echo=F, eval=T}

bDAT_rare_rhizo <- otu_table(bPHYSEQ_rhizo)
fDAT_rare_rhizo <- otu_table(fPHYSEQ_rhizo)

# library(vegan)
bDAT_rare_rhizo_dist <- vegdist(t(bDAT_rare_rhizo), method="bray")
bDAT_rare_rhizo_dist_paov <- adonis(bDAT_rare_rhizo_dist ~ location * genotype, data=bDESIGN[colnames(bDAT_rare_rhizo),], permutations=999)
pander(bDAT_rare_rhizo_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_rhizo_dist <- vegdist(t(fDAT_rare_rhizo), method="bray")
fDAT_rare_rhizo_dist_paov <- adonis(fDAT_rare_rhizo_dist ~ location * genotype, data=fDESIGN[colnames(fDAT_rare_rhizo),], permutations=999)
pander(fDAT_rare_rhizo_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_rhizo_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               file="tables/Table_S8_PERMANOVA_BX_stats_rhizo_bacteria.doc")
capture.output(pander(fDAT_rare_rhizo_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S8_PERMANOVA_BX_stats_rhizo_fungi.doc")

```

```{r PERMANOVA Changins rhizo samples to test for genotype, warning=F, echo=F, eval=T}

# check for reviewer's comment on the difference between genotype impact on bacteria and fungi
bPHYSEQ_Ch_rhizo <- prune_samples(sample_data(bPHYSEQ_rhizo)$location == "Changins", bPHYSEQ_rhizo)
fPHYSEQ_Ch_rhizo <- prune_samples(sample_data(fPHYSEQ_rhizo)$location == "Changins", fPHYSEQ_rhizo)

bDAT_rare_Ch_rhizo <- otu_table(bPHYSEQ_Ch_rhizo)
fDAT_rare_Ch_rhizo <- otu_table(fPHYSEQ_Ch_rhizo)

# library(vegan)
bDAT_rare_Ch_rhizo_dist <- vegdist(t(bDAT_rare_Ch_rhizo), method="bray")
bDAT_rare_Ch_rhizo_dist_paov <- adonis(bDAT_rare_Ch_rhizo_dist ~ genotype, data=bDESIGN[colnames(bDAT_rare_Ch_rhizo),], permutations=999)
# library(pander)
pander(bDAT_rare_Ch_rhizo_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Ch_rhizo_dist <- vegdist(t(fDAT_rare_Ch_rhizo), method="bray")
fDAT_rare_Ch_rhizo_dist_paov <- adonis(fDAT_rare_Ch_rhizo_dist ~ genotype, data=fDESIGN[colnames(fDAT_rare_Ch_rhizo),], permutations=999)
pander(fDAT_rare_Ch_rhizo_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_Ch_rhizo_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               file="tables/Table_S8_PERMANOVA_BX_stats_Ch_rhizo_bacteria.doc")
capture.output(pander(fDAT_rare_Ch_rhizo_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S8_PERMANOVA_BX_stats_Ch_rhizo_fungi.doc")
```

#### PCoA ordination (rhizosphere samples)

```{r ordination rhizo bacteria, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T}

bCAP_rhizo <- ordination_paperBX(PHYSEQ_object=bPHYSEQ_rhizo, 
                      model= ~genotype*location, method ="CAP", 
                      shape="genotype", color="location", title_var=T, 
                      shape_values= c(15,0), color_values=level_cols_loc)


```

```{r ordination rhizo fungi, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T}

fCAP_rhizo <- ordination_paperBX(PHYSEQ_object=fPHYSEQ_rhizo, 
                      model= ~genotype*location, method ="CAP", 
                      shape="genotype", color="location", title_var=T, 
                      shape_values= c(15,0), color_values=level_cols_loc)

```


```{r Figure PCoA rhizo, echo=F, warning=F, fig.height=3, fig.width=6, eval=T}
prow <- plot_grid(bCAP_rhizo 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  fCAP_rhizo 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  align='vh',
                  rel_widths=c(1,1),
                  labels=c("A", "B"),
                  hjust=-1,
                  nrow=1
)

legend_b <- get_legend(bCAP_rhizo
                       + theme(legend.position="right"))

Fig_rhizo <- plot_grid( prow, legend_b, ncol=2, rel_widths=c(1, .2))
print(Fig_rhizo)
```


\pagebreak

### ROOT samples | only use WT and bx1 lines 

```{r subsetting to root samples, echo=F, warning=F, message=F, eval=T}
### Bacteria
bPHYSEQ_root <- prune_samples(sample_data(bPHYSEQ)$compartment=="root", bPHYSEQ)
bPHYSEQ_root <- prune_samples(sample_data(bPHYSEQ_root)$genotype=="WT" | 
                                 sample_data(bPHYSEQ_root)$genotype=="bx1", bPHYSEQ_root)
### Fungi
fPHYSEQ_root <- prune_samples(sample_data(fPHYSEQ)$compartment=="root", fPHYSEQ)
fPHYSEQ_root <- prune_samples(sample_data(fPHYSEQ_root)$genotype=="WT" | 
                                       sample_data(fPHYSEQ_root)$genotype=="bx1", fPHYSEQ_root)
```

#### Table S8 | PERMANOVA (root samples)

We use the function 'adonis()' from the package vegan and ask for location and genotype effects using the formula [~ location + genotype].  

```{r PERMANOVA root samples to test for genotype, warning=F, echo=F, eval=T}
bDAT_rare_root <- otu_table(bPHYSEQ_root)
fDAT_rare_root <- otu_table(fPHYSEQ_root)

# library(vegan)
bDAT_rare_root_dist <- vegdist(t(bDAT_rare_root), method="bray")
bDAT_rare_root_dist_paov <- adonis(bDAT_rare_root_dist ~ location * genotype, data=bDESIGN[colnames(bDAT_rare_root),], permutations=999)
pander(bDAT_rare_root_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_root_dist <- vegdist(t(fDAT_rare_root), method="bray")
fDAT_rare_root_dist_paov <- adonis(fDAT_rare_root_dist ~ location * genotype, data=fDESIGN[colnames(fDAT_rare_root),], permutations=999)
pander(fDAT_rare_root_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_root_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               file="tables/Table_S8_PERMANOVA_BX_stats_root_bacteria.doc")
capture.output(pander(fDAT_rare_root_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S8_PERMANOVA_BX_stats_root_fungi.doc")
```

```{r PERMANOVA Changins root samples to test for genotype, warning=F, echo=F, eval=T}

# check for reviewer's comment on the difference between genotype impact on bacteria and fungi
bPHYSEQ_Ch_root <- prune_samples(sample_data(bPHYSEQ_root)$location == "Changins", bPHYSEQ_root)
fPHYSEQ_Ch_root <- prune_samples(sample_data(fPHYSEQ_root)$location == "Changins", fPHYSEQ_root)

bDAT_rare_Ch_root <- otu_table(bPHYSEQ_Ch_root)
fDAT_rare_Ch_root <- otu_table(fPHYSEQ_Ch_root)

# library(vegan)
bDAT_rare_Ch_root_dist <- vegdist(t(bDAT_rare_Ch_root), method="bray")
bDAT_rare_Ch_root_dist_paov <- adonis(bDAT_rare_Ch_root_dist ~ genotype, data=bDESIGN[colnames(bDAT_rare_Ch_root),], permutations=999)
# library(pander)
pander(bDAT_rare_Ch_root_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Ch_root_dist <- vegdist(t(fDAT_rare_Ch_root), method="bray")
fDAT_rare_Ch_root_dist_paov <- adonis(fDAT_rare_Ch_root_dist ~ genotype, data=fDESIGN[colnames(fDAT_rare_Ch_root),], permutations=999)
pander(fDAT_rare_Ch_root_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_Ch_root_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               file="tables/Table_S8_PERMANOVA_BX_stats_Ch_root_bacteria.doc")
capture.output(pander(fDAT_rare_Ch_root_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S8_PERMANOVA_BX_stats_Ch_root_fungi.doc")
```


#### PCoA ordination (root samples)

```{r ordination root bacteria, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T}

bCAP_root <- ordination_paperBX(PHYSEQ_object=bPHYSEQ_root, 
                      model= ~genotype*location, method ="CAP", 
                      shape="genotype", color="location", title_var=T, 
                      shape_values= c(15,0), color_values=level_cols_loc)

```


```{r ordination root fungi, echo=F, warning=F, message=F, fig.height=3, fig.width=5, eval=T}

fCAP_root <- ordination_paperBX(PHYSEQ_object=fPHYSEQ_root, 
                      model= ~genotype*location, method ="CAP", 
                      shape="genotype", color="location", title_var=T, 
                      shape_values= c(15,0), color_values=level_cols_loc)

```

```{r Figure PCoA root, echo=F, warning=F, fig.height=3, fig.width=6, eval=T}
prow <- plot_grid(bCAP_root 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  fCAP_root 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  align='vh',
                  rel_widths=c(1,1),
                  labels=c("A", "B"),
                  hjust=-1,
                  nrow=1
)

legend_b <- get_legend(bCAP_root
                       + theme(legend.position="right"))

Fig_root <- plot_grid( prow, legend_b, ncol=2, rel_widths=c(1, .2))
print(Fig_root)
```


#### Figure 2 | combined


```{r Figure CAP combined, echo=F, warning=F, fig.height=3, fig.width=6, eval=T}

prow_All <- plot_grid(bCAP_soil 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  fCAP_soil 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  bCAP_rhizo 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  fCAP_rhizo 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  bCAP_root 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
                  fCAP_root 
                  + theme(legend.position="none")
                  + panel_border(colour="black"),
           align='vh',
           rel_widths=c(1,1),
           labels=c("A", " ", "B", " ", "C", " "),
           hjust=-1,
           nrow=3,
           ncol=2
           )

legend_b <- get_legend(bCAP_rhizo
                       + theme(legend.position="right"))

Fig_all <- plot_grid(prow_All, legend_b, rel_widths=c(1, .2))

#Fig_all <- Fig_all + plot_annotation(
 # subtitle = 'model = ~genotype*location',
  #caption = 'model = ~genotype*location')

pdf("figures/Fig_2_CAP_by_compartments.pdf", width=9, height=11)
Fig_all
noshow <- dev.off()


```




\pagebreak

## Q2: genetic background vs mutation?  (Reckenholz samples)  

To answer the second research question – How do the genetic backgrounds compare in their BX exudation effects on microbial communities? – we studied the WT and bx1 mutant lines of the two genetic backgrounds B73 and W22 in the Reckenholz experiment. 

```{r subsetting to Re samples, echo=F, warning=F, message=F, eval=T}
### Bacteria
bPHYSEQ_Re_ <- prune_samples(sample_data(bPHYSEQ)$location=="Reckenholz", bPHYSEQ)
# bPHYSEQ_Re_ <- prune_samples(sample_data(bPHYSEQ_Re_)$compartment=="root", bPHYSEQ_Re_)
bPHYSEQ_Re_ <- prune_samples(sample_data(bPHYSEQ_Re_)$compartment=="root" | 
                                 sample_data(bPHYSEQ_Re_)$compartment=="rhizo", bPHYSEQ_Re_)
# subsets of root and rhizo compartments
bPHYSEQ_Re_root <- prune_samples(sample_data(bPHYSEQ_Re_)$compartment=="root", bPHYSEQ_Re_)
bPHYSEQ_Re_rhizo <- prune_samples(sample_data(bPHYSEQ_Re_)$compartment=="rhizo", bPHYSEQ_Re_)

# sample_data(bPHYSEQ_Re_)$location
# sample_data(bPHYSEQ_Re_)$compartment

### Fungi
fPHYSEQ_Re_ <- prune_samples(sample_data(fPHYSEQ)$location=="Reckenholz", fPHYSEQ)
fPHYSEQ_Re_ <- prune_samples(sample_data(fPHYSEQ_Re_)$compartment=="root" | 
                              sample_data(fPHYSEQ_Re_)$compartment=="rhizo", fPHYSEQ_Re_)
# subsets of root and rhizo compartments
fPHYSEQ_Re_root <- prune_samples(sample_data(fPHYSEQ_Re_)$compartment=="root", fPHYSEQ_Re_)
fPHYSEQ_Re_rhizo <- prune_samples(sample_data(fPHYSEQ_Re_)$compartment=="rhizo", fPHYSEQ_Re_)
```


### Table S10 | PERMANOVA

We use the function 'adonis()' from the package vegan and ask for background and genotype effects together for root and rhizosphere compartments using the formula [~ compartment + background + genotype].  

#### factorial PERMANOVA [~ compartment + background + genotype]

```{r PERMANOVA Re samples to test for background and genotype, warning=F, echo=F, eval=T}
bDAT_rare_Re_ <- otu_table(bPHYSEQ_Re_)
fDAT_rare_Re_ <- otu_table(fPHYSEQ_Re_)

# library(vegan)
bDAT_rare_Re_dist <- vegdist(t(bDAT_rare_Re_), method="bray")
bDAT_rare_Re_dist_paov <- adonis(bDAT_rare_Re_dist ~ compartment + background + genotype, data=bDESIGN[colnames(bDAT_rare_Re_),], permutations=999)
pander(bDAT_rare_Re_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Re_dist <- vegdist(t(fDAT_rare_Re_), method="bray")
fDAT_rare_Re_dist_paov <- adonis(fDAT_rare_Re_dist ~ compartment + background + genotype, data=fDESIGN[colnames(fDAT_rare_Re_),], permutations=999)
pander(fDAT_rare_Re_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_Re_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               file="tables/Table_S10_PERMANOVA_background_genotype_bacteria.doc")
capture.output(pander(fDAT_rare_Re_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S10_PERMANOVA_background_genotype_fungi.doc")
```

#### pairwise PERMANOVA Rhizosphere [~ background * genotype]

```{r pairwise PERMANOVA Re samples rhizo, warning=F, echo=F, eval=T}
## Subsetting to Re rhizo samples
bDAT_rare_Re_rhizo <- otu_table(bPHYSEQ_Re_rhizo)
bDAT_rare_Re_rhizo <- bDAT_rare_Re_rhizo[rowSums(bDAT_rare_Re_rhizo) > 0,]
bDESIGN_rhizo_Re_ <- sample_data(bPHYSEQ_Re_rhizo)
  
fDAT_rare_Re_rhizo <- otu_table(fPHYSEQ_Re_rhizo)
fDAT_rare_Re_rhizo <- fDAT_rare_Re_rhizo[rowSums(fDAT_rare_Re_rhizo) > 0,]
fDESIGN_rhizo_Re_ <- sample_data(fPHYSEQ_Re_rhizo)

## Subsetting to lowest nr of samples to have balanced sample numbers
set.seed(74500)     # 74500=zip code of Evian
bDESIGN_rhizo_Re_sub <- bDESIGN_rhizo_Re_ %>% group_by(genotype_background) %>% sample_n(size=6) %>% as.data.frame()
# as.data.frame(bDESIGN_rhizo_Re_sub)
rownames(bDESIGN_rhizo_Re_sub) <- bDESIGN_rhizo_Re_sub$sample_ID_bac
samples_overlap <- rownames(bDESIGN_rhizo_Re_sub)[rownames(bDESIGN_rhizo_Re_sub) %in% colnames(bDAT_rare_Re_rhizo)]
bDAT_rare_Re_rhizo_sub <- bDAT_rare_Re_rhizo[, samples_overlap ]

set.seed(74500)     # 74500=zip code of Evian
fDESIGN_rhizo_Re_sub <- fDESIGN_rhizo_Re_ %>% group_by(genotype_background) %>% sample_n(size=5) %>% as.data.frame()
# as.data.frame(fDESIGN_rhizo_Re_sub)
rownames(fDESIGN_rhizo_Re_sub) <- fDESIGN_rhizo_Re_sub$sample_ID_fun
samples_overlap <- rownames(fDESIGN_rhizo_Re_sub)[rownames(fDESIGN_rhizo_Re_sub) %in% colnames(fDAT_rare_Re_rhizo)]
fDAT_rare_Re_rhizo_sub <- fDAT_rare_Re_rhizo[, samples_overlap ]

## pairwise PERMANOVA Rhizosphere [~ background * genotype]
# source("functions/fun_pairwise_adonis.R")
bDAT_rare_Re_rhizo_dist_PW_adonis <- pairwise.adonis(t(bDAT_rare_Re_rhizo_sub), bDESIGN[colnames(bDAT_rare_Re_rhizo_sub), ]$genotype_background, sim.method="bray", p.adjust.m="bonferroni")
pander(bDAT_rare_Re_rhizo_dist_PW_adonis[,c(1:5)], caption="Rhizosphere bacteria")

fDAT_rare_Re_rhizo_dist_PW_adonis <- pairwise.adonis(t(fDAT_rare_Re_rhizo_sub), fDESIGN[colnames(fDAT_rare_Re_rhizo_sub), ]$genotype_background, sim.method="bray", p.adjust.m="bonferroni")
pander(fDAT_rare_Re_rhizo_dist_PW_adonis[,c(1:5)], caption="Rhizosphere fungi")

capture.output(pander(bDAT_rare_Re_rhizo_dist_PW_adonis[,c(1:5)], caption="Bacteria"), 
               file="tables/Table_S10_PERMANOVA_PW_rhizo_background_genotype_bacteria.doc")

capture.output(pander(fDAT_rare_Re_rhizo_dist_PW_adonis[,c(1:5)], caption="Fungi"),
               file="tables/Table_S10_PERMANOVA_PW_rhizo_background_genotype_fungi.doc")


# Permanova (check for reviewer)

bDAT_rare_Re_rhizo_dist <- vegdist(t(bDAT_rare_Re_rhizo), method="bray")
bDAT_rare_Re_rhizo_dist_paov <- adonis(bDAT_rare_Re_rhizo_dist ~ background + genotype, data=bDESIGN[colnames(bDAT_rare_Re_rhizo),], permutations=999)
pander(bDAT_rare_Re_rhizo_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Re_rhizo_dist <- vegdist(t(fDAT_rare_Re_rhizo), method="bray")
fDAT_rare_Re_rhizo_dist_paov <- adonis(fDAT_rare_Re_rhizo_dist ~ background + genotype, data=fDESIGN[colnames(fDAT_rare_Re_rhizo),], permutations=999)
pander(fDAT_rare_Re_rhizo_dist_paov$aov.tab[1:6], caption="Fungi")

```

#### pairwise PERMANOVA Root [~ background * genotype]

```{r pairwise PERMANOVA Re samples root, warning=F, echo=F, eval=T}
## Subsetting to Re root samples
bDAT_rare_Re_root <- otu_table(bPHYSEQ_Re_root)
bDAT_rare_Re_root <- bDAT_rare_Re_root[rowSums(bDAT_rare_Re_root) > 0,]
bDESIGN_root_Re_ <- sample_data(bPHYSEQ_Re_root)

fDAT_rare_Re_root <- otu_table(fPHYSEQ_Re_root)
fDAT_rare_Re_root <- fDAT_rare_Re_root[rowSums(fDAT_rare_Re_root) > 0,]
fDESIGN_root_Re_ <- sample_data(fPHYSEQ_Re_root)

## Subsetting to lowest nr of samples to have balanced sample numbers
set.seed(74500)     # 74500=zip code of Evian
bDESIGN_root_Re_sub <- bDESIGN_root_Re_ %>% group_by(genotype_background) %>% sample_n(size=6) %>% as.data.frame()
# as.data.frame(bDESIGN_root_Re_sub)
rownames(bDESIGN_root_Re_sub) <- bDESIGN_root_Re_sub$sample_ID_bac
samples_overlap <- rownames(bDESIGN_root_Re_sub)[rownames(bDESIGN_root_Re_sub) %in% colnames(bDAT_rare_Re_root)]
bDAT_rare_Re_root_sub <- bDAT_rare_Re_root[, samples_overlap ]

set.seed(74500)     # 74500=zip code of Evian
fDESIGN_root_Re_sub <- fDESIGN_root_Re_ %>% group_by(genotype_background) %>% sample_n(size=5) %>% as.data.frame()
# as.data.frame(fDESIGN_root_Re_sub)
rownames(fDESIGN_root_Re_sub) <- fDESIGN_root_Re_sub$sample_ID_fun
samples_overlap <- rownames(fDESIGN_root_Re_sub)[rownames(fDESIGN_root_Re_sub) %in% colnames(fDAT_rare_Re_root)]
fDAT_rare_Re_root_sub <- fDAT_rare_Re_root[, samples_overlap ]

## pairwise PERMANOVA Root [~ background * genotype]
bDAT_rare_Re_root_dist_PW_adonis <- pairwise.adonis(t(bDAT_rare_Re_root_sub), bDESIGN[colnames(bDAT_rare_Re_root_sub), ]$genotype_background, sim.method="bray", p.adjust.m="bonferroni")
pander(bDAT_rare_Re_root_dist_PW_adonis[,c(1:5)], caption="Root bacteria")

fDAT_rare_Re_root_dist_PW_adonis <- pairwise.adonis(t(fDAT_rare_Re_root_sub), fDESIGN[colnames(fDAT_rare_Re_root_sub), ]$genotype_background, sim.method="bray", p.adjust.m="bonferroni")
pander(fDAT_rare_Re_root_dist_PW_adonis[,c(1:5)], caption="Root fungi")

capture.output(pander(bDAT_rare_Re_root_dist_PW_adonis[,c(1:5)], caption="Bacteria"), 
               file="tables/Table_S10_PERMANOVA_PW_root_background_genotype_bacteria.doc")

capture.output(pander(fDAT_rare_Re_root_dist_PW_adonis[,c(1:5)], caption="Fungi"),
               file="tables/Table_S10_PERMANOVA_PW_root_background_genotype_fungi.doc")

# Permanova (check for reviewer)

bDAT_rare_Re_root_dist <- vegdist(t(bDAT_rare_Re_root), method="bray")
bDAT_rare_Re_root_dist_paov <- adonis(bDAT_rare_Re_root_dist ~ background + genotype, data=bDESIGN[colnames(bDAT_rare_Re_root),], permutations=999)
pander(bDAT_rare_Re_root_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Re_root_dist <- vegdist(t(fDAT_rare_Re_root), method="bray")
fDAT_rare_Re_root_dist_paov <- adonis(fDAT_rare_Re_root_dist ~ background + genotype, data=fDESIGN[colnames(fDAT_rare_Re_root),], permutations=999)
pander(fDAT_rare_Re_root_dist_paov$aov.tab[1:6], caption="Fungi")


```

### CAP (Reckenholz: background and genotype), each compartment for bacteria and fungi

```{r LEGEND combined Figure PCoA ZH, echo=F, warning=F, fig.height=3, fig.width=6, eval=T}
pdf("figures/Fig_3A_CAP_background_legend.pdf", width=5, height=5)
mat2 <- cbind(c(1,1,1,1,2,2,2,2), c(1,1.5,2,2.5,1,1.5,2,2.5,1,1.5,2,2.5))
cols <- c("palegreen3", "gold2", "palegreen3", "gold2","palegreen3", "gold2", "palegreen3", "gold2")
names(cols) <- c("B73", "bx1(B73)", "W22", "bx1(W22)")
pchlegend <- c(2,2,17,17, 0,0,15,15)
plot(mat2, col=cols, pch=pchlegend, cex=5, lwd=4, xlim=c(.5,5), ylim=c(.5,3.5), frame=F, axes=F, xlab=NA, ylab=NA)
# points(mat2+0.2, col=cols, pch=4, cex=5)
text(1:2, rep(3, 4), label=c("Rhizosphere","Root"), adj=0.1, cex=2, srt=45, xpd=T)
text(rep(2.5, 4), c(1,1.5,2,2.5), label=c( "bx1(W22)", "W22", "bx1(B73)","B73"), adj=0,cex=2)
noshow <- dev.off()
```


#### Root bacteria

```{r ordination root bacteria: Reckenholz, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}

bCAP_Re_rhizo <- ordination_paperBX(PHYSEQ_object=bPHYSEQ_Re_rhizo, 
                      model= ~background*genotype, method ="CAP", 
                      shape="background", color="genotype", 
                      shape_values= c(16,1), title_var=T,
                      color_values=c("gold2","palegreen3"))

```

#### Root fungi

```{r ordination root fungi: Reckenholz, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}

fCAP_Re_rhizo <- ordination_paperBX(PHYSEQ_object=fPHYSEQ_Re_rhizo, 
                      model= ~background*genotype, method ="CAP", 
                      shape="background", color="genotype", 
                      shape_values= c(16,1), title_var=T,
                      color_values=c("gold2","palegreen3"))

```


#### Rhizosphere bacteria

```{r ordination rhizo bacteria: Reckenholz, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}


bCAP_Re_root <- ordination_paperBX(PHYSEQ_object=bPHYSEQ_Re_root, 
                      model= ~background*genotype, method ="CAP",
                      shape="background", color="genotype", 
                      shape_values= c(16,1),title_var=T, 
                      color_values=c("gold2","palegreen3"))

```

#### Rhizosphere fungi

```{r ordination rhizo fungi: Reckenholz, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}

fCAP_Re_root <- ordination_paperBX(PHYSEQ_object=fPHYSEQ_Re_root, 
                      model= ~background*genotype, method ="CAP", 
                      shape="background", color="genotype", 
                      shape_values= c(16,1),title_var=T, 
                      color_values=c("gold2","palegreen3"))

```

### Figure 3 | CAP ordination (review 11.20)

```{r Figure 3 CAP Re background and genotype compartments separate, echo=F, warning=F, eval=T}

text_model = text_graph("CAP on Bray-Curtis distance. Model = ~genotype/genetic background", size=4)

bCAP_Re_rhizo <- bCAP_Re_rhizo + #theme(legend.position="none") +
  panel_border(colour="black")
fCAP_Re_rhizo <- fCAP_Re_rhizo + theme(legend.position="none") + panel_border(colour="black")
bCAP_Re_root <- bCAP_Re_root + theme(legend.position="none") + panel_border(colour="black")
fCAP_Re_root <- fCAP_Re_root + theme(legend.position="none") + panel_border(colour="black")

p <- text_null + text_bacteria + text_fungi +
  text_rhizo + bCAP_Re_rhizo + fCAP_Re_rhizo +
  text_root + bCAP_Re_root + fCAP_Re_root +
  plot_layout(ncol=3, nrow=3, heights=c(.2,1,1), widths = c(.2, 1.3, 1.3)) +
  plot_layout(guides = 'collect') + plot_annotation(
  #subtitle = 'model = ~genotype*genetic background',
  #caption = 'model = ~genotype*genetic background',
  tag_levels = list(c('A', '') )) & 
  theme(plot.tag = element_text(face = "bold"))

# labels
pdf("figures/Fig_3_CAP_Reckenholz.pdf", width=10, height=9)
p
noshow <- dev.off()

```


\pagebreak

## Q3: various BX biosynthesis mutants? (Aurora samples)  

We approached the third research question – How do various mutations in the BX biosynthesis pathway compare in their impacts on rhizosphere and root microbial communities? – by comparing the wildtype W22 to the mutant lines bx1, bx2 and bx6 in the experiment conducted in Aurora. 

```{r subsetting to Au samples, echo=F, warning=F, message=F, eval=T}
### Bacteria
bPHYSEQ_Au_ <- prune_samples(sample_data(bPHYSEQ)$location=="Aurora", bPHYSEQ)
bPHYSEQ_Au_ <- prune_samples(sample_data(bPHYSEQ_Au_)$compartment=="root" | 
                                 sample_data(bPHYSEQ_Au_)$compartment=="rhizo", bPHYSEQ_Au_)
# # subsets of root and rhizo compartments
bPHYSEQ_Au_root <- prune_samples(sample_data(bPHYSEQ_Au_)$compartment=="root", bPHYSEQ_Au_)
bPHYSEQ_Au_rhizo <- prune_samples(sample_data(bPHYSEQ_Au_)$compartment=="rhizo", bPHYSEQ_Au_)

# sample_data(bPHYSEQ_Au_)$location
# sample_data(bPHYSEQ_Au_)$compartment

### Fungi
fPHYSEQ_Au_ <- prune_samples(sample_data(fPHYSEQ)$location=="Aurora", fPHYSEQ)
fPHYSEQ_Au_ <- prune_samples(sample_data(fPHYSEQ_Au_)$compartment=="root" | 
                              sample_data(fPHYSEQ_Au_)$compartment=="rhizo", fPHYSEQ_Au_)
# # subsets of root and rhizo compartments
fPHYSEQ_Au_root <- prune_samples(sample_data(fPHYSEQ_Au_)$compartment=="root", fPHYSEQ_Au_)
fPHYSEQ_Au_rhizo <- prune_samples(sample_data(fPHYSEQ_Au_)$compartment=="rhizo", fPHYSEQ_Au_)
```

### Figure 4 | CAP (Aurora: various mutants) - both compartment separated

```{r CAP root bacteria Aurora, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}

bCAP_Au_rhizo <- ordination_paperBX(PHYSEQ_object=bPHYSEQ_Au_rhizo, 
                      model= ~genotype, method ="CAP", 
                       color="genotype", 
                      shape_values= c(16,1),title_var=T, 
                      color_values=c("gold2","palegreen3", "palegreen1", "palegreen4"))

```

```{r CAP rhizo bacteria Aurora, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}

fCAP_Au_rhizo <- ordination_paperBX(PHYSEQ_object=fPHYSEQ_Au_rhizo, 
                      model= ~genotype, method ="CAP", 
                       color="genotype", 
                      shape_values= c(16,1),title_var=T, 
                      color_values=c("gold2","palegreen3", "palegreen1", "palegreen4"))

```

```{r CAP root fungi Aurora, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}

bCAP_Au_root <- ordination_paperBX(PHYSEQ_object=bPHYSEQ_Au_root, 
                      model= ~genotype, method ="CAP", 
                       color="genotype", 
                      shape_values= c(16,1),title_var=T, 
                      color_values=c("gold2","palegreen3", "palegreen1", "palegreen4"))
```

```{r CAP rhizo fungi Aurora, warning=F, echo=F, results="asis", warning=F, fig.align="center", fig.height=3, fig.width=5, eval=T}

fCAP_Au_root <- ordination_paperBX(PHYSEQ_object=fPHYSEQ_Au_root, 
                      model= ~genotype, method ="CAP", 
                       color="genotype", 
                      shape_values= c(16,1),title_var=T, 
                      color_values=c("gold2","palegreen3", "palegreen1", "palegreen4"))

```




### Figure 4 | CAP Aurora (review 11.20)

```{r Figure 4 CAP Re background and genotype compartments separate, echo=F, warning=F, eval=T}

bCAP_Au_rhizo <- bCAP_Au_rhizo + #theme(legend.position="none") +
  panel_border(colour="black")
fCAP_Au_rhizo <- fCAP_Au_rhizo + theme(legend.position="none") + panel_border(colour="black")
bCAP_Au_root <- bCAP_Au_root + theme(legend.position="none") + panel_border(colour="black")
fCAP_Au_root <- fCAP_Au_root + theme(legend.position="none") + panel_border(colour="black")

prow <- plot_grid(text_null, text_bacteria, text_fungi, 
              text_rhizo, bCAP_Au_rhizo , fCAP_Au_rhizo ,
           text_root, bCAP_Au_root, fCAP_Au_root ,
           align='vh',
           labels=c("A", "", "", ""),
           hjust=-1,
           nrow=3,  rel_widths=c(.7,1, 1), rel_heights =c(.4,1,1)
           )

p <- text_null + text_bacteria + text_fungi +
  text_rhizo + bCAP_Au_rhizo + fCAP_Au_rhizo +
  text_root + bCAP_Au_root + fCAP_Au_root +
  plot_layout(ncol=3, nrow=3, heights=c(.2,1,1), widths = c(.2, 1.3, 1.3)) +
  plot_layout(guides = 'collect') + plot_annotation(
 # subtitle = 'model = ~genotype',
 # caption = 'model = ~genotype',
 tag_levels = list(c('A', '') )) & 
  theme(plot.tag = element_text(face = "bold"))

pdf("figures/Fig_4_CAP_Aurora.pdf", width=10, height=9)
p
noshow <- dev.off()


```


### Table S12 | PERMANOVA

We use the function 'adonis()' from the package vegan and ask for genotype effects between the mutants for both the root and rhizosphere compartments using the formula [~ compartment + genotype].  

#### factorial PERMANOVA [~ compartment + genotype]

```{r PERMANOVA Aurora samples to test for genotype effect size, warning=F, echo=F, eval=T}
bDAT_rare_Au_ <- otu_table(bPHYSEQ_Au_)
fDAT_rare_Au_ <- otu_table(fPHYSEQ_Au_)

# library(vegan)
bDAT_rare_Au_dist <- vegdist(t(bDAT_rare_Au_), method="bray")
bDAT_rare_Au_dist_paov <- adonis(bDAT_rare_Au_dist ~ compartment + genotype, data=bDESIGN[colnames(bDAT_rare_Au_),], permutations=999)
pander(bDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Au_dist <- vegdist(t(fDAT_rare_Au_), method="bray")
fDAT_rare_Au_dist_paov <- adonis(fDAT_rare_Au_dist ~ compartment + genotype, data=fDESIGN[colnames(fDAT_rare_Au_),], permutations=999)
pander(fDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Fungi")

bDAT_rare_Au_dist <- vegdist(t(bDAT_rare_Au_), method="bray")
bDAT_rare_Au_dist_paov <- adonis(bDAT_rare_Au_dist ~ compartment * genotype, data=bDESIGN[colnames(bDAT_rare_Au_),], permutations=999)
pander(bDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Au_dist <- vegdist(t(fDAT_rare_Au_), method="bray")
fDAT_rare_Au_dist_paov <- adonis(fDAT_rare_Au_dist ~ compartment * genotype, data=fDESIGN[colnames(fDAT_rare_Au_),], permutations=999)
pander(fDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Fungi")

capture.output(pander(bDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Bacteria"), 
               file="tables/Table_S12_PERMANOVA_mutants_bacteria.doc")
capture.output(pander(fDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Fungi"),
               file="tables/Table_S12_PERMANOVA_mutants_fungi.doc")



bDAT_rare_Au_root <- otu_table(bPHYSEQ_Au_root)
fDAT_rare_Au_root <- otu_table(fPHYSEQ_Au_root)
bDAT_rare_Au_rhizo <- otu_table(bPHYSEQ_Au_rhizo)
fDAT_rare_Au_rhizo <- otu_table(fPHYSEQ_Au_rhizo)


# library(vegan)
bDAT_rare_Au_dist <- vegdist(t(bDAT_rare_Au_root), method="bray")
bDAT_rare_Au_dist_paov <- adonis(bDAT_rare_Au_dist ~ genotype, data=bDESIGN[colnames(bDAT_rare_Au_root),], permutations=999)
pander(bDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Au_dist <- vegdist(t(fDAT_rare_Au_root), method="bray")
fDAT_rare_Au_dist_paov <- adonis(fDAT_rare_Au_dist ~ genotype, data=fDESIGN[colnames(fDAT_rare_Au_root),], permutations=999)
pander(fDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Fungi")

bDAT_rare_Au_dist <- vegdist(t(bDAT_rare_Au_rhizo), method="bray")
bDAT_rare_Au_dist_paov <- adonis(bDAT_rare_Au_dist ~ genotype, data=bDESIGN[colnames(bDAT_rare_Au_rhizo),], permutations=999)
pander(bDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Au_dist <- vegdist(t(fDAT_rare_Au_rhizo), method="bray")
fDAT_rare_Au_dist_paov <- adonis(fDAT_rare_Au_dist ~ genotype, data=fDESIGN[colnames(fDAT_rare_Au_rhizo),], permutations=999)
pander(fDAT_rare_Au_dist_paov$aov.tab[1:6], caption="Fungi")

```



#### pairwise PERMANOVA Rhizosphere [~ genotype]

```{r pairwise PERMANOVA Aurora samples rhizo, warning=F, echo=F, eval=T}
bDAT_rare_Au_rhizo <- otu_table(bPHYSEQ_Au_rhizo)
bDAT_rare_Au_rhizo <- bDAT_rare_Au_rhizo[rowSums(bDAT_rare_Au_rhizo) > 0,]
fDAT_rare_Au_rhizo <- otu_table(fPHYSEQ_Au_rhizo)
fDAT_rare_Au_rhizo <- fDAT_rare_Au_rhizo[rowSums(fDAT_rare_Au_rhizo) > 0,]

# pairwise comparisons
bDAT_rare_Au_rhizo_dist_PW_adonis <- pairwise.adonis(t(bDAT_rare_Au_rhizo), bDESIGN[colnames(bDAT_rare_Au_rhizo), ]$genotype, sim.method="bray", p.adjust.m="bonferroni")
pander(bDAT_rare_Au_rhizo_dist_PW_adonis[,c(1:5)], caption="Rhizosphere bacteria")
fDAT_rare_Au_rhizo_dist_PW_adonis <- pairwise.adonis(t(fDAT_rare_Au_rhizo), fDESIGN[colnames(fDAT_rare_Au_rhizo), ]$genotype, sim.method="bray", p.adjust.m="bonferroni")
pander(fDAT_rare_Au_rhizo_dist_PW_adonis[,c(1:5)], caption="Rhizosphere fungi")

capture.output(pander(bDAT_rare_Au_rhizo_dist_PW_adonis[,c(1:5)], caption="Rhizosphere bacteria"), 
               file="tables/Table_S12_PERMANOVA_PW_rhizo_mutants_bacteria.doc")
capture.output(pander(fDAT_rare_Au_rhizo_dist_PW_adonis[,c(1:5)], caption="Rhizosphere fungi"),
               file="tables/Table_S12_PERMANOVA_PW_rhizo_mutants_fungi.doc")

bDAT_rare_Au_rhizo_dist <- vegdist(t(bDAT_rare_Au_rhizo), method="bray")
bDAT_rare_Au_rhizo_dist_paov <- adonis(bDAT_rare_Au_rhizo_dist ~ genotype, data=bDESIGN[colnames(bDAT_rare_Au_rhizo),], permutations=999)
pander(bDAT_rare_Au_rhizo_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Au_rhizo_dist <- vegdist(t(fDAT_rare_Au_rhizo), method="bray")
fDAT_rare_Au_rhizo_dist_paov <- adonis(fDAT_rare_Au_rhizo_dist ~ genotype, data=fDESIGN[colnames(fDAT_rare_Au_rhizo),], permutations=999)
pander(fDAT_rare_Au_rhizo_dist_paov$aov.tab[1:6], caption="Fungi")

```

#### pairwise PERMANOVA Root [~ background * genotype]

```{r pairwise PERMANOVA Aurora sampes root, warning=F, echo=F, eval=T}
bDAT_rare_Au_root <- otu_table(bPHYSEQ_Au_root)
bDAT_rare_Au_root <- bDAT_rare_Au_root[rowSums(bDAT_rare_Au_root) > 0,]
fDAT_rare_Au_root <- otu_table(fPHYSEQ_Au_root)
fDAT_rare_Au_root <- fDAT_rare_Au_root[rowSums(fDAT_rare_Au_root) > 0,]

# pairwise comparisons
bDAT_rare_Au_root_dist_PW_adonis <- pairwise.adonis(t(bDAT_rare_Au_root), bDESIGN[colnames(bDAT_rare_Au_root), ]$genotype, sim.method="bray", p.adjust.m="bonferroni")
pander(bDAT_rare_Au_root_dist_PW_adonis[,c(1:5)], caption="Root bacteria")
fDAT_rare_Au_root_dist_PW_adonis <- pairwise.adonis(t(fDAT_rare_Au_root), fDESIGN[colnames(fDAT_rare_Au_root), ]$genotype, sim.method="bray", p.adjust.m="bonferroni")
pander(fDAT_rare_Au_root_dist_PW_adonis[,c(1:5)], caption="Root fungi")

capture.output(pander(bDAT_rare_Au_root_dist_PW_adonis[,c(1:5)], caption="Root bacteria"), 
               file="tables/Table_S12_PERMANOVA_PW_root_mutants_bacteria.doc")
capture.output(pander(fDAT_rare_Au_root_dist_PW_adonis[,c(1:5)], caption="Root fungi"),
               file="tables/Table_S12_PERMANOVA_PW_root_mutants_fungi.doc")



bDAT_rare_Au_root_dist <- vegdist(t(bDAT_rare_Au_root), method="bray")
bDAT_rare_Au_root_dist_paov <- adonis(bDAT_rare_Au_root_dist ~ genotype, data=bDESIGN[colnames(bDAT_rare_Au_root),], permutations=999)
pander(bDAT_rare_Au_root_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Au_root_dist <- vegdist(t(fDAT_rare_Au_root), method="bray")
fDAT_rare_Au_root_dist_paov <- adonis(fDAT_rare_Au_root_dist ~ genotype, data=fDESIGN[colnames(fDAT_rare_Au_root),], permutations=999)
pander(fDAT_rare_Au_root_dist_paov$aov.tab[1:6], caption="Fungi")

bPHYSEQ_Au_root_bx1WT <- prune_samples(sample_data(bPHYSEQ_Au_root)$genotype=="WT"|sample_data(bPHYSEQ_Au_root)$genotype=="bx1", bPHYSEQ_Au_root)
fPHYSEQ_Au_root_bx1WT <- prune_samples(sample_data(fPHYSEQ_Au_root)$genotype=="WT"|sample_data(fPHYSEQ_Au_root)$genotype=="bx1", fPHYSEQ_Au_root)
bDAT_rare_Au_root_bx1WT <- otu_table(bPHYSEQ_Au_root)
bDAT_rare_Au_root_bx1WT <- bDAT_rare_Au_root[rowSums(bDAT_rare_Au_root) > 0,]
fDAT_rare_Au_root_bx1WT <- otu_table(fPHYSEQ_Au_root)
fDAT_rare_Au_root_bx1WT <- fDAT_rare_Au_root[rowSums(fDAT_rare_Au_root) > 0,]

bDAT_rare_Au_root_dist <- vegdist(t(bDAT_rare_Au_root_bx1WT), method="bray")
bDAT_rare_Au_root_dist_paov <- adonis(bDAT_rare_Au_root_dist ~ genotype, data=bDESIGN[colnames(bDAT_rare_Au_root),], permutations=999)
pander(bDAT_rare_Au_root_dist_paov$aov.tab[1:6], caption="Bacteria")

fDAT_rare_Au_root_dist <- vegdist(t(fDAT_rare_Au_root_bx1WT), method="bray")
fDAT_rare_Au_root_dist_paov <- adonis(fDAT_rare_Au_root_dist ~ genotype, data=fDESIGN[colnames(fDAT_rare_Au_root),], permutations=999)
pander(fDAT_rare_Au_root_dist_paov$aov.tab[1:6], caption="Fungi")

```




\pagebreak

# OTU level analysis

Statistical analyses at OTU level are performed using the R-package edgeR (Robinson et al., 2010). The likelihood ratio tests were performed on data that was i) filtered to contain OTUs with minimal abundance according to the replicate number (e.g. 5 replicates, min. OTU abundance 5 sequences) and ii) normalized using the TMM (trimmed mean of M value) method.  
Of note, TMM normalization assumes a constant abundance of a majority of species. To avoid violating this assumption and because many microbial compartments (soil, rhizosphere and roots) are variable in their composition, we first subset the data to their compartments and then applied TMM normalization. See Weiss et al. (2017, Microbiome Journal).   


## Q2: genetic background vs mutation?  (Reckenholz samples)

To answer the second research question – How do the genetic backgrounds compare in their BX exudation effects on microbial communities? – we studied the WT and bx1 mutant lines of the two genetic backgrounds B73 and W22 in the Reckenholz experiment. 

### Rhizosphere bacteria

```{r venn diagramm rhizo bacteria Re genotype vs background, warning=F, echo=F, fig.height=3, fig.width=3}
bDESIGN$background <- as.factor(bDESIGN$background)
### Subsetting to Reckenholz rhizosphere samples
# design file
bDESIGN_Re_rhizo <- subset(bDESIGN, bDESIGN$compartment == "rhizo" & bDESIGN$location == "Reckenholz")
bDESIGN_Re_rhizo$groups <- droplevels(bDESIGN_Re_rhizo)$groups
pander(table(droplevels(bDESIGN_Re_rhizo$background_genotype)), caption="nr of replicates")
# OTU table
bDAT_Re_rhizo <- bDAT[, rownames(bDESIGN_Re_rhizo) ]
#dim(bDAT_Re_rhizo)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Re_rhizo <- edgeR.object.TMM(dat=bDAT_Re_rhizo,
                                    threshold.filter=5,  # for comparable threshold use 5 (fungi samples have only 5 reps)
                                    groups=bDESIGN_Re_rhizo$groups,
                                    tax=bTAX)
# dim(bEDGER_Re_rhizo) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
## by genotype
bDESIGN_Re_rhizo$genotype <- droplevels(bDESIGN_Re_rhizo$genotype)
bEDGER_Re_rhizo_genotype_fit <- fit.model.edgeR(bEDGER_Re_rhizo, bDESIGN_Re_rhizo$genotype)
## by background
bDESIGN_Re_rhizo$background <- droplevels(bDESIGN_Re_rhizo$background)
bEDGER_Re_rhizo_background_fit <- fit.model.edgeR(bEDGER_Re_rhizo, bDESIGN_Re_rhizo$background)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Re_genotype <- makeContrasts(WT - bx1, levels=bDESIGN_Re_rhizo$genotype)
bEDGER_Re_rhizo_fit_genotype_lrt <- glmLRT(bEDGER_Re_rhizo_genotype_fit, contrast=contrasts_rhizo_Re_genotype)
bEDGER_Re_rhizo_fit_genotype_lrt_table <- topTags(bEDGER_Re_rhizo_fit_genotype_lrt, n=nrow(bEDGER_Re_rhizo_genotype_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Re_rhizo_fit_genotype_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Re_rhizo_fit_genotype_lrt$table, topTags_table=bEDGER_Re_rhizo_fit_genotype_lrt_table)

write.csv(bEDGER_Re_rhizo_fit_genotype_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_rhizo_bacteria_genotype.csv")
bEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs <- rownames(bEDGER_Re_rhizo_fit_genotype_lrt_table)

# B73 vs W22
contrasts_rhizo_Re_background <- makeContrasts(B73 - W22, levels=bDESIGN_Re_rhizo$background) 
bEDGER_Re_rhizo_fit_background_lrt <- glmLRT(bEDGER_Re_rhizo_background_fit, contrast=contrasts_rhizo_Re_background)
bEDGER_Re_rhizo_fit_background_lrt_table <- topTags(bEDGER_Re_rhizo_fit_background_lrt, n=nrow(bEDGER_Re_rhizo_background_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
# add percent sequence at the end of all lines
bEDGER_Re_rhizo_fit_background_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Re_rhizo_fit_background_lrt$table, topTags_table=bEDGER_Re_rhizo_fit_background_lrt_table)

write.csv(bEDGER_Re_rhizo_fit_background_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_rhizo_bacteria_background.csv")
bEDGER_Re_rhizo_fit_background_lrt_table_OTUs <- rownames(bEDGER_Re_rhizo_fit_background_lrt_table)


# percent abundance common / specific zOTUs (reviewer comment 06.2020)

# common and specific OTUs
common <- intersect(bEDGER_Re_rhizo_fit_background_lrt_table_OTUs, bEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs)
spec_backgr_OTUs <- bEDGER_Re_rhizo_fit_background_lrt_table_OTUs[!bEDGER_Re_rhizo_fit_background_lrt_table_OTUs %in% common]
spec_geno_OTUs <- bEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs[!bEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs %in% common]

# relative abundance of all seq
geno_percent_brhizo <-  sum(bEDGER_Re_rhizo_fit_genotype_lrt_table[,"percentAllSeq"])
background_percent_brhizo <-  sum(bEDGER_Re_rhizo_fit_background_lrt_table[,"percentAllSeq"])

spec_geno_percent_brhizo <-  sum(bEDGER_Re_rhizo_fit_genotype_lrt_table[spec_geno_OTUs,"percentAllSeq"])
comm_geno_percent_brhizo <- sum(bEDGER_Re_rhizo_fit_genotype_lrt_table[common,"percentAllSeq"])
spec_backgr_percent_brhizo <-  sum(bEDGER_Re_rhizo_fit_background_lrt_table[spec_backgr_OTUs,"percentAllSeq"])
comm_backgr_percent_brhizo <- sum(bEDGER_Re_rhizo_fit_background_lrt_table[common,"percentAllSeq"])

## plot
par(mar=c(1,1,1,1))
Fig_3B_venn_rhizo_bacteria <- venndiagram(bEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs, bEDGER_Re_rhizo_fit_background_lrt_table_OTUs, type="2", xpd=T, printsub=F, tcol="black", diacol="black", labels=c(paste("genotype (", geno_percent_brhizo,"%) ", sep=""),paste("background (", background_percent_brhizo,"%) ", sep="")), title="Rhizosphere bacteria")

```

Differentially abundant OTUs that are specific to the genotype contrast account for `r spec_geno_percent_brhizo` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are specific to the background contrast account for `r spec_backgr_percent_brhizo` % relative abundance of total OTU abundance (average?) among the two backgrounds.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_geno_percent_brhizo` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_backgr_percent_brhizo` % relative abundance of total OTU abundance (average?) among the two bakcgrounds



### Rhizosphere fungi

```{r venn diagramm rhizo fungi Re genotype vs background, warning=F, echo=F, fig.height=3, fig.width=3}
fDESIGN$background <- as.factor(fDESIGN$background)
### Subsetting to Reckenholz rhizosphere samples
# design file
fDESIGN_Re_rhizo <- subset(fDESIGN, fDESIGN$compartment=="rhizo" & fDESIGN$location=="Reckenholz")
fDESIGN_Re_rhizo$groups <- droplevels(fDESIGN_Re_rhizo)$groups
pander(table(droplevels(fDESIGN_Re_rhizo$background_genotype)), caption="nr of replicates")
# OTU table
fDAT_Re_rhizo <- fDAT[, rownames(fDESIGN_Re_rhizo) ]
#dim(fDAT_Re_rhizo)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Re_rhizo <- edgeR.object.TMM(dat=fDAT_Re_rhizo,
                                    threshold.filter=5, # for comparable threshold use 5 (fungi samples have only 5 reps)
                                    groups=fDESIGN_Re_rhizo$groups,
                                    tax=fTAX)
# dim(fEDGER_Re_rhizo) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
## by genotype
fDESIGN_Re_rhizo$genotype <- droplevels(fDESIGN_Re_rhizo$genotype)
fEDGER_Re_rhizo_genotype_fit <- fit.model.edgeR(fEDGER_Re_rhizo, fDESIGN_Re_rhizo$genotype)
## by background
fDESIGN_Re_rhizo$background <- droplevels(fDESIGN_Re_rhizo$background)
fEDGER_Re_rhizo_background_fit <- fit.model.edgeR(fEDGER_Re_rhizo, fDESIGN_Re_rhizo$background)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Re_genotype <- makeContrasts(WT - bx1, levels=fDESIGN_Re_rhizo$genotype) 
fEDGER_Re_rhizo_fit_genotype_lrt <- glmLRT(fEDGER_Re_rhizo_genotype_fit, contrast=contrasts_rhizo_Re_genotype)
fEDGER_Re_rhizo_fit_genotype_lrt_table <- topTags(fEDGER_Re_rhizo_fit_genotype_lrt, n=nrow(fEDGER_Re_rhizo_genotype_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Re_rhizo_fit_genotype_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Re_rhizo_fit_genotype_lrt$table, topTags_table=fEDGER_Re_rhizo_fit_genotype_lrt_table)

write.csv(fEDGER_Re_rhizo_fit_genotype_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_rhizo_fungi_genotype.csv")
fEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs <- rownames(fEDGER_Re_rhizo_fit_genotype_lrt_table)

# B73 vs W22
contrasts_rhizo_Re_background <- makeContrasts(B73 - W22, levels=fDESIGN_Re_rhizo$background) 
fEDGER_Re_rhizo_fit_background_lrt <- glmLRT(fEDGER_Re_rhizo_background_fit, contrast=contrasts_rhizo_Re_background)
fEDGER_Re_rhizo_fit_background_lrt_table <- topTags(fEDGER_Re_rhizo_fit_background_lrt, n=nrow(fEDGER_Re_rhizo_background_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Re_rhizo_fit_background_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Re_rhizo_fit_background_lrt$table, topTags_table=fEDGER_Re_rhizo_fit_background_lrt_table)

write.csv(fEDGER_Re_rhizo_fit_background_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_rhizo_fungi_background.csv")
fEDGER_Re_rhizo_fit_background_lrt_table_OTUs <- rownames(fEDGER_Re_rhizo_fit_background_lrt_table)


# percent abundance common / specific zOTUs (reviewer comment 06.2020)

# common and specific OTUs
common <- intersect(fEDGER_Re_rhizo_fit_background_lrt_table_OTUs, fEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs)
spec_backgr_OTUs <- fEDGER_Re_rhizo_fit_background_lrt_table_OTUs[!fEDGER_Re_rhizo_fit_background_lrt_table_OTUs %in% common]
spec_geno_OTUs <- fEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs[!fEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs %in% common]

# relative abundance of all seq
geno_percent_frhizo <-  sum(fEDGER_Re_rhizo_fit_genotype_lrt_table[,"percentAllSeq"])
background_percent_frhizo <-  sum(fEDGER_Re_rhizo_fit_background_lrt_table[,"percentAllSeq"])
spec_geno_percent_frhizo <- sum(fEDGER_Re_rhizo_fit_genotype_lrt_table[spec_geno_OTUs,"percentAllSeq"])
comm_geno_percent_frhizo <- sum(fEDGER_Re_rhizo_fit_genotype_lrt_table[common,"percentAllSeq"])
spec_backgr_percent_frhizo <-  sum(fEDGER_Re_rhizo_fit_background_lrt_table[spec_backgr_OTUs,"percentAllSeq"])
comm_backgr_percent_frhizo <- sum(fEDGER_Re_rhizo_fit_background_lrt_table[common,"percentAllSeq"])

## plot
par(mar=c(1,1,1,1))
Fig_3B_venn_rhizo_fungi <- venndiagram(fEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs, fEDGER_Re_rhizo_fit_background_lrt_table_OTUs,
                         type="2", xpd=T, printsub=F, tcol="black", diacol="black",
                         labels=c(paste("genotype (", geno_percent_frhizo,"%) ", sep=""),paste("background (", background_percent_frhizo,"%) ", sep="")), title="Rhizosphere fungi")



```
Differentially abundant OTUs that are specific to the genotype contrast account for `r spec_geno_percent_frhizo` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are specific to the background contrast account for `r spec_backgr_percent_frhizo` % relative abundance of total OTU abundance (average?) among the two backgrounds.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_geno_percent_frhizo` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_backgr_percent_frhizo` % relative abundance of total OTU abundance (average?) among the two bakcgrounds


### Root bacteria

```{r venn diagramm root bacteria Re genotype vs background, warning=F, echo=F, fig.height=3, fig.width=3}
### Subsetting to Reckenholz root samples
# design file
bDESIGN_Re_root <- subset(bDESIGN, bDESIGN$compartment=="root" & bDESIGN$location=="Reckenholz")
bDESIGN_Re_root$groups <- droplevels(bDESIGN_Re_root)$groups
pander(table(droplevels(bDESIGN_Re_root$background_genotype)), caption="nr of replicates")
# OTU table
bDAT_Re_root <- bDAT[, rownames(bDESIGN_Re_root) ]
#dim(bDAT_Re_root)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Re_root <- edgeR.object.TMM(dat=bDAT_Re_root,
                                    threshold.filter=5, # for comparable threshold use 5 (fungi samples have only 5 reps)
                                    groups=bDESIGN_Re_root$groups,
                                    tax=bTAX)
# dim(bEDGER_Re_root) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
## by genotype
bDESIGN_Re_root$genotype <- droplevels(bDESIGN_Re_root$genotype)
bEDGER_Re_root_genotype_fit <- fit.model.edgeR(bEDGER_Re_root, bDESIGN_Re_root$genotype)
## by background
bDESIGN_Re_root$background <- droplevels(bDESIGN_Re_root$background)
bEDGER_Re_root_background_fit <- fit.model.edgeR(bEDGER_Re_root, bDESIGN_Re_root$background)

## glmLRT 
# WT vs bx1
contrasts_root_Re_genotype <- makeContrasts(WT - bx1, levels=bDESIGN_Re_root$genotype) 
bEDGER_Re_root_fit_genotype_lrt <- glmLRT(bEDGER_Re_root_genotype_fit, contrast=contrasts_root_Re_genotype)
bEDGER_Re_root_fit_genotype_lrt_table <- topTags(bEDGER_Re_root_fit_genotype_lrt, n=nrow(bEDGER_Re_root_genotype_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Re_root_fit_genotype_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Re_root_fit_genotype_lrt$table, topTags_table=bEDGER_Re_root_fit_genotype_lrt_table)

write.csv(bEDGER_Re_root_fit_genotype_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_root_bacteria_genotype.csv")
bEDGER_Re_root_fit_genotype_lrt_table_OTUs <- rownames(bEDGER_Re_root_fit_genotype_lrt_table)

# B73 vs W22
contrasts_root_Re_background <- makeContrasts(B73 - W22, levels=bDESIGN_Re_root$background) 
bEDGER_Re_root_fit_background_lrt <- glmLRT(bEDGER_Re_root_background_fit, contrast=contrasts_root_Re_background)
bEDGER_Re_root_fit_background_lrt_table <- topTags(bEDGER_Re_root_fit_background_lrt, n=nrow(bEDGER_Re_root_background_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Re_root_fit_background_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Re_root_fit_background_lrt$table, topTags_table=bEDGER_Re_root_fit_background_lrt_table)

write.csv(bEDGER_Re_root_fit_background_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_root_bacteria_background.csv")
bEDGER_Re_root_fit_background_lrt_table_OTUs <- rownames(bEDGER_Re_root_fit_background_lrt_table)


# percent abundance common / specific zOTUs (reviewer comment 06.2020)

# common and specific OTUs
common <- intersect(bEDGER_Re_root_fit_background_lrt_table_OTUs, bEDGER_Re_root_fit_genotype_lrt_table_OTUs)

spec_backgr_OTUs <- bEDGER_Re_root_fit_background_lrt_table_OTUs[!bEDGER_Re_root_fit_background_lrt_table_OTUs %in% common]
spec_geno_OTUs <- bEDGER_Re_root_fit_genotype_lrt_table_OTUs[!bEDGER_Re_root_fit_genotype_lrt_table_OTUs %in% common]

# relative abundance of all seq
geno_percent_broot <-  sum(bEDGER_Re_root_fit_genotype_lrt_table[,"percentAllSeq"])
background_percent_broot <-  sum(bEDGER_Re_root_fit_background_lrt_table[,"percentAllSeq"])
spec_geno_percent_broot <-  sum(bEDGER_Re_root_fit_genotype_lrt_table[spec_geno_OTUs,"percentAllSeq"])
comm_geno_percent_broot <- sum(bEDGER_Re_root_fit_genotype_lrt_table[common,"percentAllSeq"])
spec_backgr_percent_broot <-  sum(bEDGER_Re_root_fit_background_lrt_table[spec_backgr_OTUs,"percentAllSeq"])
comm_backgr_percent_broot <- sum(bEDGER_Re_root_fit_background_lrt_table[common,"percentAllSeq"])

par(mar=c(1,1,1,1))
Fig_3B_venn_root_bacteria <- venndiagram(bEDGER_Re_root_fit_genotype_lrt_table_OTUs, bEDGER_Re_root_fit_background_lrt_table_OTUs,
                         type="2", xpd=T, printsub=F, tcol="black", diacol="black",
                          labels=c(paste("genotype (", geno_percent_broot,"%) ", sep=""),paste("background (", background_percent_broot,"%) ", sep="")),  title="Root bacteria")

```
Differentially abundant OTUs that are specific to the genotype contrast account for `r spec_geno_percent_broot` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are specific to the background contrast account for `r spec_backgr_percent_broot` % relative abundance of total OTU abundance (average?) among the two backgrounds.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_geno_percent_broot` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_backgr_percent_broot` % relative abundance of total OTU abundance (average?) among the two bakcgrounds



### Root fungi

```{r venn diagramm root fungi Re genotype vs background, warning=F, echo=F, fig.height=3, fig.width=3}
### Subsetting to Reckenholz rootsphere samples
# design file
fDESIGN_Re_root <- subset(fDESIGN, fDESIGN$compartment=="root" & fDESIGN$location=="Reckenholz")
fDESIGN_Re_root$groups <- droplevels(fDESIGN_Re_root)$groups
pander(table(droplevels(fDESIGN_Re_root$background_genotype)), caption="nr of replicates")
# OTU table
fDAT_Re_root <- fDAT[, rownames(fDESIGN_Re_root) ]
#dim(fDAT_Re_root)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Re_root <- edgeR.object.TMM(dat=fDAT_Re_root,
                                    threshold.filter=5, # for comparable threshold use 5 (fungi samples have only 5 reps)
                                    groups=fDESIGN_Re_root$groups,
                                    tax=fTAX)
# dim(fEDGER_Re_root) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
## by genotype
fDESIGN_Re_root$genotype <- droplevels(fDESIGN_Re_root$genotype)
fEDGER_Re_root_genotype_fit <- fit.model.edgeR(fEDGER_Re_root, fDESIGN_Re_root$genotype)
## by background
fDESIGN_Re_root$background <- droplevels(fDESIGN_Re_root$background)
fEDGER_Re_root_background_fit <- fit.model.edgeR(fEDGER_Re_root, fDESIGN_Re_root$background)

## glmLRT 
# WT vs bx1
contrasts_root_Re_genotype <- makeContrasts(WT - bx1, levels=fDESIGN_Re_root$genotype) 
fEDGER_Re_root_fit_genotype_lrt <- glmLRT(fEDGER_Re_root_genotype_fit, contrast=contrasts_root_Re_genotype)
fEDGER_Re_root_fit_genotype_lrt_table <- topTags(fEDGER_Re_root_fit_genotype_lrt, n=nrow(fEDGER_Re_root_genotype_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Re_root_fit_genotype_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Re_root_fit_genotype_lrt$table, topTags_table=fEDGER_Re_root_fit_genotype_lrt_table)

write.csv(fEDGER_Re_root_fit_genotype_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_root_fungi_genotype.csv")
fEDGER_Re_root_fit_genotype_lrt_table_OTUs <- rownames(fEDGER_Re_root_fit_genotype_lrt_table)

# B73 vs W22
contrasts_root_Re_background <- makeContrasts(B73 - W22, levels=fDESIGN_Re_root$background) 
fEDGER_Re_root_fit_background_lrt <- glmLRT(fEDGER_Re_root_background_fit, contrast=contrasts_root_Re_background)
fEDGER_Re_root_fit_background_lrt_table <- topTags(fEDGER_Re_root_fit_background_lrt, n=nrow(fEDGER_Re_root_background_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Re_root_fit_background_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Re_root_fit_background_lrt$table, topTags_table=fEDGER_Re_root_fit_background_lrt_table)

write.csv(fEDGER_Re_root_fit_background_lrt_table[,c(1:7,11:16)], "tables/Table_S10_edgeR_Re_root_fungi_background.csv")
fEDGER_Re_root_fit_background_lrt_table_OTUs <- rownames(fEDGER_Re_root_fit_background_lrt_table)


# percent abundance common / specific zOTUs (reviewer comment 06.2020)

# common and specific OTUs
common <- intersect(fEDGER_Re_root_fit_background_lrt_table_OTUs, fEDGER_Re_root_fit_genotype_lrt_table_OTUs)
spec_backgr_OTUs <- fEDGER_Re_root_fit_background_lrt_table_OTUs[!fEDGER_Re_root_fit_background_lrt_table_OTUs %in% common]
spec_geno_OTUs <- fEDGER_Re_root_fit_genotype_lrt_table_OTUs[!fEDGER_Re_root_fit_genotype_lrt_table_OTUs %in% common]

# relative abundance of all seq
geno_percent_froot <-  sum(fEDGER_Re_root_fit_genotype_lrt_table[,"percentAllSeq"])
background_percent_froot <-  sum(fEDGER_Re_root_fit_background_lrt_table[,"percentAllSeq"])

spec_geno_percent_froot <-  sum(fEDGER_Re_root_fit_genotype_lrt_table[spec_geno_OTUs,"percentAllSeq"])
comm_geno_percent_froot <- sum(fEDGER_Re_root_fit_genotype_lrt_table[common,"percentAllSeq"])
spec_backgr_percent_froot <-  sum(fEDGER_Re_root_fit_background_lrt_table[spec_backgr_OTUs,"percentAllSeq"])
comm_backgr_percent_froot <- sum(fEDGER_Re_root_fit_background_lrt_table[common,"percentAllSeq"])

# plot
par(mar=c(1,1,1,1))
Fig_3B_venn_root_fungi <- venndiagram(fEDGER_Re_root_fit_genotype_lrt_table_OTUs, fEDGER_Re_root_fit_background_lrt_table_OTUs,
                         type="2", xpd=T, printsub=F, tcol="black", diacol="black", labels=c(paste("genotype (", geno_percent_froot,"%) ", sep=""),paste("background (", background_percent_froot,"%) ", sep="")), title="Root fungi")


```
Differentially abundant OTUs that are specific to the genotype contrast account for `r spec_geno_percent_froot` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are specific to the background contrast account for `r spec_backgr_percent_froot` % relative abundance of total OTU abundance (average?) among the two backgrounds.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_geno_percent_froot` % relative abundance of total OTU abundance (average?) among the two genotypes.

Differentially abundant OTUs that are common to both genotype and background contrast account for `r comm_backgr_percent_froot` % relative abundance of total OTU abundance (average?) among the two bakcgrounds

### Fig 3B | All Venn percent review, 11.20)
```{r all Venn percent 11.20, warning=F, echo=F}

## plot
pdf("figures/Fig_3B_venn_percentSeq.pdf", width=10, height=2)

par(mfrow=c(1,5), mar=c(0.2, 0.2, 1.2, 0.2))
Fig_3B_venn_rhizo_bacteria <- venndiagram(bEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs, bEDGER_Re_rhizo_fit_background_lrt_table_OTUs,
                         type="2", xpd=T, printsub=F, 
                         lines=c("red4", "steelblue3"),
                         tcol=c( "purple3", "red4",  "steelblue3"), 
                         lcol = c("red4", "steelblue3"), # potato label
                         lwd=2, cex=1.2,
                         labels=c(paste(geno_percent_brhizo,"%", 
                                        sep=" "),paste(background_percent_brhizo,"% cRA", sep=" ")), 
                         title="Rhizosphere bacteria", font.main=1
                        ); grid.echo()

Fig_3B_venn_rhizo_fungi <- venndiagram(fEDGER_Re_rhizo_fit_genotype_lrt_table_OTUs, fEDGER_Re_rhizo_fit_background_lrt_table_OTUs,
                         type="2", xpd=T, printsub=F, 
                         lines=c("red4", "steelblue3"),
                         tcol=c( "purple3", "red4",  "steelblue3"), 
                         lcol = c("red4", "steelblue3"), # potato label
                         lwd=2,
                         labels=c(paste(geno_percent_frhizo,"%", 
                                        sep=" "),paste(background_percent_frhizo,"% cRA", sep=" ")), cex=1.2,
                         title="Rhizosphere fungi", font.main=1
                        ); grid.echo()

Fig_3B_venn_root_bacteria <- venndiagram(bEDGER_Re_root_fit_genotype_lrt_table_OTUs, bEDGER_Re_root_fit_background_lrt_table_OTUs,
                         type="2", xpd=T, printsub=F, 
                         lines=c("red4", "steelblue3"),
                         tcol=c( "purple3", "red4",  "steelblue3"), 
                         lcol = c("red4", "steelblue3"), # potato label
                         lwd=2,
                         labels=c(paste(geno_percent_broot,"%", 
                                        sep=" "),paste(background_percent_broot,"% cRA", sep=" ")), cex=1.2,
                         title="Root bacteria", font.main=1
                        ); grid.echo()

Fig_3B_venn_root_fungi <- venndiagram(fEDGER_Re_root_fit_genotype_lrt_table_OTUs, fEDGER_Re_root_fit_background_lrt_table_OTUs,
                         type="2", xpd=T, printsub=F, 
                         lines=c("red4", "steelblue3"),
                         tcol=c( "purple3", "red4",  "steelblue3"), 
                         lcol = c("red4", "steelblue3"), # potato label
                         lwd=2, cex=1.2,
                         labels=c(paste(geno_percent_froot,"%", 
                                        sep=" "),paste(background_percent_froot,"% cRA", sep=" ")), 
                         title="Root fungi", font.main=1
                        ); grid.echo()


# legend of sample PCoA
cols <- c("black", "red4", "steelblue")
names(cols) <- c("cRA = count rel. abundance", "- genotype contrast", "- genetic background contrast")
plot(c(0, 1), c(0, 1), ann = F, bty = 'n', type = 'n', xaxt = 'n', yaxt = 'n')
text(rep(0,3), c(0.6, 0.7, 0.8), names(cols), col=cols, adj=0,cex=1)

dev.off()
 
 
```


\pagebreak

## Q3: various BX biosynthesis mutants? (Aurora samples)

We approached the third research question – How do various mutations in the BX biosynthesis pathway compare in their impacts on rhizosphere and root microbial communities? – by comparing the wildtype W22 to the mutant lines bx1, bx2 and bx6 in the experiment conducted in Aurora. 

### Rhizosphere bacteria

```{r Aurora rhizosphere bacteria, fig.height=3, fig.width=3, warning=F, echo=F}
### Subsetting to Aurora rhizosphere samples
# design file
bDESIGN_Au_rhizo <- subset(bDESIGN, bDESIGN$compartment=="rhizo" & bDESIGN$location=="Aurora")
bDESIGN_Au_rhizo$groups <- droplevels(bDESIGN_Au_rhizo)$groups
pander(table(droplevels(bDESIGN_Au_rhizo$background_genotype)), caption="nr of replicates")
# OTU table
bDAT_Au_rhizo <- bDAT[, rownames(bDESIGN_Au_rhizo) ]
#dim(bDAT_Au_rhizo)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_rhizo <- edgeR.object.TMM(dat=bDAT_Au_rhizo,
                                    threshold.filter=6, # 6 samples, lowest number of reps for comparative analysis
                                    groups=bDESIGN_Au_rhizo$groups,
                                    tax=bTAX)
# dim(bEDGER_Au_rhizo) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_rhizo_fit <- fit.model.edgeR(bEDGER_Au_rhizo, bDESIGN_Au_rhizo$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Au_bx1 <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx1, levels=bDESIGN_Au_rhizo$groups) 
bEDGER_Au_rhizo_fit_bx1_lrt <- glmLRT(bEDGER_Au_rhizo_fit, contrast=contrasts_rhizo_Au_bx1)
bEDGER_Au_rhizo_fit_bx1_lrt_table <- topTags(bEDGER_Au_rhizo_fit_bx1_lrt, n=nrow(bEDGER_Au_rhizo_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Au_rhizo_fit_bx1_lrt_table$contrast <- "WT_vs_bx1"
bEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs <- rownames(bEDGER_Au_rhizo_fit_bx1_lrt_table)
bEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_rhizo_fit$counts) * length(bEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs),1)

# WT vs bx2
contrasts_rhizo_Au_bx2 <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx2, levels=bDESIGN_Au_rhizo$groups) 
bEDGER_Au_rhizo_fit_bx2_lrt <- glmLRT(bEDGER_Au_rhizo_fit, contrast=contrasts_rhizo_Au_bx2)
bEDGER_Au_rhizo_fit_bx2_lrt_table <- topTags(bEDGER_Au_rhizo_fit_bx2_lrt, n=nrow(bEDGER_Au_rhizo_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Au_rhizo_fit_bx2_lrt_table$contrast <- "WT_vs_bx2"
bEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs <- rownames(bEDGER_Au_rhizo_fit_bx2_lrt_table)
bEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_rhizo_fit$counts) * length(bEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs),1)

# WT vs bx6
contrasts_rhizo_Au_bx6 <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx6, levels=bDESIGN_Au_rhizo$groups) 
bEDGER_Au_rhizo_fit_bx6_lrt <- glmLRT(bEDGER_Au_rhizo_fit, contrast=contrasts_rhizo_Au_bx6)
bEDGER_Au_rhizo_fit_bx6_lrt_table <- topTags(bEDGER_Au_rhizo_fit_bx6_lrt, n=nrow(bEDGER_Au_rhizo_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Au_rhizo_fit_bx6_lrt_table$contrast <- "WT_vs_bx6"
bEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs <- rownames(bEDGER_Au_rhizo_fit_bx6_lrt_table)
bEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_rhizo_fit$counts) * length(bEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs),1)


# add percent sequence at the end of all lines
bEDGER_Au_rhizo_fit_bx1_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Au_rhizo_fit_bx1_lrt$table, topTags_table=bEDGER_Au_rhizo_fit_bx1_lrt_table)
bEDGER_Au_rhizo_fit_bx1_lrt_sumPercent <- sum(bEDGER_Au_rhizo_fit_bx1_lrt_table$percentAllSeq)

# add percent sequence at the end of all lines
bEDGER_Au_rhizo_fit_bx2_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Au_rhizo_fit_bx2_lrt$table, topTags_table=bEDGER_Au_rhizo_fit_bx2_lrt_table)
bEDGER_Au_rhizo_fit_bx2_lrt_sumPercent <- sum(bEDGER_Au_rhizo_fit_bx2_lrt_table$percentAllSeq)

# add percent sequence at the end of all lines
bEDGER_Au_rhizo_fit_bx6_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Au_rhizo_fit_bx6_lrt$table, topTags_table=bEDGER_Au_rhizo_fit_bx6_lrt_table)
bEDGER_Au_rhizo_fit_bx6_lrt_sumPercent <- sum(bEDGER_Au_rhizo_fit_bx6_lrt_table$percentAllSeq)



## Plot with % total sequences
# source("functions/vennDia.R")
par(mar=c(1,1,1,1))
x <- venndiagram(bEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs, bEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs, bEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs, 
type="3", xpd=T, printsub=F, tcol="black", diacol="black", lcol=c("palegreen3", "palegreen1", "palegreen4"), lines=c("palegreen3", "palegreen1", "palegreen4"),
 labels=c(paste(bEDGER_Au_rhizo_fit_bx1_lrt_sumPercent, "%", sep=""), paste(bEDGER_Au_rhizo_fit_bx2_lrt_sumPercent, "%", sep=""), paste(bEDGER_Au_rhizo_fit_bx6_lrt_sumPercent, "% cRA", sep="")), title="Rhizosphere bacteria")

### Table S13
write.csv(rbind(bEDGER_Au_rhizo_fit_bx1_lrt_table[,c(1:7,11:17)],
                bEDGER_Au_rhizo_fit_bx2_lrt_table[,c(1:7,11:17)],
                bEDGER_Au_rhizo_fit_bx6_lrt_table[,c(1:7,11:17)]), "tables/Table_S13_edgeR_Au_rhizo_bacteria.csv")

```

Differentially abundant OTUs that are specific to the WT-bx1 contrast account for `r bEDGER_Au_rhizo_fit_bx1_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx2 contrast account for `r bEDGER_Au_rhizo_fit_bx2_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx6 contrast account for `r bEDGER_Au_rhizo_fit_bx6_lrt_sumPercent` % relative abundance of total OTU abundance (average?).


### Rhizosphere fungi

```{r Aurora rhizosphere fungi, fig.height=3, fig.width=3, warning=F, echo=F}
### Subsetting to Aurora rhizosphere samples
# design file
fDESIGN_Au_rhizo <- subset(fDESIGN, fDESIGN$compartment=="rhizo" & fDESIGN$location=="Aurora")
fDESIGN_Au_rhizo$groups <- droplevels(fDESIGN_Au_rhizo)$groups
pander(table(droplevels(fDESIGN_Au_rhizo$background_genotype)), caption="nr of replicates")
# OTU table
fDAT_Au_rhizo <- fDAT[, rownames(fDESIGN_Au_rhizo) ]
#dim(fDAT_Au_rhizo)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_rhizo <- edgeR.object.TMM(dat=fDAT_Au_rhizo,
                                    threshold.filter=6, # 6 samples, lowest number of reps for comparative analysis
                                    groups=fDESIGN_Au_rhizo$groups,
                                    tax=fTAX)
# dim(fEDGER_Au_rhizo) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_rhizo_fit <- fit.model.edgeR(fEDGER_Au_rhizo, fDESIGN_Au_rhizo$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Au_bx1 <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx1, levels=fDESIGN_Au_rhizo$groups) 
fEDGER_Au_rhizo_fit_bx1_lrt <- glmLRT(fEDGER_Au_rhizo_fit, contrast=contrasts_rhizo_Au_bx1)
fEDGER_Au_rhizo_fit_bx1_lrt_table <- topTags(fEDGER_Au_rhizo_fit_bx1_lrt, n=nrow(fEDGER_Au_rhizo_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Au_rhizo_fit_bx1_lrt_table$contrast <- "WT_vs_bx1"
fEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs <- rownames(fEDGER_Au_rhizo_fit_bx1_lrt_table)
fEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_rhizo_fit$counts) * length(fEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs),1)

# WT vs bx2
contrasts_rhizo_Au_bx2 <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx2, levels=fDESIGN_Au_rhizo$groups) 
fEDGER_Au_rhizo_fit_bx2_lrt <- glmLRT(fEDGER_Au_rhizo_fit, contrast=contrasts_rhizo_Au_bx2)
fEDGER_Au_rhizo_fit_bx2_lrt_table <- topTags(fEDGER_Au_rhizo_fit_bx2_lrt, n=nrow(fEDGER_Au_rhizo_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Au_rhizo_fit_bx2_lrt_table$contrast <- "WT_vs_bx2"
fEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs <- rownames(fEDGER_Au_rhizo_fit_bx2_lrt_table)
fEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_rhizo_fit$counts) * length(fEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs),1)

# WT vs bx6
contrasts_rhizo_Au_bx6 <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx6, levels=fDESIGN_Au_rhizo$groups) 
fEDGER_Au_rhizo_fit_bx6_lrt <- glmLRT(fEDGER_Au_rhizo_fit, contrast=contrasts_rhizo_Au_bx6)
fEDGER_Au_rhizo_fit_bx6_lrt_table <- topTags(fEDGER_Au_rhizo_fit_bx6_lrt, n=nrow(fEDGER_Au_rhizo_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Au_rhizo_fit_bx6_lrt_table$contrast <- "WT_vs_bx6"
fEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs <- rownames(fEDGER_Au_rhizo_fit_bx6_lrt_table)
fEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_rhizo_fit$counts) * length(fEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs),1)

# add percent sequence at the end of all lines
fEDGER_Au_rhizo_fit_bx1_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Au_rhizo_fit_bx1_lrt$table, topTags_table=fEDGER_Au_rhizo_fit_bx1_lrt_table)
fEDGER_Au_rhizo_fit_bx1_lrt_sumPercent <- sum(fEDGER_Au_rhizo_fit_bx1_lrt_table$percentAllSeq)

# add percent sequence at the end of all lines
fEDGER_Au_rhizo_fit_bx2_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Au_rhizo_fit_bx2_lrt$table, topTags_table=fEDGER_Au_rhizo_fit_bx2_lrt_table)
fEDGER_Au_rhizo_fit_bx2_lrt_sumPercent <- sum(fEDGER_Au_rhizo_fit_bx2_lrt_table$percentAllSeq)

# add percent sequence at the end of all lines
fEDGER_Au_rhizo_fit_bx6_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Au_rhizo_fit_bx6_lrt$table, topTags_table=fEDGER_Au_rhizo_fit_bx6_lrt_table)
fEDGER_Au_rhizo_fit_bx6_lrt_sumPercent <- sum(fEDGER_Au_rhizo_fit_bx6_lrt_table$percentAllSeq)


## Plot percent seq
# source("functions/vennDia.R")
par(mar=c(1,1,1,1))
x <- venndiagram(fEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs, fEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs, fEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs, 
            type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
            lcol=c("palegreen3", "palegreen1", "palegreen4"), lines=c("palegreen3", "palegreen1", "palegreen4"),
            labels=c(paste(fEDGER_Au_rhizo_fit_bx1_lrt_sumPercent, "%", sep=""),
                     paste(fEDGER_Au_rhizo_fit_bx2_lrt_sumPercent, "%", sep=""),
                     paste(fEDGER_Au_rhizo_fit_bx6_lrt_sumPercent, "% cRA", sep="")), title="Rhizosphere fungi")
noshow <- dev.off()

### Table S13
write.csv(rbind(fEDGER_Au_rhizo_fit_bx1_lrt_table[,c(1:7,11:17)],
                fEDGER_Au_rhizo_fit_bx2_lrt_table[,c(1:7,11:17)],
                fEDGER_Au_rhizo_fit_bx6_lrt_table[,c(1:7,11:17)]), "tables/Table_S13_edgeR_Au_rhizo_fungi.csv")

```

Differentially abundant OTUs that are specific to the WT-bx1 contrast account for `r fEDGER_Au_rhizo_fit_bx1_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx2 contrast account for `r fEDGER_Au_rhizo_fit_bx2_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx6 contrast account for `r fEDGER_Au_rhizo_fit_bx6_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

### Root bacteria

```{r Aurora root bacteria, fig.height=3, fig.width=3, warning=F, echo=F}
### Subsetting to Aurora root samples
# design file
bDESIGN_Au_root <- subset(bDESIGN, bDESIGN$compartment=="root" & bDESIGN$location=="Aurora")
bDESIGN_Au_root$groups <- droplevels(bDESIGN_Au_root)$groups
pander(table(droplevels(bDESIGN_Au_root$background_genotype)), caption="nr of replicates")
# OTU table
bDAT_Au_root <- bDAT[, rownames(bDESIGN_Au_root) ]
#dim(bDAT_Au_root)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_root <- edgeR.object.TMM(dat=bDAT_Au_root,
                                    threshold.filter=6, # 6 samples, lowest number of reps for comparative analysis
                                    groups=bDESIGN_Au_root$groups,
                                    tax=bTAX)
# dim(bEDGER_Au_root) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_root_fit <- fit.model.edgeR(bEDGER_Au_root, bDESIGN_Au_root$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Au_bx1 <- makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx1, levels=bDESIGN_Au_root$groups) 
bEDGER_Au_root_fit_bx1_lrt <- glmLRT(bEDGER_Au_root_fit, contrast=contrasts_root_Au_bx1)
bEDGER_Au_root_fit_bx1_lrt_table <- topTags(bEDGER_Au_root_fit_bx1_lrt, n=nrow(bEDGER_Au_root_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Au_root_fit_bx1_lrt_table$contrast <- "WT_vs_bx1"
bEDGER_Au_root_fit_bx1_lrt_table_OTUs <- rownames(bEDGER_Au_root_fit_bx1_lrt_table)
bEDGER_Au_root_fit_bx1_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_root_fit$counts) * length(bEDGER_Au_root_fit_bx1_lrt_table_OTUs),1)

# WT vs bx2
contrasts_root_Au_bx2<-makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx2, levels=bDESIGN_Au_root$groups) 
bEDGER_Au_root_fit_bx2_lrt <- glmLRT(bEDGER_Au_root_fit, contrast=contrasts_root_Au_bx2)
bEDGER_Au_root_fit_bx2_lrt_table <- topTags(bEDGER_Au_root_fit_bx2_lrt, n=nrow(bEDGER_Au_root_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Au_root_fit_bx2_lrt_table$contrast <- "WT_vs_bx2"
bEDGER_Au_root_fit_bx2_lrt_table_OTUs <- rownames(bEDGER_Au_root_fit_bx2_lrt_table)
bEDGER_Au_root_fit_bx2_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_root_fit$counts) * length(bEDGER_Au_root_fit_bx2_lrt_table_OTUs),1)

# WT vs bx6
contrasts_root_Au_bx6<-makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx6, levels=bDESIGN_Au_root$groups) 
bEDGER_Au_root_fit_bx6_lrt <- glmLRT(bEDGER_Au_root_fit, contrast=contrasts_root_Au_bx6)
bEDGER_Au_root_fit_bx6_lrt_table <- topTags(bEDGER_Au_root_fit_bx6_lrt, n=nrow(bEDGER_Au_root_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
bEDGER_Au_root_fit_bx6_lrt_table$contrast <- "WT_vs_bx6"
bEDGER_Au_root_fit_bx6_lrt_table_OTUs <- rownames(bEDGER_Au_root_fit_bx6_lrt_table)
bEDGER_Au_root_fit_bx6_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_root_fit$counts) * length(bEDGER_Au_root_fit_bx6_lrt_table_OTUs),1)

# add percent sequence at the end of all lines
bEDGER_Au_root_fit_bx1_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Au_root_fit_bx1_lrt$table, topTags_table=bEDGER_Au_root_fit_bx1_lrt_table)
bEDGER_Au_root_fit_bx1_lrt_sumPercent <- sum(bEDGER_Au_root_fit_bx1_lrt_table$percentAllSeq)

# add percent sequence at the end of all lines
bEDGER_Au_root_fit_bx2_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Au_root_fit_bx2_lrt$table, topTags_table=bEDGER_Au_root_fit_bx2_lrt_table)
bEDGER_Au_root_fit_bx2_lrt_sumPercent <- sum(bEDGER_Au_root_fit_bx2_lrt_table$percentAllSeq)

# add percent sequence at the end of all lines
bEDGER_Au_root_fit_bx6_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Au_root_fit_bx6_lrt$table, topTags_table=bEDGER_Au_root_fit_bx6_lrt_table)
bEDGER_Au_root_fit_bx6_lrt_sumPercent <- sum(bEDGER_Au_root_fit_bx6_lrt_table$percentAllSeq)

## Plot
# source("functions/vennDia.R")
par(mar=c(1,1,1,1))
x <- venndiagram(bEDGER_Au_root_fit_bx1_lrt_table_OTUs, bEDGER_Au_root_fit_bx2_lrt_table_OTUs, bEDGER_Au_root_fit_bx6_lrt_table_OTUs, 
            type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
            lcol=c("palegreen3", "palegreen1", "palegreen4"), lines=c("palegreen3", "palegreen1", "palegreen4"),
            labels=c(paste(bEDGER_Au_root_fit_bx1_lrt_sumPercent, "%", sep=""),
                     paste(bEDGER_Au_root_fit_bx2_lrt_sumPercent, "%", sep=""),
                     paste(bEDGER_Au_root_fit_bx6_lrt_sumPercent, "%", sep="")), title="Root bacteria")

### Table S13
write.csv(rbind(bEDGER_Au_root_fit_bx1_lrt_table[,c(1:7,11:17)],
                bEDGER_Au_root_fit_bx2_lrt_table[,c(1:7,11:17)],
                bEDGER_Au_root_fit_bx6_lrt_table[,c(1:7,11:17)]), "tables/Table_S13_edgeR_Au_root_bacteria.csv")
```

Differentially abundant OTUs that are specific to the WT-bx1 contrast account for `r bEDGER_Au_root_fit_bx1_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx2 contrast account for `r bEDGER_Au_root_fit_bx2_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx6 contrast account for `r bEDGER_Au_root_fit_bx6_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

### Root fungi

```{r Aurora root fungi, fig.height=3, fig.width=3, warning=F, echo=F}
### Subsetting to Aurora root samples
# design file
fDESIGN_Au_root <- subset(fDESIGN, fDESIGN$compartment=="root" & fDESIGN$location=="Aurora")
fDESIGN_Au_root$groups <- droplevels(fDESIGN_Au_root)$groups
pander(table(droplevels(fDESIGN_Au_root$background_genotype)), caption="nr of replicates")
# OTU table
fDAT_Au_root <- fDAT[, rownames(fDESIGN_Au_root) ]
#dim(fDAT_Au_root)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_root <- edgeR.object.TMM(dat=fDAT_Au_root,
                                    threshold.filter=6, # 6 samples, lowest number of reps for comparative analysis
                                    groups=fDESIGN_Au_root$groups,
                                    tax=fTAX)
# dim(fEDGER_Au_root) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_root_fit <- fit.model.edgeR(fEDGER_Au_root, fDESIGN_Au_root$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Au_bx1 <- makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx1, levels=fDESIGN_Au_root$groups) 
fEDGER_Au_root_fit_bx1_lrt <- glmLRT(fEDGER_Au_root_fit, contrast=contrasts_root_Au_bx1)
fEDGER_Au_root_fit_bx1_lrt_table <- topTags(fEDGER_Au_root_fit_bx1_lrt, n=nrow(fEDGER_Au_root_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Au_root_fit_bx1_lrt_table$contrast <- "WT_vs_bx1"
fEDGER_Au_root_fit_bx1_lrt_table_OTUs <- rownames(fEDGER_Au_root_fit_bx1_lrt_table)
fEDGER_Au_root_fit_bx1_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_root_fit$counts) * length(fEDGER_Au_root_fit_bx1_lrt_table_OTUs),1)

# WT vs bx2
contrasts_root_Au_bx2 <- makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx2, levels=fDESIGN_Au_root$groups) 
fEDGER_Au_root_fit_bx2_lrt <- glmLRT(fEDGER_Au_root_fit, contrast=contrasts_root_Au_bx2)
fEDGER_Au_root_fit_bx2_lrt_table <- topTags(fEDGER_Au_root_fit_bx2_lrt, n=nrow(fEDGER_Au_root_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Au_root_fit_bx2_lrt_table$contrast <- "WT_vs_bx2"
fEDGER_Au_root_fit_bx2_lrt_table_OTUs <- rownames(fEDGER_Au_root_fit_bx2_lrt_table)
fEDGER_Au_root_fit_bx2_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_root_fit$counts) * length(fEDGER_Au_root_fit_bx2_lrt_table_OTUs),1)

# WT vs bx6
contrasts_root_Au_bx6 <- makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx6, levels=fDESIGN_Au_root$groups) 
fEDGER_Au_root_fit_bx6_lrt <- glmLRT(fEDGER_Au_root_fit, contrast=contrasts_root_Au_bx6)
fEDGER_Au_root_fit_bx6_lrt_table <- topTags(fEDGER_Au_root_fit_bx6_lrt, n=nrow(fEDGER_Au_root_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
fEDGER_Au_root_fit_bx6_lrt_table$contrast <- "WT_vs_bx6"
fEDGER_Au_root_fit_bx6_lrt_table_OTUs <- rownames(fEDGER_Au_root_fit_bx6_lrt_table)
fEDGER_Au_root_fit_bx6_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_root_fit$counts) * length(fEDGER_Au_root_fit_bx6_lrt_table_OTUs),1)

## Plot

# add percent sequence at the end of all lines
fEDGER_Au_root_fit_bx1_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Au_root_fit_bx1_lrt$table, topTags_table=fEDGER_Au_root_fit_bx1_lrt_table)
fEDGER_Au_root_fit_bx1_lrt_sumPercent <- sum(fEDGER_Au_root_fit_bx1_lrt_table$percentAllSeq)

fEDGER_Au_root_fit_bx2_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Au_root_fit_bx2_lrt$table, topTags_table=fEDGER_Au_root_fit_bx2_lrt_table)
fEDGER_Au_root_fit_bx2_lrt_sumPercent <- sum(fEDGER_Au_root_fit_bx2_lrt_table$percentAllSeq)

fEDGER_Au_root_fit_bx6_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Au_root_fit_bx6_lrt$table, topTags_table=fEDGER_Au_root_fit_bx6_lrt_table)
fEDGER_Au_root_fit_bx6_lrt_sumPercent <- sum(fEDGER_Au_root_fit_bx6_lrt_table$percentAllSeq)

# source("functions/vennDia.R")
par(mar=c(1,1,1,1))
x <- venndiagram(fEDGER_Au_root_fit_bx1_lrt_table_OTUs, fEDGER_Au_root_fit_bx2_lrt_table_OTUs, fEDGER_Au_root_fit_bx6_lrt_table_OTUs, 
            type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
            lcol=c("palegreen3", "palegreen1", "palegreen4"), lines=c("palegreen3", "palegreen1", "palegreen4"),
            labels=c(paste(fEDGER_Au_root_fit_bx1_lrt_sumPercent, "%", sep=""),
                     paste(fEDGER_Au_root_fit_bx2_lrt_sumPercent, "%", sep=""),
                     paste("WT-bx6 (", fEDGER_Au_root_fit_bx6_lrt_sumPercent, "% cRA", sep="")), title="Root fungi")

### Table S13
write.csv(rbind(fEDGER_Au_root_fit_bx1_lrt_table[,c(1:7,11:17)],
                fEDGER_Au_root_fit_bx2_lrt_table[,c(1:7,11:17)],
                fEDGER_Au_root_fit_bx6_lrt_table[,c(1:7,11:17)]), "tables/Table_S13_edgeR_Au_root_fungi.csv")

```
### Figure 4B | Venn Aurora (rev 11.20)
```{r Aurora Figure 4B Venn review 11.20, fig.height=3, fig.width=3, warning=F, echo=F}

pdf("figures/Fig_4B_venn_percentSeq.pdf", width=9, height=1.9)
par(mfrow=c(1,5), mar=c(0.2, 0.2, 1.2, 0.2))


x <- venndiagram(bEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs, bEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs, bEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs, 
            type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
            lcol=c("palegreen3", "palegreen1", "palegreen4"), 
            lines=c("palegreen3", "palegreen1", "palegreen4"),
            labels=c(paste(bEDGER_Au_rhizo_fit_bx1_lrt_sumPercent, "% cRA", sep=""), 
                     paste(bEDGER_Au_rhizo_fit_bx2_lrt_sumPercent, "% cRA", sep=""),
                     paste(bEDGER_Au_rhizo_fit_bx6_lrt_sumPercent, "% cRA", sep="")), title="Rhizosphere bacteria", font.main=1)


x <- venndiagram(fEDGER_Au_rhizo_fit_bx1_lrt_table_OTUs, fEDGER_Au_rhizo_fit_bx2_lrt_table_OTUs, fEDGER_Au_rhizo_fit_bx6_lrt_table_OTUs, 
            type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
            lcol=c("palegreen3", "palegreen1", "palegreen4"), lines=c("palegreen3", "palegreen1", "palegreen4"),
            labels=c(paste(fEDGER_Au_rhizo_fit_bx1_lrt_sumPercent, "% cRA", sep=""),
                     paste(fEDGER_Au_rhizo_fit_bx2_lrt_sumPercent, "% cRA", sep=""),
                     paste(fEDGER_Au_rhizo_fit_bx6_lrt_sumPercent, "% cRA", sep="")), title="Rhizosphere fungi", font.main=1)

x <- venndiagram(bEDGER_Au_root_fit_bx1_lrt_table_OTUs, bEDGER_Au_root_fit_bx2_lrt_table_OTUs, bEDGER_Au_root_fit_bx6_lrt_table_OTUs, 
            type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
            lcol=c("palegreen3", "palegreen1", "palegreen4"), lines=c("palegreen3", "palegreen1", "palegreen4"),
            labels=c(paste(bEDGER_Au_root_fit_bx1_lrt_sumPercent, "% cRA", sep=""),
                     paste(bEDGER_Au_root_fit_bx2_lrt_sumPercent, "% cRA", sep=""),
                     paste(bEDGER_Au_root_fit_bx6_lrt_sumPercent, "% cRA", sep="")), title="Root bacteria", font.main=1)

x <- venndiagram(fEDGER_Au_root_fit_bx1_lrt_table_OTUs, fEDGER_Au_root_fit_bx2_lrt_table_OTUs, fEDGER_Au_root_fit_bx6_lrt_table_OTUs, 
            type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
            lcol=c("palegreen3", "palegreen1", "palegreen4"), lines=c("palegreen3", "palegreen1", "palegreen4"),
            labels=c(paste(fEDGER_Au_root_fit_bx1_lrt_sumPercent, "% cRA", sep=""),
                     paste(fEDGER_Au_root_fit_bx2_lrt_sumPercent, "% cRA", sep=""),
                     paste(fEDGER_Au_root_fit_bx6_lrt_sumPercent, "% cRA", sep="")), title="Root fungi", font.main=1)


# legend of sample PCoA
cols <- c("palegreen3", "palegreen1", "palegreen4","black")
names(cols) <- c( "- WT-bx1 contrast", "- WT-bx2 contrast", "- WT-bx6 contrast", "cRA = count rel. abundance")
## grab the scene as a grid object
library(gridGraphics)
library(grid)
plot(c(0, 1), c(0, 1), ann = F, bty = 'n', type = 'n', xaxt = 'n', yaxt = 'n')
text(rep(0,4), c(0.6,0.5,0.4,0.3), names(cols), col=cols, adj=0,cex=1)

dev.off()


```
Differentially abundant OTUs that are specific to the WT-bx1 contrast account for `r fEDGER_Au_root_fit_bx1_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx2 contrast account for `r fEDGER_Au_root_fit_bx2_lrt_sumPercent` % relative abundance of total OTU abundance (average?).

Differentially abundant OTUs that are specific to the WT-bx6 contrast account for `r fEDGER_Au_root_fit_bx6_lrt_sumPercent` % relative abundance of total OTU abundance (average?).



\pagebreak

## Q4: Core taxa responding to BX exudation?  (all locations samples)

To answer the fourth research question – Is there a core of microbial taxa that responds to BX exudation? – we compared the wildtype and bx1 mutants lines grown in the 3 field locations Changins, Aurora and Reckenholz. We limited these comparisons to WT and bx1 mutant lines, as these plant lines were present at all locations but we excluded W22 samples at the Reckenholz location because of unbalanced sample numbers. We searched for BX-sensitive OTUs – OTUs being differentially abundant between BX producing and defective plants – using edgeR (Robinson et al., 2010). 


### Rhizosphere bacteria

```{r Changins rhizo bacteria WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Changins rhizo samples
# design file
bDESIGN_Ch_rhizo_BX <- subset(bDESIGN, bDESIGN$compartment=="rhizo" & bDESIGN$location=="Changins" & bDESIGN$genotype=="WT" | 
                                bDESIGN$compartment=="rhizo" & bDESIGN$location=="Changins" & bDESIGN$genotype=="bx1")
bDESIGN_Ch_rhizo_BX$groups <- droplevels(bDESIGN_Ch_rhizo_BX)$groups
table(bDESIGN_Ch_rhizo_BX$groups)
# OTU table
bDAT_Ch_rhizo_BX <- bDAT[, rownames(bDESIGN_Ch_rhizo_BX) ]
# dim(bDAT_Ch_rhizo_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Ch_rhizo_BX <- edgeR.object.TMM(dat=bDAT_Ch_rhizo_BX,
                                       threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                       groups=bDESIGN_Ch_rhizo_BX$groups,
                                       tax=bTAX)
# dim(bEDGER_Ch_rhizo_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Ch_rhizo_BX_fit <- fit.model.edgeR(bEDGER_Ch_rhizo_BX, bDESIGN_Ch_rhizo_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Ch_BX <- makeContrasts(rhizo_Changins_B73_WT - rhizo_Changins_B73_bx1, levels=bDESIGN_Ch_rhizo_BX$groups) 
bEDGER_Ch_rhizo_BX_fit_lrt <- glmLRT(bEDGER_Ch_rhizo_BX_fit, contrast=contrasts_rhizo_Ch_BX)
bEDGER_Ch_rhizo_BX_fit_lrt_topTags <- topTags(bEDGER_Ch_rhizo_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
bEDGER_Ch_rhizo_BX_fit_lrt_table <- topTags(bEDGER_Ch_rhizo_BX_fit_lrt, n=nrow(bEDGER_Ch_rhizo_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
bEDGER_Ch_rhizo_BX_fit_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Ch_rhizo_BX_fit_lrt$table, topTags_table=bEDGER_Ch_rhizo_BX_fit_lrt_table)
bEDGER_Ch_rhizo_BX_fit_lrt_sumPercent <- sum(bEDGER_Ch_rhizo_BX_fit_lrt_table$percentAllSeq)

write.csv(bEDGER_Ch_rhizo_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_bEDGER_Ch_rhizo.csv")
bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs <- rownames(bEDGER_Ch_rhizo_BX_fit_lrt_table)
bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Ch_rhizo_BX_fit$counts) * length(bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs), 1)
```

In *Changins* rhizosphere samples, `r paste(length(bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  

BX-sensitive OTUs represent `r bEDGER_Ch_rhizo_BX_fit_lrt_sumPercent` % relative abundance of average total sequence number.

```{r Aurora rhizo bacteria WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Aurora rhizo samples
# design file
bDESIGN_Au_rhizo_BX <- subset(bDESIGN, bDESIGN$compartment=="rhizo" & bDESIGN$location=="Aurora" & bDESIGN$genotype=="WT" | 
                                 bDESIGN$compartment=="rhizo" & bDESIGN$location=="Aurora" & bDESIGN$genotype=="bx1")
bDESIGN_Au_rhizo_BX$groups <- droplevels(bDESIGN_Au_rhizo_BX)$groups
table(bDESIGN_Au_rhizo_BX$groups)
# OTU table
bDAT_Au_rhizo_BX <- bDAT[, rownames(bDESIGN_Au_rhizo_BX) ]
# dim(bDAT_Au_rhizo_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_rhizo_BX <- edgeR.object.TMM(dat=bDAT_Au_rhizo_BX,
                                    threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                    groups=bDESIGN_Au_rhizo_BX$groups,
                                    tax=bTAX)
# dim(bEDGER_Au_rhizo_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_rhizo_BX_fit <- fit.model.edgeR(bEDGER_Au_rhizo_BX, bDESIGN_Au_rhizo_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Au_BX <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx1, levels=bDESIGN_Au_rhizo_BX$groups) 
bEDGER_Au_rhizo_BX_fit_lrt <- glmLRT(bEDGER_Au_rhizo_BX_fit, contrast=contrasts_rhizo_Au_BX)
bEDGER_Au_rhizo_BX_fit_lrt_topTags <- topTags(bEDGER_Au_rhizo_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
bEDGER_Au_rhizo_BX_fit_lrt_table <- topTags(bEDGER_Au_rhizo_BX_fit_lrt, n=nrow(bEDGER_Au_rhizo_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
bEDGER_Au_rhizo_BX_fit_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Au_rhizo_BX_fit_lrt$table, topTags_table=bEDGER_Au_rhizo_BX_fit_lrt_table)
bEDGER_Au_rhizo_BX_fit_lrt_sumPercent <- sum(bEDGER_Au_rhizo_BX_fit_lrt_table$percentAllSeq)

write.csv(bEDGER_Au_rhizo_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_bEDGER_Au_rhizo.csv")
bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs <- rownames(bEDGER_Au_rhizo_BX_fit_lrt_table)
bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_rhizo_BX_fit$counts) * length(bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs), 1)
```

In *Aurora* rhizosphere samples, `r paste(length(bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type W22 and mutant bx1(W22).  

BX-sensitive OTUs represent `r bEDGER_Au_rhizo_BX_fit_lrt_sumPercent` % relative abundance of average total sequence number.

```{r Reckenholz rhizo bacteria WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Reckenholz rhizo samples
# design file
bDESIGN_Re_rhizo_BX <- subset(bDESIGN, bDESIGN$compartment=="rhizo" & bDESIGN$location=="Reckenholz" & bDESIGN$background_genotype=="B73" | 
                                bDESIGN$compartment=="rhizo" & bDESIGN$location=="Reckenholz" & bDESIGN$background_genotype=="bx1(B73)")
bDESIGN_Re_rhizo_BX$groups <- droplevels(bDESIGN_Re_rhizo_BX)$groups
table(bDESIGN_Re_rhizo_BX$groups)
# OTU table
bDAT_Re_rhizo_BX <- bDAT[, rownames(bDESIGN_Re_rhizo_BX) ]
# dim(bDAT_Re_rhizo_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Re_rhizo_BX <- edgeR.object.TMM(dat=bDAT_Re_rhizo_BX,
                                       threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                       groups=bDESIGN_Re_rhizo_BX$groups,
                                       tax=bTAX)
# dim(bEDGER_Re_rhizo_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Re_rhizo_BX_fit <- fit.model.edgeR(bEDGER_Re_rhizo_BX, bDESIGN_Re_rhizo_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Re_BX <- makeContrasts(rhizo_Reckenholz_B73_WT - rhizo_Reckenholz_B73_bx1, levels=bDESIGN_Re_rhizo_BX$groups) 
bEDGER_Re_rhizo_BX_fit_lrt <- glmLRT(bEDGER_Re_rhizo_BX_fit, contrast=contrasts_rhizo_Re_BX)
bEDGER_Re_rhizo_BX_fit_lrt_topTags <- topTags(bEDGER_Re_rhizo_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
bEDGER_Re_rhizo_BX_fit_lrt_table <- topTags(bEDGER_Re_rhizo_BX_fit_lrt, n=nrow(bEDGER_Re_rhizo_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
bEDGER_Re_rhizo_BX_fit_lrt_table <- topTags_percent_abundance(edgeR_table=bEDGER_Re_rhizo_BX_fit_lrt$table, topTags_table=bEDGER_Re_rhizo_BX_fit_lrt_table)
bEDGER_Re_rhizo_BX_fit_lrt_sumPercent <- sum(bEDGER_Re_rhizo_BX_fit_lrt_table$percentAllSeq)

write.csv(bEDGER_Re_rhizo_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_bEDGER_Re_rhizo.csv")
bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs <- rownames(bEDGER_Re_rhizo_BX_fit_lrt_table)
bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Re_rhizo_BX_fit$counts) * length(bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs), 1)
```

In *Reckenholz* rhizosphere samples, `r paste(length(bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  

BX-sensitive OTUs represent `r bEDGER_Re_rhizo_BX_fit_lrt_sumPercent` % relative abundance of average total sequence number.

#### MA plot of rhizosphere bacteria

```{r MAplot all locations rhizo bacteria WT vs bx1, fig.height=3, fig.width=5, warning=F, echo=F, results="hide"}
# custom function fun_maplot() to plot edgeR results as MA-plots
# source("functions/fun_maplot.R")

# define plotting parameters
par(mfrow=c(1,3), mar=c(4.5,4.5,1.6,1.6))

## Transform CPM abundances to % abundances 
bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$CPM <- 2^bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$logCPM # non log2 CPM
bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$percent <- bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$CPM/10000 # %

bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$CPM <- 2^bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$logCPM
bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$percent <- bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$CPM/10000

bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$CPM <- 2^bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logCPM
bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$percent <- bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$CPM/10000

# axis range
xlim_ma <- c(0.01, max(bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$percent))
ylim_ma <- c(min(bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$logFC), max(bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logFC))

# plot

fun_maplot(bEDGER_Ch_rhizo_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change (M)", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma) 

fun_maplot(bEDGER_Au_rhizo_BX_fit_lrt_topTags, title="Aurora", ylab="", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)

fun_maplot(bEDGER_Re_rhizo_BX_fit_lrt_topTags, title="Reckenholz", ylab="", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
```

```{r LEGEND maplot, echo=F, warning=F, fig.height=3, fig.width=6, eval=T}

pdf("figures/Fig_5_S7_Maplot_legend.pdf", width=5, height=5)
mat2 <- cbind(c(1,1), c(1.5,1))
cols <- c("gold2", "palegreen3")
names(cols) <- c("BX-enriched", "BX-depleted")
pchlegend <- c(16,16)
plot(mat2, col=cols, pch=pchlegend, cex=5, lwd=4, xlim=c(.5,5), ylim=c(.5,3.5), frame=F, axes=F, xlab=NA, ylab=NA)
# points(mat2+0.2, col=cols, pch=4, cex=5)
text(1, 2.2, label=c("BX-sensitive OTU"), adj=0.1, cex=2, xpd=T)
text(rep(2, 2), c(1.5,1), label=c("BX-enriched", "BX-depleted"), adj = 0.2, cex=2)
noshow <- dev.off()



pdf("figures/Fig_5_S7_Venn_legend.pdf", width=5, height=5)
mat2 <- cbind(c(1,1,1), c(1.6,1.3,1))
cols <- level_cols_loc
names(cols) <- c("Changins", "Reckenholz", "Aurora")
pchlegend <- c("-")
plot(mat2, col=cols, pch=pchlegend, cex=5, lwd=4, xlim=c(.5,5), ylim=c(.5,3.5), frame=F, axes=F, xlab=NA, ylab=NA)
# points(mat2+0.2, col=cols, pch=4, cex=5)
text(1, 2.2, label=c("Location"), adj=0.2, cex=2, xpd=T)
text(rep(2,2,2), c(1.6, 1.3,1), label=c("Changins", "Reckenholz", "Aurora"), adj = c(0,0), cex=2)
noshow <- dev.off()



```

\vspace{20pt}
Inspecting the OTU overlap of rhizosphere bacteria between the locations:  

```{r Venn all locations rhizo bacteria WT vs bx1, fig.height=3, fig.width=3, warning=F, echo=F, results="hide"}
par(mar=c(1,1,1,1))
Core_rhizo_bacteria_venn <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, 
                                          type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                          lcol=level_cols_loc, lines=level_cols_loc,
                                          labels=c(paste("Changins (", bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs_percent, "%)", sep=""),
                                                   paste("Aurora (", bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs_percent, "%)", sep=""),
                                                   paste("Reckenholz (", bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs_percent, "%)", sep="")), 
                                          title="")

## plot with percent sequence
Core_rhizo_bacteria_venn_percentSeq <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, 
                                          type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                          lcol=level_cols_loc, lines=level_cols_loc,
                                          labels=c(paste("Changins (", bEDGER_Ch_rhizo_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                   paste("Aurora (", bEDGER_Au_rhizo_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                   paste("Reckenholz (", bEDGER_Re_rhizo_BX_fit_lrt_sumPercent, "%)", sep="")), 
                                          title="")
```



\pagebreak

Inspecting the FAMILY overlap of BX-sensitve OTUs of rhizosphere bacteria between the locations:  

```{r Venn all locations rhizo bacteria WT vs bx1 family level, fig.height=3, fig.width=6, warning=F, echo=F, results="hide"}
bEDGER_Ch_rhizo_BX_fit_lrt_table_fams_enriched <- unique(bEDGER_Ch_rhizo_BX_fit_lrt_table$family[bEDGER_Ch_rhizo_BX_fit_lrt_table$logFC>0])
bEDGER_Ch_rhizo_BX_fit_lrt_table_fams_depleted <- unique(bEDGER_Ch_rhizo_BX_fit_lrt_table$family[bEDGER_Ch_rhizo_BX_fit_lrt_table$logFC<0])

bEDGER_Au_rhizo_BX_fit_lrt_table_fams_enriched <- unique(bEDGER_Au_rhizo_BX_fit_lrt_table$family[bEDGER_Au_rhizo_BX_fit_lrt_table$logFC>0])
bEDGER_Au_rhizo_BX_fit_lrt_table_fams_depleted <- unique(bEDGER_Au_rhizo_BX_fit_lrt_table$family[bEDGER_Au_rhizo_BX_fit_lrt_table$logFC<0])

bEDGER_Re_rhizo_BX_fit_lrt_table_fams_enriched <- unique(bEDGER_Re_rhizo_BX_fit_lrt_table$family[bEDGER_Re_rhizo_BX_fit_lrt_table$logFC>0])
bEDGER_Re_rhizo_BX_fit_lrt_table_fams_depleted <- unique(bEDGER_Re_rhizo_BX_fit_lrt_table$family[bEDGER_Re_rhizo_BX_fit_lrt_table$logFC<0])

par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_rhizo_bacteria_fams_enriched <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Au_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Re_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c("Changins", "Aurora", "Reckenholz"), 
                                                 title="Family taxonomy of enriched OTUs", cex.main=0.7)
Core_rhizo_bacteria_fams_depleted <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Au_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Re_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c("Changins", "Aurora", "Reckenholz"), lwd=2,
                                                 title="Family taxonomy of depleted OTUs", cex.main=0.7)

pdf("figures/Fig_S8_Venn_BXsens_bOTUs_rhizo.pdf", width = 6, height = 3)
par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_rhizo_bacteria_fams_enriched <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Au_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Re_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", lwd=2, labels="",
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                            
                                                 title="Family taxonomy of enriched OTUs", cex.main=0.7)

Core_rhizo_bacteria_fams_depleted <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Au_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Re_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                lwd=2, labels="",
                                                 title="Family taxonomy of depleted OTUs", cex.main=0.7)
noshow <- dev.off()
```

The families of the BX enriched rhizosphere bOTUs are: `r paste(Core_rhizo_bacteria_fams_enriched$q1)`.  

The families of the BX depleted rhizosphere bOTUs are: `r paste(Core_rhizo_bacteria_fams_depleted$q1)`. 



\pagebreak

### Rhizosphere fungi

```{r Changins rhizo fungi WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Changins rhizo samples
# design file
fDESIGN_Ch_rhizo_BX <- subset(fDESIGN, fDESIGN$compartment=="rhizo" & fDESIGN$location=="Changins" & fDESIGN$genotype=="WT" | 
                               fDESIGN$compartment=="rhizo" & fDESIGN$location=="Changins" & fDESIGN$genotype=="bx1")
fDESIGN_Ch_rhizo_BX$groups <- droplevels(fDESIGN_Ch_rhizo_BX)$groups
table(fDESIGN_Ch_rhizo_BX$groups)
# OTU table
fDAT_Ch_rhizo_BX <- fDAT[, rownames(fDESIGN_Ch_rhizo_BX) ]
# dim(fDAT_Ch_rhizo_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Ch_rhizo_BX <- edgeR.object.TMM(dat=fDAT_Ch_rhizo_BX,
                                      threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                      groups=fDESIGN_Ch_rhizo_BX$groups,
                                      tax=fTAX)
# dim(fEDGER_Ch_rhizo_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Ch_rhizo_BX_fit <- fit.model.edgeR(fEDGER_Ch_rhizo_BX, fDESIGN_Ch_rhizo_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Ch_BX <- makeContrasts(rhizo_Changins_B73_WT - rhizo_Changins_B73_bx1, levels=fDESIGN_Ch_rhizo_BX$groups) 
fEDGER_Ch_rhizo_BX_fit_lrt <- glmLRT(fEDGER_Ch_rhizo_BX_fit, contrast=contrasts_rhizo_Ch_BX)
fEDGER_Ch_rhizo_BX_fit_lrt_topTags <- topTags(fEDGER_Ch_rhizo_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
fEDGER_Ch_rhizo_BX_fit_lrt_table <- topTags(fEDGER_Ch_rhizo_BX_fit_lrt, n=nrow(fEDGER_Ch_rhizo_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
fEDGER_Ch_rhizo_BX_fit_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Ch_rhizo_BX_fit_lrt$table, topTags_table=fEDGER_Ch_rhizo_BX_fit_lrt_table)
fEDGER_Ch_rhizo_BX_fit_lrt_sumPercent <- sum(fEDGER_Ch_rhizo_BX_fit_lrt_table$percentAllSeq)

write.csv(fEDGER_Ch_rhizo_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_fEDGER_Ch_rhizo.csv")
fEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs <- rownames(fEDGER_Ch_rhizo_BX_fit_lrt_table)
fEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Ch_rhizo_BX_fit$counts) * length(fEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs), 1)
```

In *Changins* rhizosphere samples, `r paste(length(fEDGER_Ch_rhizo_BX_fit_lrt_sumPercent))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  


```{r Aurora rhizo fungi WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Aurora rhizo samples
# design file
fDESIGN_Au_rhizo_BX <- subset(fDESIGN, fDESIGN$compartment=="rhizo" & fDESIGN$location=="Aurora" & fDESIGN$genotype=="WT" | 
                               fDESIGN$compartment=="rhizo" & fDESIGN$location=="Aurora" & fDESIGN$genotype=="bx1")
fDESIGN_Au_rhizo_BX$groups <- droplevels(fDESIGN_Au_rhizo_BX)$groups
table(fDESIGN_Au_rhizo_BX$groups)
# OTU table
fDAT_Au_rhizo_BX <- fDAT[, rownames(fDESIGN_Au_rhizo_BX) ]
# dim(fDAT_Au_rhizo_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_rhizo_BX <- edgeR.object.TMM(dat=fDAT_Au_rhizo_BX,
                                      threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                      groups=fDESIGN_Au_rhizo_BX$groups,
                                      tax=fTAX)
# dim(fEDGER_Au_rhizo_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_rhizo_BX_fit <- fit.model.edgeR(fEDGER_Au_rhizo_BX, fDESIGN_Au_rhizo_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Au_BX <- makeContrasts(rhizo_Aurora_W22_WT - rhizo_Aurora_W22_bx1, levels=fDESIGN_Au_rhizo_BX$groups) 
fEDGER_Au_rhizo_BX_fit_lrt <- glmLRT(fEDGER_Au_rhizo_BX_fit, contrast=contrasts_rhizo_Au_BX)
fEDGER_Au_rhizo_BX_fit_lrt_topTags <- topTags(fEDGER_Au_rhizo_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
fEDGER_Au_rhizo_BX_fit_lrt_table <- topTags(fEDGER_Au_rhizo_BX_fit_lrt, n=nrow(fEDGER_Au_rhizo_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
fEDGER_Au_rhizo_BX_fit_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Au_rhizo_BX_fit_lrt$table, topTags_table=fEDGER_Au_rhizo_BX_fit_lrt_table)
fEDGER_Au_rhizo_BX_fit_lrt_sumPercent <- sum(fEDGER_Au_rhizo_BX_fit_lrt_table$percentAllSeq)

write.csv(fEDGER_Au_rhizo_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_fEDGER_Au_rhizo.csv")
fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs <- rownames(fEDGER_Au_rhizo_BX_fit_lrt_table)
fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_rhizo_BX_fit$counts) * length(fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs), 1)
```

In *Aurora* rhizosphere samples, `r paste(length(fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type W22 and mutant bx1(W22).  


```{r Reckenholz rhizo fungi WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Reckenholz rhizo samples
# design file
fDESIGN_Re_rhizo_BX <- subset(fDESIGN, fDESIGN$compartment=="rhizo" & fDESIGN$location=="Reckenholz" & fDESIGN$background_genotype=="B73" | 
                               fDESIGN$compartment=="rhizo" & fDESIGN$location=="Reckenholz" & fDESIGN$background_genotype=="bx1(B73)")
fDESIGN_Re_rhizo_BX$groups <- droplevels(fDESIGN_Re_rhizo_BX)$groups
table(fDESIGN_Re_rhizo_BX$groups)
# OTU table
fDAT_Re_rhizo_BX <- fDAT[, rownames(fDESIGN_Re_rhizo_BX) ]
# dim(fDAT_Re_rhizo_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Re_rhizo_BX <- edgeR.object.TMM(dat=fDAT_Re_rhizo_BX,
                                      threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                      groups=fDESIGN_Re_rhizo_BX$groups,
                                      tax=fTAX)
# dim(fEDGER_Re_rhizo_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Re_rhizo_BX_fit <- fit.model.edgeR(fEDGER_Re_rhizo_BX, fDESIGN_Re_rhizo_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_rhizo_Re_BX <- makeContrasts(rhizo_Reckenholz_B73_WT - rhizo_Reckenholz_B73_bx1, levels=fDESIGN_Re_rhizo_BX$groups) 
fEDGER_Re_rhizo_BX_fit_lrt <- glmLRT(fEDGER_Re_rhizo_BX_fit, contrast=contrasts_rhizo_Re_BX)
fEDGER_Re_rhizo_BX_fit_lrt_topTags <- topTags(fEDGER_Re_rhizo_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
fEDGER_Re_rhizo_BX_fit_lrt_table <- topTags(fEDGER_Re_rhizo_BX_fit_lrt, n=nrow(fEDGER_Re_rhizo_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]
# add percent sequence at the end of all lines
fEDGER_Re_rhizo_BX_fit_lrt_table <- topTags_percent_abundance(edgeR_table=fEDGER_Re_rhizo_BX_fit_lrt$table, topTags_table=fEDGER_Re_rhizo_BX_fit_lrt_table)
fEDGER_Re_rhizo_BX_fit_lrt_sumPercent <- sum(fEDGER_Re_rhizo_BX_fit_lrt_table$percentAllSeq)

write.csv(fEDGER_Re_rhizo_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_fEDGER_Re_rhizo.csv")
fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs <- rownames(fEDGER_Re_rhizo_BX_fit_lrt_table)
fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Re_rhizo_BX_fit$counts) * length(fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs), 1)


```

In *Reckenholz* rhizosphere samples, `r paste(length(fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  


#### MA plot of rhizosphere fungi

```{r MAplot all locations rhizo fungi WT vs bx1, fig.height=3, fig.width=5, warning=F, echo=F, results="hide"}
# custom function fun_maplot() to plot edgeR results as MA-plots
# source("functions/fun_maplot.R")

# define plotting parameters
par(mfrow=c(1,3), mar=c(4.5,4.5,1.6,1.6))

## Transform CPM abundances to % abundances 
fEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$CPM <- 2^fEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$logCPM # non log2 CPM
fEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$percent <- fEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$CPM/10000 # %

fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$CPM <- 2^fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$logCPM
fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$percent <- fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$CPM/10000

fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$CPM <- 2^fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logCPM
fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$percent <- fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$CPM/10000

# axis range
xlim_ma <- c(0.01, max(fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$percent))
ylim_ma <- c(min(fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$logFC), max(fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logFC))
   
fun_maplot(fEDGER_Ch_rhizo_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change (M)", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)

fun_maplot(fEDGER_Au_rhizo_BX_fit_lrt_topTags, title="Aurora", ylab="", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)

fun_maplot(fEDGER_Re_rhizo_BX_fit_lrt_topTags, title="Reckenholz", ylab="", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
```

\vspace{20pt}
Inspecting the OTU overlap of rhizosphere fungi between the locations:  
  
```{r Venn all locations rhizo fungi WT vs bx1, fig.height=3, fig.width=3, warning=F, echo=F, results="hide"}
par(mar=c(1,1,1,1))
Core_rhizo_fungi_venn <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, 
                                          type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                          lcol=level_cols_loc, lines=level_cols_loc,
                                          labels=c(paste("Changins (", fEDGER_Ch_rhizo_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                   paste("Aurora (", fEDGER_Au_rhizo_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                   paste("Reckenholz (", fEDGER_Re_rhizo_BX_fit_lrt_sumPercent, "%)", sep="")), 
                                          title="")



# percent of total abundance (reviewer comment 06.2020)

# # all location diff. OTUs abundances
# fEDGER_percent_Re_rhizo_BX <- topTags_percent_abundance(fEDGER_Re_rhizo_BX_fit_lrt$table, fEDGER_Re_rhizo_BX_fit_lrt_table)
# fEDGER_percent_Au_rhizo_BX <- topTags_percent_abundance(fEDGER_Au_rhizo_BX_fit_lrt$table, fEDGER_Au_rhizo_BX_fit_lrt_table)
# fEDGER_percent_Ch_rhizo_BX <- topTags_percent_abundance(fEDGER_Ch_rhizo_BX_fit_lrt$table, fEDGER_Ch_rhizo_BX_fit_lrt_table)
```
Differentially abundant OTUs account for `r round(fEDGER_Re_rhizo_BX_fit_lrt_sumPercent,1)` % of total relative abundance in Reckenholz, `r round(fEDGER_Ch_rhizo_BX_fit_lrt_sumPercent,1)` % in Changins and `r round(fEDGER_Au_rhizo_BX_fit_lrt_sumPercent,1)` in Aurora.


\pagebreak

Inspecting the FAMILY overlap of BX-sensitve OTUs of rhizosphere fungi between the locations:  

```{r Venn all locations rhizo fungi WT vs bx1 family level, fig.height=3, fig.width=6, warning=F, echo=F, results="hide"}
fEDGER_Ch_rhizo_BX_fit_lrt_table_fams_enriched <- unique(fEDGER_Ch_rhizo_BX_fit_lrt_table$family[fEDGER_Ch_rhizo_BX_fit_lrt_table$logFC>0])
fEDGER_Ch_rhizo_BX_fit_lrt_table_fams_depleted <- unique(fEDGER_Ch_rhizo_BX_fit_lrt_table$family[fEDGER_Ch_rhizo_BX_fit_lrt_table$logFC<0])

fEDGER_Au_rhizo_BX_fit_lrt_table_fams_enriched <- unique(fEDGER_Au_rhizo_BX_fit_lrt_table$family[fEDGER_Au_rhizo_BX_fit_lrt_table$logFC>0])
fEDGER_Au_rhizo_BX_fit_lrt_table_fams_depleted <- unique(fEDGER_Au_rhizo_BX_fit_lrt_table$family[fEDGER_Au_rhizo_BX_fit_lrt_table$logFC<0])

fEDGER_Re_rhizo_BX_fit_lrt_table_fams_enriched <- unique(fEDGER_Re_rhizo_BX_fit_lrt_table$family[fEDGER_Re_rhizo_BX_fit_lrt_table$logFC>0])
fEDGER_Re_rhizo_BX_fit_lrt_table_fams_depleted <- unique(fEDGER_Re_rhizo_BX_fit_lrt_table$family[fEDGER_Re_rhizo_BX_fit_lrt_table$logFC<0])

par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_rhizo_fungi_fams_enriched <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Au_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Re_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                  labels=c(""), lwd=2,           title="Family taxonomy of enriched OTUs", cex.main=0.7)
Core_rhizo_fungi_fams_depleted <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Au_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Re_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                  labels=c(""), lwd=2,
                                                 title="Family taxonomy of depleted OTUs", cex.main=0.7)


pdf("figures/Fig_S8_Venn_BXsens_fOTUs_rhizo.pdf", width=6, height=3)
par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_rhizo_fungi_fams_enriched <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Au_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Re_rhizo_BX_fit_lrt_table_fams_enriched, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                  labels=c(""), lwd=2,
                                                 title="Family taxonomy of enriched OTUs", cex.main=0.7)

Core_rhizo_fungi_fams_depleted <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Au_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Re_rhizo_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c(""), lwd=2,
                                                 title="Family taxonomy of depleted OTUs", cex.main=0.7)
noshow <- dev.off()
```

The families of the BX enriched rhizosphere fOTUs are: `r paste(Core_rhizo_fungi_fams_enriched$q1)` fungi.  

The families of the BX depleted rhizosphere fOTUs are: `r paste(Core_rhizo_fungi_fams_depleted$q1)` fungi. 


### Figure S7 | MA plot and Venn

```{r Fig S7 MAplot all locations rhizo bacteria WT vs bx1 rev 1, fig.height=5, fig.width=15, warning=F, echo=F, results="hide"}
# custom function fun_maplot() to plot edgeR results as MA-plots
# source("functions/fun_maplot.R")

# redefine MAX for all plots the same
max_rhizo_all_percent <- max(c(bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$percent,
                           bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$percent,
                           bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$percent,
                           fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$percent,
                           fEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$percent,
                           fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$percent))
max_rhizo_all_logFC <- max(c(bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$logFC))
min_rhizo_all_logFC <- min(c(bEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Re_rhizo_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Au_rhizo_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Ch_rhizo_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Re_rhizo_BX_fit_lrt_topTags$table$logFC))

xlim_ma <- c(0.01, max_rhizo_all_percent)
ylim_ma <- c(min_rhizo_all_logFC, max_rhizo_all_logFC)

pdf("figures/Fig_S7_MAplot_rhizosphere_percentSeq.pdf", width=12, height=6)
par(mfrow=c(2,4), mar=c(4.5,4.5,1.6,1.6))

# upper panel
fun_maplot(bEDGER_Ch_rhizo_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change (M)", xlab="", ylim = ylim_ma, xlim=xlim_ma)
mtext("A", font=2, side=3, line=0.5, adj=-0.25, cex=1)
fun_maplot(bEDGER_Au_rhizo_BX_fit_lrt_topTags, title="Aurora", ylab=" ", xlab="", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(bEDGER_Re_rhizo_BX_fit_lrt_topTags, title="Reckenholz", ylab=" ", xlab="", ylim = ylim_ma, xlim=xlim_ma)
Core_rhizo_bacteria_venn <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, 
                                          type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                          lcol=level_cols_loc, lines=level_cols_loc,
                                          labels=c(paste( bEDGER_Ch_rhizo_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                   paste( bEDGER_Au_rhizo_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                   paste( bEDGER_Re_rhizo_BX_fit_lrt_sumPercent, "% cRA", sep="")), 
                                          title="Rhizosphere bacteria")

# lower panel
fun_maplot(fEDGER_Ch_rhizo_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change (M)", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
mtext("B",font=2, side=3, line=0.5, adj=-0.25, cex=1)
fun_maplot(fEDGER_Au_rhizo_BX_fit_lrt_topTags, title="Aurora", ylab=" ", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(fEDGER_Re_rhizo_BX_fit_lrt_topTags, title="Reckenholz", ylab=" ", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
Core_rhizo_fungi_venn <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, 
                                          type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                          lcol=level_cols_loc, lines=level_cols_loc,
                                          labels=c(paste( fEDGER_Ch_rhizo_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                   paste( fEDGER_Au_rhizo_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                   paste( fEDGER_Re_rhizo_BX_fit_lrt_sumPercent, "% cRA", sep="")), 
                                          title="Rhizosphere fungi")
noshow <- dev.off()




```


\pagebreak

### Root bacteria

```{r Changins root bacteria WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Changins root samples
# design file
bDESIGN_Ch_root_BX <- subset(bDESIGN, bDESIGN$compartment=="root" & bDESIGN$location=="Changins" & bDESIGN$genotype=="WT" | 
                                bDESIGN$compartment=="root" & bDESIGN$location=="Changins" & bDESIGN$genotype=="bx1")
bDESIGN_Ch_root_BX$groups <- droplevels(bDESIGN_Ch_root_BX)$groups
table(bDESIGN_Ch_root_BX$groups)
# OTU table
bDAT_Ch_root_BX <- bDAT[, rownames(bDESIGN_Ch_root_BX) ]
# dim(bDAT_Ch_root_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Ch_root_BX <- edgeR.object.TMM(dat=bDAT_Ch_root_BX,
                                       threshold.filter=6,   # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                       groups=bDESIGN_Ch_root_BX$groups,
                                       tax=bTAX)
# dim(bEDGER_Ch_root_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Ch_root_BX_fit <- fit.model.edgeR(bEDGER_Ch_root_BX, bDESIGN_Ch_root_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Ch_BX <- makeContrasts(root_Changins_B73_WT - root_Changins_B73_bx1, levels=bDESIGN_Ch_root_BX$groups) 
bEDGER_Ch_root_BX_fit_lrt <- glmLRT(bEDGER_Ch_root_BX_fit, contrast=contrasts_root_Ch_BX)
bEDGER_Ch_root_BX_fit_lrt_topTags <- topTags(bEDGER_Ch_root_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
bEDGER_Ch_root_BX_fit_lrt_table <- topTags(bEDGER_Ch_root_BX_fit_lrt, n=nrow(bEDGER_Ch_root_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
# de-log abundance
bEDGER_Ch_root_BX_fit_lrt$table$CPM <- 2^(bEDGER_Ch_root_BX_fit_lrt$table$logCPM) 
# calculate percent Chlative abundance for all otus
bEDGER_Ch_root_BX_fit_lrt$table$percentAllSeq <- round( bEDGER_Ch_root_BX_fit_lrt$table$CPM*100/sum(bEDGER_Ch_root_BX_fit_lrt$table$CPM),2)
# select only diffeChntial otus
bEDGER_Ch_root_BX_fit_lrt_table$percentAllSeq <- bEDGER_Ch_root_BX_fit_lrt$table [rownames(bEDGER_Ch_root_BX_fit_lrt_table),]$percentAllSeq
# sum of Chl abundance of diffeChntial OTUs
bEDGER_Ch_root_BX_fit_lrt_sumPercent <- sum(bEDGER_Ch_root_BX_fit_lrt_table$percentAllSeq)

write.csv(bEDGER_Ch_root_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_bEDGER_Ch_root.csv") 
bEDGER_Ch_root_BX_fit_lrt_table_OTUs <- rownames(bEDGER_Ch_root_BX_fit_lrt_table)
bEDGER_Ch_root_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Ch_root_BX_fit$counts) * length(bEDGER_Ch_root_BX_fit_lrt_table_OTUs), 1)

```

In *Changins* root samples, `r paste(length(bEDGER_Ch_root_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  


```{r Aurora root bacteria WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Aurora root samples
# design file
bDESIGN_Au_root_BX <- subset(bDESIGN, bDESIGN$compartment=="root" & bDESIGN$location=="Aurora" & bDESIGN$genotype=="WT" | 
                                bDESIGN$compartment=="root" & bDESIGN$location=="Aurora" & bDESIGN$genotype=="bx1")
bDESIGN_Au_root_BX$groups <- droplevels(bDESIGN_Au_root_BX)$groups
table(bDESIGN_Au_root_BX$groups)
# OTU table
bDAT_Au_root_BX <- bDAT[, rownames(bDESIGN_Au_root_BX) ]
# dim(bDAT_Au_root_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_root_BX <- edgeR.object.TMM(dat=bDAT_Au_root_BX,
                                       threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                       groups=bDESIGN_Au_root_BX$groups,
                                       tax=bTAX)
# dim(bEDGER_Au_root_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Au_root_BX_fit <- fit.model.edgeR(bEDGER_Au_root_BX, bDESIGN_Au_root_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Au_BX <- makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx1, levels=bDESIGN_Au_root_BX$groups) 
bEDGER_Au_root_BX_fit_lrt <- glmLRT(bEDGER_Au_root_BX_fit, contrast=contrasts_root_Au_BX)
bEDGER_Au_root_BX_fit_lrt_topTags <- topTags(bEDGER_Au_root_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
bEDGER_Au_root_BX_fit_lrt_table <- topTags(bEDGER_Au_root_BX_fit_lrt, n=nrow(bEDGER_Au_root_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
bEDGER_Au_root_BX_fit_lrt$table$CPM <- 2^(bEDGER_Au_root_BX_fit_lrt$table$logCPM) 
bEDGER_Au_root_BX_fit_lrt$table$percentAllSeq <- round( bEDGER_Au_root_BX_fit_lrt$table$CPM*100/sum(bEDGER_Au_root_BX_fit_lrt$table$CPM),2)
bEDGER_Au_root_BX_fit_lrt_table$percentAllSeq <- bEDGER_Au_root_BX_fit_lrt$table [rownames(bEDGER_Au_root_BX_fit_lrt_table),]$percentAllSeq
bEDGER_Au_root_BX_fit_lrt_sumPercent <- sum(bEDGER_Au_root_BX_fit_lrt_table$percentAllSeq)

write.csv(bEDGER_Au_root_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_bEDGER_Au_root.csv") 
bEDGER_Au_root_BX_fit_lrt_table_OTUs <- rownames(bEDGER_Au_root_BX_fit_lrt_table)
bEDGER_Au_root_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Au_root_BX_fit$counts) * length(bEDGER_Au_root_BX_fit_lrt_table_OTUs), 1)
```

In *Aurora* root samples, `r paste(length(bEDGER_Au_root_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type W22 and mutant bx1(W22).  


```{r Reckenholz root bacteria WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Reckenholz root samples
# design file
bDESIGN_Re_root_BX <- subset(bDESIGN, bDESIGN$compartment=="root" & bDESIGN$location=="Reckenholz" & bDESIGN$background_genotype=="B73" | 
                                bDESIGN$compartment=="root" & bDESIGN$location=="Reckenholz" & bDESIGN$background_genotype=="bx1(B73)")
bDESIGN_Re_root_BX$groups <- droplevels(bDESIGN_Re_root_BX)$groups
table(bDESIGN_Re_root_BX$groups)
# OTU table
bDAT_Re_root_BX <- bDAT[, rownames(bDESIGN_Re_root_BX) ]
# dim(bDAT_Re_root_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Re_root_BX <- edgeR.object.TMM(dat=bDAT_Re_root_BX,
                                       threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                       groups=bDESIGN_Re_root_BX$groups,
                                       tax=bTAX)
# dim(bEDGER_Re_root_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Re_root_BX_fit <- fit.model.edgeR(bEDGER_Re_root_BX, bDESIGN_Re_root_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Re_BX <- makeContrasts(root_Reckenholz_B73_WT - root_Reckenholz_B73_bx1, levels=bDESIGN_Re_root_BX$groups) 
bEDGER_Re_root_BX_fit_lrt <- glmLRT(bEDGER_Re_root_BX_fit, contrast=contrasts_root_Re_BX)
bEDGER_Re_root_BX_fit_lrt_topTags <- topTags(bEDGER_Re_root_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
bEDGER_Re_root_BX_fit_lrt_table <- topTags(bEDGER_Re_root_BX_fit_lrt, n=nrow(bEDGER_Re_root_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
bEDGER_Re_root_BX_fit_lrt$table$CPM <- 2^(bEDGER_Re_root_BX_fit_lrt$table$logCPM) 
bEDGER_Re_root_BX_fit_lrt$table$percentAllSeq <- round( bEDGER_Re_root_BX_fit_lrt$table$CPM*100/sum(bEDGER_Re_root_BX_fit_lrt$table$CPM),2)
bEDGER_Re_root_BX_fit_lrt_table$percentAllSeq <- bEDGER_Re_root_BX_fit_lrt$table [rownames(bEDGER_Re_root_BX_fit_lrt_table),]$percentAllSeq
bEDGER_Re_root_BX_fit_lrt_sumPercent <- sum(bEDGER_Re_root_BX_fit_lrt_table$percentAllSeq)

write.csv(bEDGER_Re_root_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_bEDGER_Re_root.csv")
bEDGER_Re_root_BX_fit_lrt_table_OTUs <- rownames(bEDGER_Re_root_BX_fit_lrt_table)
bEDGER_Re_root_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(bEDGER_Re_root_BX_fit$counts) * length(bEDGER_Re_root_BX_fit_lrt_table_OTUs), 1)
```

In *Reckenholz* root samples, `r paste(length(bEDGER_Re_root_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  


#### MA plot of root bacteria

```{r MAplot all locations root bacteria WT vs bx1, fig.height=3, fig.width=5, warning=F, echo=F, results="hide"}
# custom function fun_maplot() to plot edgeR results as MA-plots
# source("functions/fun_maplot.R")

# define plotting parameters
par(mfrow=c(1,3), mar=c(4.5,4.5,1.6,1.6))

## Transform CPM abundances to % abundances 
bEDGER_Ch_root_BX_fit_lrt_topTags$table$CPM <- 2^bEDGER_Ch_root_BX_fit_lrt_topTags$table$logCPM # non log2 CPM
bEDGER_Ch_root_BX_fit_lrt_topTags$table$percent <- bEDGER_Ch_root_BX_fit_lrt_topTags$table$CPM/10000 # %

bEDGER_Re_root_BX_fit_lrt_topTags$table$CPM <- 2^bEDGER_Re_root_BX_fit_lrt_topTags$table$logCPM
bEDGER_Re_root_BX_fit_lrt_topTags$table$percent <- bEDGER_Re_root_BX_fit_lrt_topTags$table$CPM/10000

bEDGER_Au_root_BX_fit_lrt_topTags$table$CPM <- 2^bEDGER_Au_root_BX_fit_lrt_topTags$table$logCPM
bEDGER_Au_root_BX_fit_lrt_topTags$table$percent <- bEDGER_Au_root_BX_fit_lrt_topTags$table$CPM/10000

# axis range
xlim_ma <- c(0.01, max(bEDGER_Au_root_BX_fit_lrt_topTags$table$percent))
ylim_ma <- c(min(bEDGER_Au_root_BX_fit_lrt_topTags$table$logFC), max(bEDGER_Re_root_BX_fit_lrt_topTags$table$logFC))

# plot
fun_maplot(bEDGER_Ch_root_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change", xlab="Average abundance [%]", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(bEDGER_Au_root_BX_fit_lrt_topTags, title="Aurora", ylab="Log fold change", xlab="Average abundance [%]", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(bEDGER_Re_root_BX_fit_lrt_topTags, title="Reckenholz", ylab="Log fold change", xlab="Average abundance [%]", ylim = ylim_ma, xlim=xlim_ma)
```
  
\vspace{20pt}
Inspecting the OTU overlap of root bacteria between the locations:  
  
```{r Venn all locations root bacteria WT vs bx1, fig.height=3, fig.width=3, warning=F, echo=F}

par(mar=c(1,1,1,1))
Core_root_bacteria_venn <- venndiagram(bEDGER_Ch_root_BX_fit_lrt_table_OTUs, bEDGER_Au_root_BX_fit_lrt_table_OTUs, bEDGER_Re_root_BX_fit_lrt_table_OTUs, 
                                        type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                        lcol=level_cols_loc, lines=level_cols_loc,
                                        labels=c(paste("Changins (", bEDGER_Ch_root_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                 paste("Aurora (", bEDGER_Au_root_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                 paste("Reckenholz (", bEDGER_Re_root_BX_fit_lrt_sumPercent, "%)", sep="")), 
                                        title="")

pander(bTAX[Core_root_bacteria_venn$q1, c(3,5,6)], caption="shared bOTU across all locations")


# percent of total abundance (reviewer comment 06.2020)

# all location diff. OTUs abundances
bEDGER_percent_Re_root_BX <- topTags_percent_abundance(bEDGER_Re_root_BX_fit_lrt$table, bEDGER_Re_root_BX_fit_lrt_table)
bEDGER_percent_Au_root_BX <- topTags_percent_abundance(bEDGER_Au_root_BX_fit_lrt$table, bEDGER_Au_root_BX_fit_lrt_table)
bEDGER_percent_Ch_root_BX <- topTags_percent_abundance(bEDGER_Ch_root_BX_fit_lrt$table, bEDGER_Ch_root_BX_fit_lrt_table)

```
Differentially abundant OTUs account for `r round(bEDGER_Re_root_BX_fit_lrt_sumPercent,1)` % of total relative abundance in Reckenholz, `r round(bEDGER_Ch_root_BX_fit_lrt_sumPercent,1)` % in Changins and `r round(bEDGER_Au_root_BX_fit_lrt_sumPercent,1)` in Aurora.




\pagebreak

Inspecting the FAMILY overlap of BX-sensitve OTUs of root bacteria between the locations:  

```{r Venn all locations root bacteria WT vs bx1 family level, fig.height=3, fig.width=6, warning=F, echo=F, results="hide"}
bEDGER_Ch_root_BX_fit_lrt_table_fams_enriched <- unique(bEDGER_Ch_root_BX_fit_lrt_table$family[bEDGER_Ch_root_BX_fit_lrt_table$logFC>0])
bEDGER_Ch_root_BX_fit_lrt_table_fams_depleted <- unique(bEDGER_Ch_root_BX_fit_lrt_table$family[bEDGER_Ch_root_BX_fit_lrt_table$logFC<0])

bEDGER_Au_root_BX_fit_lrt_table_fams_enriched <- unique(bEDGER_Au_root_BX_fit_lrt_table$family[bEDGER_Au_root_BX_fit_lrt_table$logFC>0])
bEDGER_Au_root_BX_fit_lrt_table_fams_depleted <- unique(bEDGER_Au_root_BX_fit_lrt_table$family[bEDGER_Au_root_BX_fit_lrt_table$logFC<0])

bEDGER_Re_root_BX_fit_lrt_table_fams_enriched <- unique(bEDGER_Re_root_BX_fit_lrt_table$family[bEDGER_Re_root_BX_fit_lrt_table$logFC>0])
bEDGER_Re_root_BX_fit_lrt_table_fams_depleted <- unique(bEDGER_Re_root_BX_fit_lrt_table$family[bEDGER_Re_root_BX_fit_lrt_table$logFC<0])

par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_root_bacteria_fams_enriched <- venndiagram(bEDGER_Ch_root_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Au_root_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Re_root_BX_fit_lrt_table_fams_enriched, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c("Changins", "Aurora", "Reckenholz"), 
                                                 title="Family taxonomy of enriched OTUs", cex.main=0.7)
# Core_root_bacteria_fams_enriched$q1
Core_root_bacteria_fams_depleted <- venndiagram(bEDGER_Ch_root_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Au_root_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Re_root_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c("Changins", "Aurora", "Reckenholz"), 
                                                 title="Family taxonomy of depleted OTUs", cex.main=0.7)

# pander(Core_root_bacteria_fams_depleted$q1, caption="shared bOTU across all locations")

pdf("figures/Fig_6_Venn_BXsens_bOTUs_root.pdf", width=6, height=3)
par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_root_bacteria_fams_enriched <- venndiagram(bEDGER_Ch_root_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Au_root_BX_fit_lrt_table_fams_enriched, 
                                                 bEDGER_Re_root_BX_fit_lrt_table_fams_enriched, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", lwd=3,
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c(""), 
                                                 title="Family taxonomy of enriched OTUs", cex.main=0.7)

Core_root_bacteria_fams_depleted <- venndiagram(bEDGER_Ch_root_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Au_root_BX_fit_lrt_table_fams_depleted, 
                                                 bEDGER_Re_root_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", lwd=3,
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c(""), 
                                                 title="Family taxonomy of depleted OTUs", cex.main=0.7)
noshow <- dev.off()
```

The families of the BX enriched root bOTUs are: NA.  

The families of the BX depleted root bOTUs are: `r paste(Core_root_bacteria_fams_depleted$q1)`. 



\pagebreak

### Root fungi

```{r Changins root fungi WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Changins root samples
# design file
fDESIGN_Ch_root_BX <- subset(fDESIGN, fDESIGN$compartment=="root" & fDESIGN$location=="Changins" & fDESIGN$genotype=="WT" | 
                               fDESIGN$compartment=="root" & fDESIGN$location=="Changins" & fDESIGN$genotype=="bx1")
fDESIGN_Ch_root_BX$groups <- droplevels(fDESIGN_Ch_root_BX)$groups
table(fDESIGN_Ch_root_BX$groups)
# OTU table
fDAT_Ch_root_BX <- fDAT[, rownames(fDESIGN_Ch_root_BX) ]
# dim(fDAT_Ch_root_BX)


### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Ch_root_BX <- edgeR.object.TMM(dat=fDAT_Ch_root_BX,
                                      threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                      groups=fDESIGN_Ch_root_BX$groups,
                                      tax=fTAX)
# dim(fEDGER_Ch_root_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Ch_root_BX_fit <- fit.model.edgeR(fEDGER_Ch_root_BX, fDESIGN_Ch_root_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Ch_BX <- makeContrasts(root_Changins_B73_WT - root_Changins_B73_bx1, levels=fDESIGN_Ch_root_BX$groups) 
fEDGER_Ch_root_BX_fit_lrt <- glmLRT(fEDGER_Ch_root_BX_fit, contrast=contrasts_root_Ch_BX)
fEDGER_Ch_root_BX_fit_lrt_topTags <- topTags(fEDGER_Ch_root_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
fEDGER_Ch_root_BX_fit_lrt_table <- topTags(fEDGER_Ch_root_BX_fit_lrt, n=nrow(fEDGER_Ch_root_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
fEDGER_Ch_root_BX_fit_lrt$table$CPM <- 2^(fEDGER_Ch_root_BX_fit_lrt$table$logCPM) 
fEDGER_Ch_root_BX_fit_lrt$table$percentAllSeq <- round( fEDGER_Ch_root_BX_fit_lrt$table$CPM*100/sum(fEDGER_Ch_root_BX_fit_lrt$table$CPM),2)
fEDGER_Ch_root_BX_fit_lrt_table$percentAllSeq <- fEDGER_Ch_root_BX_fit_lrt$table [rownames(fEDGER_Ch_root_BX_fit_lrt_table),]$percentAllSeq
fEDGER_Ch_root_BX_fit_lrt_sumPercent <- sum(fEDGER_Ch_root_BX_fit_lrt_table$percentAllSeq)

write.csv(fEDGER_Ch_root_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_fEDGER_Ch_root.csv")
fEDGER_Ch_root_BX_fit_lrt_table_OTUs <- rownames(fEDGER_Ch_root_BX_fit_lrt_table)
fEDGER_Ch_root_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Ch_root_BX_fit$counts) * length(fEDGER_Ch_root_BX_fit_lrt_table_OTUs), 1)
```

In *Changins* root samples, `r paste(length(fEDGER_Ch_root_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  


```{r Aurora root fungi WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Aurora root samples
# design file
fDESIGN_Au_root_BX <- subset(fDESIGN, fDESIGN$compartment=="root" & fDESIGN$location=="Aurora" & fDESIGN$genotype=="WT" | 
                               fDESIGN$compartment=="root" & fDESIGN$location=="Aurora" & fDESIGN$genotype=="bx1")
fDESIGN_Au_root_BX$groups <- droplevels(fDESIGN_Au_root_BX)$groups
table(fDESIGN_Au_root_BX$groups)
# OTU table
fDAT_Au_root_BX <- fDAT[, rownames(fDESIGN_Au_root_BX) ]
# dim(fDAT_Au_root_BX)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_root_BX <- edgeR.object.TMM(dat=fDAT_Au_root_BX,
                                      threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                      groups=fDESIGN_Au_root_BX$groups,
                                      tax=fTAX)
# dim(fEDGER_Au_root_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Au_root_BX_fit <- fit.model.edgeR(fEDGER_Au_root_BX, fDESIGN_Au_root_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Au_BX <- makeContrasts(root_Aurora_W22_WT - root_Aurora_W22_bx1, levels=fDESIGN_Au_root_BX$groups) 
fEDGER_Au_root_BX_fit_lrt <- glmLRT(fEDGER_Au_root_BX_fit, contrast=contrasts_root_Au_BX)
fEDGER_Au_root_BX_fit_lrt_topTags <- topTags(fEDGER_Au_root_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
fEDGER_Au_root_BX_fit_lrt_table <- topTags(fEDGER_Au_root_BX_fit_lrt, n=nrow(fEDGER_Au_root_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
fEDGER_Au_root_BX_fit_lrt$table$CPM <- 2^(fEDGER_Au_root_BX_fit_lrt$table$logCPM) 
fEDGER_Au_root_BX_fit_lrt$table$percentAllSeq <- round( fEDGER_Au_root_BX_fit_lrt$table$CPM*100/sum(fEDGER_Au_root_BX_fit_lrt$table$CPM),2)
fEDGER_Au_root_BX_fit_lrt_table$percentAllSeq <- fEDGER_Au_root_BX_fit_lrt$table [rownames(fEDGER_Au_root_BX_fit_lrt_table),]$percentAllSeq
fEDGER_Au_root_BX_fit_lrt_sumPercent <- sum(fEDGER_Au_root_BX_fit_lrt_table$percentAllSeq)

write.csv(fEDGER_Au_root_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_fEDGER_Au_root.csv")
fEDGER_Au_root_BX_fit_lrt_table_OTUs <- rownames(fEDGER_Au_root_BX_fit_lrt_table)
fEDGER_Au_root_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Au_root_BX_fit$counts) * length(fEDGER_Au_root_BX_fit_lrt_table_OTUs), 1)
```

In *Aurora* root samples, `r paste(length(fEDGER_Au_root_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type W22 and mutant bx1(W22).  


```{r Reckenholz root fungi WT vs bx1, warning=F, echo=F, results="hide"}
### Subsetting to Reckenholz root samples
# design file
fDESIGN_Re_root_BX <- subset(fDESIGN, fDESIGN$compartment=="root" & fDESIGN$location=="Reckenholz" & fDESIGN$background_genotype=="B73" | 
                               fDESIGN$compartment=="root" & fDESIGN$location=="Reckenholz" & fDESIGN$background_genotype=="bx1(B73)")
fDESIGN_Re_root_BX$groups <- droplevels(fDESIGN_Re_root_BX)$groups
table(fDESIGN_Re_root_BX$groups)
# OTU table
fDAT_Re_root_BX <- fDAT[, rownames(fDESIGN_Re_root_BX) ]
# dim(fDAT_Re_root_BX)

### edgeR statistics
## Preparing EdgeR object, filtering and normalization (TMM) using
# custom function edgeR.object.TMM() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Re_root_BX <- edgeR.object.TMM(dat=fDAT_Re_root_BX,
                                      threshold.filter=6, # for comparable threshold use 6 (Aurora root WT samples have only 6 reps)
                                      groups=fDESIGN_Re_root_BX$groups,
                                      tax=fTAX)
# dim(fEDGER_Re_root_BX) 

## fit model, estimate dispersion (on edgeR object) using
# custom function fit.model.edgeR() - source("functions/fun_norm_edgeR_obj.R") 
fEDGER_Re_root_BX_fit <- fit.model.edgeR(fEDGER_Re_root_BX, fDESIGN_Re_root_BX$groups)

## glmLRT 
# WT vs bx1
contrasts_root_Re_BX <- makeContrasts(root_Reckenholz_B73_WT - root_Reckenholz_B73_bx1, levels=fDESIGN_Re_root_BX$groups) 
fEDGER_Re_root_BX_fit_lrt <- glmLRT(fEDGER_Re_root_BX_fit, contrast=contrasts_root_Re_BX)
fEDGER_Re_root_BX_fit_lrt_topTags <- topTags(fEDGER_Re_root_BX_fit_lrt, n=Inf, p.value=1)  # all stat tests
fEDGER_Re_root_BX_fit_lrt_table <- topTags(fEDGER_Re_root_BX_fit_lrt, n=nrow(fEDGER_Re_root_BX_fit$counts), adjust.method="BH", sort.by="PValue", p.value=0.05)[[1]]

# add percent sequence at the end of all lines
fEDGER_Re_root_BX_fit_lrt$table$CPM <- 2^(fEDGER_Re_root_BX_fit_lrt$table$logCPM) 
fEDGER_Re_root_BX_fit_lrt$table$percentAllSeq <- round( fEDGER_Re_root_BX_fit_lrt$table$CPM*100/sum(fEDGER_Re_root_BX_fit_lrt$table$CPM),2)
fEDGER_Re_root_BX_fit_lrt_table$percentAllSeq <- fEDGER_Re_root_BX_fit_lrt$table [rownames(fEDGER_Re_root_BX_fit_lrt_table),]$percentAllSeq
fEDGER_Re_root_BX_fit_lrt_sumPercent <- sum(fEDGER_Re_root_BX_fit_lrt_table$percentAllSeq)

write.csv(fEDGER_Re_root_BX_fit_lrt_table[,c(1:7,11:16)], "tables/Table_S13_fEDGER_Re_root.csv")
fEDGER_Re_root_BX_fit_lrt_table_OTUs <- rownames(fEDGER_Re_root_BX_fit_lrt_table)
fEDGER_Re_root_BX_fit_lrt_table_OTUs_percent <- round(100 / nrow(fEDGER_Re_root_BX_fit$counts) * length(fEDGER_Re_root_BX_fit_lrt_table_OTUs), 1)
```

In *Reckenholz* root samples, `r paste(length(fEDGER_Re_root_BX_fit_lrt_table_OTUs))` bOTUs vary significantly between wild-type B73 and mutant bx1(B73).  


#### MA plot of root fungi

```{r MAplot all locations root fungi WT vs bx1, fig.height=3, fig.width=5, warning=F, echo=F, results="hide"}
# custom function fun_maplot() to plot edgeR results as MA-plots
# source("functions/fun_maplot.R")

# define plotting parameters
par(mfrow=c(1,3), mar=c(4.5,4.5,1.6,1.6))

## Transform CPM abundances to % abundances 
fEDGER_Ch_root_BX_fit_lrt_topTags$table$CPM <- 2^fEDGER_Ch_root_BX_fit_lrt_topTags$table$logCPM # non log2 CPM
fEDGER_Ch_root_BX_fit_lrt_topTags$table$percent <- fEDGER_Ch_root_BX_fit_lrt_topTags$table$CPM/10000 # %

fEDGER_Re_root_BX_fit_lrt_topTags$table$CPM <- 2^fEDGER_Re_root_BX_fit_lrt_topTags$table$logCPM
fEDGER_Re_root_BX_fit_lrt_topTags$table$percent <- fEDGER_Re_root_BX_fit_lrt_topTags$table$CPM/10000

fEDGER_Au_root_BX_fit_lrt_topTags$table$CPM <- 2^fEDGER_Au_root_BX_fit_lrt_topTags$table$logCPM
fEDGER_Au_root_BX_fit_lrt_topTags$table$percent <- fEDGER_Au_root_BX_fit_lrt_topTags$table$CPM/10000

# axis range
xlim_ma <- c(0.01, max(fEDGER_Au_root_BX_fit_lrt_topTags$table$percent))
ylim_ma <- c(min(fEDGER_Au_root_BX_fit_lrt_topTags$table$logFC), max(fEDGER_Au_root_BX_fit_lrt_topTags$table$logFC))

# plot
fun_maplot(fEDGER_Ch_root_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change", xlab="Average abundance [%]", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(fEDGER_Au_root_BX_fit_lrt_topTags, title="Aurora", ylab="Log fold change", xlab="Average abundance [%]", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(fEDGER_Re_root_BX_fit_lrt_topTags, title="Reckenholz", ylab="Log fold change", xlab="Average abundance [%]", ylim = ylim_ma, xlim=xlim_ma)
```

\vspace{20pt}
Inspecting the OTU overlap of root fungi between the locations:  
  
```{r Venn all locations root fungi WT vs bx1, fig.height=3, fig.width=3, warning=F, echo=F, results="hide"}
par(mar=c(1,1,1,1))
Core_root_fungi_venn <- venndiagram(fEDGER_Ch_root_BX_fit_lrt_table_OTUs, fEDGER_Au_root_BX_fit_lrt_table_OTUs, fEDGER_Re_root_BX_fit_lrt_table_OTUs, 
                                       type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                       lcol=level_cols_loc, lines=level_cols_loc,
                                       labels=c(paste("Changins (", fEDGER_Ch_root_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                 paste("Aurora (", fEDGER_Au_root_BX_fit_lrt_sumPercent, "%)", sep=""),
                                                 paste("Reckenholz (", fEDGER_Re_root_BX_fit_lrt_sumPercent, "%)", sep="")), 
                                       title="")



# percent of total abundance (reviewer comment 06.2020)

# all location diff. OTUs abundances
fEDGER_percent_Re_root_BX <- topTags_percent_abundance(fEDGER_Re_root_BX_fit_lrt$table, fEDGER_Re_root_BX_fit_lrt_table)
fEDGER_percent_Au_root_BX <- topTags_percent_abundance(fEDGER_Au_root_BX_fit_lrt$table, fEDGER_Au_root_BX_fit_lrt_table)
fEDGER_percent_Ch_root_BX <- topTags_percent_abundance(fEDGER_Ch_root_BX_fit_lrt$table, fEDGER_Ch_root_BX_fit_lrt_table)
```


\pagebreak

Inspecting the FAMILY overlap of BX-sensitve OTUs of root fungi between the locations:  

```{r Venn all locations root fungi WT vs bx1 family level, fig.height=3, fig.width=6, warning=F, echo=F, results="hide"}
fEDGER_Ch_root_BX_fit_lrt_table_fams_enriched <- unique(fEDGER_Ch_root_BX_fit_lrt_table$family[fEDGER_Ch_root_BX_fit_lrt_table$logFC>0])
fEDGER_Ch_root_BX_fit_lrt_table_fams_depleted <- unique(fEDGER_Ch_root_BX_fit_lrt_table$family[fEDGER_Ch_root_BX_fit_lrt_table$logFC<0])

fEDGER_Au_root_BX_fit_lrt_table_fams_enriched <- unique(fEDGER_Au_root_BX_fit_lrt_table$family[fEDGER_Au_root_BX_fit_lrt_table$logFC>0])
fEDGER_Au_root_BX_fit_lrt_table_fams_depleted <- unique(fEDGER_Au_root_BX_fit_lrt_table$family[fEDGER_Au_root_BX_fit_lrt_table$logFC<0])

fEDGER_Re_root_BX_fit_lrt_table_fams_enriched <- unique(fEDGER_Re_root_BX_fit_lrt_table$family[fEDGER_Re_root_BX_fit_lrt_table$logFC>0])
fEDGER_Re_root_BX_fit_lrt_table_fams_depleted <- unique(fEDGER_Re_root_BX_fit_lrt_table$family[fEDGER_Re_root_BX_fit_lrt_table$logFC<0])

par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_root_fungi_fams_enriched <- venndiagram(fEDGER_Ch_root_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Au_root_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Re_root_BX_fit_lrt_table_fams_enriched, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc,
                                                 labels=c("Changins", "Aurora", "Reckenholz"), 
                                                 title="Family taxonomy of enriched OTUs", cex.main=0.7)
# Core_root_fungi_fams_enriched$q1
Core_root_fungi_fams_depleted <- venndiagram(fEDGER_Ch_root_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Au_root_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Re_root_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                                 lcol=level_cols_loc, lines=level_cols_loc, lwd=3,
                                                 labels=c("Changins", "Aurora", "Reckenholz"), 
                                                 title="Family taxonomy of depleted OTUs", cex.main=0.7)
# Core_root_fungi_fams_depleted$q1

pdf("figures/Fig_6_Venn_BXsens_fOTUs_root.pdf", width=6, height=3)
par(mfrow=c(1,2), mar=c(1,1,1,1))
Core_root_fungi_fams_enriched <- venndiagram(fEDGER_Ch_root_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Au_root_BX_fit_lrt_table_fams_enriched, 
                                                 fEDGER_Re_root_BX_fit_lrt_table_fams_enriched, type="3", lwd=3, xpd=F, printsub=F, tcol="black", diacol="black",  lcol=level_cols_loc, lines=level_cols_loc, labels="", title="Family taxonomy of enriched OTUs", cex.main=0.7)

Core_root_fungi_fams_depleted <- venndiagram(fEDGER_Ch_root_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Au_root_BX_fit_lrt_table_fams_depleted, 
                                                 fEDGER_Re_root_BX_fit_lrt_table_fams_depleted, 
                                                 type="3", xpd=T, printsub=F, tcol="black", diacol="black",  #lcol=level_cols_loc,
                                             lwd=3,
       lines=level_cols_loc,labels="", # labels=c("Changins", "Aurora", "Reckenholz"), 
title="Family taxonomy of depleted OTUs", cex.main=0.7)
noshow <- dev.off()

```

The families of the BX enriched root fOTUs are: `r paste(Core_root_fungi_fams_enriched$q1)` fungi.  

The families of the BX depleted root fOTUs are: NA. 


### Figure 5AB | MA plot and Venn

```{r Fig 5 MAplot all locations root bacteria and fungi WT vs bx1, fig.height=5, fig.width=15, warning=F, echo=F, results="hide"}
# custom function fun_maplot() to plot edgeR results as MA-plots
# source("functions/fun_maplot.R")

# redefine MAX for all plots the same
max_root_all_percent <- max(c(bEDGER_Au_root_BX_fit_lrt_topTags$table$percent,
                           bEDGER_Ch_root_BX_fit_lrt_topTags$table$percent,
                           bEDGER_Re_root_BX_fit_lrt_topTags$table$percent,
                           fEDGER_Au_root_BX_fit_lrt_topTags$table$percent,
                           fEDGER_Ch_root_BX_fit_lrt_topTags$table$percent,
                           fEDGER_Re_root_BX_fit_lrt_topTags$table$percent))
max_root_all_logFC <- max(c(bEDGER_Au_root_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Ch_root_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Re_root_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Au_root_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Ch_root_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Re_root_BX_fit_lrt_topTags$table$logFC))
min_root_all_logFC <- min(c(bEDGER_Au_root_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Ch_root_BX_fit_lrt_topTags$table$logFC,
                           bEDGER_Re_root_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Au_root_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Ch_root_BX_fit_lrt_topTags$table$logFC,
                           fEDGER_Re_root_BX_fit_lrt_topTags$table$logFC))

xlim_ma <- c(0.01, max_root_all_percent)
ylim_ma <- c(min_root_all_logFC, max_root_all_logFC)


pdf("figures/Fig_5AB_MAplot_root_percentSeq.pdf", width=12, height=6)
par(mfrow=c(2,4), mar=c(4.5,4.5,1.7,1.6))

# upper panel
fun_maplot(bEDGER_Ch_root_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change (M)", xlab="", ylim = ylim_ma, xlim=xlim_ma)
mtext("A", font=2, side=3, line=0.5, adj=-0.25, cex=1)
fun_maplot(bEDGER_Au_root_BX_fit_lrt_topTags, title="Aurora", ylab=" ", xlab="", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(bEDGER_Re_root_BX_fit_lrt_topTags, title="Reckenholz", ylab=" ", xlab="", ylim = ylim_ma, xlim=xlim_ma)
Core_root_bacteria_venn <- venndiagram(bEDGER_Ch_root_BX_fit_lrt_table_OTUs, bEDGER_Au_root_BX_fit_lrt_table_OTUs, bEDGER_Re_root_BX_fit_lrt_table_OTUs, 
                                        type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                        lcol=level_cols_loc, lines=level_cols_loc,
                                        labels=c(paste( bEDGER_Ch_root_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                 paste( bEDGER_Au_root_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                 paste( bEDGER_Re_root_BX_fit_lrt_sumPercent, "% cRA", sep="")), 
                                        title="Root bacteria")

# lower panel
fun_maplot(fEDGER_Ch_root_BX_fit_lrt_topTags, title="Changins", ylab="Log fold change (M)", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
mtext("B",font = 2, side=3, line=0.5, adj=-0.25, cex=1)
fun_maplot(fEDGER_Au_root_BX_fit_lrt_topTags, title="Aurora", ylab=" ", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
fun_maplot(fEDGER_Re_root_BX_fit_lrt_topTags, title="Reckenholz", ylab=" ", xlab="Average abundance [%] (A)", ylim = ylim_ma, xlim=xlim_ma)
Core_root_fungi_venn <- venndiagram(fEDGER_Ch_root_BX_fit_lrt_table_OTUs, fEDGER_Au_root_BX_fit_lrt_table_OTUs, fEDGER_Re_root_BX_fit_lrt_table_OTUs, 
                                       type="3", xpd=T, printsub=F, tcol="black", diacol="black", 
                                       lcol=level_cols_loc, lines=level_cols_loc,
                                       labels=c(paste( fEDGER_Ch_root_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                 paste( fEDGER_Au_root_BX_fit_lrt_sumPercent, "% cRA", sep=""),
                                                 paste( fEDGER_Re_root_BX_fit_lrt_sumPercent, "% cRA", sep="")), 
                                       title="Root fungi")
noshow <- dev.off()

```


\pagebreak

### Comparing compartments at each location

Inspecting the overlap between rhizosphere and root in *Changins*:  

```{r Venn rhizosphere and root bacteria in Changins, fig.height=3, fig.width=6, warning=F, echo=F, results="hide"}


par(mfrow=c(1,2), mar=c(1,1,1,1))
Ch_bacteria_venn <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Ch_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[1], lines=level_cols_loc[1],
                                labels=c("Rhizosphere", "Root"), title="Bacteria", cex.main=0.7)

Ch_fungi_venn <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Ch_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[1], lines=level_cols_loc[1],
                                labels=c("Rhizosphere", "Root"), title="Fungi", cex.main=0.7)
```

\vspace{20pt}
Inspecting the overlap between rhizosphere and root bacteria in *Aurora*:  

```{r Venn rhizosphere and root bacteria in Aurora, fig.height=3, fig.width=6, warning=F, echo=F, results="hide"}
par(mfrow=c(1,2), mar=c(1,1,1,1))
Au_bacteria_venn <- venndiagram(bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Au_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[2], lines=level_cols_loc[2],
                                labels=c("Rhizosphere", "Root"), title="Bacteria", cex.main=0.7)

Au_fungi_venn <- venndiagram(fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Au_root_BX_fit_lrt_table_OTUs,
                             type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                             lcol=level_cols_loc[2], lines=level_cols_loc[2],
                             labels=c("Rhizosphere", "Root"), title="Fungi", cex.main=0.7)
```

\vspace{20pt}
Inspecting the overlap between rhizosphere and root bacteria in *Reckenholz*:  
  
```{r Venn rhizosphere and root bacteria in Reckenholz, fig.height=3, fig.width=6, warning=F, echo=F, results="hide"}
par(mfrow=c(1,2), mar=c(1,1,1,1))
Re_bacteria_venn <- venndiagram(bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Re_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[3], lines=level_cols_loc[3],
                                labels=c("Rhizosphere", "Root"), title="Bacteria", cex.main=0.7)

Re_fungi_venn <- venndiagram(fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Re_root_BX_fit_lrt_table_OTUs,
                             type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                             lcol=level_cols_loc[3], lines=level_cols_loc[3],
                             labels=c("Rhizosphere", "Root"), title="Fungi", cex.main=0.7)
```


### Figure 5C | Rhizosphere vs root Venn Diagrams

```{r Figure 5C Venn rhizosphere and root bacteria all sites, fig.height=5, fig.width=5, warning=F, echo=F, results="hide"}
pdf("figures/Fig_5C_rhizosphere_vs_root.pdf", width=12, height=4)
par(mfrow=c(2,4), mar=c(4.5,4.5,1.6,1.6))

# upper panel
Ch_bacteria_venn <- venndiagram(bEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Ch_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[1], lines=level_cols_loc[1],
                                labels=c("Rhizosphere", "Root"), title="Changins bacteria")
mtext("C", font=2, side=3, line=0.5, adj=-0.25, cex=1)
Au_bacteria_venn <- venndiagram(bEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Au_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[2], lines=level_cols_loc[2],
                                labels=c("Rhizosphere", "Root"), title="Aurora bacteria")
Re_bacteria_venn <- venndiagram(bEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, bEDGER_Re_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[3], lines=level_cols_loc[3],
                                labels=c("Rhizosphere", "Root"), title="Reckenholz bacteria")
plot.new()
# lower panel
Ch_fungi_venn <- venndiagram(fEDGER_Ch_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Ch_root_BX_fit_lrt_table_OTUs,
                                type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                                lcol=level_cols_loc[1], lines=level_cols_loc[1],
                                labels=c("Rhizosphere", "Root"), title="Changins fungi")
Au_fungi_venn <- venndiagram(fEDGER_Au_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Au_root_BX_fit_lrt_table_OTUs,
                             type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                             lcol=level_cols_loc[2], lines=level_cols_loc[2],
                             labels=c("Rhizosphere", "Root"), title="Aurora fungi")
Re_fungi_venn <- venndiagram(fEDGER_Re_rhizo_BX_fit_lrt_table_OTUs, fEDGER_Re_root_BX_fit_lrt_table_OTUs,
                             type="2", xpd=T, printsub=F, tcol="black", diacol="black", 
                             lcol=level_cols_loc[3], lines=level_cols_loc[3],
                             labels=c("Rhizosphere", "Root"), title="Reckenholz fungi")
noshow <- dev.off()
```


\pagebreak

### Plotting BX-sensitive OTUs with their family taxonomy

We first identify for all BX-sensitive OTUs whether they are enriched or depleted by BX exudation. This is done at each site and for each compartment.

```{r defining enriched and depleted top differential OTUs, warning=F, echo=F}
# bacteria in the rhizosphere
bEDGER_Ch_rhizo_BX_fit_lrt_OTUs_WTenriched <- rownames(bEDGER_Ch_rhizo_BX_fit_lrt_table[bEDGER_Ch_rhizo_BX_fit_lrt_table$logFC > 0,])
bEDGER_Re_rhizo_BX_fit_lrt_OTUs_WTenriched <- rownames(bEDGER_Re_rhizo_BX_fit_lrt_table[bEDGER_Re_rhizo_BX_fit_lrt_table$logFC > 0,])
bEDGER_Au_rhizo_BX_fit_lrt_OTUs_WTenriched <- rownames(bEDGER_Au_rhizo_BX_fit_lrt_table[bEDGER_Au_rhizo_BX_fit_lrt_table$logFC > 0,])
bEDGER_Ch_rhizo_BX_fit_lrt_OTUs_bx1enriched <- rownames(bEDGER_Ch_rhizo_BX_fit_lrt_table[bEDGER_Ch_rhizo_BX_fit_lrt_table$logFC < 0,])
bEDGER_Re_rhizo_BX_fit_lrt_OTUs_bx1enriched <- rownames(bEDGER_Re_rhizo_BX_fit_lrt_table[bEDGER_Re_rhizo_BX_fit_lrt_table$logFC < 0,])
bEDGER_Au_rhizo_BX_fit_lrt_OTUs_bx1enriched <- rownames(bEDGER_Au_rhizo_BX_fit_lrt_table[bEDGER_Au_rhizo_BX_fit_lrt_table$logFC < 0,])
# bacteria in roots
bEDGER_Ch_root_BX_fit_lrt_OTUs_WTenriched <- rownames(bEDGER_Ch_root_BX_fit_lrt_table[bEDGER_Ch_root_BX_fit_lrt_table$logFC > 0,])
bEDGER_Re_root_BX_fit_lrt_OTUs_WTenriched <- rownames(bEDGER_Re_root_BX_fit_lrt_table[bEDGER_Re_root_BX_fit_lrt_table$logFC > 0,])
bEDGER_Au_root_BX_fit_lrt_OTUs_WTenriched <- rownames(bEDGER_Au_root_BX_fit_lrt_table[bEDGER_Au_root_BX_fit_lrt_table$logFC > 0,])
bEDGER_Ch_root_BX_fit_lrt_OTUs_bx1enriched <- rownames(bEDGER_Ch_root_BX_fit_lrt_table[bEDGER_Ch_root_BX_fit_lrt_table$logFC < 0,])
bEDGER_Re_root_BX_fit_lrt_OTUs_bx1enriched <- rownames(bEDGER_Re_root_BX_fit_lrt_table[bEDGER_Re_root_BX_fit_lrt_table$logFC < 0,])
bEDGER_Au_root_BX_fit_lrt_OTUs_bx1enriched <- rownames(bEDGER_Au_root_BX_fit_lrt_table[bEDGER_Au_root_BX_fit_lrt_table$logFC < 0,])
# fungi in the rhizosphere
fEDGER_Ch_rhizo_BX_fit_lrt_OTUs_WTenriched <- rownames(fEDGER_Ch_rhizo_BX_fit_lrt_table[fEDGER_Ch_rhizo_BX_fit_lrt_table$logFC > 0,])
fEDGER_Re_rhizo_BX_fit_lrt_OTUs_WTenriched <- rownames(fEDGER_Re_rhizo_BX_fit_lrt_table[fEDGER_Re_rhizo_BX_fit_lrt_table$logFC > 0,])
fEDGER_Au_rhizo_BX_fit_lrt_OTUs_WTenriched <- rownames(fEDGER_Au_rhizo_BX_fit_lrt_table[fEDGER_Au_rhizo_BX_fit_lrt_table$logFC > 0,])
fEDGER_Ch_rhizo_BX_fit_lrt_OTUs_bx1enriched <- rownames(fEDGER_Ch_rhizo_BX_fit_lrt_table[fEDGER_Ch_rhizo_BX_fit_lrt_table$logFC < 0,])
fEDGER_Re_rhizo_BX_fit_lrt_OTUs_bx1enriched <- rownames(fEDGER_Re_rhizo_BX_fit_lrt_table[fEDGER_Re_rhizo_BX_fit_lrt_table$logFC < 0,])
fEDGER_Au_rhizo_BX_fit_lrt_OTUs_bx1enriched <- rownames(fEDGER_Au_rhizo_BX_fit_lrt_table[fEDGER_Au_rhizo_BX_fit_lrt_table$logFC < 0,])
# fungi in roots
fEDGER_Ch_root_BX_fit_lrt_OTUs_WTenriched <- rownames(fEDGER_Ch_root_BX_fit_lrt_table[fEDGER_Ch_root_BX_fit_lrt_table$logFC > 0,])
fEDGER_Re_root_BX_fit_lrt_OTUs_WTenriched <- rownames(fEDGER_Re_root_BX_fit_lrt_table[fEDGER_Re_root_BX_fit_lrt_table$logFC > 0,])
fEDGER_Au_root_BX_fit_lrt_OTUs_WTenriched <- rownames(fEDGER_Au_root_BX_fit_lrt_table[fEDGER_Au_root_BX_fit_lrt_table$logFC > 0,])
fEDGER_Ch_root_BX_fit_lrt_OTUs_bx1enriched <- rownames(fEDGER_Ch_root_BX_fit_lrt_table[fEDGER_Ch_root_BX_fit_lrt_table$logFC < 0,])
fEDGER_Re_root_BX_fit_lrt_OTUs_bx1enriched <- rownames(fEDGER_Re_root_BX_fit_lrt_table[fEDGER_Re_root_BX_fit_lrt_table$logFC < 0,])
fEDGER_Au_root_BX_fit_lrt_OTUs_bx1enriched <- rownames(fEDGER_Au_root_BX_fit_lrt_table[fEDGER_Au_root_BX_fit_lrt_table$logFC < 0,])

```

For plotting, we then take the TMM transformed data and express the values as % relative abundance (RA). We use phyloseq functions for plotting. Mean relativae abundances are calculated for each genotype at each location and in both compartments. The data is graphed separately for BX-enriched (left) and BX-depleted OTUs (right). 



```{r BX-sensitive bOTUs barplot prepare data 1, warning=F, echo=F}
#### Bacteria

### We take the TMM transformed data (from the edgeR objects that were split by location and compartment)
## and express the values as % relative abundance (RA)
# source("functions/fun_norm_edgeR_obj.R")
bEDGER_Ch_rhizo_BX_RA <- convert.edgeR.to.datRA(bEDGER_Ch_rhizo_BX)
bEDGER_Re_rhizo_BX_RA <- convert.edgeR.to.datRA(bEDGER_Re_rhizo_BX)
bEDGER_Au_rhizo_BX_RA <- convert.edgeR.to.datRA(bEDGER_Au_rhizo_BX)
bEDGER_Ch_root_BX_RA <- convert.edgeR.to.datRA(bEDGER_Ch_root_BX)
bEDGER_Re_root_BX_RA <- convert.edgeR.to.datRA(bEDGER_Re_root_BX)
bEDGER_Au_root_BX_RA <- convert.edgeR.to.datRA(bEDGER_Au_root_BX)

# source("functions/fun_union_intersect_merge.R") 
bEDGER_BX_RA <- merge_six_row(bEDGER_Ch_rhizo_BX_RA, bEDGER_Re_rhizo_BX_RA, bEDGER_Au_rhizo_BX_RA, 
                              bEDGER_Ch_root_BX_RA, bEDGER_Re_root_BX_RA, bEDGER_Au_root_BX_RA)
bEDGER_BX_RA[is.na(bEDGER_BX_RA)] <- 0

## adjusting color vector with different greens to differentiate classes of proteobacteria
bTAX2$cols_phyla_2 <- bTAX2$cols_phyla
bTAX2$cols_phyla_2[bTAX2$cols_phyla_2=='palegreen2'] <- 'chartreuse'
bTAX2$cols_phyla_2[bTAX2$cols_phyla_2=='palegreen3'] <- 'olivedrab'
bTAX2$cols_phyla_2[bTAX2$cols_phyla_2=='palegreen4'] <- 'darkolivegreen'
bTAX2$cols_phyla_2[bTAX2$cols_phyla_2=='lightgrey'] <- 'grey'

### for plotting, we go into Phyloseq again
# re-create a phyloseq object with corrected label vector and with 
bPHYSEQ_tmm_ra <- phyloseq(sample_data(bDESIGN),
                           otu_table(bEDGER_BX_RA, taxa_are_rows=T),
                           tax_table(as.matrix(bTAX2[rownames(bEDGER_BX_RA),])) )

### calculate mean by genotype, location and compartment
# source("functions/fun_merge_samples_mean.R") 
bPHYSEQ_tmm_ra_mean <- merge_samples_mean(bPHYSEQ_tmm_ra, "compartment_location_genotype")

# fix missing location names
sample_data(bPHYSEQ_tmm_ra_mean)$location <- 
 ifelse(sample_data(bPHYSEQ_tmm_ra_mean)$location==1, 
        levels(sample_data(bPHYSEQ_tmm_ra)$location)[1], 
        ifelse(sample_data(bPHYSEQ_tmm_ra_mean)$location==2,
               levels(sample_data(bPHYSEQ_tmm_ra)$location)[2],
               levels(sample_data(bPHYSEQ_tmm_ra)$location)[3]))

# fix missing genotype names
sample_data(bPHYSEQ_tmm_ra_mean)$genotype <- 
 ifelse(sample_data(bPHYSEQ_tmm_ra_mean)$genotype==1, 
        levels(sample_data(bPHYSEQ_tmm_ra)$genotype)[1],
        ifelse(sample_data(bPHYSEQ_tmm_ra_mean)$genotype==2,
               levels(sample_data(bPHYSEQ_tmm_ra)$genotype)[2],
               ifelse(sample_data(bPHYSEQ_tmm_ra_mean)$genotype==3,
                      levels(sample_data(bPHYSEQ_tmm_ra)$genotype)[3],
                      ifelse(sample_data(bPHYSEQ_tmm_ra_mean)$genotype==4,
                             levels(sample_data(bPHYSEQ_tmm_ra)$genotype)[4],
                             levels(sample_data(bPHYSEQ_tmm_ra)$genotype)[5]))))
sample_data(bPHYSEQ_tmm_ra_mean)$genotype <- factor(sample_data(bPHYSEQ_tmm_ra_mean)$genotype, levels = c("no", "WT", "bx1", "bx2", "bx6"))


## split to genotypes for each location and compartment 
bPHYSEQ_Ch_rhizo_WT <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "rhizo_Changins_WT", bPHYSEQ_tmm_ra_mean); 
bPHYSEQ_Ch_rhizo_bx1 <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "rhizo_Changins_bx1", bPHYSEQ_tmm_ra_mean)
bPHYSEQ_Re_rhizo_WT <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "rhizo_Reckenholz_WT", bPHYSEQ_tmm_ra_mean); 
bPHYSEQ_Re_rhizo_bx1 <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "rhizo_Reckenholz_bx1", bPHYSEQ_tmm_ra_mean)
bPHYSEQ_Au_rhizo_WT <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "rhizo_Aurora_WT", bPHYSEQ_tmm_ra_mean); 
bPHYSEQ_Au_rhizo_bx1 <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "rhizo_Aurora_bx1", bPHYSEQ_tmm_ra_mean)
bPHYSEQ_Ch_root_WT <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "root_Changins_WT", bPHYSEQ_tmm_ra_mean); 
bPHYSEQ_Ch_root_bx1 <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "root_Changins_bx1", bPHYSEQ_tmm_ra_mean)
bPHYSEQ_Re_root_WT <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "root_Reckenholz_WT", bPHYSEQ_tmm_ra_mean); 
bPHYSEQ_Re_root_bx1 <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "root_Reckenholz_bx1", bPHYSEQ_tmm_ra_mean)
bPHYSEQ_Au_root_WT <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "root_Aurora_WT", bPHYSEQ_tmm_ra_mean); 
bPHYSEQ_Au_root_bx1 <- prune_samples(rownames(sample_data(bPHYSEQ_tmm_ra_mean)) == "root_Aurora_bx1", bPHYSEQ_tmm_ra_mean)


### Identifying BX-enriched and BX-depleted OTUs in each split (genotype:location:compartment) 
# vectors of BX-enriched and -depleted OTUs are defined about
col_subset <- c("Sample", "Abundance", "location", "labels", "family", "genotype")

## Changins rhizo 
bPHYSEQ_Ch_rhizo_WT_enriched <- subset_taxa(bPHYSEQ_Ch_rhizo_WT, rownames(tax_table(bPHYSEQ_Ch_rhizo_WT)) %in% bEDGER_Ch_rhizo_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Ch_rhizo_WT_enriched_melt <- psmelt(bPHYSEQ_Ch_rhizo_WT_enriched) [,col_subset]
bPHYSEQ_Ch_rhizo_WT_depleted <- subset_taxa(bPHYSEQ_Ch_rhizo_WT, rownames(tax_table(bPHYSEQ_Ch_rhizo_WT)) %in% bEDGER_Ch_rhizo_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Ch_rhizo_WT_depleted_melt <- psmelt(bPHYSEQ_Ch_rhizo_WT_depleted) [,col_subset]
bPHYSEQ_Ch_rhizo_bx1_enriched <- subset_taxa(bPHYSEQ_Ch_rhizo_bx1, rownames(tax_table(bPHYSEQ_Ch_rhizo_bx1)) %in% bEDGER_Ch_rhizo_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Ch_rhizo_bx1_enriched_melt <- psmelt(bPHYSEQ_Ch_rhizo_bx1_enriched) [,col_subset]
bPHYSEQ_Ch_rhizo_bx1_depleted <- subset_taxa(bPHYSEQ_Ch_rhizo_bx1, rownames(tax_table(bPHYSEQ_Ch_rhizo_bx1)) %in% bEDGER_Ch_rhizo_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Ch_rhizo_bx1_depleted_melt <- psmelt(bPHYSEQ_Ch_rhizo_bx1_depleted) [,col_subset]

## Aurora rhizo
bPHYSEQ_Au_rhizo_WT_enriched <- subset_taxa(bPHYSEQ_Au_rhizo_WT, rownames(tax_table(bPHYSEQ_Au_rhizo_WT)) %in% bEDGER_Au_rhizo_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Au_rhizo_WT_enriched_melt <- psmelt(bPHYSEQ_Au_rhizo_WT_enriched) [,col_subset]
bPHYSEQ_Au_rhizo_WT_depleted <- subset_taxa(bPHYSEQ_Au_rhizo_WT, rownames(tax_table(bPHYSEQ_Au_rhizo_WT)) %in% bEDGER_Au_rhizo_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Au_rhizo_WT_depleted_melt <- psmelt(bPHYSEQ_Au_rhizo_WT_depleted) [,col_subset]
bPHYSEQ_Au_rhizo_bx1_enriched <- subset_taxa(bPHYSEQ_Au_rhizo_bx1, rownames(tax_table(bPHYSEQ_Au_rhizo_bx1)) %in% bEDGER_Au_rhizo_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Au_rhizo_bx1_enriched_melt <- psmelt(bPHYSEQ_Au_rhizo_bx1_enriched) [,col_subset]
bPHYSEQ_Au_rhizo_bx1_depleted <- subset_taxa(bPHYSEQ_Au_rhizo_bx1, rownames(tax_table(bPHYSEQ_Au_rhizo_bx1)) %in% bEDGER_Au_rhizo_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Au_rhizo_bx1_depleted_melt <- psmelt(bPHYSEQ_Au_rhizo_bx1_depleted) [,col_subset]

## Reckenholz rhizo
bPHYSEQ_Re_rhizo_WT_enriched <- subset_taxa(bPHYSEQ_Re_rhizo_WT, rownames(tax_table(bPHYSEQ_Re_rhizo_WT)) %in% bEDGER_Re_rhizo_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Re_rhizo_WT_enriched_melt <- psmelt(bPHYSEQ_Re_rhizo_WT_enriched) [,col_subset]
bPHYSEQ_Re_rhizo_WT_depleted <- subset_taxa(bPHYSEQ_Re_rhizo_WT, rownames(tax_table(bPHYSEQ_Re_rhizo_WT)) %in% bEDGER_Re_rhizo_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Re_rhizo_WT_depleted_melt <- psmelt(bPHYSEQ_Re_rhizo_WT_depleted) [,col_subset]
bPHYSEQ_Re_rhizo_bx1_enriched <- subset_taxa(bPHYSEQ_Re_rhizo_bx1, rownames(tax_table(bPHYSEQ_Re_rhizo_bx1)) %in% bEDGER_Re_rhizo_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Re_rhizo_bx1_enriched_melt <- psmelt(bPHYSEQ_Re_rhizo_bx1_enriched) [,col_subset]
bPHYSEQ_Re_rhizo_bx1_depleted <- subset_taxa(bPHYSEQ_Re_rhizo_bx1, rownames(tax_table(bPHYSEQ_Re_rhizo_bx1)) %in% bEDGER_Re_rhizo_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Re_rhizo_bx1_depleted_melt <- psmelt(bPHYSEQ_Re_rhizo_bx1_depleted) [,col_subset]


## Changins root 
bPHYSEQ_Ch_root_WT_enriched <- subset_taxa(bPHYSEQ_Ch_root_WT, rownames(tax_table(bPHYSEQ_Ch_root_WT)) %in% bEDGER_Ch_root_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Ch_root_WT_enriched_melt <- psmelt(bPHYSEQ_Ch_root_WT_enriched) [,col_subset]
bPHYSEQ_Ch_root_WT_depleted <- subset_taxa(bPHYSEQ_Ch_root_WT, rownames(tax_table(bPHYSEQ_Ch_root_WT)) %in% bEDGER_Ch_root_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Ch_root_WT_depleted_melt <- psmelt(bPHYSEQ_Ch_root_WT_depleted) [,col_subset]
bPHYSEQ_Ch_root_bx1_enriched <- subset_taxa(bPHYSEQ_Ch_root_bx1, rownames(tax_table(bPHYSEQ_Ch_root_bx1)) %in% bEDGER_Ch_root_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Ch_root_bx1_enriched_melt <- psmelt(bPHYSEQ_Ch_root_bx1_enriched) [,col_subset]
bPHYSEQ_Ch_root_bx1_depleted <- subset_taxa(bPHYSEQ_Ch_root_bx1, rownames(tax_table(bPHYSEQ_Ch_root_bx1)) %in% bEDGER_Ch_root_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Ch_root_bx1_depleted_melt <- psmelt(bPHYSEQ_Ch_root_bx1_depleted) [,col_subset]

## Aurora root
bPHYSEQ_Au_root_WT_enriched <- subset_taxa(bPHYSEQ_Au_root_WT, rownames(tax_table(bPHYSEQ_Au_root_WT)) %in% bEDGER_Au_root_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Au_root_WT_enriched_melt <- psmelt(bPHYSEQ_Au_root_WT_enriched) [,col_subset]
bPHYSEQ_Au_root_WT_depleted <- subset_taxa(bPHYSEQ_Au_root_WT, rownames(tax_table(bPHYSEQ_Au_root_WT)) %in% bEDGER_Au_root_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Au_root_WT_depleted_melt <- psmelt(bPHYSEQ_Au_root_WT_depleted) [,col_subset]
bPHYSEQ_Au_root_bx1_enriched <- subset_taxa(bPHYSEQ_Au_root_bx1, rownames(tax_table(bPHYSEQ_Au_root_bx1)) %in% bEDGER_Au_root_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Au_root_bx1_enriched_melt <- psmelt(bPHYSEQ_Au_root_bx1_enriched) [,col_subset]
bPHYSEQ_Au_root_bx1_depleted <- subset_taxa(bPHYSEQ_Au_root_bx1, rownames(tax_table(bPHYSEQ_Au_root_bx1)) %in% bEDGER_Au_root_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Au_root_bx1_depleted_melt <- psmelt(bPHYSEQ_Au_root_bx1_depleted) [,col_subset]

## Reckenholz root
bPHYSEQ_Re_root_WT_enriched <- subset_taxa(bPHYSEQ_Re_root_WT, rownames(tax_table(bPHYSEQ_Re_root_WT)) %in% bEDGER_Re_root_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Re_root_WT_enriched_melt <- psmelt(bPHYSEQ_Re_root_WT_enriched) [,col_subset]
bPHYSEQ_Re_root_WT_depleted <- subset_taxa(bPHYSEQ_Re_root_WT, rownames(tax_table(bPHYSEQ_Re_root_WT)) %in% bEDGER_Re_root_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Re_root_WT_depleted_melt <- psmelt(bPHYSEQ_Re_root_WT_depleted) [,col_subset]
bPHYSEQ_Re_root_bx1_enriched <- subset_taxa(bPHYSEQ_Re_root_bx1, rownames(tax_table(bPHYSEQ_Re_root_bx1)) %in% bEDGER_Re_root_BX_fit_lrt_OTUs_WTenriched)
bPHYSEQ_Re_root_bx1_enriched_melt <- psmelt(bPHYSEQ_Re_root_bx1_enriched) [,col_subset]
bPHYSEQ_Re_root_bx1_depleted <- subset_taxa(bPHYSEQ_Re_root_bx1, rownames(tax_table(bPHYSEQ_Re_root_bx1)) %in% bEDGER_Re_root_BX_fit_lrt_OTUs_bx1enriched)
bPHYSEQ_Re_root_bx1_depleted_melt <- psmelt(bPHYSEQ_Re_root_bx1_depleted) [,col_subset]
```


```{r combining BX-sensitive bOTUs data, warning=F, echo=F}
#### preparing plotting matrices
### Rhizosphere
## Combining BX-enriched OTUs
bPHYSEQ_all_rhizo_enriched_melt <- rbind(bPHYSEQ_Ch_rhizo_WT_enriched_melt, 
                                         bPHYSEQ_Ch_rhizo_bx1_enriched_melt, 
                                         bPHYSEQ_Re_rhizo_WT_enriched_melt,  
                                         bPHYSEQ_Re_rhizo_bx1_enriched_melt, 
                                         bPHYSEQ_Au_rhizo_WT_enriched_melt, 
                                         bPHYSEQ_Au_rhizo_bx1_enriched_melt)
bPHYSEQ_all_rhizo_enriched_melt$labels_family <- as.character(bPHYSEQ_all_rhizo_enriched_melt$family)

# define low abundant families (below 1% RA in dataset) and create vector with new plotting labels for family names
for (i in (1:length(bPHYSEQ_all_rhizo_enriched_melt$family))) {
  bPHYSEQ_all_rhizo_enriched_melt$labels_family[i] <-
    ifelse(
      sum(bPHYSEQ_all_rhizo_enriched_melt[bPHYSEQ_all_rhizo_enriched_melt$family==bPHYSEQ_all_rhizo_enriched_melt$family[i],]$Abundance) > sum(bPHYSEQ_all_rhizo_enriched_melt$Abundance)/100,
      as.character(bPHYSEQ_all_rhizo_enriched_melt$family[i]),
      "low abundant family")
}

## Combining BX-depleted OTUs
bPHYSEQ_all_rhizo_depleted_melt <- rbind(bPHYSEQ_Ch_rhizo_WT_depleted_melt, 
                                         bPHYSEQ_Ch_rhizo_bx1_depleted_melt,
                                         bPHYSEQ_Re_rhizo_WT_depleted_melt, 
                                         bPHYSEQ_Re_rhizo_bx1_depleted_melt,
                                         bPHYSEQ_Au_rhizo_WT_depleted_melt, 
                                         bPHYSEQ_Au_rhizo_bx1_depleted_melt)
bPHYSEQ_all_rhizo_depleted_melt$labels_family <- as.character(bPHYSEQ_all_rhizo_depleted_melt$family)

# define low abundant families (below 1% RA in dataset) and create vector with new plotting labels for family names
for (i in (1:length(bPHYSEQ_all_rhizo_depleted_melt$family))) {
  bPHYSEQ_all_rhizo_depleted_melt$labels_family[i] <-
    ifelse(
      sum(bPHYSEQ_all_rhizo_depleted_melt[bPHYSEQ_all_rhizo_depleted_melt$family==bPHYSEQ_all_rhizo_depleted_melt$family[i],]$Abundance) > sum(bPHYSEQ_all_rhizo_depleted_melt$Abundance)/100,
      as.character(bPHYSEQ_all_rhizo_depleted_melt$family[i]),
      "low abundant family")
}


### Root
## Combining BX-enriched OTUs
bPHYSEQ_all_root_enriched_melt <- rbind(bPHYSEQ_Ch_root_WT_enriched_melt, 
                                        bPHYSEQ_Ch_root_bx1_enriched_melt, 
                                        bPHYSEQ_Re_root_WT_enriched_melt, 
                                        bPHYSEQ_Re_root_bx1_enriched_melt, 
                                        bPHYSEQ_Au_root_WT_enriched_melt, 
                                        bPHYSEQ_Au_root_bx1_enriched_melt)
bPHYSEQ_all_root_enriched_melt$labels_family <- as.character(bPHYSEQ_all_root_enriched_melt$family)

# define low abundant families (below 1% RA in dataset) and create vector with new plotting labels for family names
for (i in (1:length(bPHYSEQ_all_root_enriched_melt$family))) {
  bPHYSEQ_all_root_enriched_melt$labels_family[i] <-
    ifelse(
      sum(bPHYSEQ_all_root_enriched_melt[bPHYSEQ_all_root_enriched_melt$family==bPHYSEQ_all_root_enriched_melt$family[i],]$Abundance) > sum(bPHYSEQ_all_root_enriched_melt$Abundance)/100,
      as.character(bPHYSEQ_all_root_enriched_melt$family[i]),
      "low abundant family")
}

## Combining BX-depleted OTUs
bPHYSEQ_all_root_depleted_melt <- rbind(bPHYSEQ_Ch_root_WT_depleted_melt, 
                                        bPHYSEQ_Ch_root_bx1_depleted_melt, 
                                        bPHYSEQ_Re_root_WT_depleted_melt, 
                                        bPHYSEQ_Re_root_bx1_depleted_melt,
                                         bPHYSEQ_Au_root_WT_depleted_melt, 
                                        bPHYSEQ_Au_root_bx1_depleted_melt)
bPHYSEQ_all_root_depleted_melt$labels_family <- as.character(bPHYSEQ_all_root_depleted_melt$family)

# define low abundant families (below 1% RA in dataset) and create vector with new plotting labels for family names
for (i in (1:length(bPHYSEQ_all_root_depleted_melt$family))) {
  bPHYSEQ_all_root_depleted_melt$labels_family[i] <-
    ifelse(
      sum(bPHYSEQ_all_root_depleted_melt[bPHYSEQ_all_root_depleted_melt$family==bPHYSEQ_all_root_depleted_melt$family[i],]$Abundance) > sum(bPHYSEQ_all_root_depleted_melt$Abundance)/100,
      as.character(bPHYSEQ_all_root_depleted_melt$family[i]),
      "low abundant family")
}

```

```{r colors and factor order bacteria, warning= F, echo=F}

## Defining OTU colors by family (that are in the same color range as the phylum colors) 
# create dataframe of unique families + corresponding phylum
labs <- c("labels", "labels_family")
rhizo_enriched_bFamily <- as.data.frame(bPHYSEQ_all_rhizo_enriched_melt[,labs])
rhizo_depleted_bFamily <- as.data.frame(bPHYSEQ_all_rhizo_depleted_melt[,labs])
root_enriched_bFamily <- as.data.frame(bPHYSEQ_all_root_enriched_melt[,labs])
root_depleted_bFamily <- as.data.frame(bPHYSEQ_all_root_depleted_melt[,labs])
# source("functions/fun_union_intersect_merge.R")
all_bFamilies <- merge_four(unique(rhizo_enriched_bFamily), unique(rhizo_depleted_bFamily), 
                            unique(root_enriched_bFamily), unique(root_depleted_bFamily))

# unique families for color assignment, without low abundant
all_bFamilies_unique <- unique(all_bFamilies[ ! all_bFamilies$labels_family=="low abundant family" & ! all_bFamilies$labels=="unassigned",])

# generate range of colors for each family, based on phylum colors
all_bFamilies_unique_abu <- list()
for (i in 1:length(unique(all_bFamilies_unique$labels))) {
    all_bFamilies_unique_abu[[i]] <-
      all_bFamilies_unique[all_bFamilies_unique$labels == unique(all_bFamilies_unique$labels)[i], ]
    col <-  gsub("[1-9]", "", unique(bTAX2[bTAX2$labels %in% unique(all_bFamilies_unique$labels)[i], ]$cols_phyla_2))
    colfunc <- colorRampPalette(c(paste(col, 4, sep=""), paste(col, 1, sep="")))
    all_bFamilies_unique_abu[[i]]$cols <- colfunc(dim(all_bFamilies_unique_abu[[i]])[1])
    }

# re-pool all families with their colors in a unique-family dataframe, with unassigned and low abundant
temp <- data.frame(c("low abundant family", "unassigned"), c( "low abundant family", "unassigned"), c("lightgrey", "dimgrey"))
colnames(temp) <- c(labs[1], labs[2], "cols")
all_bFamilies_unique_abu_cols_summary <- rbind(do.call("rbind", all_bFamilies_unique_abu), temp)
all_bFamilies_unique_abu_cols_summary$cols[all_bFamilies_unique_abu_cols_summary$labels_family=="unassigned"] <- "dimgrey"

# color matrix for plot rhizo WT
rhizo_WT_unique_col_summary <- all_bFamilies_unique_abu_cols_summary[all_bFamilies_unique_abu_cols_summary$labels_family %in% bPHYSEQ_all_rhizo_enriched_melt$labels_family,] 
b_col_family_rhizo_WT <- rhizo_WT_unique_col_summary$cols 
names(b_col_family_rhizo_WT) <- rhizo_WT_unique_col_summary$labels_family

# color matrix for plot rhizo bx1
rhizo_bx1_unique_col_summary <- all_bFamilies_unique_abu_cols_summary[all_bFamilies_unique_abu_cols_summary$labels_family %in% bPHYSEQ_all_rhizo_depleted_melt$labels_family,] 
b_col_family_rhizo_bx1 <- rhizo_bx1_unique_col_summary$cols 
names(b_col_family_rhizo_bx1) <- rhizo_bx1_unique_col_summary$labels_family

# color matrix for plot root WT
root_WT_unique_col_summary <- all_bFamilies_unique_abu_cols_summary[all_bFamilies_unique_abu_cols_summary$labels_family %in% bPHYSEQ_all_root_enriched_melt$labels_family,] 
b_col_family_root_WT <- root_WT_unique_col_summary$cols 
names(b_col_family_root_WT) <- root_WT_unique_col_summary$labels_family

# color matrix for plot root bx1
root_bx1_unique_col_summary <- all_bFamilies_unique_abu_cols_summary[all_bFamilies_unique_abu_cols_summary$labels_family %in% bPHYSEQ_all_root_depleted_melt$labels_family,] 
b_col_family_root_bx1 <- root_bx1_unique_col_summary$cols 
names(b_col_family_root_bx1) <- root_bx1_unique_col_summary$labels_family


# order families and put "low abundant family" at the end in the data for plotting

# in WT rhizo
bPHYSEQ_all_rhizo_enriched_melt$labels_family <- factor(bPHYSEQ_all_rhizo_enriched_melt$labels_family, levels=unique(c(as.character(
  all_bFamilies_unique_abu_cols_summary$labels_family[all_bFamilies_unique_abu_cols_summary$labels_family %in% unique(bPHYSEQ_all_rhizo_enriched_melt$labels_family) &! all_bFamilies_unique_abu_cols_summary$labels_family=="low abundant family"]), "low abundant family")))

# order families in bx1 rhizo
bPHYSEQ_all_rhizo_depleted_melt$labels_family <- factor(bPHYSEQ_all_rhizo_depleted_melt$labels_family, levels=unique( c(as.character(all_bFamilies_unique_abu_cols_summary$labels_family[ all_bFamilies_unique_abu_cols_summary$labels_family %in% bPHYSEQ_all_rhizo_depleted_melt$labels_family &! all_bFamilies_unique_abu_cols_summary$labels_family=="low abundant family"]),"low abundant family")))

# order families in WT root
bPHYSEQ_all_root_enriched_melt$labels_family <- factor(bPHYSEQ_all_root_enriched_melt$labels_family, levels=unique( c(as.character(as.factor(all_bFamilies_unique_abu_cols_summary$labels_family[all_bFamilies_unique_abu_cols_summary$labels_family %in% bPHYSEQ_all_root_enriched_melt$labels_family &! all_bFamilies_unique_abu_cols_summary$labels_family=="low abundant family"]))
, "low abundant family")))

# order families in bx1 root
bPHYSEQ_all_root_depleted_melt$labels_family <- factor(bPHYSEQ_all_root_depleted_melt$labels_family, levels=unique( c(as.character(as.factor(all_bFamilies_unique_abu_cols_summary$labels_family[all_bFamilies_unique_abu_cols_summary$labels_family %in% bPHYSEQ_all_root_depleted_melt$labels_family & !all_bFamilies_unique_abu_cols_summary$labels_family=="low abundant family"]
)), "low abundant family")))


bPHYSEQ_all_rhizo_enriched_melt$location <- factor(bPHYSEQ_all_rhizo_enriched_melt$location, levels=c("Changins","Aurora", "Reckenholz"))
bPHYSEQ_all_rhizo_depleted_melt$location <- factor(bPHYSEQ_all_rhizo_depleted_melt$location, levels=c("Changins","Aurora", "Reckenholz"))
bPHYSEQ_all_root_enriched_melt$location <- factor(bPHYSEQ_all_root_enriched_melt$location, levels=c("Changins","Aurora", "Reckenholz"))
bPHYSEQ_all_root_depleted_melt$location <- factor(bPHYSEQ_all_root_depleted_melt$location, levels=c("Changins","Aurora", "Reckenholz"))
```


```{r bOTU BX-sensitive barplot, warning=F, echo=F}

### print plots for enriched and depleted for both compartments
## source("functions/fun_plotting_fig6.R") 
Plot_all_rhizo_BXenriched_bOTUs <- fun_plotting_fig6(bPHYSEQ_all_rhizo_enriched_melt, b_col_family_rhizo_WT, "labels_family") + ylim(c(0,70))
# print(Plot_all_rhizo_BXenriched_bOTUs) 
Plot_all_rhizo_BXdepleted_bOTUs <- fun_plotting_fig6(bPHYSEQ_all_rhizo_depleted_melt, b_col_family_rhizo_bx1, "labels_family") + ylim(c(0,70))
# print(Plot_all_rhizo_BXdepleted_bOTUs) 
Plot_all_root_BXenriched_bOTUs <- fun_plotting_fig6(bPHYSEQ_all_root_enriched_melt, b_col_family_root_WT, "labels_family") + ylim(c(0,25))
# print(Plot_all_root_BXenriched_bOTUs) 
Plot_all_root_BXdepleted_bOTUs <- fun_plotting_fig6(bPHYSEQ_all_root_depleted_melt, b_col_family_root_bx1, "labels_family") + ylim(c(0,25))
# print(Plot_all_root_BXdepleted_bOTUs) 

## Legend for all families of BX-sensitive bacteria barplots
# dataset
# source("functions/fun_union_intersect_merge.R")
all_bact <- merge_four(bPHYSEQ_all_rhizo_enriched_melt,
                       bPHYSEQ_all_rhizo_depleted_melt,
                       bPHYSEQ_all_root_enriched_melt,
                       bPHYSEQ_all_root_depleted_melt)

# reorder families with low abundant at the bottom                
all_bact$labels_family <- factor(all_bact$labels_family, levels=unique(
  c(as.character(as.factor(all_bFamilies_unique_abu_cols_summary$labels_family[all_bFamilies_unique_abu_cols_summary$labels_family %in% all_bact$labels_family & !all_bFamilies_unique_abu_cols_summary$labels_family=="low abundant family"]
)), "low abundant family")))

# color matrix for all
all_bFamilies_unique_abu_cols_summary$labels <- as.character(all_bFamilies_unique_abu_cols_summary$labels)
all_bFamilies_unique_abu_cols_summary <- rbind(all_bFamilies_unique_abu_cols_summary, c("low abundant","low abundant family", "lightgrey"))
all_colors <- all_bFamilies_unique_abu_cols_summary$cols
names(all_colors) <- all_bFamilies_unique_abu_cols_summary$labels_family

# plot legend
all_bact_plot <- ggplot(all_bact, aes_string(x="genotype", y="Abundance", fill="labels_family")) +
  geom_bar(stat ="identity") +
  scale_colour_manual(values=all_colors) +
  scale_fill_manual(values=all_colors) +
  guides(fill=guide_legend(title="family")) 

all_bact_legend <- cowplot::get_legend(all_bact_plot)
# grid.newpage()
# grid.draw(all_bact_legend)
```

### Bacteria

#### BX enriched and depleted bOTUs in the rhizosphere 
```{r Plot_all_rhizo_BXsensitive_bOTUs, echo=F, fig.height=8, fig.width=16}

# library(magrittr)
# library(multipanelfigure)
Plot_all_rhizo_BXsensitive_bOTUs <- multi_panel_figure(height=c(80,1), width=c(100,100,100))
Plot_all_rhizo_BXsensitive_bOTUs %<>%
  fill_panel(Plot_all_rhizo_BXenriched_bOTUs, column=1, row=1) %<>%
  fill_panel(Plot_all_rhizo_BXdepleted_bOTUs, column=2, row=1) %<>%
  fill_panel(all_bact_legend, column=3, row=1)
print(Plot_all_rhizo_BXsensitive_bOTUs)

```

#### BX enriched and depleted bOTUs in roots 
```{r Plot_all_root_BXsensitive_bOTUs, echo=F, fig.height=8, fig.width=16}

Plot_all_root_BXsensitive_bOTUs <- multi_panel_figure(height=c(80,1), width=c(100,100,100))
Plot_all_root_BXsensitive_bOTUs %<>%
  fill_panel(Plot_all_root_BXenriched_bOTUs, column=1, row=1) %<>%
  fill_panel(Plot_all_root_BXdepleted_bOTUs, column=2, row=1) %<>%
  fill_panel(all_bact_legend, column=3, row=1)
print(Plot_all_root_BXsensitive_bOTUs)

```

\pagebreak

### Fungi

```{r BX-sensitive fOTUs barplot prepare data 1, warning=F, echo=F}
### We take the TMM transformed data (from the edgeR objects that were split by location and compartment)
## and express the values as % relative abundance (RA)
# source("functions/fun_norm_edgeR_obj.R")
fEDGER_Ch_rhizo_BX_RA <- convert.edgeR.to.datRA(fEDGER_Ch_rhizo_BX)
fEDGER_Re_rhizo_BX_RA <- convert.edgeR.to.datRA(fEDGER_Re_rhizo_BX)
fEDGER_Au_rhizo_BX_RA <- convert.edgeR.to.datRA(fEDGER_Au_rhizo_BX)
fEDGER_Ch_root_BX_RA <- convert.edgeR.to.datRA(fEDGER_Ch_root_BX)
fEDGER_Re_root_BX_RA <- convert.edgeR.to.datRA(fEDGER_Re_root_BX)
fEDGER_Au_root_BX_RA <- convert.edgeR.to.datRA(fEDGER_Au_root_BX)

# source("functions/fun_union_intersect_merge.R") 
fEDGER_BX_RA <- merge_six_row(fEDGER_Ch_rhizo_BX_RA, fEDGER_Re_rhizo_BX_RA, fEDGER_Au_rhizo_BX_RA, fEDGER_Ch_root_BX_RA, fEDGER_Re_root_BX_RA, fEDGER_Au_root_BX_RA)
fEDGER_BX_RA[is.na(fEDGER_BX_RA)] <- 0

### ====================== ###
### FUNGuild analysis
## Export dataset for uploading to FUNGuild app (functional guild assessment)
ftax_init <- as.character(all_fDAT[,295])
names(ftax_init) <- rownames(all_fDAT)
fotutable_tmm_ra_for_funguild <- as.data.frame(cbind(otus=rownames(fEDGER_BX_RA), 
                                                     fEDGER_BX_RA,
                                                     taxonomy=ftax_init[rownames(fEDGER_BX_RA)]))
write.table(fotutable_tmm_ra_for_funguild, file="FUNGuild/fotutable_tmm_ra.txt", sep="\t", quote=F, row.names=F)
# upload on http://www.stbates.org/guilds/app.php and download resulting file

## import resulting FUNGuild file
FUNGuild_dat <- read.delim("FUNGuild/fotutable_tmm_ra.guilds.txt", row.names=1,  head=T, "\t")

# simplified guild annotation 
# FUNGuild_dat$simplified <- ifelse(FUNGuild_dat$Guild=="-" | FUNGuild_dat$Confidence.Ranking=="Possible", "Unassigned", 
#                                   ifelse(grepl("Plant Pathogen", FUNGuild_dat$Guild)==T, "Plant pathogen", "Other" ))  # wrong: 'Possible' assignmets become 'Unassigned'
FUNGuild_dat$simplified <- ifelse(grepl("Plant Pathogen", FUNGuild_dat$Guild)==T, "Plant pathogen", "Other" )
FUNGuild_dat$simplified[FUNGuild_dat$Guild=="-"] <- "Unassigned"
## ev. add confidence threshold: only probable and highly probable
# FUNGuild_dat$simplified[FUNGuild_dat$simplified=="Plant pathogen" & FUNGuild_dat$Confidence.Ranking=="Possible"] <- "Other"
# check: FUNGuild_dat[1:40,c("Guild", "Confidence.Ranking", "simplified")]


# combining Taxonomy and FUNGuild colomns (for easy plotting in phyloseq) 
fTAX_guild <- cbind(fTAX[rownames(FUNGuild_dat),], FUNGuild_dat[,c("Guild","simplified")])
### ====================== ###


### for plotting, we go into Phyloseq again
# re-create a phyloseq object with corrected label vector and with FUNGuild columns
fPHYSEQ_tmm_ra <- phyloseq(sample_data(fDESIGN),
                           otu_table(fEDGER_BX_RA, taxa_are_rows=T),
                           tax_table(as.matrix(fTAX_guild[rownames(fEDGER_BX_RA),])))

### calculate mean by genotype, location and compartment
# source("functions/fun_merge_samples_mean.R") 
fPHYSEQ_tmm_ra_mean <- merge_samples_mean(fPHYSEQ_tmm_ra, "compartment_location_genotype")


# fix missing location names
sample_data(fPHYSEQ_tmm_ra_mean)$location <- 
  ifelse(sample_data(fPHYSEQ_tmm_ra_mean)$location==1, 
         levels(sample_data(fPHYSEQ_tmm_ra)$location)[1], 
         ifelse(sample_data(fPHYSEQ_tmm_ra_mean)$location==2,
                levels(sample_data(fPHYSEQ_tmm_ra)$location)[2],
                levels(sample_data(fPHYSEQ_tmm_ra)$location)[3]))

# fix missing genotype names
sample_data(fPHYSEQ_tmm_ra_mean)$genotype <- 
  ifelse(sample_data(fPHYSEQ_tmm_ra_mean)$genotype==1, 
         levels(sample_data(fPHYSEQ_tmm_ra)$genotype)[1],
         ifelse(sample_data(fPHYSEQ_tmm_ra_mean)$genotype==2,
                levels(sample_data(fPHYSEQ_tmm_ra)$genotype)[2],
                ifelse(sample_data(fPHYSEQ_tmm_ra_mean)$genotype==3,
                       levels(sample_data(fPHYSEQ_tmm_ra)$genotype)[3],
                       ifelse(sample_data(fPHYSEQ_tmm_ra_mean)$genotype==4,
                              levels(sample_data(fPHYSEQ_tmm_ra)$genotype)[4],
                              levels(sample_data(fPHYSEQ_tmm_ra)$genotype)[5]))))
sample_data(fPHYSEQ_tmm_ra_mean)$genotype <- factor(sample_data(fPHYSEQ_tmm_ra_mean)$genotype, levels = c("no", "WT", "bx1", "bx2", "bx6"))

sample_data(fPHYSEQ_tmm_ra_mean)$compartment <- 
 ifelse(sample_data(fPHYSEQ_tmm_ra_mean)$compartment==1, 
        levels(sample_data(fPHYSEQ_tmm_ra)$compartment)[1],
               levels(sample_data(fPHYSEQ_tmm_ra)$compartment)[2])


## split to genotypes for each location and compartment 
fPHYSEQ_Ch_rhizo_WT <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "rhizo_Changins_WT", fPHYSEQ_tmm_ra_mean); 
fPHYSEQ_Ch_rhizo_bx1 <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "rhizo_Changins_bx1", fPHYSEQ_tmm_ra_mean)
fPHYSEQ_Re_rhizo_WT <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "rhizo_Reckenholz_WT", fPHYSEQ_tmm_ra_mean); 
fPHYSEQ_Re_rhizo_bx1 <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "rhizo_Reckenholz_bx1", fPHYSEQ_tmm_ra_mean)
fPHYSEQ_Au_rhizo_WT <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "rhizo_Aurora_WT", fPHYSEQ_tmm_ra_mean); 
fPHYSEQ_Au_rhizo_bx1 <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "rhizo_Aurora_bx1", fPHYSEQ_tmm_ra_mean)
fPHYSEQ_Ch_root_WT <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "root_Changins_WT", fPHYSEQ_tmm_ra_mean); 
fPHYSEQ_Ch_root_bx1 <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "root_Changins_bx1", fPHYSEQ_tmm_ra_mean)
fPHYSEQ_Re_root_WT <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "root_Reckenholz_WT", fPHYSEQ_tmm_ra_mean); 
fPHYSEQ_Re_root_bx1 <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "root_Reckenholz_bx1", fPHYSEQ_tmm_ra_mean)
fPHYSEQ_Au_root_WT <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "root_Aurora_WT", fPHYSEQ_tmm_ra_mean); 
fPHYSEQ_Au_root_bx1 <- prune_samples(rownames(sample_data(fPHYSEQ_tmm_ra_mean)) == "root_Aurora_bx1", fPHYSEQ_tmm_ra_mean)


### Identifying BX-enriched and BX-depleted OTUs in each split (genotype:location:compartment) 
# vectors of BX-enriched and -depleted OTUs are defined about

## Changins rhizo 
fPHYSEQ_Ch_rhizo_WT_enriched <- prune_taxa(fEDGER_Ch_root_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Ch_rhizo_WT)
fPHYSEQ_Ch_rhizo_WT_depleted <- prune_taxa(fEDGER_Ch_rhizo_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Ch_rhizo_WT)
fPHYSEQ_Ch_rhizo_bx1_enriched <- prune_taxa(fEDGER_Ch_rhizo_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Ch_rhizo_bx1) 
fPHYSEQ_Ch_rhizo_bx1_depleted <- prune_taxa(fEDGER_Ch_rhizo_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Ch_rhizo_bx1) 

## Aurora rhizo
fPHYSEQ_Au_rhizo_WT_enriched <- prune_taxa(fEDGER_Au_rhizo_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Au_rhizo_WT)
fPHYSEQ_Au_rhizo_WT_depleted <- prune_taxa(fEDGER_Au_rhizo_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Au_rhizo_WT)
fPHYSEQ_Au_rhizo_bx1_enriched <- prune_taxa(fEDGER_Au_rhizo_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Au_rhizo_bx1)
fPHYSEQ_Au_rhizo_bx1_depleted <- prune_taxa(fEDGER_Au_rhizo_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Au_rhizo_bx1)

## Reckenholz rhizo
fPHYSEQ_Re_rhizo_WT_enriched <- prune_taxa(fEDGER_Re_rhizo_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Re_rhizo_WT)
fPHYSEQ_Re_rhizo_WT_depleted <- prune_taxa(fEDGER_Re_rhizo_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Re_rhizo_WT)
fPHYSEQ_Re_rhizo_bx1_enriched <- prune_taxa(fEDGER_Re_rhizo_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Re_rhizo_bx1)
fPHYSEQ_Re_rhizo_bx1_depleted <- prune_taxa(fEDGER_Re_rhizo_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Re_rhizo_bx1) 

## Changins root 
fPHYSEQ_Ch_root_WT_enriched <- prune_taxa(fEDGER_Ch_root_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Ch_root_WT)
fPHYSEQ_Ch_root_WT_depleted <- prune_taxa(fEDGER_Ch_root_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Ch_root_WT)
fPHYSEQ_Ch_root_bx1_enriched <- prune_taxa(fEDGER_Ch_root_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Ch_root_bx1)
fPHYSEQ_Ch_root_bx1_depleted <- prune_taxa(fEDGER_Ch_root_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Ch_root_bx1)

## Aurora root
fPHYSEQ_Au_root_WT_enriched <- prune_taxa(fEDGER_Au_root_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Au_root_WT)
fPHYSEQ_Au_root_WT_depleted <- prune_taxa(fEDGER_Au_root_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Au_root_WT)
fPHYSEQ_Au_root_bx1_enriched <- prune_taxa(fEDGER_Au_root_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Au_root_bx1)
fPHYSEQ_Au_root_bx1_depleted <- prune_taxa(fEDGER_Au_root_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Au_root_bx1)

## Reckenholz root
fPHYSEQ_Re_root_WT_enriched <- prune_taxa(fEDGER_Re_root_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Re_root_WT)
fPHYSEQ_Re_root_WT_depleted <- prune_taxa(fEDGER_Re_root_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Re_root_WT)
fPHYSEQ_Re_root_bx1_enriched <- prune_taxa(fEDGER_Re_root_BX_fit_lrt_OTUs_WTenriched, fPHYSEQ_Re_root_bx1)
fPHYSEQ_Re_root_bx1_depleted <- prune_taxa(fEDGER_Re_root_BX_fit_lrt_OTUs_bx1enriched, fPHYSEQ_Re_root_bx1)

col_subset <- c("Sample", "Abundance", "location", "labels", "family", "genotype", "simplified")

fPHYSEQ_Ch_rhizo_WT_enriched_melt <- psmelt(fPHYSEQ_Ch_rhizo_WT_enriched) [,col_subset]
fPHYSEQ_Ch_rhizo_WT_depleted_melt <- psmelt(fPHYSEQ_Ch_rhizo_WT_depleted) [,col_subset]
fPHYSEQ_Ch_rhizo_bx1_enriched_melt <- psmelt(fPHYSEQ_Ch_rhizo_bx1_enriched) [,col_subset]
fPHYSEQ_Ch_rhizo_bx1_depleted_melt <- psmelt(fPHYSEQ_Ch_rhizo_bx1_depleted) [,col_subset]

fPHYSEQ_Au_rhizo_WT_enriched_melt <- psmelt(fPHYSEQ_Au_rhizo_WT_enriched) [,col_subset]
fPHYSEQ_Au_rhizo_WT_depleted_melt <- psmelt(fPHYSEQ_Au_rhizo_WT_depleted) [,col_subset]
fPHYSEQ_Au_rhizo_bx1_enriched_melt <- psmelt(fPHYSEQ_Au_rhizo_bx1_enriched) [,col_subset]
fPHYSEQ_Au_rhizo_bx1_depleted_melt <- psmelt(fPHYSEQ_Au_rhizo_bx1_depleted) [,col_subset]

fPHYSEQ_Re_rhizo_WT_enriched_melt <- psmelt(fPHYSEQ_Re_rhizo_WT_enriched) [,col_subset]
fPHYSEQ_Re_rhizo_WT_depleted_melt <- psmelt(fPHYSEQ_Re_rhizo_WT_depleted) [,col_subset]
fPHYSEQ_Re_rhizo_bx1_enriched_melt <- psmelt(fPHYSEQ_Re_rhizo_bx1_enriched) [,col_subset]
fPHYSEQ_Re_rhizo_bx1_depleted_melt <- psmelt(fPHYSEQ_Re_rhizo_bx1_depleted) [,col_subset]

fPHYSEQ_Ch_root_WT_enriched_melt <- psmelt(fPHYSEQ_Ch_root_WT_enriched) [,col_subset]
fPHYSEQ_Ch_root_WT_depleted_melt <- psmelt(fPHYSEQ_Ch_root_WT_depleted) [,col_subset]
fPHYSEQ_Ch_root_bx1_enriched_melt <- psmelt(fPHYSEQ_Ch_root_bx1_enriched) [,col_subset]
fPHYSEQ_Ch_root_bx1_depleted_melt <- psmelt(fPHYSEQ_Ch_root_bx1_depleted) [,col_subset]

fPHYSEQ_Au_root_WT_enriched_melt <- psmelt(fPHYSEQ_Au_root_WT_enriched) [,col_subset]
fPHYSEQ_Au_root_WT_depleted_melt <- psmelt(fPHYSEQ_Au_root_WT_depleted) [,col_subset]
fPHYSEQ_Au_root_bx1_enriched_melt <- psmelt(fPHYSEQ_Au_root_bx1_enriched) [,col_subset]
fPHYSEQ_Au_root_bx1_depleted_melt <- psmelt(fPHYSEQ_Au_root_bx1_depleted) [,col_subset]

fPHYSEQ_Re_root_WT_enriched_melt <- psmelt(fPHYSEQ_Re_root_WT_enriched) [,col_subset]
fPHYSEQ_Re_root_WT_depleted_melt <- psmelt(fPHYSEQ_Re_root_WT_depleted) [,col_subset]
fPHYSEQ_Re_root_bx1_enriched_melt <- psmelt(fPHYSEQ_Re_root_bx1_enriched) [,col_subset]
fPHYSEQ_Re_root_bx1_depleted_melt <- psmelt(fPHYSEQ_Re_root_bx1_depleted) [,col_subset]


## Combining in rhizo / root dataset OTUs
fPHYSEQ_all_rhizo_enriched_melt <- rbind(fPHYSEQ_Ch_rhizo_WT_enriched_melt, 
                                         fPHYSEQ_Ch_rhizo_bx1_enriched_melt, 
                                         fPHYSEQ_Re_rhizo_WT_enriched_melt,  
                                         fPHYSEQ_Re_rhizo_bx1_enriched_melt, 
                                         fPHYSEQ_Au_rhizo_WT_enriched_melt, 
                                         fPHYSEQ_Au_rhizo_bx1_enriched_melt)
fPHYSEQ_all_rhizo_depleted_melt <- rbind(fPHYSEQ_Ch_rhizo_WT_depleted_melt, 
                                         fPHYSEQ_Ch_rhizo_bx1_depleted_melt,
                                         fPHYSEQ_Re_rhizo_WT_depleted_melt, 
                                         fPHYSEQ_Re_rhizo_bx1_depleted_melt,
                                         fPHYSEQ_Au_rhizo_WT_depleted_melt, 
                                         fPHYSEQ_Au_rhizo_bx1_depleted_melt)
fPHYSEQ_all_root_enriched_melt <- rbind(fPHYSEQ_Ch_root_WT_enriched_melt,
                                        fPHYSEQ_Ch_root_bx1_enriched_melt, 
                                        fPHYSEQ_Re_root_WT_enriched_melt, 
                                        fPHYSEQ_Re_root_bx1_enriched_melt, 
                                        fPHYSEQ_Au_root_WT_enriched_melt, 
                                        fPHYSEQ_Au_root_bx1_enriched_melt)
fPHYSEQ_all_root_depleted_melt <- rbind(fPHYSEQ_Ch_root_WT_depleted_melt,
                                        fPHYSEQ_Ch_root_bx1_depleted_melt, 
                                        fPHYSEQ_Re_root_WT_depleted_melt, 
                                        fPHYSEQ_Re_root_bx1_depleted_melt,
                                        fPHYSEQ_Au_root_WT_depleted_melt, 
                                        fPHYSEQ_Au_root_bx1_depleted_melt)
```


```{r define low-abundant BX-sensitive fOTUs data, warning=F, echo=F}
#### preparing plotting matrices
### Rhizosphere

fPHYSEQ_all_rhizo_enriched_melt$labels_family <- as.character(fPHYSEQ_all_rhizo_enriched_melt$family)

# define low abundant families (below 1% RA in dataset) and create vector with new plotting labels for family names
for (i in (1:length(fPHYSEQ_all_rhizo_enriched_melt$family))) {
  fPHYSEQ_all_rhizo_enriched_melt$labels_family[i] <-
    ifelse(
      sum(fPHYSEQ_all_rhizo_enriched_melt[fPHYSEQ_all_rhizo_enriched_melt$family==fPHYSEQ_all_rhizo_enriched_melt$family[i],]$Abundance) > sum(fPHYSEQ_all_rhizo_enriched_melt$Abundance)/100,
      as.character(fPHYSEQ_all_rhizo_enriched_melt$family[i]),
      "low abundant family")
}

fPHYSEQ_all_rhizo_depleted_melt$labels_family <- as.character(fPHYSEQ_all_rhizo_depleted_melt$family)

# define low abundant families (below 1% RA in dataset) and create vector with new plotting labels for family names
for (i in (1:length(fPHYSEQ_all_rhizo_depleted_melt$family))) {
  fPHYSEQ_all_rhizo_depleted_melt$labels_family[i] <-
    ifelse(
      sum(fPHYSEQ_all_rhizo_depleted_melt[fPHYSEQ_all_rhizo_depleted_melt$family==fPHYSEQ_all_rhizo_depleted_melt$family[i],]$Abundance) > sum(fPHYSEQ_all_rhizo_depleted_melt$Abundance)/100,
      as.character(fPHYSEQ_all_rhizo_depleted_melt$family[i]),
      "low abundant family")
}

### Root

fPHYSEQ_all_root_enriched_melt$labels_family <- as.character(fPHYSEQ_all_root_enriched_melt$family)

# define low abundant families (below 1% RA in dataset) and create vector with new plotting labels for family names
for (i in (1:length(fPHYSEQ_all_root_enriched_melt$family))) {
  fPHYSEQ_all_root_enriched_melt$labels_family[i] <-
    ifelse(
      sum(fPHYSEQ_all_root_enriched_melt[fPHYSEQ_all_root_enriched_melt$family==fPHYSEQ_all_root_enriched_melt$family[i],]$Abundance) > sum(fPHYSEQ_all_root_enriched_melt$Abundance)/100,
      as.character(fPHYSEQ_all_root_enriched_melt$family[i]),
      "low abundant family")
}

fPHYSEQ_all_root_depleted_melt$labels_family <- as.character(fPHYSEQ_all_root_depleted_melt$family)

for (i in (1:length(fPHYSEQ_all_root_depleted_melt$family))) {
  fPHYSEQ_all_root_depleted_melt$labels_family[i] <-
    ifelse(
      sum(fPHYSEQ_all_root_depleted_melt[fPHYSEQ_all_root_depleted_melt$family==fPHYSEQ_all_root_depleted_melt$family[i],]$Abundance) > sum(fPHYSEQ_all_root_depleted_melt$Abundance)/100,
      as.character(fPHYSEQ_all_root_depleted_melt$family[i]),
      "low abundant family")
}
```

```{r colors and factor order fungi, warning= F, echo=F}

## Defining OTU colors by family (that are in the same color range as the phylum colors) 
# create dataframe of unique families + corresponding phylum
labs <- c("labels", "labels_family")
rhizo_enriched_fFamily <- as.data.frame(fPHYSEQ_all_rhizo_enriched_melt[,labs])
rhizo_depleted_fFamily <- as.data.frame(fPHYSEQ_all_rhizo_depleted_melt[,labs])
root_enriched_fFamily <- as.data.frame(fPHYSEQ_all_root_enriched_melt[,labs])
root_depleted_fFamily <- as.data.frame(fPHYSEQ_all_root_depleted_melt[,labs])
# source("functions/fun_union_intersect_merge.R")
all_fFamilies <- merge_four(unique(rhizo_enriched_fFamily), unique(rhizo_depleted_fFamily), 
                            unique(root_enriched_fFamily), unique(root_depleted_fFamily))

# unique families for color assignment, without low abundant
all_fFamilies_unique <- unique(all_fFamilies[ ! all_fFamilies$labels_family=="low abundant family" & ! all_fFamilies$labels=="unassigned",])

# generate range of colors for each family, based on phylum colors
all_fFamilies_unique_abu <- list()
for (i in 1:length(unique(all_fFamilies_unique$labels))) {
  all_fFamilies_unique_abu[[i]] <-
    all_fFamilies_unique[all_fFamilies_unique$labels == unique(all_fFamilies_unique$labels)[i],]
  col <-  gsub("[1-9]", "", unique(fTAX[fTAX$labels %in% unique(all_fFamilies_unique$labels)[i],]$cols_phyla))
  ifelse (unique(all_fFamilies_unique$labels)[i] == "Ascomycota", 
          colfunc <- colorRampPalette(c("dodgerblue4", "lightblue1")),
          ifelse (unique(all_fFamilies_unique$labels)[i] == "Olpidiomycota",
                  colfunc <- colorRampPalette(c("ivory4", "ivory1")),
                  colfunc <- colorRampPalette(c(paste(col, 4, sep=""), paste(col, 1, sep="")))
          ))
  all_fFamilies_unique_abu[[i]]$cols <- colfunc(dim(all_fFamilies_unique_abu[[i]])[1])
}


# re-pool all families with their colors in a unique-family dataframe, with unassigned and low abundant
temp <- data.frame(c("low abundant family", "unassigned"), c( "low abundant family", "unassigned"), c("lightgrey", "dimgrey"))
colnames(temp) <- c(labs[1], labs[2], "cols")
all_fFamilies_unique_abu_cols_summary <- rbind(do.call("rbind", all_fFamilies_unique_abu), temp)
all_fFamilies_unique_abu_cols_summary$cols[all_fFamilies_unique_abu_cols_summary$labels_family=="unassigned"] <- "dimgrey"

# color matrix for plot rhizo WT
rhizo_WT_unique_col_summary <- all_fFamilies_unique_abu_cols_summary[all_fFamilies_unique_abu_cols_summary$labels_family %in% fPHYSEQ_all_rhizo_enriched_melt$labels_family,] 
f_col_family_rhizo_WT <- rhizo_WT_unique_col_summary$cols 
names(f_col_family_rhizo_WT) <- rhizo_WT_unique_col_summary$labels_family

# color matrix for plot rhizo bx1
rhizo_bx1_unique_col_summary <- all_fFamilies_unique_abu_cols_summary[all_fFamilies_unique_abu_cols_summary$labels_family %in% fPHYSEQ_all_rhizo_depleted_melt$labels_family,] 
f_col_family_rhizo_bx1 <- rhizo_bx1_unique_col_summary$cols 
names(f_col_family_rhizo_bx1) <- rhizo_bx1_unique_col_summary$labels_family

# color matrix for plot root WT
root_WT_unique_col_summary <- all_fFamilies_unique_abu_cols_summary[all_fFamilies_unique_abu_cols_summary$labels_family %in% fPHYSEQ_all_root_enriched_melt$labels_family,] 
f_col_family_root_WT <- root_WT_unique_col_summary$cols 
names(f_col_family_root_WT) <- root_WT_unique_col_summary$labels_family

# color matrix for plot root bx1
root_bx1_unique_col_summary <- all_fFamilies_unique_abu_cols_summary[all_fFamilies_unique_abu_cols_summary$labels_family %in% fPHYSEQ_all_root_depleted_melt$labels_family,] 
f_col_family_root_bx1 <- root_bx1_unique_col_summary$cols 
names(f_col_family_root_bx1) <- root_bx1_unique_col_summary$labels_family


# order families and put "low abundant family" at the end in the data for plotting
# in WT rhizo
fPHYSEQ_all_rhizo_enriched_melt$labels_family <- factor(fPHYSEQ_all_rhizo_enriched_melt$labels_family, levels=unique(c(as.character(
  all_fFamilies_unique_abu_cols_summary$labels_family[all_fFamilies_unique_abu_cols_summary$labels_family %in% unique(fPHYSEQ_all_rhizo_enriched_melt$labels_family) &! all_fFamilies_unique_abu_cols_summary$labels_family=="low abundant family"]), "low abundant family")))

# order families in bx1 rhizo
fPHYSEQ_all_rhizo_depleted_melt$labels_family <- factor(fPHYSEQ_all_rhizo_depleted_melt$labels_family, levels=unique( c(as.character(all_fFamilies_unique_abu_cols_summary$labels_family[ all_fFamilies_unique_abu_cols_summary$labels_family %in% fPHYSEQ_all_rhizo_depleted_melt$labels_family &! all_fFamilies_unique_abu_cols_summary$labels_family=="low abundant family"]),"low abundant family")))

# order families in WT root
fPHYSEQ_all_root_enriched_melt$labels_family <- factor(fPHYSEQ_all_root_enriched_melt$labels_family, levels=unique( c(as.character(as.factor(all_fFamilies_unique_abu_cols_summary$labels_family[all_fFamilies_unique_abu_cols_summary$labels_family %in% fPHYSEQ_all_root_enriched_melt$labels_family &! all_fFamilies_unique_abu_cols_summary$labels_family=="low abundant family"])), "low abundant family")))

# order families in bx1 root
fPHYSEQ_all_root_depleted_melt$labels_family <- factor(fPHYSEQ_all_root_depleted_melt$labels_family, levels=unique( c(as.character(as.factor(all_fFamilies_unique_abu_cols_summary$labels_family[all_fFamilies_unique_abu_cols_summary$labels_family %in% fPHYSEQ_all_root_depleted_melt$labels_family & !all_fFamilies_unique_abu_cols_summary$labels_family=="low abundant family"])), "low abundant family")))

fPHYSEQ_all_rhizo_enriched_melt$location <- factor(fPHYSEQ_all_rhizo_enriched_melt$location, levels=c("Changins","Aurora", "Reckenholz"))
fPHYSEQ_all_rhizo_depleted_melt$location <- factor(fPHYSEQ_all_rhizo_depleted_melt$location, levels=c("Changins","Aurora", "Reckenholz"))
fPHYSEQ_all_root_enriched_melt$location <- factor(fPHYSEQ_all_root_enriched_melt$location, levels=c("Changins","Aurora", "Reckenholz"))
fPHYSEQ_all_root_depleted_melt$location <- factor(fPHYSEQ_all_root_depleted_melt$location, levels=c("Changins","Aurora", "Reckenholz"))

```


```{r fOTU BX-sensitive barplot, warning=F, echo=F}
### print plots for enriched and depleted for both compartments
## source("functions/fun_plotting_fig6.R") 
Plot_all_rhizo_BXenriched_fOTUs <- fun_plotting_fig6(fPHYSEQ_all_rhizo_enriched_melt, f_col_family_rhizo_WT, "labels_family") + ylim(c(0,25))
# print(Plot_all_rhizo_BXenriched_fOTUs) 
Plot_all_rhizo_BXdepleted_fOTUs <- fun_plotting_fig6(fPHYSEQ_all_rhizo_depleted_melt, f_col_family_rhizo_bx1, "labels_family") + ylim(c(0,25))
# print(Plot_all_rhizo_BXdepleted_fOTUs) 
Plot_all_root_BXenriched_fOTUs <- fun_plotting_fig6(fPHYSEQ_all_root_enriched_melt, f_col_family_root_WT, "labels_family") + ylim(c(0,60))
# print(Plot_all_root_BXenriched_fOTUs) 
Plot_all_root_BXdepleted_fOTUs <- fun_plotting_fig6(fPHYSEQ_all_root_depleted_melt, f_col_family_root_bx1, "labels_family") + ylim(c(0,60))
# print(Plot_all_root_BXdepleted_fOTUs) 

## Legend for all families of BX-sensitive bacteria barplots
# dataset
# source("functions/fun_union_intersect_merge.R")
all_fun <- merge_four(fPHYSEQ_all_rhizo_enriched_melt,
                       fPHYSEQ_all_rhizo_depleted_melt,
                       fPHYSEQ_all_root_enriched_melt,
                       fPHYSEQ_all_root_depleted_melt)

# reorder families with low abundant at the bottom                
all_fun$labels_family <- factor(all_fun$labels_family, levels=unique(
  c(as.character(as.factor(all_fFamilies_unique_abu_cols_summary$labels_family[all_fFamilies_unique_abu_cols_summary$labels_family %in% all_fun$labels_family & !all_fFamilies_unique_abu_cols_summary$labels_family=="low abundant family"])), "low abundant family")))

# color matrix for all
all_fFamilies_unique_abu_cols_summary$labels <- as.character(all_fFamilies_unique_abu_cols_summary$labels)
all_fFamilies_unique_abu_cols_summary <- rbind(all_fFamilies_unique_abu_cols_summary, c("low abundant","low abundant family", "lightgrey"))
all_colors <- all_fFamilies_unique_abu_cols_summary$cols
names(all_colors) <- all_fFamilies_unique_abu_cols_summary$labels_family

# plot legend
all_fun_plot <- ggplot(all_fun, aes_string(x = "genotype", y = "Abundance", fill="labels_family")) +
  geom_bar(stat ="identity") +
  scale_colour_manual(values=all_colors) +
  scale_fill_manual(values=all_colors) +
  guides(fill=guide_legend(title="family")) 

all_fun_legend <- cowplot::get_legend(all_fun_plot)
# grid.newpage()
# grid.draw(all_fun_legend) 

```

#### BX enriched and depleted fOTUs in the rhizosphere 
```{r Plot_all_rhizo_BXsensitive_fOTUs, echo=F, fig.height=8, fig.width=16}

Plot_all_rhizo_BXsensitive_fOTUs <- multi_panel_figure(height=c(80,1), width=c(100,100,100))
Plot_all_rhizo_BXsensitive_fOTUs %<>%
  fill_panel(Plot_all_rhizo_BXenriched_fOTUs, column=1, row=1) %<>%
  fill_panel(Plot_all_rhizo_BXdepleted_fOTUs, column=2, row=1) %<>%
  fill_panel(all_fun_legend, column=3, row=1)
print(Plot_all_rhizo_BXsensitive_fOTUs)

```


#### BX enriched and depleted fOTUs in roots 
```{r Plot_all_root_BXsensitive_fOTUs, echo=F, fig.height=8, fig.width=16}

Plot_all_root_BXsensitive_fOTUs <- multi_panel_figure(height=c(80,1), width=c(100,100,100))
Plot_all_root_BXsensitive_fOTUs %<>%
  fill_panel(Plot_all_root_BXenriched_fOTUs, column=1, row=1) %<>%
  fill_panel(Plot_all_root_BXdepleted_fOTUs, column=2, row=1) %<>%
  fill_panel(all_fun_legend, column=3, row=1)
print(Plot_all_root_BXsensitive_fOTUs)

```


#### Figure 6 | Taxonomic pattern of BX-sensitive ROOT microbes
```{r figure 6 barplot BX-sensitive OTUs, warning=F, echo=F, include=F}
#library(magrittr)
#library(multipanelfigure)
Fig_6_barplot_BXsensitive_OTUs_root <- multi_panel_figure(height=c(80,80), width=c(100,100,30,100))
Fig_6_barplot_BXsensitive_OTUs_root %<>%
  fill_panel(Plot_all_root_BXenriched_bOTUs, column = 1, row = 1) %<>%
  fill_panel(Plot_all_root_BXdepleted_bOTUs, column = 2, row = 1) %<>%
  fill_panel(Plot_all_root_BXenriched_fOTUs, column = 1, row = 2) %<>%
  fill_panel(Plot_all_root_BXdepleted_fOTUs, column = 2, row = 2) %<>%
  fill_panel(all_bact_legend, column = 4, row = 1) %<>%
  fill_panel(all_fun_legend, column = 4, row = 2)

pdf("figures/Fig_6_barplot_BXsensitiveOTUs_root.pdf", width=16, height=8)
print(Fig_6_barplot_BXsensitive_OTUs_root)
dev.off()

```


#### Figure S8 | Taxonomic pattern of BX-sensitive RHIZOSPHERE microbes
```{r Figure S8 barplot BX-sensitive OTUs, warning=F, echo=F, include=F}

Fig_S8_barplot_BXsensitive_OTUs_rhizo <- multi_panel_figure(height=c(80,80), width=c(100,100,30,100))
Fig_S8_barplot_BXsensitive_OTUs_rhizo %<>%
  fill_panel(Plot_all_rhizo_BXenriched_bOTUs, column = 1, row = 1) %<>%
  fill_panel(Plot_all_rhizo_BXdepleted_bOTUs, column = 2, row = 1) %<>%
  fill_panel(Plot_all_rhizo_BXenriched_fOTUs, column = 1, row = 2) %<>%
  fill_panel(Plot_all_rhizo_BXdepleted_fOTUs, column = 2, row = 2) %<>%
  fill_panel(all_bact_legend, column = 4, row = 1) %<>%
  fill_panel(all_fun_legend, column = 4, row = 2)

pdf("figures/Fig_S8_barplot_BXsensitiveOTUs_rhizo.pdf", width=16, height=8)
print(Fig_S8_barplot_BXsensitive_OTUs_rhizo)
dev.off()

```

\pagebreak

### FUNGuild analysis

```{r FUNGuild, echo=F, warning=F}
# order variable levels
fPHYSEQ_all_root_enriched_melt$simplified <- factor(fPHYSEQ_all_root_enriched_melt$simplified, levels=c("Plant pathogen", "Other", "Unassigned"))
fPHYSEQ_all_root_depleted_melt$simplified <- factor(fPHYSEQ_all_root_depleted_melt$simplified, levels=c("Plant pathogen", "Other", "Unassigned"))
fPHYSEQ_all_rhizo_enriched_melt$simplified <- factor(fPHYSEQ_all_rhizo_enriched_melt$simplified, levels=c("Plant pathogen", "Other", "Unassigned"))
fPHYSEQ_all_rhizo_depleted_melt$simplified <- factor(fPHYSEQ_all_rhizo_depleted_melt$simplified, levels=c("Plant pathogen", "Other", "Unassigned"))

fPHYSEQ_all_root_enriched_melt$location <- factor(fPHYSEQ_all_root_enriched_melt$location, levels=c("Changins", "Aurora", "Reckenholz"))
fPHYSEQ_all_root_depleted_melt$location <- factor(fPHYSEQ_all_root_depleted_melt$location, levels=c("Changins", "Aurora", "Reckenholz"))
fPHYSEQ_all_rhizo_enriched_melt$location <- factor(fPHYSEQ_all_rhizo_enriched_melt$location, levels=c("Changins", "Aurora", "Reckenholz"))
fPHYSEQ_all_rhizo_depleted_melt$location <- factor(fPHYSEQ_all_rhizo_depleted_melt$location, levels=c("Changins", "Aurora", "Reckenholz"))

# colors
cols_guilds <- c("forestgreen", "lightgrey", "dimgrey")
names(cols_guilds) <- c("Plant pathogen", "Other", "Unassigned")

# plot legend
guild_fun_plot <- ggplot(fPHYSEQ_all_root_depleted_melt, aes_string(x="genotype", y="Abundance", fill="simplified")) +
  geom_bar(stat ="identity") +
  scale_colour_manual(values=cols_guilds) +
  scale_fill_manual(values=cols_guilds) +
  guides(fill=guide_legend(title="Possible guilds")) 
guild_fun_legend <- cowplot::get_legend(guild_fun_plot)

# Transform data for QUALITATIVE FUNGuild plot
fPHYSEQ_all_root_enriched_melt$Occurrence <- ifelse(fPHYSEQ_all_root_enriched_melt$Abundance>0, 1, 0)
fPHYSEQ_all_root_depleted_melt$Occurrence <- ifelse(fPHYSEQ_all_root_depleted_melt$Abundance>0, 1, 0)
fPHYSEQ_all_rhizo_enriched_melt$Occurrence <- ifelse(fPHYSEQ_all_rhizo_enriched_melt$Abundance>0, 1, 0)
fPHYSEQ_all_rhizo_depleted_melt$Occurrence <- ifelse(fPHYSEQ_all_rhizo_depleted_melt$Abundance>0, 1, 0)
```


#### Fig 6 | Root: BX-sensitive fOTUs per FUNGuild
```{r fig 6 FUNGuild Root, echo=F, fig.height=8, fig.width=16}

Plot_all_root_BXenriched_fOTUs_FUNGuild <- fun_plotting_fig6(fPHYSEQ_all_root_enriched_melt, cols_guilds, "simplified") + ylim(c(0,60))
Plot_all_root_BXdepleted_fOTUs_FUNGuild <- fun_plotting_fig6(fPHYSEQ_all_root_depleted_melt, cols_guilds, "simplified") + ylim(c(0,60))

Plot_all_root_BXsensitive_fOTUs_FUNGuild <- multi_panel_figure(height=c(80,80), width=c(100,100,100))
Plot_all_root_BXsensitive_fOTUs_FUNGuild %<>%
  fill_panel(Plot_all_root_BXenriched_fOTUs_FUNGuild, column=1, row=1) %<>%
  fill_panel(Plot_all_root_BXdepleted_fOTUs_FUNGuild, column=2, row=1) %<>%
  fill_panel(guild_fun_legend, column=3, row=1)
print(Plot_all_root_BXsensitive_fOTUs_FUNGuild)

pdf("figures/Fig_6_FUNGuild_root_analysis.pdf", width=16, height=8)
print(Plot_all_root_BXsensitive_fOTUs_FUNGuild)
noshow <- dev.off()

```



#### Fig S8 | Rhizosphere: BX-sensitive fOTUs per FUNGuild
```{r Fig S8 FUNGuild Rhizosphere, echo=F, fig.height=8, fig.width=16}

Plot_all_rhizo_BXenriched_fOTUs_FUNGuild <- fun_plotting_fig6(fPHYSEQ_all_rhizo_enriched_melt, cols_guilds, "simplified") + ylim(c(0,25))
Plot_all_rhizo_BXdepleted_fOTUs_FUNGuild <- fun_plotting_fig6(fPHYSEQ_all_rhizo_depleted_melt, cols_guilds, "simplified") + ylim(c(0,25))

Plot_all_rhizo_BXsensitive_fOTUs_FUNGuild <- multi_panel_figure(height=c(80,80), width=c(100,100,100))
Plot_all_rhizo_BXsensitive_fOTUs_FUNGuild %<>%
  fill_panel(Plot_all_rhizo_BXenriched_fOTUs_FUNGuild, column=1, row=1) %<>%
  fill_panel(Plot_all_rhizo_BXdepleted_fOTUs_FUNGuild, column=2, row=1) %<>%
  fill_panel(guild_fun_legend, column=3, row=1)
print(Plot_all_rhizo_BXsensitive_fOTUs_FUNGuild)

pdf("figures/Fig_S8_FUNGuild_rhizo_analysis.pdf", width=16, height=8)
print(Plot_all_rhizo_BXsensitive_fOTUs_FUNGuild)
noshow <- dev.off()

```

```{r abundance Bacteroidetes - fungal pathogens, echo=F, warning=F}

## TMM RA data
fPHYSEQ_tmm_ra_melt <- psmelt(fPHYSEQ_tmm_ra)
bPHYSEQ_tmm_ra_melt <- psmelt(bPHYSEQ_tmm_ra) 

# data fungal pathogens 
fGuild_tmm_ra_sumAbundance <- fPHYSEQ_tmm_ra_melt %>%
  dplyr::group_by(groups, rep, simplified, genotype, location, compartment) %>%
  dplyr::summarise(sum_guild = sum(Abundance) ) # create dataset with W22, mutants?

# data phyla
bLabels_tmm_ra_sumAbundance <- bPHYSEQ_tmm_ra_melt %>%
  dplyr::group_by(groups, rep, labels, genotype, location, compartment) %>%
  dplyr::summarise(sum_phyla = sum(Abundance) ) 

# unique ID
fGuild_tmm_ra_sumAbundance$unique_ID <- paste(fGuild_tmm_ra_sumAbundance$groups, fGuild_tmm_ra_sumAbundance$rep, sep="_")
bLabels_tmm_ra_sumAbundance$unique_ID <- paste(bLabels_tmm_ra_sumAbundance$groups, bLabels_tmm_ra_sumAbundance$rep, sep="_")

# select only pathogen and bacteroidetes, only groups analyzed for pathogens
sum_pathogen <- fGuild_tmm_ra_sumAbundance[fGuild_tmm_ra_sumAbundance$simplified=="Plant pathogen"&fGuild_tmm_ra_sumAbundance$unique_ID %in% bLabels_tmm_ra_sumAbundance$unique_ID,]
sum_bacteroidetes <- bLabels_tmm_ra_sumAbundance[bLabels_tmm_ra_sumAbundance$labels=="Bacteroidetes"&bLabels_tmm_ra_sumAbundance$unique_ID %in% fGuild_tmm_ra_sumAbundance$unique_ID,]

# merge
sum_bacteroidetes_pathogen <- merge(sum_pathogen, sum_bacteroidetes, by=c("unique_ID", "groups"))

# plot Bacteroidetes vs pathogen for each sample

## Explore: All
all_genotype <- ggplot(sum_bacteroidetes_pathogen, aes(x=sum_phyla, y=sum_guild)) +
  geom_point(aes(col=genotype.x)) +
  scale_color_manual(values=c("gold2", "forestgreen")) +
  geom_smooth(method="lm", formula=y~x) +
  xlab("Bacteroidetes abundance (%, TMM)") +
  ylab("Potential pathogen abundance (%, TMM)")

all_comp_loc <- ggplot(sum_bacteroidetes_pathogen, aes(x=sum_phyla, y=sum_guild)) +
  geom_point(aes(col=compartment.x, shape=location.x)) +
  scale_color_manual(values=c("red3", "cadetblue3")) +
  scale_shape_manual(values=c(0,3,16)) +
  xlab("Bacteroidetes abundance (%, TMM)") +
  ylab("Potential pathogen abundance (%, TMM)")

# Separate sub-groups compartment... 
#compartments
sum_bacteroidetes_pathogen_rhizo <- sum_bacteroidetes_pathogen[sum_bacteroidetes_pathogen$compartment.x=="rhizo",]
sum_bacteroidetes_pathogen_root <- sum_bacteroidetes_pathogen[sum_bacteroidetes_pathogen$compartment.x=="root",]
```

#### Figure 6C | Abundance Bacteroidetes vs possible pathogens in the roots
```{r plot Bacteroidetes - fungal pathogens root, echo=F, warning=F}

### Root 

# residuals non-normal -> log
sum_bacteroidetes_pathogen_root$sum_phyla_log <-log(sum_bacteroidetes_pathogen_root$sum_phyla)
sum_bacteroidetes_pathogen_root$sum_guild_log <-log(sum_bacteroidetes_pathogen_root$sum_guild)

# statistical test
fit <- lm(sum_bacteroidetes_pathogen_root$sum_phyla_log~sum_bacteroidetes_pathogen_root$sum_guild_log+sum_bacteroidetes_pathogen_root$location.x )
sum_fit_root <- summary(fit)
#library(broom)
model.metrics <- augment(fit)
#shapiro.test(model.metrics$.resid) # shapiro.test(residuals(fit))
#bartlett.test(model.metrics$.resid~model.metrics$sum_bacteroidetes_pathogen_root.location.x); plot(model.metrics$.resid)

## Roots
plot_Bact_fPatho_root <- ggplot(sum_bacteroidetes_pathogen_root, aes(x=sum_phyla_log, y=sum_guild_log)) +
  geom_point(aes(col=location.x, shape=genotype.x)) +
  scale_color_manual(values=c("goldenrod2", "dodgerblue2", "firebrick2")) +
    scale_shape_manual(values=c(4, 16)) +
  geom_smooth(method="lm", aes(x=sum_phyla_log, y=sum_guild_log, col=location.x),se=F) +
  xlab("Bacteroidetes abundance (log %)") +
  ylab("Potential pathogen abundance (log %)") +
  guides(col=guide_legend(title="Location")) 
 
print(plot_Bact_fPatho_root)


```

```{r plot Bacteroidetes - fungal pathogens rhizo, echo=F, warning=F}

#### Abundance Bacteroidetes vs possible pathogens in the rhizosphere


# residuals non-normal -> log
sum_bacteroidetes_pathogen_rhizo$sum_phyla_log <-log(sum_bacteroidetes_pathogen_rhizo$sum_phyla)
sum_bacteroidetes_pathogen_rhizo$sum_guild_log <-log(sum_bacteroidetes_pathogen_rhizo$sum_guild)

# Plot rhizo
plot_Bact_fPatho_rhizo <- ggplot(sum_bacteroidetes_pathogen_rhizo, aes(x=sum_phyla_log, y=sum_guild_log)) +
  geom_point(aes(col=location.x, shape=genotype.x)) +
  scale_color_manual(values=c("goldenrod2", "dodgerblue2", "firebrick2")) +
  scale_shape_manual(values=c(4, 16)) +
  geom_smooth(method="lm", aes(x=sum_phyla_log, y=sum_guild_log, col=location.x),se=F) +
  xlab("Bacteroidetes abundance (log %)") +
  ylab("Potential pathogen abundance (log %)") +
  guides(col=guide_legend(title="Location")) 
  
print(plot_Bact_fPatho_rhizo)


# data : tmm counts in RA (%)
# without RA: tmm counts (unit = tmm counts??)



## test without RA: same thing 
bdat_tmm <- merge_six_row(t(t(bEDGER_Ch_root_BX$counts)*bEDGER_Ch_root_BX$samples$norm.factors), t(t(bEDGER_Au_root_BX$counts)*bEDGER_Au_root_BX$samples$norm.factors), t(t(bEDGER_Re_root_BX$counts)*bEDGER_Re_root_BX$samples$norm.factors), t(t(bEDGER_Ch_rhizo_BX$counts)*bEDGER_Ch_rhizo_BX$samples$norm.factors),t(t(bEDGER_Au_rhizo_BX$counts)*bEDGER_Au_rhizo_BX$samples$norm.factors),t(t(bEDGER_Re_rhizo_BX$counts)*bEDGER_Re_rhizo_BX$samples$norm.factors))

bdat_tmm[is.na(bdat_tmm)] <- 0

bPHYSEQ_tmm <- phyloseq(sample_data(bDESIGN),
                           otu_table(bdat_tmm, taxa_are_rows=T),
                           tax_table(as.matrix(bTAX2[rownames(bdat_tmm),])))
bPHYSEQ_tmm_melt <- psmelt(bPHYSEQ_tmm)

fdat_tmm <- merge_six_row(fEDGER_Ch_root_BX$counts*fEDGER_Ch_root_BX$samples$norm.factors, fEDGER_Au_root_BX$counts*fEDGER_Au_root_BX$samples$norm.factors,
          fEDGER_Re_root_BX$counts*fEDGER_Re_root_BX$samples$norm.factors,
          fEDGER_Ch_rhizo_BX$counts*fEDGER_Ch_rhizo_BX$samples$norm.factors,
          fEDGER_Au_rhizo_BX$counts*fEDGER_Au_rhizo_BX$samples$norm.factors,
          fEDGER_Re_rhizo_BX$counts*fEDGER_Re_rhizo_BX$samples$norm.factors)

fdat_tmm[is.na(fdat_tmm)] <- 0

fPHYSEQ_tmm <- phyloseq(sample_data(fDESIGN),
                           otu_table(fdat_tmm, taxa_are_rows=T),
                           tax_table(as.matrix(fTAX_guild[rownames(fdat_tmm),])))
fPHYSEQ_tmm_melt <- psmelt(fPHYSEQ_tmm)

# data fungal pathogens 
fGuild_tmm_sumAbundance <- fPHYSEQ_tmm_melt %>%
  group_by(groups, rep, simplified, genotype, location, compartment) %>%
  summarise(sum_guild = sum(Abundance) ) 

# data phyla
bLabels_tmm_sumAbundance <- bPHYSEQ_tmm_melt %>%
  group_by(groups, rep, labels, genotype, location, compartment) %>%
  summarise(sum_phyla = sum(Abundance) ) 

# unique ID
fGuild_tmm_sumAbundance$unique_ID <- paste(fGuild_tmm_sumAbundance$groups, fGuild_tmm_sumAbundance$rep, sep="_")
bLabels_tmm_sumAbundance$unique_ID <- paste(bLabels_tmm_sumAbundance$groups, bLabels_tmm_sumAbundance$rep, sep="_")

# select only pathogen and bacteroidetes, only groups analyzed for pathogens
sum_pathogen <- fGuild_tmm_sumAbundance[fGuild_tmm_sumAbundance$simplified=="Plant pathogen"&fGuild_tmm_sumAbundance$unique_ID %in% bLabels_tmm_sumAbundance$unique_ID,]
sum_bacteroidetes <- bLabels_tmm_sumAbundance[bLabels_tmm_sumAbundance$labels=="Bacteroidetes"&bLabels_tmm_sumAbundance$unique_ID %in% fGuild_tmm_sumAbundance$unique_ID,]

# merge
sum_bacteroidetes_pathogen <- merge(sum_pathogen, sum_bacteroidetes, by=c("unique_ID", "groups"))

# Separate sub-groups # separate location_compartment... 
#compartments
sum_bacteroidetes_pathogen_rhizo <- sum_bacteroidetes_pathogen[sum_bacteroidetes_pathogen$compartment.x=="rhizo",]
sum_bacteroidetes_pathogen_root <- sum_bacteroidetes_pathogen[sum_bacteroidetes_pathogen$compartment.x=="root",]

fit <- lm(sum_bacteroidetes_pathogen_root$sum_phyla~sum_bacteroidetes_pathogen_root$sum_guild+sum_bacteroidetes_pathogen_root$location.x )
sum_fit_root <- summary(fit)

# residuals non-normal -> log
sum_bacteroidetes_pathogen_root$sum_phyla_log <-log(sum_bacteroidetes_pathogen_root$sum_phyla)
sum_bacteroidetes_pathogen_root$sum_guild_log <-log(sum_bacteroidetes_pathogen_root$sum_guild)

# stats
fit <- lm(sum_bacteroidetes_pathogen_root$sum_phyla_log~sum_bacteroidetes_pathogen_root$sum_guild_log+sum_bacteroidetes_pathogen_root$location.x )
sum_fit_root <- summary(fit)
model.metrics <- augment(fit)
#shapiro.test(model.metrics$.resid) # shapiro.test(residuals(fit))
#bartlett.test(model.metrics$.resid~model.metrics$sum_bacteroidetes_pathogen_root.location.x); plot(model.metrics$.resid)

plot_Bact_fPath_root_notRA <- ggplot(sum_bacteroidetes_pathogen_root, aes(x=sum_phyla_log, y=sum_guild_log)) +
  geom_point(aes(col=location.x)) +
  scale_color_manual(values=c("goldenrod2", "firebrick2", "dodgerblue3")) +
  geom_smooth(method="lm", aes(x=sum_phyla_log, y=sum_guild_log, col=location.x)) + xlab("Bacteroidetes abundance") + ylab("Possible pathogen abundance")


```


\pagebreak

## Abundance of Methylophilaceae bOTUs

We inspected the abundance of Methylophilaceae bOTUs, because they were previously found to be enriched by BXs in experiments conducted with Changins (Hu et al., 2018) and Sheffield soils (Cotton et al., 2019). We find the following Methylophilaceae differing significanly between WT and bx1 plants in root and rhizosphere samples:    

```{r, echo=F}
# bTAX[bTAX$order=="Methylophilales", c(3:6)]
# Methylo_bOTUs <- rownames(bTAX)[bTAX$order=="Methylophilales"]
```

#### Changins rhizosphere
```{r Methylophilaceae in Changins 1, warning=F, echo=F}
bEDGER_Ch_rhizo_BX_fit_lrt_table[bEDGER_Ch_rhizo_BX_fit_lrt_table$order=="Methylophilales", c(4:5,11:12,15)]
```
#### Changins roots
```{r Methylophilaceae in Changins 2, warning=F, echo=F}
bEDGER_Ch_root_BX_fit_lrt_table[bEDGER_Ch_root_BX_fit_lrt_table$order=="Methylophilales", c(4:5,11:12,15)]
```

#### Aurora rhizosphere
```{r Methylophilaceae in Aurora 1, warning=F, echo=F}
bEDGER_Au_rhizo_BX_fit_lrt_table[bEDGER_Au_rhizo_BX_fit_lrt_table$order=="Methylophilales", c(4:5,11:12,15)]
```
#### Aurora root
```{r Methylophilaceae in Aurora 2, warning=F, echo=F}
bEDGER_Au_root_BX_fit_lrt_table[bEDGER_Au_root_BX_fit_lrt_table$order=="Methylophilales", c(4:5,11:12,15)]
```

#### Reckenholz rhizosphere
```{r Methylophilaceae in Reckenholz 1, warning=F, echo=F}
bEDGER_Re_rhizo_BX_fit_lrt_table[bEDGER_Re_rhizo_BX_fit_lrt_table$order=="Methylophilales", c(4:5,11:12,15)]
```
#### Reckenholz root
```{r Methylophilaceae in Reckenholz 2, warning=F, echo=F}
bEDGER_Re_root_BX_fit_lrt_table[bEDGER_Re_root_BX_fit_lrt_table$order=="Methylophilales", c(4:5,11:12,15)]
```


```{r Identifying Methylophilaceae bOTUs, warning=F, echo=F}
#### Plotting

### Attention: the Methylohilaceae OTUs are low abundant and they are not seen in the previous 'data_tmm_ra' tables because of the filtering threshold (filter=present in at least 6 samples) at the TMM step. Therefore, we re-prepare 'data_TMM_ra' tables but with threshold.filter=1

### TMM normalized counts without filtering ...
# source("functions/fun_norm_edgeR_obj.R") 
bEDGER_Ch_rhizo_BX_filt1 <- edgeR.object.TMM(dat=bDAT_Ch_rhizo_BX, threshold.filter=1, groups=bDESIGN_Ch_rhizo_BX$groups, tax=bTAX2)
bEDGER_Au_rhizo_BX_filt1 <- edgeR.object.TMM(dat=bDAT_Au_rhizo_BX, threshold.filter=1, groups=bDESIGN_Au_rhizo_BX$groups, tax=bTAX2)
bEDGER_Re_rhizo_BX_filt1 <- edgeR.object.TMM(dat=bDAT_Re_rhizo_BX, threshold.filter=1, groups=bDESIGN_Re_rhizo_BX$groups, tax=bTAX2)
bEDGER_Ch_root_BX_filt1 <- edgeR.object.TMM(dat=bDAT_Ch_root_BX, threshold.filter=1, groups=bDESIGN_Ch_root_BX$groups, tax=bTAX2)
bEDGER_Au_root_BX_filt1 <- edgeR.object.TMM(dat=bDAT_Au_root_BX, threshold.filter=1, groups=bDESIGN_Au_root_BX$groups, tax=bTAX2)
bEDGER_Re_root_BX_filt1 <- edgeR.object.TMM(dat=bDAT_Re_root_BX, threshold.filter=1, groups=bDESIGN_Re_root_BX$groups, tax=bTAX2)

## ... and express them as % relative abundance (RA)
# source("functions/fun_norm_edgeR_obj.R")
bEDGER_Ch_rhizo_BX_filt1_RA <- convert.edgeR.to.datRA(bEDGER_Ch_rhizo_BX_filt1)
bEDGER_Au_rhizo_BX_filt1_RA <- convert.edgeR.to.datRA(bEDGER_Au_rhizo_BX_filt1)
bEDGER_Re_rhizo_BX_filt1_RA <- convert.edgeR.to.datRA(bEDGER_Re_rhizo_BX_filt1)
bEDGER_Ch_root_BX_filt1_RA <- convert.edgeR.to.datRA(bEDGER_Ch_root_BX_filt1)
bEDGER_Au_root_BX_filt1_RA <- convert.edgeR.to.datRA(bEDGER_Au_root_BX_filt1)
bEDGER_Re_root_BX_filt1_RA <- convert.edgeR.to.datRA(bEDGER_Re_root_BX_filt1)

### for plotting, we go into Phyloseq again
## Merge and subset to Methylo_OTUs
Methylo_bOTUs <- c("bOTU270","bOTU479","bOTU647")
# source("functions/fun_union_intersect_merge.R") 
bEDGER_BX_filt1_RA <- merge_six_row(bEDGER_Ch_rhizo_BX_filt1_RA[Methylo_bOTUs,], bEDGER_Re_rhizo_BX_filt1_RA[Methylo_bOTUs,], bEDGER_Au_rhizo_BX_filt1_RA[Methylo_bOTUs,], 
                                    bEDGER_Ch_root_BX_filt1_RA[Methylo_bOTUs,], bEDGER_Re_root_BX_filt1_RA[Methylo_bOTUs,], bEDGER_Au_root_BX_filt1_RA[Methylo_bOTUs,])
bPHYSEQ_tmm_filt1_ra <- phyloseq(sample_data(bDESIGN),
                           otu_table(bEDGER_BX_filt1_RA, taxa_are_rows=T),
                           tax_table(as.matrix(bTAX[rownames(bEDGER_BX_filt1_RA),])) )

#### split to samples for each location and compartment 
#### calculate mean and SE of each OTU for both genotypes
col_subset2 <- c("OTU", "Sample", "Abundance", "location", "genotype", "location_compartment")
# source("functions/fun_plotting_figS10.R") 

## Changins Rhizosphere
bPHYSEQ_tmm_ra_Ch_rhizo <- prune_samples(sample_data(bPHYSEQ_tmm_filt1_ra)$location_compartment=="Changins_rhizo", bPHYSEQ_tmm_filt1_ra)
bPHYSEQ_tmm_ra_Ch_rhizo_melt <- psmelt(bPHYSEQ_tmm_ra_Ch_rhizo)[,col_subset2]
bPHYSEQ_tmm_ra_Ch_rhizo_melt_mean <- bPHYSEQ_tmm_ra_Ch_rhizo_melt %>% group_by(OTU, genotype) %>% dplyr::summarize(Mean=mean(Abundance), SE=se(Abundance))
Methylophilaceae_rhizo_Changins_barplot <- fun_plotting_figS10(bPHYSEQ_tmm_ra_Ch_rhizo_melt_mean, "Rhizosphere Changins", 1.00, c("", "", "", "*", "", "*"))

## Changins Root
bPHYSEQ_tmm_ra_Ch_root <- prune_samples(sample_data(bPHYSEQ_tmm_filt1_ra)$location_compartment=="Changins_root", bPHYSEQ_tmm_filt1_ra)
bPHYSEQ_tmm_ra_Ch_root_melt <- psmelt(bPHYSEQ_tmm_ra_Ch_root)[,col_subset2]
bPHYSEQ_tmm_ra_Ch_root_melt_mean <- bPHYSEQ_tmm_ra_Ch_root_melt %>% group_by(OTU, genotype) %>% dplyr::summarize(Mean=mean(Abundance), SE=se(Abundance))
Methylophilaceae_root_Changins_barplot <- fun_plotting_figS10(bPHYSEQ_tmm_ra_Ch_root_melt_mean, "Root Changins", 1.00, c("", "", "", "", "", ""))

## Aurora Rhizosphere
bPHYSEQ_tmm_ra_Au_rhizo <- prune_samples(sample_data(bPHYSEQ_tmm_filt1_ra)$location_compartment=="Aurora_rhizo", bPHYSEQ_tmm_filt1_ra)
bPHYSEQ_tmm_ra_Au_rhizo_melt <- psmelt(bPHYSEQ_tmm_ra_Au_rhizo)[,col_subset2]
bPHYSEQ_tmm_ra_Au_rhizo_melt_mean <- bPHYSEQ_tmm_ra_Au_rhizo_melt %>% group_by(OTU, genotype) %>% dplyr::summarize(Mean=mean(Abundance), SE=se(Abundance))
Methylophilaceae_rhizo_Aurora_barplot <- fun_plotting_figS10(bPHYSEQ_tmm_ra_Au_rhizo_melt_mean, "Rhizosphere Aurora", 0.06, c("", "", "", "", "", "*"))

## Aurora Root
bPHYSEQ_tmm_ra_Au_root <- prune_samples(sample_data(bPHYSEQ_tmm_filt1_ra)$location_compartment=="Aurora_root", bPHYSEQ_tmm_filt1_ra)
bPHYSEQ_tmm_ra_Au_root_melt <- psmelt(bPHYSEQ_tmm_ra_Au_root)[,col_subset2]
bPHYSEQ_tmm_ra_Au_root_melt_mean <- bPHYSEQ_tmm_ra_Au_root_melt %>% group_by(OTU, genotype) %>% dplyr::summarize(Mean=mean(Abundance), SE=se(Abundance))
Methylophilaceae_root_Aurora_barplot <- fun_plotting_figS10(bPHYSEQ_tmm_ra_Au_root_melt_mean, "Root Aurora", 0.06, c("", "", "", "", "", "*"))

## Reckenholz Rhizosphere
bPHYSEQ_tmm_ra_Re_rhizo <- prune_samples(sample_data(bPHYSEQ_tmm_filt1_ra)$location_compartment=="Reckenholz_rhizo", bPHYSEQ_tmm_filt1_ra)
bPHYSEQ_tmm_ra_Re_rhizo_melt <- psmelt(bPHYSEQ_tmm_ra_Re_rhizo)[,col_subset2]
bPHYSEQ_tmm_ra_Re_rhizo_melt_mean <- bPHYSEQ_tmm_ra_Re_rhizo_melt %>% group_by(OTU, genotype) %>% dplyr::summarize(Mean=mean(Abundance), SE=se(Abundance))
Methylophilaceae_rhizo_Reckenholz_barplot <- fun_plotting_figS10(bPHYSEQ_tmm_ra_Re_rhizo_melt_mean, "Rhizosphere Reckenholz", 0.06, c("", "*", "", "*", "", ""))

## Reckenholz Root
bPHYSEQ_tmm_ra_Re_root <- prune_samples(sample_data(bPHYSEQ_tmm_filt1_ra)$location_compartment=="Reckenholz_root", bPHYSEQ_tmm_filt1_ra)
bPHYSEQ_tmm_ra_Re_root_melt <- psmelt(bPHYSEQ_tmm_ra_Re_root)[,col_subset2]
bPHYSEQ_tmm_ra_Re_root_melt_mean <- bPHYSEQ_tmm_ra_Re_root_melt %>% group_by(OTU, genotype) %>% dplyr::summarize(Mean=mean(Abundance), SE=se(Abundance))
Methylophilaceae_root_Reckenholz_barplot <- fun_plotting_figS10(bPHYSEQ_tmm_ra_Re_root_melt_mean, "Root Reckenholz", 0.06, c("", "", "", "", "", ""))
```

\pagebreak
```{r combined plot methylo, warning=F, echo=F, fig.height=15, fig.width=15}
Fig_S9_barplot_Methylo_OTUs <- multi_panel_figure(height=c(180), width=c(180), columns=2, rows=3)
Fig_S9_barplot_Methylo_OTUs %<>%
  fill_panel(Methylophilaceae_rhizo_Changins_barplot, column=1, row=1) %<>%
  fill_panel(Methylophilaceae_root_Changins_barplot, column=2, row=1) %<>%
  fill_panel(Methylophilaceae_rhizo_Aurora_barplot, column=1, row=2) %<>% 
  fill_panel(Methylophilaceae_root_Aurora_barplot, column=2, row=2) %<>% 
  fill_panel(Methylophilaceae_rhizo_Reckenholz_barplot, column=1, row=3) %<>%
  fill_panel(Methylophilaceae_root_Reckenholz_barplot, column=2, row=3) 
 
print(Fig_S9_barplot_Methylo_OTUs)
```

#### Figure S9 | Barplot methylotrophic bOTU abundance

```{r Fig S9 plots methylo, warning=F, echo=F, include=F}
pdf("figures/Fig_S9_barplot_methylo_OTUs.pdf", width=10, height=10)
print(Fig_S9_barplot_Methylo_OTUs)
dev.off()
```


